%% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[a4paper,11pt,oneside,english]{sphinxmanual}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
\edef\sphinxdqmaybe{\ifdefined\DeclareUnicodeCharacterAsOptional\string"\fi}
  \DeclareUnicodeCharacter{\sphinxdqmaybe00A0}{\nobreakspace}
  \DeclareUnicodeCharacter{\sphinxdqmaybe2500}{\sphinxunichar{2500}}
  \DeclareUnicodeCharacter{\sphinxdqmaybe2502}{\sphinxunichar{2502}}
  \DeclareUnicodeCharacter{\sphinxdqmaybe2514}{\sphinxunichar{2514}}
  \DeclareUnicodeCharacter{\sphinxdqmaybe251C}{\sphinxunichar{251C}}
  \DeclareUnicodeCharacter{\sphinxdqmaybe2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage[english]{babel}
\usepackage{times}
\usepackage[Bjarne]{fncychap}
\usepackage[,numfigreset=1,mathnumfig]{sphinx}

\fvset{fontsize=\small}
\usepackage{geometry}

% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}

\addto\captionsenglish{\renewcommand{\figurename}{Fig.\@ }}
\makeatletter
\def\fnum@figure{\figurename\thefigure{}}
\makeatother
\addto\captionsenglish{\renewcommand{\tablename}{Table }}
\makeatletter
\def\fnum@table{\tablename\thetable{}}
\makeatother
\addto\captionsenglish{\renewcommand{\literalblockname}{Listing}}

\addto\captionsenglish{\renewcommand{\literalblockcontinuedname}{continued from previous page}}
\addto\captionsenglish{\renewcommand{\literalblockcontinuesname}{continues on next page}}
\addto\captionsenglish{\renewcommand{\sphinxnonalphabeticalgroupname}{Non-alphabetical}}
\addto\captionsenglish{\renewcommand{\sphinxsymbolsname}{Symbols}}
\addto\captionsenglish{\renewcommand{\sphinxnumbersname}{Numbers}}

\addto\extrasenglish{\def\pageautorefname{page}}

\setcounter{tocdepth}{1}


\renewcommand{\releasename}{Version}
\let\OrigVerbatim\OriginalVerbatim
\renewcommand{\OriginalVerbatim}[1][1]{\OrigVerbatim[#1,frame=single]}
\pagestyle{normal}
\thispagestyle{normal}
\pagenumbering{arabic}
\usepackage{braket}


\title{GENESIS User Guide}
\date{May 22, 2022}
\release{1.7.1}
\author{RIKEN}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Release}
\makeindex
\begin{document}

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}

%%%%%%%
\begin{center}
{\Huge GENESIS 1.7.1}
\end{center}

\vspace{2ex}

\begin{quote}
  Project Leader: Yuji Sugita (RIKEN)

  Current main developers:
  Jaewoon Jung (RIKEN), 
  Chigusa Kobayashi (RIKEN), 
  Takaharu Mori (RIKEN), 
  Hiraku Oshima (RIKEN), 
  Kiyoshi Yagi (RIKEN)

  Current developers/contributors:
  Shingo Ito (RIKEN), 
  Motoshi Kamiya (RIKEN/IMS), 
  Kento Kasahara (RIKEN/Osaka Univ.),
  Yasuhiro Matsunaga (RIKEN/Saitama Univ.),
  Daisuke Matsuoka (RIKEN/RIST), 
  Osamu Miyashita (RIKEN), 
  Suyong Re (RIKEN/NIBIOHN),
  Ai Shinobu (RIKEN), 
  Yosuke Sumiya (RIKEN),
  Florence Tama (RIKEN/Nagoya Univ.),
  Cheng Tan (RIKEN),
  Isseki Yu (RIKEN/Maebashi Institute of Technology)

  Other developers/contributors for older versions:
  Tadashi Ando (RIKEN),
  Michael Feig (Michigan State University), 
  Raimondas Galvelis (RIKEN),
  Ryuhei Harada (RIKEN), 
  Takashi Imai (RIKEN), 
  Yasuaki Komuro (RIKEN), 
  Yasuhito Karino (RIKEN),
  Naoyuki Miyashita (RIKEN), 
  Wataru Nishima (RIKEN),
  Donatas Surblys (RIKEN), 
  Koichi Tamura (RIKEN), 
  Kenta Yamada (RIKEN),
  Takao Yoda (Nagahama Institute of Bio-Science and Technology),

  Acknowledgments: 
  Norio Takase (Isogo Soft), 
  Yasumasa Joti (RIKEN SPring8),
  Akira Naruse (NVIDIA), 
  Yukihiko Hirano (NVIDIA Japan),
  Hikaru Inoue (Fujitsu Ltd.), 
  Tomoyuki Noda (Fujitsu Ltd.),
  Kiyotaka Sakamoto (Fujitsu Ltd.), 
  Yoshinobu Akinaga (VINAS),
  Yoshitake Sakae (RIST),
  Nobuhiko Kato (ASTOM R\&D),
  Toru Shiozaki (QSimulate), 
  Klaas Gunst (QSimualte),
  Hideyo Yoshida (JSOL Corporation),
  Kenta Chaki (JSOL Corporation)

  Copyright \copyright 2014-2021 RIKEN. All Rights Reserved
\end{quote}

\vspace{3ex}

%%%%%%%
{\LARGE GENESIS website}

\begin{quote}
  \url{https://www.r-ccs.riken.jp/labs/cbrt/}
\end{quote}

\vspace{3ex}

%%%%%%%
{\LARGE Citation Information}

\begin{itemize}
 \item C. Kobayashi, J. Jung, Y. Matsunaga, T. Mori, T. Ando, K. Tamura, M. Kamiya, and Y. Sugita, "GENESIS 1.1: A hybrid-parallel molecular dynamics simulator with enhanced sampling algorithms on multiple computational platforms", J. Comput. Chem. 38, 2193-2206 (2017).

 \item J. Jung, T. Mori, C. Kobayashi, Y. Matsunaga, T. Yoda, M. Feig, and Y. Sugita, "GENESIS: A hybrid-parallel and multi-scale molecular dynamics simulator with enhanced sampling algorithms for biomolecular and cellular simulations", WIREs Computational Molecular Science 5, 310-323 (2015).
\end{itemize}

\vspace{3ex}

%%%%%%%
{\LARGE Copyright Notices}

GENESIS is distributed under the GNU Lesser General Public License
version 3.

\begin{quote}
  Copyright \copyright 2014-2021 RIKEN.

  GENESIS is free software; you can redistribute it and/or
  modify it under the terms of the GNU Lesser General Public
  License as published by the Free Software Foundation; either
  version 3 of the License, or (at your option) any later version.
  
  GENESIS is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
  See the GNU Lesser General Public License for more details.
  
  You should have received a copy of the GNU Lesser General Public
  License along with GENESIS -- see the file COPYING and COPYING.LESSER.
  If not, see \url{https://www.gnu.org/licenses/}.
 
\end{quote}

It should be mentioned this package contains the following softwares
for convenience.  Please note that these are not covered by the license
under which a copy of GENESIS is licensed to you, while neither
composition nor distribution of any derivative work of GENESIS with
these software violates the terms of each license, provided that it meets
every condition of the respective licenses.

\vspace{1ex}

{\large SIMD-oriented Fast Mersenne Twister (SFMT)}

SFMT is a new variant of Mersenne Twister (MT) introduced by Mutsuo Saito and Makoto Matsumoto in 2006. The algorithm was reported at MCQMC 2006.
The routine is distributed under the New BSD License.

\begin{quote}
  Copyright \copyright 2006,2007 Mutsuo Saito, Makoto Matsumoto and Hiroshima University.
  Copyright \copyright 2012 Mutsuo Saito, Makoto Matsumoto, Hiroshima University and The University of Tokyo.
  All rights reserved.

  Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:

  * Redistributions of source code must retain the above copyright
  notice, this list of conditions and the following disclaimer.

  * Redistributions in binary form must reproduce the above
  copyright notice, this list of conditions and the following
  disclaimer in the documentation and/or other materials provided
  with the distribution.

  * Neither the names of Hiroshima University, The University of
  Tokyo nor the names of its contributors may be used to endorse
  or promote products derived from this software without specific
  prior written permission.

  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
\end{quote}

%\vspace{2ex}
%
%{\large Mersenne Twister: A random number generator}
%
%A Mersenne Twister random number generator was originally written in C
%by Makoto Matsumoto and Takuji Nishimura, and later translated into
%Fortran by Hiroshi Takano and Richard Woloshyn. This routine is
%distributed under the GNU General Public License version 2.
%
%\begin{quote}
%  Copyright \copyright 1997 Makoto Matsumoto and Takuji Nishimura.
%
%  Copyright \copyright 1999 Hiroshi Takano.
%
%  Copyright \copyright 1999 Richard Woloshyn.
%
%  This library is free software; you can redistribute it and/or
%  modify it under the terms of the GNU Library General Public
%  License as published by the Free Software Foundation; either
%  version 2 of the License, or (at your option) any later
%  version.
%
%  This library is distributed in the hope that it will be useful,
%  but WITHOUT ANY WARRANTY; without even the implied warranty of
%  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
%  See the GNU Library General Public License for more details.
%  You should have received a copy of the GNU Library General
%  Public License along with this library; if not, write to the
%  Free Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
%  02111-1307  USA
%\end{quote}

\vspace{2ex}

{\large FFTE: A Fast Fourier Transform Package}

FFTE (\url{http://www.ffte.jp/}) is written by Daisuke Takahashi (Tsukuba
University). 

\begin{quote}
  Copyright \copyright 2000-2004, 2008-2011 Daisuke Takahashi (Tsukuba
  University). 

  You may use, copy, modify this code for any purpose (include
  commercial use) and without fee. You may distribute this ORIGINAL
  package. 
\end{quote}

\vspace{2ex}

{\large Complementary error function: erfc04}

A Complementary error function routine (erfc04) is written by SunSoft, 
a Sun Microsystems, Inc. business. 

\begin{quote}
  Copyright \copyright 1993 Sun Microsystems, Inc.

  Developed at SunSoft, a Sun Microsystems, Inc. business. 
  Permission to use, copy, modify, and distribute this 
  software is freely granted, provided that this notice 
  is preserved (see math\_libs.fpp).
\end{quote}

\vspace{2ex}

{\large L-BFGS-B (version 3.0)}

L-BFGS-B (\url{http://users.iems.northwestern.edu/~nocedal/lbfgsb.html})
is written by C. Zhu, R. Byrd, J. Nocedal and J. L. Morales. 

\begin{quote}
   This software is freely available, but we expect that all publications 
   describing work using this software, or all commercial products using 
   it, quote at least one of the references given below. This software is 
   released under the "New BSD License" (aka "Modified BSD License" or 
   "3-clause license"). 

   R. H. Byrd, P. Lu and J. Nocedal. A Limited Memory Algorithm for Bound Constrained Optimization, (1995), SIAM Journal on Scientific and Statistical Computing, 16, 5, pp. 1190-1208.

   C. Zhu, R. H. Byrd and J. Nocedal. L-BFGS-B: Algorithm 778: L-BFGS-B, FORTRAN routines for large scale bound constrained optimization (1997), ACM Transactions on Mathematical Software, Vol 23, Num. 4, pp. 550-560.

   J.L. Morales and J. Nocedal. L-BFGS-B: Remark on Algorithm 778: L-BFGS-B, FORTRAN routines for large scale bound constrained optimization (2011), ACM Transactions on Mathematical Software, Vol 38, Num. 1, Article No. 7.

\end{quote}

\vspace{3ex}

\tableofcontents

\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}



\chapter{Introduction}
\label{\detokenize{00_Introduction:introduction}}\label{\detokenize{00_Introduction:id1}}\label{\detokenize{00_Introduction::doc}}
\sphinxstylestrong{GENESIS} (\sphinxstyleemphasis{Gen}eralized-\sphinxstyleemphasis{E}nsemble \sphinxstyleemphasis{Si}mulation \sphinxstyleemphasis{S}ystem)
is a suite of
computer programs for carrying out molecular dynamics (MD) simulations
of biomolecular systems. MD simulations of biomolecules such as proteins,
nucleic acids, lipid bilayers, N-glycans, are used as
important research tools in structural and molecular biology. Many
useful MD simulation packages \sphinxcite{99_References:amber18-program}
\sphinxcite{99_References:charmm-program} \sphinxcite{99_References:gromacs-program} \sphinxcite{99_References:desmond-program}
\sphinxcite{99_References:namd-program} are now available together with accurate
molecular force field parameter sets \sphinxcite{99_References:case-2005}
\sphinxcite{99_References:mackerell-1998} \sphinxcite{99_References:mackerell-2004} \sphinxcite{99_References:jorgensen-1996}
\sphinxcite{99_References:oostenbrink-2004}. Most of the MD software have been optimized
and parallelized for distributed-memory parallel supercomputers or
PC-clusters. Therefore hundreds of CPUs or CPU cores can be used
efficiently for a single MD simulation of a relatively large
biomolecular system, typically composed of several hundred
thousands of atoms. In recent years, the number of available CPUs
or CPU cores is rapidly increasing. The implmentation of highly
efficient parallel schemes is therefore required in modern MD simulation
programs. Accelerators such as GPGPU (General-Purpose
computing on Graphics Processing Units) also become popular,
and thus their utilization is also desired.
Actually, many MD program packages support various accelerators.

Our major motivation is to develop MD simulation software
with a scalable performance on such modern supercomputers. For this
purpose, we have developed the software from scratch, introducing the
hybrid (MPI + OpenMP) parallelism, several new parallel
algorithms \sphinxcite{99_References:jung-2013} \sphinxcite{99_References:jung-2014}, and GPGPU calculation.
Another motivation is to develop a simple MD program, which can be
easily understood and modified for methodological developments. These two
policies (high parallel performance and simplicity) usually conflict
each other in computer software.
To avoid the conflict, we have developed two MD programs in \sphinxstylestrong{GENESIS},
namely \sphinxstylestrong{SPDYN} (\sphinxstyleemphasis{Sp}atial decomposition \sphinxstyleemphasis{dyn}amics)
and \sphinxstylestrong{ATDYN} (\sphinxstyleemphasis{At}omic decomposition \sphinxstyleemphasis{dyn}amics).

\sphinxstylestrong{SPDYN} and \sphinxstylestrong{ATDYN} share almost the same data structures, subroutines,
and modules, but differ in their parallelization schemes.
In \sphinxstylestrong{SPDYN}, the spatial decomposition scheme is implemented with new
parallel algorithms \sphinxcite{99_References:jung-2013} \sphinxcite{99_References:jung-2014} and GPGPU calculation.
In \sphinxstylestrong{ATDYN}, the atomic decomposition scheme is introduced for simplicity.
The performance of \sphinxstylestrong{ATDYN} is not comparable to \sphinxstylestrong{SPDYN} due to the
simple parallelization scheme.
However, \sphinxstylestrong{ATDYN} is easier to modify for development of new algorithms
or novel molecular models. We hope that users develop new methodologies
in \sphinxstylestrong{ATDYN} at first and, eventually, port them to \sphinxstylestrong{SPDYN} for the
better performance.  As we maintain consistency between the source codes
of \sphinxstylestrong{ATDYN} and \sphinxstylestrong{SPDYN}, switching from \sphinxstylestrong{ATDYN} to \sphinxstylestrong{SPDYN} is
not quite hard.

Other features in \sphinxstylestrong{GENESIS} are listed below:
\begin{itemize}
\item {} 
Not only atomistic molecular force field (CHARMM, AMBER) but
also some coarse-grained models are available in \sphinxstylestrong{ATDYN}.

\item {} 
For extremely large biomolecular systems (more than 10 million
atoms), parallel input/output (I/O) scheme is implemented.

\item {} 
\sphinxstylestrong{GENESIS} is optimized for K and Fugaku computer (developed by RIKEN and
Fujitsu company), but it is also available on Intel-based supercomputers
and PC-clusters.

\item {} 
\sphinxstylestrong{GENESIS} is written in modern Fortran language (90/95/2003)
using modules and dynamic memory allocation. No common blocks are used.

\item {} 
\sphinxstylestrong{GENESIS} is free software under the GNU Lesser General Public License (LGPL)
version 3 or later. We allow users to use/modify \sphinxstylestrong{GENESIS} and
redistribute the modified version under the same license.

\end{itemize}

This user manual mainly provides detailed description of
keywords used in the control file. Tutorials for standard MD
simulations, REMD simulations, and some analyses are
\sphinxhref{https://www.r-ccs.riken.jp/labs/cbrt/}{available online} (https://www.r-ccs.riken.jp/labs/cbrt/).
We recommend new users of \sphinxstylestrong{GENESIS} to start from
the next chapter to learn a basic idea, installation, and work flow of the program.

Comparing to other MD software, e.g. AMBER, CHARMM, or NAMD,
\sphinxstylestrong{GENESIS} is a very young MD simulation program. Before releasing the
program, the developers and contributors in \sphinxstylestrong{GENESIS} development team
worked hard to fix all bugs in the program, and performed a bunch of
test simulations. Still, there might be defects or bugs in
\sphinxstylestrong{GENESIS}. Since we cannot bear any responsibility for the
simulation results produced by \sphinxstylestrong{GENESIS}, we strongly recommend
the users to check the results carefully.

The \sphinxstylestrong{GENESIS} development team has a rich plan for future development
of methodology and molecular models. We would like to
make \sphinxstylestrong{GENESIS} one of the most powerful and feasible MD software packages,
contributing to computational chemistry and biophysics. Computational
studies in life science is still at a very early stage (like
‘GENESIS’) compared to established experimental researches. We hope
that \sphinxstylestrong{GENESIS} pushes forward the computational science and
contribute to bio-tech and medical applications in the future.


\chapter{Getting Started}
\label{\detokenize{01_Getting_Started:getting-started}}\label{\detokenize{01_Getting_Started:id1}}\label{\detokenize{01_Getting_Started::doc}}

\section{Installation of GENESIS}
\label{\detokenize{01_Getting_Started:installation-of-genesis}}

\subsection{Requirements}
\label{\detokenize{01_Getting_Started:requirements}}

\subsubsection{Compilers}
\label{\detokenize{01_Getting_Started:compilers}}
\sphinxstylestrong{GENESIS} works on various systems: laptop PCs, workstations, cluster machines, and supercomputers.
Since the source code of \sphinxstylestrong{GENESIS} is mainly written in
Fortran language, Fortran compiler is the first requirement for installation.
In addition, “preprocessor” is required, because the source code is
“processed” according to the user’s computer environment before the compilation.
One of the commonly used Fortran compilers is \sphinxcode{\sphinxupquote{gfortran}}, which
is freely available as part of the GNU Compiler Collection (GCC).
In this case, \sphinxcode{\sphinxupquote{cpp}} is selected as a preprocessor, which is also available freely.
Another recommended Fortran compiler is \sphinxcode{\sphinxupquote{ifort}} provided
by Intel Corporation which enables us to run the program much faster on Intel CPU.
In the Intel compiler package, \sphinxcode{\sphinxupquote{fpp}} is provided as a preprocessor.
Fujitsu compiler \sphinxcode{\sphinxupquote{frtpx}}, which also functions as a preprocessor,
is suitable for Fujitsu machines like FX100.


\subsubsection{MPI and OpenMP}
\label{\detokenize{01_Getting_Started:mpi-and-openmp}}
Both \sphinxstylestrong{ATDYN} and \sphinxstylestrong{SPDYN} work on multiple CPU cores using MPI
(Message Passing Interface) and OpenMP protocols (hybrid MPI+OpenMP).
MPI and OpenMP are commonly used for parallel computing.
In general, MPI is employed for communication between different machines,
nodes, or processors, where the memory is not shared among them (distributed-memory).
On the other hand, OpenMP is employed in a single processor, and thus,
memory is shared in the parallel computation.

OpenMP is natively supported in most modern Fortran compilers.
As for MPI, however, the users may have to install MPI libraries by
themselves, especially, in the case of laptop PCs and workstations.
One of the commonly used MPI software is OpenMPI (\sphinxurl{https://www.open-mpi.org/}).
When the users install the OpenMPI libraries in the computer, the users must
specify Fortran and C compilers (e.g., \sphinxcode{\sphinxupquote{gfortran}} and \sphinxcode{\sphinxupquote{gcc}}) to be used with MPI.
After installing the libraries, the users can use \sphinxcode{\sphinxupquote{mpif90}}, \sphinxcode{\sphinxupquote{mpicc}}, and \sphinxcode{\sphinxupquote{mpirun}},
which are necessary to compile and run the program that is parallelized with MPI.
OpenMPI is available freely, and the example installation scheme is shown in {\hyperref[\detokenize{98_Appendix:appendix}]{\sphinxcrossref{\DUrole{std,std-ref}{Appendix}}}}.
Intel and Fujitsu Corporations are also providing their own MPI libraries for parallel computation.


\subsubsection{Mathematical libraries}
\label{\detokenize{01_Getting_Started:mathematical-libraries}}
\sphinxstylestrong{GENESIS} utilizes mathematical libraries such as LAPACK and BLAS (\sphinxurl{http://www.netlib.org/lapack/}).
These libraries enable us to efficiently solve complicated mathematical
equations such as eigenvalue problems and singular value decomposition.
The users have to install these libraries by themselves,
if they are not installed in the computer (see {\hyperref[\detokenize{98_Appendix:appendix}]{\sphinxcrossref{\DUrole{std,std-ref}{Appendix}}}}).
In the case of the Intel and Fujitsu compilers, Intel MKL and Fujitsu SSL II are automatically selected, respectively.


\subsubsection{GPGPU}
\label{\detokenize{01_Getting_Started:gpgpu}}
\sphinxstylestrong{SPDYN} works not only with CPU but also with CPU+GPU.
Some of the source code in \sphinxstylestrong{SPDYN} are written in CUDA,
which enables us to effectively run the program on NVIDIA GPU cards.
If the users want to run \sphinxstylestrong{SPDYN} with GPGPU calculations,
the CUDA toolkit (\sphinxurl{https://developer.nvidia.com/cuda-toolkit}) must be also installed in the computer.
Note that OpenACC is not employed in \sphinxstylestrong{GENESIS} currently.


\bigskip\hrule\bigskip


The recommended compilers, preprocessors, and libraries for \sphinxstylestrong{GENESIS} are listed below.
Please make sure that at least one of them in each section is installed on your system (GPU is optional).
If the users do not use the Intel or Fujitsu compilers,
the combination of GCC compiler, GCC preprocessor, and OpenMPI is recommended.
\begin{itemize}
\item {} 
Operating systems (see {\hyperref[\detokenize{98_Appendix:appendix}]{\sphinxcrossref{\DUrole{std,std-ref}{Appendix}}}})
\begin{itemize}
\item {} 
Linux

\item {} 
macOS

\end{itemize}

\item {} 
Fortran and C compilers
\begin{itemize}
\item {} 
GCC compiler \sphinxcode{\sphinxupquote{gfortran}}, \sphinxcode{\sphinxupquote{gcc}} (version 4.4.7 or higher is required)

\item {} 
Intel compiler \sphinxcode{\sphinxupquote{ifort}}, \sphinxcode{\sphinxupquote{icc}}

\item {} 
Fujitsu compiler \sphinxcode{\sphinxupquote{frtpx}}, \sphinxcode{\sphinxupquote{fccpx}}

\end{itemize}

\item {} 
Preprocessors
\begin{itemize}
\item {} 
GCC preprocessor \sphinxcode{\sphinxupquote{cpp}}

\item {} 
Intel preprocessor \sphinxcode{\sphinxupquote{fpp}}

\item {} 
Fujitsu compiler \sphinxcode{\sphinxupquote{frtpx}}

\end{itemize}

\item {} 
MPI libraries for parallel computing
\begin{itemize}
\item {} 
OpenMPI \sphinxcode{\sphinxupquote{mpirun}}, \sphinxcode{\sphinxupquote{mpif90}}, \sphinxcode{\sphinxupquote{mpicc}}

\item {} 
Intel MPI

\item {} 
Fujitsu MPI

\end{itemize}

\item {} 
Numerical libraries for mathematical algorithms
\begin{itemize}
\item {} 
LAPACK/BLAS

\item {} 
Intel Math Kernel Library (MKL)

\item {} 
Fujitsu Scientific Subroutine Library (SSL II)

\end{itemize}

\item {} 
GPU \sphinxstylestrong{(Optional)}
\begin{itemize}
\item {} 
NVIDIA GPU cards which support Compute Capability (CC) 3.5 or higher

\item {} 
The following GPU cards and CUDA versions have been tested by the GENESIS developers
\begin{itemize}
\item {} 
NVIDIA K20, K40, P100, TITAN V, GTX 1080, GTX 1080Ti, RTX 2080, RTX 2080Ti

\item {} 
CUDA ver. 8.0, 9.0, 9.1, 9.2, 10.0

\end{itemize}

\end{itemize}

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
If you are using a supercomputer in universities or research institutes,
there is a high chance that the system already provides the above requirements so
that you don’t need to install yourself.
Please refer to the users’ guide of the supercomputer, or consult the system administrator.

In general, the latest version of CUDA does not support the latest version of GCC compiler.
If you cannot compile GENESIS with new CUDA (ver. 10) and new GCC compiler (ver. 8.0 or higher),
please first make an attempt to install CUDA with older GCC compilers (ver. 7.0 or older),
and then install GENESIS with those CUDA and GCC compilers.
\end{sphinxadmonition}

\clearpage


\subsection{General scheme for installation}
\label{\detokenize{01_Getting_Started:general-scheme-for-installation}}

\subsubsection{Step1. Download the source code}
\label{\detokenize{01_Getting_Started:step1-download-the-source-code}}
The source code of \sphinxstylestrong{GENESIS} is available in the GENESIS website (\sphinxurl{https://www.r-ccs.riken.jp/labs/cbrt/download/}).
The users have to first uncompress the download file in an appropriate directory.
Here, we assume that the users install \sphinxstylestrong{GENESIS} in “\$HOME/genesis”.
The “src” directory contains the source code, and “COPYING” is the software license.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} mkdir \PYG{n+nv}{\PYGZdl{}HOME}/genesis
\PYGZdl{} \PYG{n+nb}{cd} \PYG{n+nv}{\PYGZdl{}HOME}/genesis
\PYGZdl{} mv \PYGZti{}/Downloads/genesis\PYGZhy{}1.7.1.tar.bz2 ./
\PYGZdl{} tar xvfj genesis\PYGZhy{}1.7.1.tar.bz2
\PYGZdl{} \PYG{n+nb}{cd} genesis\PYGZhy{}1.7.1
\PYGZdl{} ls
AUTHORS      Makefile.am    aclocal.m4      depcomp       src
COPYING      Makefile.in    compile         fortdep.py
ChangeLog    NEWS           configure       install\PYGZhy{}sh
INSTALL      README         configure.ac    missing
\end{sphinxVerbatim}


\subsubsection{Step2. Configure}
\label{\detokenize{01_Getting_Started:step2-configure}}
In order to compile the source code, the users execute the “configure” script in the directory.
This script automatically detects appropriate compilers, preprocessors, and libraries
in the users’ computer, and create “\sphinxcode{\sphinxupquote{Makefile}}”.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./configure
\end{sphinxVerbatim}

If you encountered a failure in the configure command, please check the error message carefully.
You may have to add appropriate options in this command according to your computer environment (see {\hyperref[\detokenize{01_Getting_Started:advanced-installation}]{\sphinxcrossref{\DUrole{std,std-ref}{Advanced installation}}}}).
The followings are possible suggestions to solve frequent problems.
Other solutions might be found in the online page (\sphinxurl{https://www.r-ccs.riken.jp/labs/cbrt/installation/}).
\begin{itemize}
\item {} 
First of all, please check whether the Fortran and C compilers are installed in your computer.
If you are going to run GENESIS with multiple CPUs, you should additionally install
MPI libraries such as OpenMPI before compiling GENESIS (see {\hyperref[\detokenize{98_Appendix:appendix}]{\sphinxcrossref{\DUrole{std,std-ref}{Appendix}}}}).

\item {} 
If you see the error message “configure: error: Fortran compiler cannot create executables”,
it may imply that the path to the installed compilers or MPI libraries might not be correctly
set in “\textasciitilde{}/.bashrc” or “\textasciitilde{}/.bash\_profile” (see {\hyperref[\detokenize{98_Appendix:appendix}]{\sphinxcrossref{\DUrole{std,std-ref}{Appendix}}}}).
This configure script automatically detects “mpif90”, “mpifrtpx”, or “mpifrt” for Fortran compiler,
and “mpicc”, “mpifccpx”, or “mpifrt” for C compiler.
The error message may indicate that the detection was failed due to some reasons.
For example, if you installed OpenMPI in your computer, both “mpif90” and “mpicc” should be detected.
Please check the path to these executables by typing the “which” command (e.g., which mpif90) in the terminal window.
If you cannot see any paths, setting of the path in “\textasciitilde{}/.bashrc” or “\textasciitilde{}/.bash\_profile” might have a mistake (see {\hyperref[\detokenize{98_Appendix:appendix}]{\sphinxcrossref{\DUrole{std,std-ref}{Appendix}}}}).
You should also check typing mistakes of the path.

\item {} 
If the recommended software are not used in compilation, warning messages might be
displayed in the terminal when the configure command is executed.
Those messages are just a warning (not an error), and you may continue the compilation.
However, we strongly recommended you to verity the installation in such cases
(see {\hyperref[\detokenize{01_Getting_Started:verify-installation}]{\sphinxcrossref{\DUrole{std,std-ref}{Verify the installation}}}}).

\item {} 
In some supercomputer systems, “module load {[}module{]}” command is required to use
compliers, and need to be set before the configure. See the user guide of the system.

\item {} 
Try “autoreconf” or “./bootstrap” before the configure command,
if your computer environment is significantly different from what we assume
and/or if you modify “configure.ac” or “Makefile.am” by yourself.

\end{itemize}


\subsubsection{Step3. Make install}
\label{\detokenize{01_Getting_Started:step3-make-install}}
After the “configure” command is successful, type the following command to compile and install \sphinxstylestrong{GENESIS}.
All programs in \sphinxstylestrong{GENESIS} are compiled and installed into the “./bin” directory by default.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} make install
\end{sphinxVerbatim}

If you encountered a failure, please check the error message carefully.
In many cases, errors are caused by invalid path of compilers and libraries.
The followings are possible suggestions to solve frequent problems.
Other solutions might be found in the online page
(\sphinxurl{https://www.r-ccs.riken.jp/labs/cbrt/installation/}).
\begin{itemize}
\item {} 
If the error message is like “/usr/bin/ld: cannot find -lblas” or “/usr/bin/ld: cannot find -llapack”,
make sure that the BLAS or LAPACK libraries are installed in the computer (see also {\hyperref[\detokenize{98_Appendix:appendix}]{\sphinxcrossref{\DUrole{std,std-ref}{Appendix}}}}).
The users may also have to set the path to the libraries in the “configure” command
with the “\sphinxcode{\sphinxupquote{LAPACK\_LIBS}}” or “\sphinxcode{\sphinxupquote{LAPACK\_PATH}}” option (see {\hyperref[\detokenize{01_Getting_Started:advanced-installation}]{\sphinxcrossref{\DUrole{std,std-ref}{Advanced installation}}}}).

\item {} 
If you have installed additional software or libraries to solve a make error,
please execute “make clean”, and try Step2 and “make install” again.

\end{itemize}


\subsubsection{Step4. Confirmation}
\label{\detokenize{01_Getting_Started:step4-confirmation}}
After the installation is successfully finished, the following binary files are found in the “bin” directory.
There are 42 programs in total. Brief description of each program is shown in {\hyperref[\detokenize{02_Available_Programs:available-programs}]{\sphinxcrossref{\DUrole{std,std-ref}{Available Programs}}}}.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ls ./bin
atdyn               fret\PYGZus{}analysis        qmmm\PYGZus{}generator
avecrd\PYGZus{}analysis     hb\PYGZus{}analysis          qval\PYGZus{}analysis
comcrd\PYGZus{}analysis     hbond\PYGZus{}analysis       rdf\PYGZus{}analysis
contact\PYGZus{}analysis    kmeans\PYGZus{}clustering    remd\PYGZus{}convert
crd\PYGZus{}convert         lipidthick\PYGZus{}analysis  rg\PYGZus{}analysis
density\PYGZus{}analysis    mbar\PYGZus{}analysis        rmsd\PYGZus{}analysis
diffusion\PYGZus{}analysis  meanforce\PYGZus{}analysis   rpath\PYGZus{}generator
distmat\PYGZus{}analysis    msd\PYGZus{}analysis         rst\PYGZus{}convert
drms\PYGZus{}analysis       pathcv\PYGZus{}analysis      rst\PYGZus{}upgrade
dssp\PYGZus{}interface      pcavec\PYGZus{}drawer        sasa\PYGZus{}analysis
eigmat\PYGZus{}analysis     pcrd\PYGZus{}convert         spdyn
emmap\PYGZus{}generator     pmf\PYGZus{}analysis         tilt\PYGZus{}analysis
prjcrd\PYGZus{}analysis     trj\PYGZus{}analysis         flccrd\PYGZus{}analysis
prst\PYGZus{}setup          wham\PYGZus{}analysis
\end{sphinxVerbatim}

\clearpage


\subsection{Advanced installation}
\label{\detokenize{01_Getting_Started:advanced-installation}}\label{\detokenize{01_Getting_Started:id2}}
In the above scheme, \sphinxstylestrong{GENESIS} is installed with default options,
and all installed programs run on CPU with double precision calculation.
The users can specify additional options in the configure command
according to the users’ computer environment or desired conditions.
The full lists of the available options are obtained by “\sphinxcode{\sphinxupquote{./configure -{-}help}}”.
The representative options are as follows.

\sphinxcode{\sphinxupquote{-{-}enable-single}}
\begin{quote}

Turn on single precision calculation. In this case, only \sphinxstylestrong{SPDYN} is installed.
\end{quote}

\sphinxcode{\sphinxupquote{-{-}enable-gpu}}
\begin{quote}

Turn on GPGPU calculation. In this case, only \sphinxstylestrong{SPDYN} is installed.
\end{quote}

\sphinxcode{\sphinxupquote{-{-}with-cuda=PATH}}
\begin{quote}

Define path to the CUDA libraries manually.
\end{quote}

\sphinxcode{\sphinxupquote{-{-}disable-mpi}}
\begin{quote}

Turn off MPI parallelization. In this case, \sphinxstylestrong{SPDYN} is not installed.
\end{quote}

\sphinxcode{\sphinxupquote{-{-}disable-openmp}}
\begin{quote}

Turn off OpenMP parallelization.
\end{quote}

\sphinxcode{\sphinxupquote{-{-}disable-parallel\_IO}}
\begin{quote}

Do not install the parallel I/O tool (\sphinxstylestrong{prst\_setup})
\end{quote}

\sphinxcode{\sphinxupquote{-{-}enable-debug}}
\begin{quote}

Turn on program debugging (see below)
\end{quote}

\sphinxcode{\sphinxupquote{-{-}prefix=PREFIX}}
\begin{quote}

Install the programs in the directory designated by \sphinxcode{\sphinxupquote{PREFIX}}
\end{quote}

\sphinxcode{\sphinxupquote{-{-}with-msmpi}}
\begin{quote}

Turn on use of MSMPI. Compilation and execution must be done on Windows10.
\end{quote}


\bigskip\hrule\bigskip



\subsubsection{Configuration with non-default compilers}
\label{\detokenize{01_Getting_Started:configuration-with-non-default-compilers}}
Although the compilers are set to “mpif90” and “mpicc” by default,
the users may specify different compilers by configure commands.
Fortran compiler is specified with \sphinxcode{\sphinxupquote{FC}} and \sphinxcode{\sphinxupquote{F77}}, and C compiler with \sphinxcode{\sphinxupquote{CC}}.
For example, in the case of “mpiifort” and “mpiicc”, the following options are added:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./configure \PYG{n+nv}{CC}\PYG{o}{=}mpiicc \PYG{n+nv}{FC}\PYG{o}{=}mpiifort \PYG{n+nv}{F77}\PYG{o}{=}mpiifort
\end{sphinxVerbatim}


\subsubsection{Configuration with an explicit path to LAPACK/BLAS libraries}
\label{\detokenize{01_Getting_Started:configuration-with-an-explicit-path-to-lapack-blas-libraries}}
The following is an example command to set the path to LAPACK and BLAS libraries
that are installed in \$HOME/Software/lapack-3.8.0/ (see also {\hyperref[\detokenize{98_Appendix:appendix}]{\sphinxcrossref{\DUrole{std,std-ref}{Appendix}}}}).
Please be careful about the filename of the installed libraries.
If the BLAS libraries are installed as “librefblas.a”, the option “-lrefblas” must be used.
If “librefblas.a” is renamed to “libblas.a”, the following command can be used.
Linking with the reverse order of “-llapack” and “-lblas” might
also cause a failure of installation of \sphinxstylestrong{GENESIS}.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./configure \PYG{n+nv}{LAPACK\PYGZus{}LIBS}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{\PYGZhy{}L}\PYG{n+nv}{\PYGZdl{}HOME}\PYG{l+s+s2}{/Software/lapack\PYGZhy{}3.8.0 \PYGZhy{}llapack \PYGZhy{}lblas}\PYG{l+s+s2}{\PYGZdq{}}
\end{sphinxVerbatim}

or use the “\sphinxcode{\sphinxupquote{LAPACK\_PATH}}” option:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./configure \PYG{n+nv}{LAPACK\PYGZus{}PATH}\PYG{o}{=}\PYG{n+nv}{\PYGZdl{}HOME}/Software/lapack\PYGZhy{}3.8.0
\end{sphinxVerbatim}


\subsubsection{Configuration for single-precision calculation on CPU}
\label{\detokenize{01_Getting_Started:configuration-for-single-precision-calculation-on-cpu}}
The following command is used to turn on single-precision calculation in \sphinxstylestrong{SPDYN}.
In this case, force calculations are carried out with single precision,
while integration of the equations of motion as well as
accumulation of the force and energy are still done with double-precision.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./configure \PYGZhy{}\PYGZhy{}enable\PYGZhy{}single
\end{sphinxVerbatim}

Only \sphinxstylestrong{SPDYN} that works on CPU will be installed with this option.
If the user additionally needs analysis tools as well as \sphinxstylestrong{ATDYN},
one must prepare another GENESIS directory, and install without the “\sphinxcode{\sphinxupquote{-{-}enable-single}}” option.


\subsubsection{Configuration for GPGPU calculation}
\label{\detokenize{01_Getting_Started:configuration-for-gpgpu-calculation}}
In the following command, the users install \sphinxstylestrong{SPDYN} that works on CPU+GPU with single-precision calculation.
If “\sphinxcode{\sphinxupquote{-{-}enable-single}}” is omitted in the command, \sphinxstylestrong{SPDYN} works on CPU+GPU with double-precision calculation.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./configure \PYGZhy{}\PYGZhy{}enable\PYGZhy{}single \PYGZhy{}\PYGZhy{}enable\PYGZhy{}gpu
\end{sphinxVerbatim}

Here, if the users encountered an error message like “nvcc: command not found”,
make sure that the CUDA Toolkit is installed in the computer.
In typical Linux workstations or cluster machines,
CUDA is installed in “/usr/local/cuda-x.x/” or “/usr/lib/x86\_64-linux-gnu/”,
and “nvcc” should be in a “bin” directory of the install directory.
The path to “nvcc” and CUDA libraries should be set in a startup file
such as “\textasciitilde{}/.bashrc”.  For example, add the following information to “\textasciitilde{}/.bashrc”
in the case of CUDA 9.0,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n+nv}{CUDAROOT}\PYG{o}{=}/usr/local/cuda\PYGZhy{}9.0
\PYG{n+nb}{export} \PYG{n+nv}{PATH}\PYG{o}{=}\PYG{n+nv}{\PYGZdl{}CUDAROOT}/bin:\PYG{n+nv}{\PYGZdl{}PATH}
\PYG{n+nb}{export} \PYG{n+nv}{LD\PYGZus{}LIBRARY\PYGZus{}PATH}\PYG{o}{=}\PYG{n+nv}{\PYGZdl{}CUDAROOT}/lib64:/lib:\PYG{n+nv}{\PYGZdl{}LD\PYGZus{}LIBRARY\PYGZus{}PATH}
\end{sphinxVerbatim}

then reload “\textasciitilde{}/.bashrc” and try the configure command again.
If there still remain some troubles, explicitly specify a path to CUDA
libraries in the configure command by:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./configure \PYGZhy{}\PYGZhy{}enable\PYGZhy{}single \PYGZhy{}\PYGZhy{}enable\PYGZhy{}gpu \PYGZhy{}\PYGZhy{}with\PYGZhy{}cuda\PYG{o}{=}/usr/local/cuda\PYGZhy{}9.0
\end{sphinxVerbatim}


\subsubsection{Configuration for supercomputer systems}
\label{\detokenize{01_Getting_Started:configuration-for-supercomputer-systems}}
The configuration for supercomputer systems may require non-standard setups.
In the online usage page, we describe recommended configure options for some supercomputers
(\sphinxurl{https://www.r-ccs.riken.jp/labs/cbrt/usage/}).

For example, the following commands are used to compile \sphinxstylestrong{GENESIS} on HOKUSAI GreatWave
(FX100) in RIKEN.
Note that the parallel I/O tool (\sphinxstylestrong{prst\_setup}) is not compiled in this configuration,
because Fujitsu compiler has a trouble in compiling \sphinxstylestrong{prst\_setup} (see aldo {\hyperref[\detokenize{02_Available_Programs:available-programs}]{\sphinxcrossref{\DUrole{std,std-ref}{Available Programs}}}}).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} module load sparc
\PYGZdl{} ./configure \PYGZhy{}\PYGZhy{}host\PYG{o}{=}k
\end{sphinxVerbatim}


\subsubsection{Configuration for single CPU calculations}
\label{\detokenize{01_Getting_Started:configuration-for-single-cpu-calculations}}
By specifying the “\sphinxcode{\sphinxupquote{-{-}disable-mpi}}” option, the users can install \sphinxstylestrong{GENESIS} that can work on one CPU.
The configure script automatically looks for “gfortran”, “ifort”, “frt”, or “frtpx” for Fortran compiler,
and “gcc”, “icc”, “fcc”, or “fccpx” for C compiler. Therefore, in this case MPI libraries are not required
for the installation and execution of \sphinxstylestrong{GENESIS}. \sphinxstylestrong{ATDYN} and analysis tools are installed.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./configure \PYGZhy{}\PYGZhy{}disable\PYGZhy{}mpi
\end{sphinxVerbatim}


\subsubsection{Configuration for program debugging}
\label{\detokenize{01_Getting_Started:configuration-for-program-debugging}}
If the users encountered memory leak errors during the simulation using \sphinxstylestrong{GENESIS},
the origin of the error might be tracked by using a program compiled with a debug option.
Note that the debug option makes the calculation much slow.
In this case, the runtime check is activated only for CPU codes,
even if the “\textendash{}enable-gpu” option is added to the command.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./configure \PYGZhy{}\PYGZhy{}enable\PYGZhy{}debug\PYG{o}{=}\PYG{l+m}{3}
\end{sphinxVerbatim}

Note that \sphinxcode{\sphinxupquote{-{-}enable-debug}} corresponds to \sphinxcode{\sphinxupquote{-{-}enable-debug=1}}.
\begin{itemize}
\item {} 
0 = no debugging (default)

\item {} 
1 = debugging without intensive optimization

\item {} 
2 = LEVEL1 + debug information (\sphinxcode{\sphinxupquote{-g}} and \sphinxcode{\sphinxupquote{-DDEBUG}})

\item {} 
3 = LEVEL2 + memory check (if possible)

\item {} 
4 = LEVEL3 + full check (intel compiler only)

\end{itemize}


\bigskip\hrule\bigskip



\subsubsection{Installation using multiple CPU cores (parallel compile)}
\label{\detokenize{01_Getting_Started:installation-using-multiple-cpu-cores-parallel-compile}}
In Step3, \sphinxcode{\sphinxupquote{-j}} option is available, which enables quick compilation
of the program using multiple CPU cores. The following command uses 4 CPU cores.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} make \PYGZhy{}j4 install
\end{sphinxVerbatim}

If you encountered an error message like “Fatal Error: Can’t delete temporary module
file ‘…’: No such file or directory”, please try “make install”
without the “\sphinxcode{\sphinxupquote{-j}}” option.

\clearpage


\subsection{Verify the installation}
\label{\detokenize{01_Getting_Started:verify-the-installation}}\label{\detokenize{01_Getting_Started:verify-installation}}
The users can verify the installation of \sphinxstylestrong{GENESIS} by using test sets
which are available in the \sphinxstylestrong{GENESIS} website
(\sphinxurl{https://www.r-ccs.riken.jp/labs/cbrt/download/}).
Please uncompress the downloaded file in an appropriate directory,
and move to the “regression\_test” directory.
Note that the file name of the tar.bz2 file contains the date (year, month, and day),
so please change the following execution commands accordingly.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{cd} \PYG{n+nv}{\PYGZdl{}HOME}/genesis
\PYGZdl{} mv \PYGZti{}/Downloads/tests\PYGZhy{}1.7.1\PYGZus{}YYMMDD.tar.bz2 ./
\PYGZdl{} tar xvfj tests\PYGZhy{}1.7.1\PYGZus{}YYMMDD.tar.bz2
\PYGZdl{} \PYG{n+nb}{cd} tests\PYGZhy{}1.7.1\PYGZus{}YYMMDD/regression\PYGZus{}test
\PYGZdl{} ls
build       test\PYGZus{}analysis    test\PYGZus{}gamd\PYGZus{}spdyn    test\PYGZus{}rpath\PYGZus{}atdyn
charmm.py   test\PYGZus{}atdyn       test\PYGZus{}nonstrict.py  test\PYGZus{}rpath\PYGZus{}spdyn
cleanup.sh  test\PYGZus{}common      test\PYGZus{}parallel\PYGZus{}IO   test\PYGZus{}spana
fep.py      test\PYGZus{}fep         test\PYGZus{}remd.py       test\PYGZus{}spdyn
genesis.py  test\PYGZus{}fep.py      test\PYGZus{}remd\PYGZus{}common   test\PYGZus{}vib
param       test\PYGZus{}gamd.py     test\PYGZus{}remd\PYGZus{}spdyn    test\PYGZus{}vib.py
test.py     test\PYGZus{}gamd\PYGZus{}atdyn  test\PYGZus{}rpath.py
\end{sphinxVerbatim}

In the sub-directories in “regression\_test”, the users can find a lot of input files (“\sphinxcode{\sphinxupquote{inp}}”),
in which various combinations of simulation parameters are specified.
In addition, each sub-directory contains output file (“\sphinxcode{\sphinxupquote{ref}}”) obtained by the developers.
The users run “test.py”, “test\_remd.py”, “test\_rpath.py”, and so on,
which enable automatic comparison between the users’ and developers’
results for each MD algorithm.


\subsubsection{Run the basic tests}
\label{\detokenize{01_Getting_Started:run-the-basic-tests}}
The following is an example command to verify the two simulators
\sphinxstylestrong{atdyn} and \sphinxstylestrong{spdyn} for basic MD and energy minimization.
Here, the programs are executed using 1 CPU core with the “mpirun” command.
The users can increase the number of MPI processors according to
the users’ computer environment, but only 1, 2, 4, or 8 are allowed in these tests.
Other MPI launchers such as “mpiexec” are also available in the command.
There are about 50 test sets, and each test should finish in a few seconds.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{export} \PYG{n+nv}{OMP\PYGZus{}NUM\PYGZus{}THREADS}\PYG{o}{=}\PYG{l+m}{1}
\PYGZdl{} ./test.py \PYG{l+s+s2}{\PYGZdq{}mpirun \PYGZhy{}np 1 \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/atdyn\PYGZdq{}}
\PYGZdl{} ./test.py \PYG{l+s+s2}{\PYGZdq{}mpirun \PYGZhy{}np 1 \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/spdyn\PYGZdq{}}
\end{sphinxVerbatim}

If any tests cannot run, please check the following points:
\begin{itemize}
\item {} 
Number of OpenMP threads should be specified before running the tests (one is recommended).

\item {} 
Original executable file name (e.g., \sphinxstylestrong{spdyn} and \sphinxstylestrong{atdyn}) must not be changed.

\item {} 
Python ver. 3 does not work. Please use ver. 2.x, and run “\sphinxcode{\sphinxupquote{python2.x ./test.py ...}}”

\item {} 
Regression tests via a queuing system or batch script may not work.

\end{itemize}

The “test.py” script compares energy in log file between the developer’s and user’s ones.
If the energy differences are less than the tolerance (default = 1.00e-08),
“Passed” is displayed for each test. Among the physical quantities in the log file,
virial is the most sensitive to numerical factors, and thus, the tolerance
for virial is set to a larger value (1.00e-06).
After all tests are finished, the total number of succeeded,
failed, and aborted runs will be displayed at the end.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Passed  \PYG{l+m}{46} / \PYG{l+m}{46}
Failed  \PYG{l+m}{0} / \PYG{l+m}{46}
Aborted \PYG{l+m}{0} / \PYG{l+m}{46}
\end{sphinxVerbatim}

If all tests were passed, it means that your \sphinxstylestrong{GENESIS} can generate
identical results to the developer’s \sphinxstylestrong{GENESIS}.
Note that the developer’s \sphinxstylestrong{GENESIS} was compiled with Intel compilers,
Intel MKL, OpenMPI library, and the double precision option on Intel CPUs.
If your computer system is significantly different from the developer’s one,
unexpected numerical errors may happen, which can cause failures in some tests.
If there were any aborted tests, the users had better check their log or error files carefully,
which exist in the tested sub-directory, and figure out why the error happened.
The followings are suggestions to solve typical problems:
\begin{itemize}
\item {} 
If some tests were aborted due to “memory allocation error”,
the reason might come from limitation of the memory size.
Namely, those tested systems were too large for your computer.
The problem should not be so serious.

\item {} 
Available number of MPI slots in your computer might be actually
smaller than the given number of MPI processors.
Try to use less number of MPI processors.

\item {} 
Try to specify the “absolute path” to the program instead of using “relative path”.

\item {} 
Make sure that the MPI environment is properly set.

\item {} 
Detailed solutions in specific supercomputer systems might be found in the GENESIS website
(\sphinxurl{https://www.r-ccs.riken.jp/labs/cbrt/usage/}).

\end{itemize}


\subsubsection{Run the additional tests}
\label{\detokenize{01_Getting_Started:run-the-additional-tests}}
By using a similar way, the users can check other functions in \sphinxstylestrong{atdyn} and \sphinxstylestrong{spdyn},
such as GaMD, REMD, path sampling, vibrational analysis, parallel I/O, and GPGPU calculation.
Available number of MPI processors depends on each test (test\_gamd: 1, 2, 4, 8;
test\_remd: 4, 8, 16, 32; test\_rpath: 8; test\_vib: 8; parallel\_io: 8; gpu: 1, 2, 4, 8).
As for the GPGPU tests, the users must use \sphinxstylestrong{spdyn} that was installed with the “\textendash{}enable-gpu” option.
The parallel\_io tests require both \sphinxstylestrong{spdyn} and \sphinxstylestrong{prst\_setup}.
Note that \sphinxstylestrong{prst\_setup} is not installed in some cases according to the configure options or compilers (see {\hyperref[\detokenize{01_Getting_Started:advanced-installation}]{\sphinxcrossref{\DUrole{std,std-ref}{Advanced installation}}}}).
In order to run the analysis tool tests, the users first move to “test\_analysis”,
and then execute “./test\_analysis.py”. Note that MPI is not used in the analysis tool tests.
In a similar way, the users can test the SPANA (spatial decomposition analysis) tool sets.
SPANA tool sets are tested with 1, 2, 4, and 8 MPI processes.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{export} \PYG{n+nv}{OMP\PYGZus{}NUM\PYGZus{}THREADS}\PYG{o}{=}\PYG{l+m}{1}
\PYGZdl{} ./test\PYGZus{}gamd.py \PYG{l+s+s2}{\PYGZdq{}mpirun \PYGZhy{}np 1 \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/atdyn\PYGZdq{}}
\PYGZdl{} ./test\PYGZus{}gamd.py \PYG{l+s+s2}{\PYGZdq{}mpirun \PYGZhy{}np 1 \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/spdyn\PYGZdq{}}
\PYGZdl{} ./test\PYGZus{}remd.py \PYG{l+s+s2}{\PYGZdq{}mpirun \PYGZhy{}np 4 \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/atdyn\PYGZdq{}}
\PYGZdl{} ./test\PYGZus{}remd.py \PYG{l+s+s2}{\PYGZdq{}mpirun \PYGZhy{}np 4 \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/spdyn\PYGZdq{}}
\PYGZdl{} ./test\PYGZus{}fep.py \PYG{l+s+s2}{\PYGZdq{}mpirun \PYGZhy{}np 8 \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/spdyn\PYGZdq{}}
\PYGZdl{} ./test\PYGZus{}rpath.py \PYG{l+s+s2}{\PYGZdq{}mpirun \PYGZhy{}np 8 \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/atdyn\PYGZdq{}}
\PYGZdl{} ./test\PYGZus{}rpath.py \PYG{l+s+s2}{\PYGZdq{}mpirun \PYGZhy{}np 8 \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/spdyn\PYGZdq{}}
\PYGZdl{} ./test\PYGZus{}vib.py   \PYG{l+s+s2}{\PYGZdq{}mpirun \PYGZhy{}np 8 \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/atdyn\PYGZdq{}}
\PYGZdl{} ./test.py \PYG{l+s+s2}{\PYGZdq{}mpirun \PYGZhy{}np 8 \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/spdyn\PYGZdq{}} parallel\PYGZus{}io
\PYGZdl{} ./test.py \PYG{l+s+s2}{\PYGZdq{}mpirun \PYGZhy{}np 8 \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/spdyn\PYGZdq{}} gpu

\PYGZdl{} \PYG{n+nb}{cd} test\PYGZus{}analysis
\PYGZdl{} ./cleanup.sh
\PYGZdl{} \PYG{n+nb}{export} \PYG{n+nv}{OMP\PYGZus{}NUM\PYGZus{}THREADS}\PYG{o}{=}\PYG{l+m}{1}
\PYGZdl{} ./test\PYGZus{}analysis.py \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/

\PYGZdl{} \PYG{n+nb}{cd} test\PYGZus{}spana
\PYGZdl{} ./cleanup.sh
\PYGZdl{} \PYG{n+nb}{export} \PYG{n+nv}{OMP\PYGZus{}NUM\PYGZus{}THREADS}\PYG{o}{=}\PYG{l+m}{1}
\PYGZdl{} ./test\PYGZus{}spana.py \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
Some tests might be using “abnormal” parameters or conditions in the input files
for the sake of simple tests. Do not use such parameters in your research.
“Normal” parameters are mainly introduced in this user manual or online tutorials.
\end{sphinxadmonition}

\clearpage


\subsection{Clean install and re-compilation}
\label{\detokenize{01_Getting_Started:clean-install-and-re-compilation}}
The following commands are used to fully recompile \sphinxstylestrong{GENESIS}.
Note that the direct “make clean” command may not work
in the case where \sphinxcode{\sphinxupquote{Makefiles}} were created in another machine.
In this case, the users must run the “./configure” command before “make clean”.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} make clean
\PYGZdl{} make distclean
\PYGZdl{} ./configure \PYG{o}{[}option\PYG{o}{]}
\PYGZdl{} make install
\end{sphinxVerbatim}


\subsection{Uninstall}
\label{\detokenize{01_Getting_Started:uninstall}}
The user can uninstall \sphinxstylestrong{GENESIS} by just removing the program directory.
If the user changed the install directory by specifying “\sphinxcode{\sphinxupquote{-{-}prefix=PREFIX}}” in the configure command,
please remove the programs (\sphinxstylestrong{atdyn}, \sphinxstylestrong{spdyn}, and so on) in the “\sphinxcode{\sphinxupquote{PREFIX}}” directory.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} rm \PYGZhy{}rf \PYG{n+nv}{\PYGZdl{}HOME}/genesis/genesis\PYGZhy{}1.7.1
\end{sphinxVerbatim}

\clearpage


\section{Basic usage of GENESIS}
\label{\detokenize{01_Getting_Started:basic-usage-of-genesis}}

\subsection{Running GENESIS on a command line}
\label{\detokenize{01_Getting_Started:running-genesis-on-a-command-line}}
The \sphinxstylestrong{GENESIS} programs are executed on a command line.
The first argument is basically interpreted as an input file of the program.
The input file, which we call \sphinxstyleemphasis{control file} hereafter, contains parameters for simulations.
The following examples show typical usage of the \sphinxstylestrong{GENESIS} programs.
In the case of serial execution,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{o}{[}program\PYGZus{}name\PYG{o}{]} \PYG{o}{[}control\PYGZus{}file\PYG{o}{]}
\end{sphinxVerbatim}

In the case of parallel execution with “mpirun”,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} mpirun \PYGZhy{}np n \PYG{o}{[}program\PYGZus{}name\PYG{o}{]} \PYG{o}{[}control\PYGZus{}file\PYG{o}{]}
\end{sphinxVerbatim}

For example, \sphinxstylestrong{SPDYN} is executed in the following way using 8 MPI processors:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} mpirun \PYGZhy{}np \PYG{l+m}{8} \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/spdyn INP
\end{sphinxVerbatim}

The users should specify an OpenMP thread number explicitly before running the program.
Appropriate number of CPU cores must be used according to the user’s computer environment (see also {\hyperref[\detokenize{02_Available_Programs:available-programs}]{\sphinxcrossref{\DUrole{std,std-ref}{Available Programs}}}}).
For example, if the users want to use 32 CPU cores in the calculation,
the following command might be executed.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{export} \PYG{n+nv}{OMP\PYGZus{}NUM\PYGZus{}THREADS}\PYG{o}{=}\PYG{l+m}{4}
\PYGZdl{} mpirun \PYGZhy{}np \PYG{l+m}{8} \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/spdyn INP
\end{sphinxVerbatim}

As for the analysis tools, the usage is almost same, but mpirun is not used.
Note that some analysis tools (e.g., mbar\_analysis, wham\_analysis, msd\_analysis, and drms\_analysis) are
parallelized with OpenMP.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} RMSD analysis tool}
\PYGZdl{} \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/rmsd\PYGZus{}analysis INP

\PYG{c+c1}{\PYGZsh{} MBAR analysis}
\PYGZdl{} \PYG{n+nb}{export} \PYG{n+nv}{OMP\PYGZus{}NUM\PYGZus{}THREADS}\PYG{o}{=}\PYG{l+m}{4}
\PYGZdl{} \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/mbar\PYGZus{}analysis INP
\end{sphinxVerbatim}


\subsection{Automatic generation of a template control file}
\label{\detokenize{01_Getting_Started:automatic-generation-of-a-template-control-file}}
Basic usage of each program is shown by executing the program with the \sphinxcode{\sphinxupquote{-h}} option. In addition, sample control file of each program can be obtained with the \sphinxcode{\sphinxupquote{-h ctrl}} option:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Show the usage of the program}
\PYGZdl{} \PYG{o}{[}program\PYGZus{}name\PYG{o}{]} \PYGZhy{}h

\PYG{c+c1}{\PYGZsh{} Display a template control file}
\PYGZdl{} \PYG{o}{[}program\PYGZus{}name\PYG{o}{]} \PYGZhy{}h ctrl \PYG{o}{[}module\PYGZus{}name\PYG{o}{]}
\end{sphinxVerbatim}

For example, in the case of \sphinxstylestrong{SPDYN}, the following messages are displayed:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} spdyn \PYGZhy{}h

\PYG{c+c1}{\PYGZsh{} normal usage}
  \PYGZpc{} mpirun \PYGZhy{}np XX ./spdyn INP

\PYG{c+c1}{\PYGZsh{} check control parameters of md}
  \PYGZpc{} ./spdyn \PYGZhy{}h ctrl md

\PYG{c+c1}{\PYGZsh{} check control parameters of min}
  \PYGZpc{} ./spdyn \PYGZhy{}h ctrl min

\PYG{c+c1}{\PYGZsh{} check control parameters of remd}
  \PYGZpc{} ./spdyn \PYGZhy{}h ctrl remd

\PYG{c+c1}{\PYGZsh{} check control parameters of rpath}
  \PYGZpc{} ./spdyn \PYGZhy{}h ctrl rpath

\PYG{c+c1}{\PYGZsh{} check all control parameters of md}
  \PYGZpc{} ./spdyn \PYGZhy{}h ctrl\PYGZus{}all md

\PYG{o}{(}skipped...\PYG{o}{)}
\end{sphinxVerbatim}

This message tells the users that \sphinxstylestrong{SPDYN} can be executed with mpirun.
A template control file for molecular dynamics simulation (md) can be
generated by executing \sphinxstylestrong{SPDYN} with the \sphinxcode{\sphinxupquote{-h ctrl md}} option.
The same way is applicable for energy minimization (min),
replica exchange simulation (remd), and replica path sampling simulation (rpath).
The template control file for energy minimization is shown below.
If the users want to show all available options, please specify \sphinxcode{\sphinxupquote{ctrl\_all}} instead of \sphinxcode{\sphinxupquote{ctrl}}.
The users can edit this template control file to perform the simulation that the users want to do.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/spdyn \PYGZhy{}h ctrl min \PYGZgt{} INPMIN

\PYGZdl{} less INPMIN

\PYG{o}{[}INPUT\PYG{o}{]}
\PYG{n+nv}{topfile} \PYG{o}{=} sample.top      \PYG{c+c1}{\PYGZsh{} topology file}
\PYG{n+nv}{parfile} \PYG{o}{=} sample.par      \PYG{c+c1}{\PYGZsh{} parameter file}
\PYG{n+nv}{psffile} \PYG{o}{=} sample.psf      \PYG{c+c1}{\PYGZsh{} protein structure file}
\PYG{n+nv}{pdbfile} \PYG{o}{=} sample.pdb      \PYG{c+c1}{\PYGZsh{} PDB file}

\PYG{o}{[}ENERGY\PYG{o}{]}
\PYG{n+nv}{forcefield}    \PYG{o}{=} CHARMM    \PYG{c+c1}{\PYGZsh{} [CHARMM,AMBER,GROAMBER,GROMARTINI]}
\PYG{n+nv}{electrostatic} \PYG{o}{=} PME       \PYG{c+c1}{\PYGZsh{} [CUTOFF,PME]}
\PYG{n+nv}{switchdist}    \PYG{o}{=} \PYG{l+m}{10}.0      \PYG{c+c1}{\PYGZsh{} switch distance}
\PYG{n+nv}{cutoffdist}    \PYG{o}{=} \PYG{l+m}{12}.0      \PYG{c+c1}{\PYGZsh{} cutoff distance}
\PYG{n+nv}{pairlistdist}  \PYG{o}{=} \PYG{l+m}{13}.5      \PYG{c+c1}{\PYGZsh{} pair\PYGZhy{}list distance}

\PYG{o}{[}MINIMIZE\PYG{o}{]}
\PYG{n+nv}{method}        \PYG{o}{=} SD        \PYG{c+c1}{\PYGZsh{} [SD]}
\PYG{n+nv}{nsteps}        \PYG{o}{=} \PYG{l+m}{100}       \PYG{c+c1}{\PYGZsh{} number of minimization steps}

\PYG{o}{[}BOUNDARY\PYG{o}{]}
\PYG{n+nb}{type}          \PYG{o}{=} PBC       \PYG{c+c1}{\PYGZsh{} [PBC, NOBC]}
\end{sphinxVerbatim}

\clearpage


\section{Control file}
\label{\detokenize{01_Getting_Started:control-file}}
In the control file, detailed simulation conditions are specified.
The control file consists of several sections (e.g., \sphinxstylestrong{{[}INPUT{]}},
\sphinxstylestrong{{[}OUTPUT{]}}, \sphinxstylestrong{{[}ENERGY{]}}, \sphinxstylestrong{{[}ENSEMBLE{]}}, and so on),
each of which contains closely-related keywords.
For example, in the \sphinxstylestrong{{[}ENERGY{]}} section, parameters are specified for the potential energy calculation
such as a force field type and cut-off distance.
In the \sphinxstylestrong{{[}ENSEMBLE{]}} section, there are parameters to select the algorithm to control the temperature
and pressure in addition to the target temperature and pressure of the system.
Here, we show example control files for the energy minimization and normal molecular dynamics simulations.


\subsection{Example control file for the energy minimization}
\label{\detokenize{01_Getting_Started:example-control-file-for-the-energy-minimization}}
The control file for the energy minimization must include a \sphinxstylestrong{{[}MINIMIZE{]}} section (see {\hyperref[\detokenize{07_Minimize:minimize}]{\sphinxcrossref{\DUrole{std,std-ref}{Minimize section}}}}).
By using the following control file, the users carry out 2,000-step
energy minimization with the steepest descent algorithm (SD).
The CHARMM36m force field is used, and the particle mesh Ewald (PME) method
is employed for the calculation of long-range interaction.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}INPUT\PYG{o}{]}
\PYG{n+nv}{topfile} \PYG{o}{=} top\PYGZus{}all36\PYGZus{}prot.rtf     \PYG{c+c1}{\PYGZsh{} topology file}
\PYG{n+nv}{parfile} \PYG{o}{=} par\PYGZus{}all36m\PYGZus{}prot.prm    \PYG{c+c1}{\PYGZsh{} parameter file}
\PYG{n+nv}{strfile} \PYG{o}{=} toppar\PYGZus{}water\PYGZus{}ions.str  \PYG{c+c1}{\PYGZsh{} stream file}
\PYG{n+nv}{psffile} \PYG{o}{=} build.psf              \PYG{c+c1}{\PYGZsh{} protein structure file}
\PYG{n+nv}{pdbfile} \PYG{o}{=} build.pdb              \PYG{c+c1}{\PYGZsh{} PDB file}

\PYG{o}{[}OUTPUT\PYG{o}{]}
\PYG{n+nv}{dcdfile} \PYG{o}{=} min.dcd                \PYG{c+c1}{\PYGZsh{} coordinates trajectory file}
\PYG{n+nv}{rstfile} \PYG{o}{=} min.rst                \PYG{c+c1}{\PYGZsh{} restart file}

\PYG{o}{[}ENERGY\PYG{o}{]}
\PYG{n+nv}{forcefield}        \PYG{o}{=} CHARMM       \PYG{c+c1}{\PYGZsh{} CHARMM force field}
\PYG{n+nv}{electrostatic}     \PYG{o}{=} PME          \PYG{c+c1}{\PYGZsh{} Particl mesh Ewald method}
\PYG{n+nv}{switchdist}        \PYG{o}{=} \PYG{l+m}{10}.0         \PYG{c+c1}{\PYGZsh{} switch distance (Ang)}
\PYG{n+nv}{cutoffdist}        \PYG{o}{=} \PYG{l+m}{12}.0         \PYG{c+c1}{\PYGZsh{} cutoff distance (Ang)}
\PYG{n+nv}{pairlistdist}      \PYG{o}{=} \PYG{l+m}{13}.5         \PYG{c+c1}{\PYGZsh{} pair\PYGZhy{}list cutoff distance (Ang)}
\PYG{n+nv}{pme\PYGZus{}nspline}       \PYG{o}{=} \PYG{l+m}{4}            \PYG{c+c1}{\PYGZsh{} order of B\PYGZhy{}spline in PME}
\PYG{n+nv}{pme\PYGZus{}max\PYGZus{}spacing}   \PYG{o}{=} \PYG{l+m}{1}.2          \PYG{c+c1}{\PYGZsh{} max grid spacing allowed (Ang)}
\PYG{n+nv}{vdw\PYGZus{}force\PYGZus{}switch}  \PYG{o}{=} YES          \PYG{c+c1}{\PYGZsh{} turn on van der Waals force switch}
\PYG{n+nv}{contact\PYGZus{}check}     \PYG{o}{=} YES          \PYG{c+c1}{\PYGZsh{} turn on clash checker}

\PYG{o}{[}MINIMIZE\PYG{o}{]}
\PYG{n+nv}{method}            \PYG{o}{=} SD           \PYG{c+c1}{\PYGZsh{} Steepest descent method}
\PYG{n+nv}{nsteps}            \PYG{o}{=} \PYG{l+m}{2000}         \PYG{c+c1}{\PYGZsh{} number of steps}
\PYG{n+nv}{eneout\PYGZus{}period}     \PYG{o}{=}  \PYG{l+m}{100}         \PYG{c+c1}{\PYGZsh{} energy output freq}
\PYG{n+nv}{crdout\PYGZus{}period}     \PYG{o}{=}  \PYG{l+m}{100}         \PYG{c+c1}{\PYGZsh{} coordinates output frequency}
\PYG{n+nv}{rstout\PYGZus{}period}     \PYG{o}{=} \PYG{l+m}{2000}         \PYG{c+c1}{\PYGZsh{} restart output frequency}
\PYG{n+nv}{nbpdate\PYGZus{}period}    \PYG{o}{=}   \PYG{l+m}{10}         \PYG{c+c1}{\PYGZsh{} pairlist update frequency}

\PYG{o}{[}BOUNDARY\PYG{o}{]}
\PYG{n+nb}{type}              \PYG{o}{=} PBC          \PYG{c+c1}{\PYGZsh{} periodic boundary condition}
\PYG{n+nv}{box\PYGZus{}size\PYGZus{}x}        \PYG{o}{=} \PYG{l+m}{64}.0         \PYG{c+c1}{\PYGZsh{} Box size in X dimension (Ang)}
\PYG{n+nv}{box\PYGZus{}size\PYGZus{}y}        \PYG{o}{=} \PYG{l+m}{64}.0         \PYG{c+c1}{\PYGZsh{} Box size in Y dimension (Ang)}
\PYG{n+nv}{box\PYGZus{}size\PYGZus{}z}        \PYG{o}{=} \PYG{l+m}{64}.0         \PYG{c+c1}{\PYGZsh{} Box size in Z dimension (Ang)}
\end{sphinxVerbatim}

\clearpage


\subsection{Example control file for normal MD simulations}
\label{\detokenize{01_Getting_Started:example-control-file-for-normal-md-simulations}}
The control file for normal MD simulations must include a \sphinxstylestrong{{[}DYNAMICS{]}} section (see {\hyperref[\detokenize{06_Dynamics:dynamics}]{\sphinxcrossref{\DUrole{std,std-ref}{Dynamics section}}}}).
By using the following control file, the users carry out a
100-ps MD simulation at \(T\) = 298.15 K and \(P\) = 1 atm in the NPT ensemble.
The equations of motion are integrated by the RESPA algorithm
with a time step of 2.5 fs, and the bonds of light atoms (hydrogen atoms)
are constrained using the SHAKE/RATTLE and SETTLE algorithms.
The temperature and pressure are controlled with the Bussi thermostat and barostat.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}INPUT\PYG{o}{]}
\PYG{n+nv}{topfile} \PYG{o}{=} top\PYGZus{}all36\PYGZus{}prot.rtf     \PYG{c+c1}{\PYGZsh{} topology file}
\PYG{n+nv}{parfile} \PYG{o}{=} par\PYGZus{}all36m\PYGZus{}prot.prm    \PYG{c+c1}{\PYGZsh{} parameter file}
\PYG{n+nv}{strfile} \PYG{o}{=} toppar\PYGZus{}water\PYGZus{}ions.str  \PYG{c+c1}{\PYGZsh{} stream file}
\PYG{n+nv}{psffile} \PYG{o}{=} build.psf              \PYG{c+c1}{\PYGZsh{} protein structure file}
\PYG{n+nv}{pdbfile} \PYG{o}{=} build.pdb              \PYG{c+c1}{\PYGZsh{} PDB file}
\PYG{n+nv}{rstfile} \PYG{o}{=} min.rst                \PYG{c+c1}{\PYGZsh{} restart file}

\PYG{o}{[}OUTPUT\PYG{o}{]}
\PYG{n+nv}{dcdfile} \PYG{o}{=} md.dcd                 \PYG{c+c1}{\PYGZsh{} coordinates trajectory file}
\PYG{n+nv}{rstfile} \PYG{o}{=} md.rst                 \PYG{c+c1}{\PYGZsh{} restart file}

\PYG{o}{[}ENERGY\PYG{o}{]}
\PYG{n+nv}{forcefield}        \PYG{o}{=} CHARMM       \PYG{c+c1}{\PYGZsh{} CHARMM force field}
\PYG{n+nv}{electrostatic}     \PYG{o}{=} PME          \PYG{c+c1}{\PYGZsh{} Particl mesh Ewald method}
\PYG{n+nv}{switchdist}        \PYG{o}{=} \PYG{l+m}{10}.0         \PYG{c+c1}{\PYGZsh{} switch distance (Ang)}
\PYG{n+nv}{cutoffdist}        \PYG{o}{=} \PYG{l+m}{12}.0         \PYG{c+c1}{\PYGZsh{} cutoff distance (Ang)}
\PYG{n+nv}{pairlistdist}      \PYG{o}{=} \PYG{l+m}{13}.5         \PYG{c+c1}{\PYGZsh{} pair\PYGZhy{}list cutoff distance (Ang)}
\PYG{n+nv}{pme\PYGZus{}nspline}       \PYG{o}{=} \PYG{l+m}{4}            \PYG{c+c1}{\PYGZsh{} order of B\PYGZhy{}spline in PME}
\PYG{n+nv}{pme\PYGZus{}max\PYGZus{}spacing}   \PYG{o}{=} \PYG{l+m}{1}.2          \PYG{c+c1}{\PYGZsh{} max grid spacing allowed (Ang)}
\PYG{n+nv}{vdw\PYGZus{}force\PYGZus{}switch}  \PYG{o}{=} YES          \PYG{c+c1}{\PYGZsh{} turn on van der Waals force switch}

\PYG{o}{[}DYNAMICS\PYG{o}{]}
\PYG{n+nv}{integrator}        \PYG{o}{=}   VRES       \PYG{c+c1}{\PYGZsh{} RESPA integrator}
\PYG{n+nv}{timestep}          \PYG{o}{=} \PYG{l+m}{0}.0025       \PYG{c+c1}{\PYGZsh{} timestep (2.5fs)}
\PYG{n+nv}{nsteps}            \PYG{o}{=}  \PYG{l+m}{40000}       \PYG{c+c1}{\PYGZsh{} number of MD steps (100ps)}
\PYG{n+nv}{eneout\PYGZus{}period}     \PYG{o}{=}    \PYG{l+m}{400}       \PYG{c+c1}{\PYGZsh{} energy output period (1ps)}
\PYG{n+nv}{crdout\PYGZus{}period}     \PYG{o}{=}    \PYG{l+m}{400}       \PYG{c+c1}{\PYGZsh{} coordinates output period (1ps)}
\PYG{n+nv}{rstout\PYGZus{}period}     \PYG{o}{=}  \PYG{l+m}{40000}       \PYG{c+c1}{\PYGZsh{} restart output period}
\PYG{n+nv}{nbupdate\PYGZus{}period}   \PYG{o}{=}     \PYG{l+m}{10}       \PYG{c+c1}{\PYGZsh{} nonbond update period}
\PYG{n+nv}{elec\PYGZus{}long\PYGZus{}period}  \PYG{o}{=}      \PYG{l+m}{2}       \PYG{c+c1}{\PYGZsh{} period of reciprocal space calculation}
\PYG{n+nv}{thermostat\PYGZus{}period} \PYG{o}{=}     \PYG{l+m}{10}       \PYG{c+c1}{\PYGZsh{} period of thermostat update}
\PYG{n+nv}{barostat\PYGZus{}period}   \PYG{o}{=}     \PYG{l+m}{10}       \PYG{c+c1}{\PYGZsh{} period of barostat update}

\PYG{o}{[}CONSTRAINTS\PYG{o}{]}
\PYG{n+nv}{rigid\PYGZus{}bond}        \PYG{o}{=} YES          \PYG{c+c1}{\PYGZsh{} constraint all bonds involving hydrogen}

\PYG{o}{[}ENSEMBLE\PYG{o}{]}
\PYG{n+nv}{ensemble}          \PYG{o}{=} NPT          \PYG{c+c1}{\PYGZsh{} NPT ensemble}
\PYG{n+nv}{tpcontrol}         \PYG{o}{=} BUSSI        \PYG{c+c1}{\PYGZsh{} BUSSI thermostat and barostat}
\PYG{n+nv}{temperature}       \PYG{o}{=} \PYG{l+m}{300}          \PYG{c+c1}{\PYGZsh{} target temperature (K)}
\PYG{n+nv}{pressure}          \PYG{o}{=} \PYG{l+m}{1}.0          \PYG{c+c1}{\PYGZsh{} target pressure (atm)}

\PYG{o}{[}BOUNDARY\PYG{o}{]}
\PYG{n+nb}{type}              \PYG{o}{=} PBC          \PYG{c+c1}{\PYGZsh{} periodic boundary condition}
\end{sphinxVerbatim}


\chapter{Available Programs}
\label{\detokenize{02_Available_Programs:available-programs}}\label{\detokenize{02_Available_Programs:id1}}\label{\detokenize{02_Available_Programs::doc}}

\section{Simulators}
\label{\detokenize{02_Available_Programs:simulators}}

\subsection{Basic functions}
\label{\detokenize{02_Available_Programs:basic-functions}}
\sphinxstylestrong{atdyn}
\begin{quote}

The simulator that is parallelized with the atomic decomposition scheme.
In most cases, \sphinxstylestrong{atdyn} is applied to small systems or coarse-grained systems.
The program runs on CPU with the hybrid MPI+OpenMP protocol, where
only double-precision calculation is available.
Since the atomic decomposition is a simple parallelization scheme,
the source code is actually simple compared to that for the domain decomposition.
Therefore, this program is also useful to develop a new function of \sphinxstylestrong{GENESIS}.
\end{quote}

\sphinxstylestrong{spdyn}
\begin{quote}

The simulator that is parallelized with the domain decomposition scheme.
The program is designed to achieve high-performance molecular dynamics simulations,
such as micro-second simulations and cellular-scale simulations.
The program runs on not only CPU but also CPU+GPU with the hybrid MPI+OpenMP protocol.
Here, beside double-precision, mixed-precision calculations are also available.
In the mixed-precision model, force calculations are carried out with single precision,
while integration of the equations of motion as well as accumulation of the force and
energy are done with double-precision.
\end{quote}


\begin{savenotes}\sphinxattablestart
\centering
\sphinxcapstartof{table}
\sphinxthecaptionisattop
\sphinxcaption{Available functions in \sphinxstylestrong{atdyn} and \sphinxstylestrong{spdyn}}\label{\detokenize{02_Available_Programs:id2}}
\sphinxaftertopcaption
\begin{tabular}[t]{|\X{13}{33}|\X{10}{33}|\X{10}{33}|}
\hline
\sphinxstyletheadfamily 
Function
&\sphinxstyletheadfamily 
atdyn
&\sphinxstyletheadfamily 
spdyn
\\
\hline
Energy minimization
&
\(\bigcirc\) (SD and LBFGS)
&
\(\bigcirc\) (SD)
\\
\hline
All-atom molecular dynamics
&
\(\bigcirc\)
&
\(\bigcirc\)
\\
\hline
Coarse-grained molecular dynamics
&
\(\bigcirc\)
&
\(\bigcirc\) (All-atom Go-model)
\\
\hline
Implicit solvent model
&
\(\bigcirc\)
&
\(-\)
\\
\hline
Replica-exchange method
&
\(\bigcirc\)
&
\(\bigcirc\)
\\
\hline
Gaussian accelerated MD
&
\(\bigcirc\)
&
\(\bigcirc\)
\\
\hline
Reaction path search
&
\(\bigcirc\) (MEP and MFEP)
&
\(\bigcirc\) (MFEP)
\\
\hline
QM/MM calculation
&
\(\bigcirc\)
&
\(-\)
\\
\hline
Vibrational analysis
&
\(\bigcirc\)
&
\(-\)
\\
\hline
Cryo-EM flexible fitting
&
\(\bigcirc\)
&
\(\bigcirc\)
\\
\hline
Precision
&
double
&
double/mixed
\\
\hline
GPGPU calculation
&
\(-\)
&
\(\bigcirc\) (All-atom MD)
\\
\hline
Parallel I/O
&
\(-\)
&
\(\bigcirc\)
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}


\subsection{Atomic and domain decomposition schemes}
\label{\detokenize{02_Available_Programs:atomic-and-domain-decomposition-schemes}}
In the atomic decomposition MD, which is also called a replicated-data MD algorithm,
all MPI processors have the same coordinates data of all atoms in the system.
MPI parallelization is mainly applied to the “DO loops” of the bonded and
non-bonded interaction pair lists for the energy and force calculations.
\hyperref[\detokenize{02_Available_Programs:parallel-scheme}]{Fig.\@ \ref{\detokenize{02_Available_Programs:parallel-scheme}}} (a) shows a schematic representation of the atomic decomposition scheme
for the non-bonded interaction calculation in a Lennard-Jones system,
where 2 MPI processors are used.
In this scheme, MPI\_ALLREDUCE must be used to accumulate all the atomic forces every step,
resulting in large communication cost.

In the domain-decomposition MD, which is also called a
distributed-data MD algorithm, the whole system is decomposed into domains
according to the number of MPI processors, and each MPI processor is assigned to a specific domain.
Each MPI processor handles the coordinates data of the atoms
in the assigned domain and in the buffer regions near the domain boundary,
and carries out the calculation of the bonded and non-bonded interactions
in the assigned domain, enabling us to reduce computational cost.
In this scheme, communication of the atomic coordinates and
forces in the buffer region is essential.
\hyperref[\detokenize{02_Available_Programs:parallel-scheme}]{Fig.\@ \ref{\detokenize{02_Available_Programs:parallel-scheme}}} (b) is a schematic representation of the domain decomposition scheme,
where the system is decomposed into two domains to use 2 MPI processors.
Note that in the figure the system periodicity is not considered for simplicity.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=0.800\linewidth]{{Parallel_scheme}.png}
\caption{Parallelization scheme in the (a) atomic decomposition and (b) domain decomposition.}\label{\detokenize{02_Available_Programs:parallel-scheme}}\end{figure}


\subsection{Hybrid MPI+OpenMP calculation in SPDYN}
\label{\detokenize{02_Available_Programs:hybrid-mpi-openmp-calculation-in-spdyn}}
The users had better understand a basic scheme of parallel calculation in \sphinxstylestrong{SPDYN}
to get the best performance in the calculation.
As described above, the simulation box is divided into domains
according to the number of MPI processors.
Each domain is further divided into smaller cells, each of whose size is adjusted to be
approximately equal to or larger than the half of “pairlistdist + \(\alpha\)”.
Here, “pairlistdist” is specified in the control file, and \(\alpha\) depends on the
algorithms used in the simulation (see the next subsection).
Note that all domains or cells have the same size with a rectangular or cubic shape.
Each MPI processor is assigned to each domain, and data transfer or communication
about atomic coordinates and forces is achieved between only neighboring domains.
In addition, calculation of bonded and non-bonded interactions in each domain is
parallelized based on the OpenMP protocol.
These schemes realize hybrid MPI+OpenMP calculation, which is more efficient than
flat MPI calculation on recent computers with multiple CPU cores.
Because MPI and OpenMP are designed for distributed-memory and shared-memory
architectures, respectively, MPI is mainly used for parallelization between
nodes and OpenMP is used within one node.

The following figures illustrate how the hybrid MPI+OpenMP calculations are achieved in \sphinxstylestrong{SPDYN}.
In \hyperref[\detokenize{02_Available_Programs:domain-decomposition}]{Fig.\@ \ref{\detokenize{02_Available_Programs:domain-decomposition}}} (a) and 2(b), 8 MPI processors with 4 OpenMP threads (32 CPU cores
in total), and 27 MPI processors with 2 OpenMP threads
(54 CPU cores in total) are used, respectively.
In these Figures, only XY dimensions are shown for simplicity.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=0.800\linewidth]{{Domain_decomposition}.png}
\caption{Schematic representation of the hybrid MPI+OpenMP calculation in SPDYN.}\label{\detokenize{02_Available_Programs:domain-decomposition}}\end{figure}

For Case (a), the following commands are used:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{export} \PYG{n+nv}{OMP\PYGZus{}NUM\PYGZus{}THREADS}\PYG{o}{=}\PYG{l+m}{4}
\PYGZdl{} mpirun \PYGZhy{}np \PYG{l+m}{8} \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/spdyn INP \PYGZgt{} log
\end{sphinxVerbatim}

For Case (b), the following commands are used:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{export} \PYG{n+nv}{OMP\PYGZus{}NUM\PYGZus{}THREADS}\PYG{o}{=}\PYG{l+m}{2}
\PYGZdl{} mpirun \PYGZhy{}np \PYG{l+m}{27} \PYGZti{}/genesis/genesis\PYGZhy{}1.7.1/bin/spdyn INP \PYGZgt{} log
\end{sphinxVerbatim}

In the log file, the users can check whether the given numbers of MPI processors
and OpenMP threads are actually employed or not.
The following information should be found in the log file for instance for Case (a):

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}STEP2\PYG{o}{]} Setup MPI

Setup\PYGZus{}Mpi\PYGZus{}Md\PYGZgt{} Summary of Setup MPI
  number of MPI \PYG{n+nv}{processes}   \PYG{o}{=}          \PYG{l+m}{8}
  number of OpenMP \PYG{n+nv}{threads}  \PYG{o}{=}          \PYG{l+m}{4}
  total number of CPU \PYG{n+nv}{cores} \PYG{o}{=}         \PYG{l+m}{32}
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
In most cases, the number of domains in each dimension is automatically
determined according to the given number of MPI processors.
However, if such automatic determination is failed, the users must specify
the number of domains explicitly in the control file (see {\hyperref[\detokenize{10_Boundary:boundary}]{\sphinxcrossref{\DUrole{std,std-ref}{Boundary section}}}}).
\end{sphinxadmonition}


\subsection{Limitation of the available MPI processors}
\label{\detokenize{02_Available_Programs:limitation-of-the-available-mpi-processors}}
Basically, there is no strict limitation in the available number of MPI processors in \sphinxstylestrong{ATDYN}.
However, there are a few limitations in \sphinxstylestrong{SPDYN}.
First, the number of domains must be equal to the number of MPI processors.
Second, one domain must be composed of at least 8 cells (= \(2{\times}2{\times}2\)),
where the cell size in one dimension is automatically set to be larger than
the half of “pairlistdist + \(\alpha\)”,
The following table summarizes the \(\alpha\) value in each algorithm.
According to these rules, the available “maximum” number of MPI processors (\(N_{\rm max}\))
for a certain target system is mainly determined by the simulation box size and “pairlistdist”.
For example, if the box size of your target system is \(64{\times}64{\times}64 \text{\AA}^3\),
and “pairlistdist=13.5” is specified in the control file,
\(N_{\rm max}\) is \(4{\times}4{\times}4 = 64\) in the case of NVT ensemble and “rigid\_bond=YES”.
If the users want to use much more CPU cores than \(N_{\rm max}\),
the number of OpenMP threads should be increased instead of the MPI processors.


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabulary}{\linewidth}[t]{|T|T|T|}
\hline
\sphinxstyletheadfamily 
Rigid bond
&\sphinxstyletheadfamily 
Ensemble
&\sphinxstyletheadfamily 
\(\alpha\) \((\text{\AA})\)
\\
\hline
NO
&
NVE/NVT
&
0.0
\\
\hline
NO
&
NPT/NPAT/NPgT
&
0.6
\\
\hline
YES
&
NVE/NVT
&
2.0
\\
\hline
YES
&
NPT/NPAT/NPgT
&
2.6
\\
\hline
\end{tabulary}
\par
\sphinxattableend\end{savenotes}

In the MD simulation with the NPT ensemble, these rules become more important,
because the box size (or cell size) can change during the simulation.
In fact, the number of domains in each dimension is initially fixed, but the number of cells can be
changed and adjusted to keep the cell size larger than the half of “pairlistdist + \(\alpha\)”.
If the box size is decreased during the simulation, and the number of cells in one dimension of the
domain unfortunately becomes one, the calculation stops immediately because of the violation of the above rule.
The users may often encounter this situation if the number of cells in one dimension of the domain
is just two at the beginning of the MD simulation, and the simulation box has significantly shrunk during the simulation.
To avoid such problems, the users may have to use smaller number of MPI processors
(which makes cells larger) or shorter pairlistdist (making much cells in one domain),
or reconstruct a larger system.

If the users encountered the following error message in the simulation,
the problem is probably related to the above rules, where the specified number
of MPI processors might exceed \(N_{\rm max}\).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Setup\PYGZus{}Processor\PYGZus{}Number\PYGZgt{} Cannot define domains and cells. Smaller or
adjusted MPI processors, or shorter pairlistdist, or larger boxsize
should be used.
\end{sphinxVerbatim}

In this case, please make sure that one domain can be composed of at least 8 cells.
If the domains and cells are successfully determined,
they can be seen in the early part of the log file.
The following example is corresponding to the situation in \hyperref[\detokenize{02_Available_Programs:domain-decomposition}]{Fig.\@ \ref{\detokenize{02_Available_Programs:domain-decomposition}}} (b).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Setup\PYGZus{}Boundary\PYGZus{}Cell\PYGZgt{} Set Variables For Boundary Condition
  domains \PYG{o}{(}x,y,z\PYG{o}{)} \PYG{o}{=}          \PYG{l+m}{3}         \PYG{l+m}{3}         \PYG{l+m}{3}
  ncells \PYG{o}{(}x,y,z\PYG{o}{)}  \PYG{o}{=}          \PYG{l+m}{6}         \PYG{l+m}{6}         \PYG{l+m}{6}
\end{sphinxVerbatim}


\subsection{Available sections}
\label{\detokenize{02_Available_Programs:available-sections}}
Fundamental functions in \sphinxstylestrong{SPDYN} and \sphinxstylestrong{ATDYN} are energy minimization (Min),
molecular dynamics method (MD), replica-exchange method (REMD),
string method (String), and vibrational analysis (Vib).
As shown in the last part of the previous chapter, the users carry out simulations
of these methods by writing related sections in the control file.
The users can extend these fundamental functions by combining various sections.
For example, to run a “restrained MD simulation”, the users
add \sphinxstylestrong{{[}SELECTION{]}} and \sphinxstylestrong{{[}RESTRAINTS{]}} sections in the control file of the “normal MD simulation”.
In fact, there are 17 individual sections in \sphinxstylestrong{GENESIS} version 1.4.
The following table summarizes the available sections in each function.
Detailed usage of each section is described in this user guide,
and also in the online tutorials (\sphinxurl{https://www.r-ccs.riken.jp/labs/cbrt/tutorials2019/}).


\begin{savenotes}\sphinxattablestart
\centering
\sphinxcapstartof{table}
\sphinxthecaptionisattop
\sphinxcaption{Available sections in each algorithm and method}\label{\detokenize{02_Available_Programs:id3}}
\sphinxaftertopcaption
\begin{tabular}[t]{|\X{25}{100}|\X{10}{100}|\X{10}{100}|\X{10}{100}|\X{10}{100}|\X{10}{100}|\X{25}{100}|}
\hline
\sphinxstyletheadfamily 
Section
&\sphinxstyletheadfamily 
Min
&\sphinxstyletheadfamily 
MD
&\sphinxstyletheadfamily 
REMD
&\sphinxstyletheadfamily 
String
&\sphinxstyletheadfamily 
Vib
&\sphinxstyletheadfamily 
Description
\\
\hline
{[}INPUT{]}
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
{\hyperref[\detokenize{03_Input:input}]{\sphinxcrossref{\DUrole{std,std-ref}{Input section}}}}
\\
\hline
{[}OUTPUT{]}
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
{\hyperref[\detokenize{04_Output:output}]{\sphinxcrossref{\DUrole{std,std-ref}{Output section}}}}
\\
\hline
{[}ENERGY{]}
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
{\hyperref[\detokenize{05_Energy:energy}]{\sphinxcrossref{\DUrole{std,std-ref}{Energy section}}}}
\\
\hline
{[}BOUNDARY{]}
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
{\hyperref[\detokenize{10_Boundary:boundary}]{\sphinxcrossref{\DUrole{std,std-ref}{Boundary section}}}}
\\
\hline
{[}DYNAMICS{]}
&
\(-\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(-\)
&
{\hyperref[\detokenize{06_Dynamics:dynamics}]{\sphinxcrossref{\DUrole{std,std-ref}{Dynamics section}}}}
\\
\hline
{[}CONSTRAINTS{]}
&
\(-\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(-\)
&
{\hyperref[\detokenize{08_Constraints:constraints}]{\sphinxcrossref{\DUrole{std,std-ref}{Constraints section}}}}
\\
\hline
{[}ENSEMBLE{]}
&
\(-\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(-\)
&
{\hyperref[\detokenize{09_Ensemble:ensemble}]{\sphinxcrossref{\DUrole{std,std-ref}{Ensemble section}}}}
\\
\hline
{[}MINIMIZE{]}
&
\(\bigcirc\)
&
\(-\)
&
\(-\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
{\hyperref[\detokenize{07_Minimize:minimize}]{\sphinxcrossref{\DUrole{std,std-ref}{Minimize section}}}}
\\
\hline
{[}REMD{]}
&
\(-\)
&
\(-\)
&
\(\bigcirc\)
&
\(-\)
&
\(-\)
&
{\hyperref[\detokenize{14_REMD:remd}]{\sphinxcrossref{\DUrole{std,std-ref}{REMD section}}}}
\\
\hline
{[}RPATH{]}
&
\(-\)
&
\(-\)
&
\(-\)
&
\(\bigcirc\)
&
\(-\)
&
{\hyperref[\detokenize{15_RPath:rpath}]{\sphinxcrossref{\DUrole{std,std-ref}{RPATH section}}}}
\\
\hline
{[}VIBRATION{]}
&
\(-\)
&
\(-\)
&
\(-\)
&
\(-\)
&
\(\bigcirc\)
&
{\hyperref[\detokenize{18_Vibration:vibration}]{\sphinxcrossref{\DUrole{std,std-ref}{Vibration section}}}}
\\
\hline
{[}SELECTION{]}
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
{\hyperref[\detokenize{11_Selection:selection}]{\sphinxcrossref{\DUrole{std,std-ref}{Selection section}}}}
\\
\hline
{[}RESTRAINTS{]}
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
{\hyperref[\detokenize{12_Restraints:restraints}]{\sphinxcrossref{\DUrole{std,std-ref}{Restraints section}}}}
\\
\hline
{[}FITTING{]}
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
{\hyperref[\detokenize{13_Fitting:fitting}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting section}}}}
\\
\hline
{[}GAMD{]}
&
\(-\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(-\)
&
\(-\)
&
{\hyperref[\detokenize{16_GaMD:gamd}]{\sphinxcrossref{\DUrole{std,std-ref}{GAMD section}}}}
\\
\hline
{[}QMMM{]}
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
{\hyperref[\detokenize{17_QMMM:qmmm}]{\sphinxcrossref{\DUrole{std,std-ref}{QMMM section}}}}
\\
\hline
{[}EXPERIMENTS{]}
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(\bigcirc\)
&
\(-\)
&
\(-\)
&
{\hyperref[\detokenize{19_Experiments:experiments}]{\sphinxcrossref{\DUrole{std,std-ref}{Experiments section}}}}
\\
\hline
\end{tabular}
\par
\sphinxattableend\end{savenotes}


\section{Analysis tools}
\label{\detokenize{02_Available_Programs:analysis-tools}}
The following programs are available as the trajectory analysis tools in \sphinxstylestrong{GENESIS}.
Basic usage of each tool is similar to that of \sphinxstylestrong{spdyn} or \sphinxstylestrong{atdyn}.
The users can automatically generate a template control file for each program
by using the “{[}program\_name{]} -h ctrl” command.
The control file is mainly composed of INPUT, OUTPUT, TRAJECTORY,
FITTING, SELECTION, and OPTION sections.
The trajectory files to be analyzed are specified in the \sphinxstylestrong{{[}TRAJECTORY{]}} section,
and the parameters used in the analysis are specified in the \sphinxstylestrong{{[}OPTION{]}} section.
Note that the required sections are depending on the program.
For example, \sphinxstylestrong{eigmat\_analysis} requires only INPUT and OUTPUT sections.
Detailed usage of each tool is described in the online tutorial.


\subsection{Trajectory analysis}
\label{\detokenize{02_Available_Programs:trajectory-analysis}}
\sphinxstylestrong{comcrd\_analysis}
\begin{quote}

Analyze the coordinates of the center of mass of the selected atoms.
\end{quote}

\sphinxstylestrong{diffusion\_analysis}
\begin{quote}

Analyze the diffusion constant.
\end{quote}

\sphinxstylestrong{distmat\_analysis}
\begin{quote}

Analyze the matrix of the averaged distance of the selected atoms.
\end{quote}

\sphinxstylestrong{drms\_analysis}
\begin{quote}

Analyze the distance RMSD of the selected atoms
with respect to the initial structure.
\end{quote}

\sphinxstylestrong{fret\_analysis}
\begin{quote}

Analyze the FRET efficiency.
\end{quote}

\sphinxstylestrong{hb\_analysis}
\begin{quote}

Analyze the hydrogen bond.
\end{quote}

\sphinxstylestrong{lipidthick\_analysis}
\begin{quote}

Analyze the membrane thickness.
\end{quote}

\sphinxstylestrong{msd\_analysis}
\begin{quote}

Analyze the mean-square displacement (MSD) of the selected atoms or molecules.
\end{quote}

\sphinxstylestrong{qval\_analysis}
\begin{quote}

Analyze the fraction of native contacts (Q-value).
\end{quote}

\sphinxstylestrong{rg\_analysis}
\begin{quote}

Analyze the radius of gyration of the selected atoms.
\end{quote}

\sphinxstylestrong{rmsd\_analysis}
\begin{quote}

Analyze the root-mean-square deviation (RMSD) of the selected atoms
with respect to the initial structure.
\end{quote}

\sphinxstylestrong{tilt\_analysis}
\begin{quote}

Analyze the tilt angle
\end{quote}

\sphinxstylestrong{trj\_analysis}
\begin{quote}

Analyze the distance, angle, dihedral angle, distance of the centers of mass (COM)
of the selected atom groups, angle of the COM of the selected atom groups,
and dihedral angle of the COM of the selected atom groups.
\end{quote}


\subsection{Principal component analysis (PCA)}
\label{\detokenize{02_Available_Programs:principal-component-analysis-pca}}
\sphinxstylestrong{avecrd\_analysis}
\begin{quote}

Calculate the average structure of the target molecule.
\end{quote}

\sphinxstylestrong{flccrd\_analysis}
\begin{quote}

Calculate the variance-covariance matrix from the trajectories and averaged coordinates.
This tool can be also used to calculate root-mean-square fluctuation (RMSF).
\end{quote}

\sphinxstylestrong{eigmat\_analysis}
\begin{quote}

Diagonalize the variance-covariance matrix in PCA.
\end{quote}

\sphinxstylestrong{prjcrd\_analysis}
\begin{quote}

Project the trajectories onto PC axes.
\end{quote}

\sphinxstylestrong{pcavec\_drawer}
\begin{quote}

Create a script for VMD and PyMol to visualize PC vectors obtained from eigmat\_analysis.
\end{quote}


\subsection{Trajectory and restart file converter}
\label{\detokenize{02_Available_Programs:trajectory-and-restart-file-converter}}
\sphinxstylestrong{crd\_convert}
\begin{quote}

Convert trajectories to PDB/DCD formats.
This tool can do centering of the target molecule,
fitting of a given atom group to the initial structure,
wrapping of molecules into the unit cell,
combining multiple trajectory files into a single file,
extraction of coordinates of selected atoms, and so on.
\end{quote}

\sphinxstylestrong{remd\_convert}
\begin{quote}

Convert REMD trajectories to those sorted by parameters.
Since the trajectory files are generated from each replica during the REMD simulations,
the obtained “raw” trajectories are composed of “mixed” data at various conditions (replica parameters).
\sphinxstylestrong{remd\_convert} enables the users to sort the REMD trajectories by parameters.
This is applicable to not only dcdfile but also energy log files.
\end{quote}

\sphinxstylestrong{rst\_convert}
\begin{quote}

Convert GENESIS restart file (rstfile) to the PDB file.
\end{quote}

\sphinxstylestrong{rst\_upgrade}
\begin{quote}

Convert old restart file (version \textless{} 1.1.0) to that in the new format (version \textgreater{}= 1.1.0).
\end{quote}


\subsection{Free energy calculation}
\label{\detokenize{02_Available_Programs:free-energy-calculation}}
\sphinxstylestrong{wham\_analysis}
\begin{quote}

Free energy analysis using the Weighted Histogram Analysis Method (WHAM).
\end{quote}

\sphinxstylestrong{mbar\_analysis}
\begin{quote}

Free energy analysis using the Multistate Bennett Acceptance Ratio (MBAR) method.
\end{quote}

\sphinxstylestrong{pmf\_analysis}
\begin{quote}

Calculate free energy profile using MBAR output.
\end{quote}

\sphinxstylestrong{meanforce\_analysis}
\begin{quote}

Calculate free energy profile from RPATH.
\end{quote}


\subsection{Clustering}
\label{\detokenize{02_Available_Programs:clustering}}
\sphinxstylestrong{kmeans\_clustering}
\begin{quote}

Carry out k-means clustering for coordinates trajectories
\end{quote}


\subsection{Interface program}
\label{\detokenize{02_Available_Programs:interface-program}}
\sphinxstylestrong{dssp\_interface}
\begin{quote}

Interface program to analyze the protein secondary structure
in the DCD trajectory file
using the DSSP program (\sphinxurl{https://swift.cmbi.umcn.nl/gv/dssp/}).
\end{quote}


\subsection{SPANA}
\label{\detokenize{02_Available_Programs:spana}}
SPANA (SPatial decomposition ANAlysis) is developed to carry out trajectory analyses of large-scale biological simulations using multiple CPU cores in parallel. SPANA employs a spatial decomposition of a system to distribute structural and dynamical analyses into the individual CPU core and allows us to reduce the computational time for the analysis significantly. SPANA is suitable for the analysis of systems with multiple macromolecules (such as cellular crowding systems) under the periodic boundary condition.

\sphinxstylestrong{contact\_analysis}
\begin{quote}

Calculate the number of close atomic pairs between given molecules. The close atomic pairs (or atomic contacts) are defined if the closest atom-atom distance between two macromolecules is shorter than given cutoff distance. This program also finds the closest atom pairs between macromolecule pairs within the cutoff distance.
\end{quote}

\sphinxstylestrong{density\_analysis}
\begin{quote}

Calculate 3D density distribution of atoms and output the density in X-PLOR/CCP4/DX format.
\end{quote}

\sphinxstylestrong{hbond\_analysis}
\begin{quote}

Analyze hydrogen bonds
\end{quote}

\sphinxstylestrong{rdf\_analysis}
\begin{quote}

Calculate the radial distribution function (RDF) and proximal distribution function (PDF) of molecules (as solvent) around the target group (as solute). PDF provides density of solvent as function of the distance to the surface of macromolecules.
\end{quote}

\sphinxstylestrong{sasa\_analysis}
\begin{quote}

Calculate solvent accessible surface area (SASA) of the target molecules. This program outputs not only the total SASA but also the SASA for each atom in the target molecules.
\end{quote}


\subsection{Other utilities}
\label{\detokenize{02_Available_Programs:other-utilities}}
\sphinxstylestrong{rpath\_generator}
\begin{quote}

Generate inputs for the string method.
This tool is usually used after targeted MD simulation
for generating an initial pathway for the subsequent string method.
\end{quote}

\sphinxstylestrong{pathcv\_analysis}
\begin{quote}

Calculate tangential and orthogonal coordinates to a pathway from samples.
\end{quote}

\sphinxstylestrong{qmmm\_generator}
\begin{quote}

Generate a system for QM/MM calculation from MD data.
\end{quote}

\sphinxstylestrong{emmap\_generator}
\begin{quote}

Generate cryo-EM density map from PDB file.
\end{quote}


\section{Parallel I/O tools}
\label{\detokenize{02_Available_Programs:parallel-i-o-tools}}
\sphinxstylestrong{SPDYN} can be employed with the parallel I/O protocol to achieve massively parallel computation.
Since \sphinxstylestrong{SPDYN} is parallelized with the domain decomposition scheme,
each MPI processor has the coordinates of atoms in the assigned domain.
Therefore, large ammount of communication is needed between MPI processors
to write the coordinates in a single DCD file, which is a waste of time
in the case of the simulations for a huge system like 100,000,000 atoms.
To avoid such situations, file I/O in each node (parallel I/O) is useful.
The following tools are used to handle the files generated from parallel I/O simulations.

\sphinxstylestrong{prst\_setup}
\begin{quote}

This tool divides input files (PDB and PSF) for a huge system into multiple files,
where each file is assinged to each domain.
The obtained files can be read as restart files in the \sphinxstylestrong{{[}INPUT{]}} section.
Note that \sphinxstylestrong{prst\_setup} is not compiled with Fujitsu compilers.
Therefore, if the users are going to perform MD simulations with parallel I/O in Fujitsu supercomputers,
the users must create the files without using Fujitsu compilers elsewhere in advance.
Even if \sphinxstylestrong{prst\_setup} and \sphinxstylestrong{SPDYN} are compiled with different compilers,
there is no problem to execute \sphinxstylestrong{SPDYN} with parallel I/O.
\end{quote}

\sphinxstylestrong{pcrd\_convert}
\begin{quote}

Convert multiple trajectory files obtained from the parallel I/O simulation to a single DCD file.
This tool has a similar function to \sphinxstylestrong{crd\_convert}.
\end{quote}


\chapter{Input section}
\label{\detokenize{03_Input:input-section}}\label{\detokenize{03_Input:input}}\label{\detokenize{03_Input::doc}}

\section{How to prepare input files}
\label{\detokenize{03_Input:how-to-prepare-input-files}}
In order to run MD simulations, the users have to prepare input files that contains
information about the coordinates of the initial structure as well as
topology of the system and force field parameters.
The users first create those input files by using a setup tool,
and their filenames are specified in the \sphinxstylestrong{{[}INPUT{]}} section of the control file.
\sphinxstylestrong{GENESIS} supports various input file formats such as CHARMM, AMBER, and GROMACS.
Basically, required input files depend on the force field to be used in the simulation.
The following table summarizes the essential input files and setup tools for each force field.


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabulary}{\linewidth}[t]{|T|T|T|}
\hline
\sphinxstyletheadfamily 
Force field
&\sphinxstyletheadfamily 
Input files
&\sphinxstyletheadfamily 
Setup tool
\\
\hline
CHARMM
&
top, par, psf, pdb (or crd), str
&
VMD, PSFGEN, CHARMM-GUI, CHARMM
\\
\hline
AMBER
&
prmtop, pdb, (or ambcrd)
&
LEaP
\\
\hline
KB Go-model
&
top, par, psf, pdb
&
MMTSB server
\\
\hline
All-atom Go-model
&
grotop, grocrd (or pdb)
&
SMOG server, SMOG2
\\
\hline
\end{tabulary}
\par
\sphinxattableend\end{savenotes}


\subsection{CHARMM force field}
\label{\detokenize{03_Input:charmm-force-field}}
One of the commonly used parameters for biomolecules is the CHARMM force field,
which was originally developed by the Karplus group at the Harvard University \sphinxcite{99_References:mackerell-1998}.
The users can obtain the files that contain the force field parameters from the CHARMM group’s web site.
At this momemt, the latest version of the CHARMM force field is C36m \sphinxcite{99_References:huang-2017}.
In the download file, there are topology and parameter files (e.g., top\_all36\_prot.rtf and par\_all36m\_prot.prm).

In order to run the MD simulation with the CHARMM force field, the users have to additionally make
a new file that holds the information about the atom connectivity of the “whole” target system.
Note that the topology file (e.g., top\_all36\_prot.rtf) does not contain such information,
because it is designed to generally define the topology of proteins by dealing with
the 20 amino acid residues as “fragments”. In order to hold the topology information
of the target system, the users will create a “PSF” file (protein structure file).
It is commonly used in other MD software, and can be generated from the PDB and
topology files by using VMD/PSFGEN \sphinxcite{99_References:humphrey-1996}, CHARMM-GUI \sphinxcite{99_References:jo-2008}, or the CHARMM program \sphinxcite{99_References:charmm-program}.

When the PSF file is created, “processed PDB” file is also obtained,
where the atom name or residue name might be changed from those
in the original PDB file. The users must use this PDB file as the input
of the MD simulation, because it has a consistency with the information in PSF.
Consequently, the users need four files (processed PDB, parameter, topology, and PSF)
as the inputs of \sphinxstylestrong{GENESIS}. These files are specified in the {[}INPUT{]} section
of the control file of \sphinxstylestrong{GENESIS}.

\noindent{\hspace*{\fill}\sphinxincludegraphics[width=0.800\linewidth]{{Input_fig1}.png}\hspace*{\fill}}


\subsection{AMBER force field}
\label{\detokenize{03_Input:amber-force-field}}
The AMBER force field has been also commonly used for the MD simulations of
biomolecules, which was originally developed by the Kollman group at the
University of California, San Francisco \sphinxcite{99_References:cornell-1995}. \sphinxstylestrong{GENESIS} can deal with the
AMBER force fields. Basic scheme to prepare the input files for \sphinxstylestrong{GENESIS}
is similar to that in the case of CHARMM. The users utilize the LEaP program in AmberTools \sphinxcite{99_References:amber18-program}.
LEaP has a similar function to PSFGEN. After building the target system using LEaP,
the users obtain PRMTOP, CRD, and PDB files. PRMTOP contains the information about
parameter and topology of the target system, and CRD and PDB include
the coordinates of atoms in the initial structure. \sphinxstylestrong{GENESIS} uses
these files as the inputs.


\subsection{Other force fields}
\label{\detokenize{03_Input:other-force-fields}}
\sphinxstylestrong{GENESIS} can deal with coarse-grained (CG) models such as the Go-model \sphinxcite{99_References:taketomi-1975} and MARTINI \sphinxcite{99_References:marrink-2007}.
In this case, the users again use external setup tools to build the system and prepare
the parameter and topology files. For the all-atom Go-model \sphinxcite{99_References:whitford-2009},
the users use the SMOG server \sphinxcite{99_References:noel-2010} or SMOG2 program \sphinxcite{99_References:noel-2016},
which generates grotop and grocrd files. The grotop file contains
the information about parameter and topology, and the grocrd file includes
the coordinates of the initial structure, both of which are the file formats
used in the GROMACS program. For the Karanicolas-Brooks (KB) Go-model \sphinxcite{99_References:karanicolas-2002hl} \sphinxcite{99_References:karanicolas2003improved},
the users use the MMTSB server \sphinxcite{99_References:feig-2004}, which generates par, top, pdb, and psf files.


\section{General input files}
\label{\detokenize{03_Input:general-input-files}}
\sphinxstylestrong{topfile}
\begin{quote}

CHARMM topology file containing information about atom connectivity
of residues and other molecules.
For details on the format, see the CHARMM web site \sphinxcite{99_References:charmm-online}.
\end{quote}

\sphinxstylestrong{parfile}
\begin{quote}

CHARMM parameter file containing force field parameters,
e.g. force constants and equilibrium geometries.
\end{quote}

\sphinxstylestrong{strfile}
\begin{quote}

CHARMM stream file containing both topology information and parameters.
\end{quote}

\sphinxstylestrong{psffile}
\begin{quote}

CHARMM/X-PLOR ‘psffile{}` containing information of the system
such as atomic masses, charges, and atom connectivities.
\end{quote}

\sphinxstylestrong{prmtopfile}
\begin{quote}

AMBER ‘PARM’ or ‘prmtop’ file (AMBER7 or later format) containing
information of the system such as atomic masses, charges, and atom
connectivities. For details about this format, see the AMBER web site
\sphinxcite{99_References:amber-online}.
\end{quote}

\sphinxstylestrong{grotopfile}
\begin{quote}

Gromacs ‘top’ file containing information of the system
such as atomic masses, charges, atom connectivities.
For details about this format, see the Gromacs web site
\sphinxcite{99_References:gromacs-online}.
\end{quote}

\sphinxstylestrong{pdbfile}
\begin{quote}

Coordinates file in the PDB format.
If \sphinxstyleemphasis{rstfile} is also specified in the \sphinxstylestrong{{[}INPUT{]}} section,
coordinates in \sphinxstyleemphasis{pdbfile} are replaced with those in \sphinxstyleemphasis{rstfile}.
\end{quote}

\sphinxstylestrong{crdfile}
\begin{quote}

Coordinates file in the CHARMM format.
If \sphinxstyleemphasis{pdbfile} is also specified in the \sphinxstylestrong{{[}INPUT{]}} section, coordinates in \sphinxstyleemphasis{crdfile} are NOT used.
However, if \sphinxstyleemphasis{pdbfile} is not specified, coordinates in \sphinxstyleemphasis{crdfile} are used.
If \sphinxstyleemphasis{rstfile} is further specified, coordinates in \sphinxstyleemphasis{rstfile} are used.
\end{quote}

\sphinxstylestrong{ambcrdfile}
\begin{quote}

Coordinates file in the AMBER format (ascii).
If \sphinxstyleemphasis{pdbfile} is also specified in the \sphinxstylestrong{{[}INPUT{]}} section, coordinates in \sphinxstyleemphasis{ambcrdfile} are NOT used.
However, if \sphinxstyleemphasis{pdbfile} is not specified, coordinates in \sphinxstyleemphasis{ambcrdfile} are used.
If \sphinxstyleemphasis{rstfile} is further specified, coordinates in \sphinxstyleemphasis{rstfile} are used.
\end{quote}

\sphinxstylestrong{grocrdfile}
\begin{quote}

Coordinates file in the GROMACS format (.gro file).
If \sphinxstyleemphasis{pdbfile} is also specified in the \sphinxstylestrong{{[}INPUT{]}} section, coordinates in \sphinxstyleemphasis{grocrdfile} are NOT used.
However, if \sphinxstyleemphasis{pdbfile} is not specified, coordinates in \sphinxstyleemphasis{grocrdfile} are used.
If \sphinxstyleemphasis{rstfile} is further specified, coordinates in \sphinxstyleemphasis{rstfile} are used.
Note that velocites and simulation box size in \sphinxstyleemphasis{grocrdfile} are NOT used.
\end{quote}

\sphinxstylestrong{rstfile}
\begin{quote}

Restart file in the GENESIS format.
This file contains atomic coordinates, velocities, simulation box size, and
other variables which are essential to restart the simulation continuously.
If \sphinxstyleemphasis{rstfile} is specified in the \sphinxstylestrong{{[}INPUT{]}} section, coordinates in
\sphinxstyleemphasis{pdbfile}, \sphinxstyleemphasis{crdfile}, \sphinxstyleemphasis{grocrdfile}, or \sphinxstyleemphasis{ambcrdfile} are replaced with those in \sphinxstyleemphasis{rstfile}.
The box size specified in the \sphinxstylestrong{{[}BOUNDARY{]}} seciton is also overwritten.
Note that \sphinxstyleemphasis{pdbfile}, \sphinxstyleemphasis{crdfile}, \sphinxstyleemphasis{grocrdfile}, or \sphinxstyleemphasis{ambcrdfile} should be still specified
in the \sphinxstylestrong{{[}INPUT{]}} section, even if \sphinxstyleemphasis{rstfile} is specified.

Note that the file format of \sphinxstyleemphasis{rstfile} was changed after ver. 1.1.0.
The \sphinxstyleemphasis{rst\_upgrade} tool enables us to change
the old format used in ver. 1.0.0 or older to the new one.
\end{quote}


\section{Input files for implicit solvent models}
\label{\detokenize{03_Input:input-files-for-implicit-solvent-models}}
\sphinxstylestrong{eef1file} (for \sphinxstylestrong{ATDYN} only)
\begin{quote}

If the users employ the EEF1, IMM1, or IMIC model, the \sphinxstylestrong{eef1file} should be specified in the \sphinxstylestrong{{[}INPUT{]}} section.
This file contains the parameters for the solvation free energy calculation.
Note that the file is not provided in GENESIS, but available in the CHARMM program package.
To get the file, the users first have to download CHARMM, and then find \sphinxstyleemphasis{solver.inp} and \sphinxstyleemphasis{solvpar22.inp} in the support/aspara directory.
The corresponding topology and parameter files are also available in the same directory.
\end{quote}


\section{Input files for restraint}
\label{\detokenize{03_Input:input-files-for-restraint}}
\sphinxstylestrong{reffile}
\begin{quote}

Reference coordinates (PDB file format) for positional restraints and
coordinate fitting.
This file should contain the same total number of atoms as \sphinxstyleemphasis{pdbfile},
\sphinxstyleemphasis{crdfile}, \sphinxstyleemphasis{ambcrdfile}, or \sphinxstyleemphasis{grocrdfile}.
\end{quote}

\sphinxstylestrong{ambreffile}
\begin{quote}

Reference coordinates (‘amber crd’ file format) for positional
restraints and coordinate fitting.
This file should contain the same total number of atoms as
\sphinxstyleemphasis{pdbfile} or \sphinxstyleemphasis{ambcrdfile}.
\end{quote}

\sphinxstylestrong{groreffile}
\begin{quote}

Reference coordinates (‘gro’ file format) for positional
restraints and coordinate fitting.
This file should contain the same total number of atoms as
\sphinxstyleemphasis{pdbfile} or \sphinxstyleemphasis{grocrdfile}.
\end{quote}

\sphinxstylestrong{modefile}
\begin{quote}

Principal modes used with principal component (PC) restraints.
This file contains only single column ascii data.
The XYZ values of each atom’s mode vector are stored from the low-index modes.
\end{quote}

\sphinxstylestrong{localresfile} (for \sphinxstylestrong{SPDYN} only)
\begin{quote}

This file defines restraints to be applied in the system.
If you are not an expert of GENESIS, we strongly recommend you to simply use
the \sphinxstylestrong{{[}RESTRAINTS{]}} section for restraint instead of using \sphinxstylestrong{localresfile}.

In \sphinxstylestrong{localresfile}, only bond, angle, and dihedral angle restraints can be defined.
In addition, selected atoms in \sphinxstylestrong{localresfile} must exist in the same cell
in the domain decomposition scheme.
The restraint energy calculated for the lists in \sphinxstylestrong{localresfile} is NOT explicitly
displayed in the log file. Instead, the local restraint energy is hidden
in the conventional bond, angle, and dihedral angle energy terms of the log file.

The restraint potentials defined in \sphinxstylestrong{localresfile} are given by harmonic potentials:

\(U(r) = k\;(r-r_{0})^{2}\) for bonds

\(U({\theta}) = k\;({\theta}-{\theta}_{0})^{2}\) for bond angles

\(U({\phi}) = k\;({\phi}-{\phi}_{0})^{2}\) for dihedral angles

Here, \(r\), \({\theta}\), and \({\phi}\) are bond distance,
angle, and dihedral angles, respectively; subscript \(0\) denotes their
reference values; and \(k\) is the force constant.

The syntax in \sphinxstylestrong{localresfile} is as follows:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}BOND/ANGLE/DIHEDRAL\PYG{o}{]}     atom atom \PYG{o}{[}atom \PYG{o}{[}atom\PYG{o}{]}\PYG{o}{]}  k r0
\end{sphinxVerbatim}

The users must carefully specify the atom index in this file.
The atom indexes in \sphinxstylestrong{localresfile} must be consistent with
those in the other input files such as \sphinxstylestrong{psffile}.

The following is an example of \sphinxstylestrong{localresfile}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
BOND     \PYG{l+m}{139} \PYG{l+m}{143}          \PYG{l+m}{2}.0 \PYG{l+m}{10}.0
ANGLE    \PYG{l+m}{233} \PYG{l+m}{231} \PYG{l+m}{247}      \PYG{l+m}{3}.0 \PYG{l+m}{10}.0
DIHEDRAL  \PYG{l+m}{22}  \PYG{l+m}{24}  \PYG{l+m}{41}  \PYG{l+m}{43}  \PYG{l+m}{2}.0 \PYG{l+m}{10}.0
\end{sphinxVerbatim}
\end{quote}


\section{Input files for REMD and RPATH simulations}
\label{\detokenize{03_Input:input-files-for-remd-and-rpath-simulations}}
In the REMD or RPATH simulations, input files (mainly coordinates and restart files)
should be prepared for each replica.
In GENESIS, we can easily specify those multiple files in the \sphinxstylestrong{{[}INPUT{]}} section.
If we include ‘\{\}’ in the input filename, \{\} is automatically replaced with the replica index.
For example, in the case of REMD simulations with 4 replicas,
we prepare input\_1.pdb, input\_2.pdb, input\_3.pdb, and input\_3.pdb,
and specify \sphinxcode{\sphinxupquote{pdbfile = input\_\{\}.pdb}} in the \sphinxstylestrong{{[}INPUT{]}} section.
This rule is also applicable to the restart filename.

\sphinxstylestrong{fitfile} (for RPATH only; GENESIS 1.1.5 or later)
\begin{quote}

Reference coordinates for structure fitting.
This file is only used in the string method.
For other cases (MD, MIN, or REMD),
\sphinxstyleemphasis{reffile}, \sphinxstyleemphasis{groreffile}, or \sphinxstyleemphasis{ambreffile} is used for reference coordinates for fitting,
and this \sphinxstyleemphasis{fitfile} is simply ignored, even if it is specified in the \sphinxstylestrong{{[}INPUT{]}} section.
\end{quote}


\section{Examples}
\label{\detokenize{03_Input:examples}}
MD simulations of proteins in explicit solvent with the CHARMM36m force field:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}INPUT\PYG{o}{]}
\PYG{n+nv}{topfile} \PYG{o}{=} ../toppar/top\PYGZus{}all36\PYGZus{}prot.rtf
\PYG{n+nv}{parfile} \PYG{o}{=} ../toppar/par\PYGZus{}all36m\PYGZus{}prot.prm
\PYG{n+nv}{strfile} \PYG{o}{=} ../toppar/toppar\PYGZus{}water\PYGZus{}ions.str
\PYG{n+nv}{psffile} \PYG{o}{=} ../build/input.psf
\PYG{n+nv}{pdbfile} \PYG{o}{=} ../build/input.pdb
\end{sphinxVerbatim}

MD simulations with positional restraint:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}INPUT\PYG{o}{]}
\PYG{n+nv}{topfile} \PYG{o}{=} ../toppar/top\PYGZus{}all36\PYGZus{}prot.rtf
\PYG{n+nv}{parfile} \PYG{o}{=} ../toppar/par\PYGZus{}all36m\PYGZus{}prot.prm
\PYG{n+nv}{strfile} \PYG{o}{=} ../toppar/toppar\PYGZus{}water\PYGZus{}ions.str
\PYG{n+nv}{psffile} \PYG{o}{=} ../build/input.psf
\PYG{n+nv}{pdbfile} \PYG{o}{=} ../build/input.pdb
\PYG{n+nv}{reffile} \PYG{o}{=} ../build/input.pdb
\end{sphinxVerbatim}

MD simulations of membrane proteins with the CHARMM36m force field:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}INPUT\PYG{o}{]}
\PYG{n+nv}{topfile} \PYG{o}{=} ../toppar/top\PYGZus{}all36\PYGZus{}prot.rtf, ../toppar/top\PYGZus{}all36\PYGZus{}lipid.rtf
\PYG{n+nv}{parfile} \PYG{o}{=} ../toppar/par\PYGZus{}all36m\PYGZus{}prot.prm, ../toppar/par\PYGZus{}all36\PYGZus{}lipid.prm
\PYG{n+nv}{strfile} \PYG{o}{=} ../toppar/toppar\PYGZus{}water\PYGZus{}ions.str
\PYG{n+nv}{psffile} \PYG{o}{=} ../build/input.psf
\PYG{n+nv}{pdbfile} \PYG{o}{=} ../build/input.pdb
\end{sphinxVerbatim}

In this case, we specify multiple top and par files for proteins and lipids separated by commas.

If one line becomes very long, backslash ” \textbackslash{} ” can be used as a line continuation character:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}INPUT\PYG{o}{]}
\PYG{n+nv}{topfile} \PYG{o}{=} ../toppar/top\PYGZus{}all36\PYGZus{}prot.rtf, \PYG{l+s+se}{\PYGZbs{}}
          ../toppar/par\PYGZus{}all36\PYGZus{}na.prm,   \PYG{l+s+se}{\PYGZbs{}}
          ../toppar/top\PYGZus{}all36\PYGZus{}lipid.rtf
\PYG{n+nv}{parfile} \PYG{o}{=} ../toppar/par\PYGZus{}all36m\PYGZus{}prot.prm, \PYG{l+s+se}{\PYGZbs{}}
          ../toppar/top\PYGZus{}all36\PYGZus{}na.rtf,    \PYG{l+s+se}{\PYGZbs{}}
          ../toppar/par\PYGZus{}all36\PYGZus{}lipid.prm
\PYG{n+nv}{strfile} \PYG{o}{=} ../toppar/toppar\PYGZus{}water\PYGZus{}ions.str
\PYG{n+nv}{psffile} \PYG{o}{=} ../build/input.psf
\PYG{n+nv}{pdbfile} \PYG{o}{=} ../build/input.pdb
\end{sphinxVerbatim}

MD simulations with the AMBER force field:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}INPUT\PYG{o}{]}
\PYG{n+nv}{prmtopfile} \PYG{o}{=} ../build/input.prmtop
\PYG{n+nv}{ambcrdfile} \PYG{o}{=} ../build/input.crd
\end{sphinxVerbatim}

MD simulations with the all-atom Go-model:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}INPUT\PYG{o}{]}
\PYG{n+nv}{grotopfile} \PYG{o}{=} ../build/input.top
\PYG{n+nv}{grocrdfile} \PYG{o}{=} ../build/input.gro
\end{sphinxVerbatim}

In this case, we specify grotop and grocrd files obtained from the SMOG server or SMOG2 software.

MD simulation with the EEF1/IMM1/IMIC implicit solvent models (CHARMM19):

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}INPUT\PYG{o}{]}
\PYG{n+nv}{topfile}  \PYG{o}{=} ../support/aspara/toph19\PYGZus{}eef1.1.inp
\PYG{n+nv}{parfile}  \PYG{o}{=} ../support/aspara/param19\PYGZus{}eef1.1.inp
\PYG{n+nv}{eef1file} \PYG{o}{=} ../support/aspara/solvpar.inp
\PYG{n+nv}{psffile}  \PYG{o}{=} ../build/input.psf
\PYG{n+nv}{pdbfile}  \PYG{o}{=} ../build/input.pdb
\end{sphinxVerbatim}

MD simulation with the EEF1/IMM1/IMIC implicit solvent models (CHARMM C36):

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}INPUT\PYG{o}{]}
\PYG{n+nv}{topfile}  \PYG{o}{=} ../support/aspara/top\PYGZus{}all36\PYGZus{}prot\PYGZus{}eef1.1.rtf
\PYG{n+nv}{parfile}  \PYG{o}{=} ../toppar/par\PYGZus{}all36\PYGZus{}prot.prm
\PYG{n+nv}{eef1file} \PYG{o}{=} ../support/aspara/solvpar22.inp
\PYG{n+nv}{psffile}  \PYG{o}{=} ../build/input.psf
\PYG{n+nv}{pdbfile}  \PYG{o}{=} ../build/input.pdb
\end{sphinxVerbatim}

REMD simulations starting from the same initial structure:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}INPUT\PYG{o}{]}
\PYG{n+nv}{topfile} \PYG{o}{=} ../toppar/top\PYGZus{}all36\PYGZus{}prot.rtf
\PYG{n+nv}{parfile} \PYG{o}{=} ../toppar/par\PYGZus{}all36m\PYGZus{}prot.prm
\PYG{n+nv}{strfile} \PYG{o}{=} ../toppar/toppar\PYGZus{}water\PYGZus{}ions.str
\PYG{n+nv}{psffile} \PYG{o}{=} ../build/input.psf
\PYG{n+nv}{pdbfile} \PYG{o}{=} ../build/input.pdb
\end{sphinxVerbatim}

REMD simulations starting from different initial structures:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}INPUT\PYG{o}{]}
\PYG{n+nv}{topfile} \PYG{o}{=} ../toppar/top\PYGZus{}all36\PYGZus{}prot.rtf
\PYG{n+nv}{parfile} \PYG{o}{=} ../toppar/par\PYGZus{}all36m\PYGZus{}prot.prm
\PYG{n+nv}{strfile} \PYG{o}{=} ../toppar/toppar\PYGZus{}water\PYGZus{}ions.str
\PYG{n+nv}{psffile} \PYG{o}{=} ../build/input.psf
\PYG{n+nv}{pdbfile} \PYG{o}{=} ../build/input\PYGZus{}rep\PYG{o}{\PYGZob{}}\PYG{o}{\PYGZcb{}}.pdb
\end{sphinxVerbatim}

REMD simulations with restarting:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}INPUT\PYG{o}{]}
\PYG{n+nv}{topfile} \PYG{o}{=} ../toppar/top\PYGZus{}all36\PYGZus{}prot.rtf
\PYG{n+nv}{parfile} \PYG{o}{=} ../toppar/par\PYGZus{}all36m\PYGZus{}prot.prm
\PYG{n+nv}{strfile} \PYG{o}{=} ../toppar/toppar\PYGZus{}water\PYGZus{}ions.str
\PYG{n+nv}{psffile} \PYG{o}{=} ../build/input.psf
\PYG{n+nv}{pdbfile} \PYG{o}{=} ../build/input.pdb
\PYG{n+nv}{rstfile} \PYG{o}{=} run\PYGZus{}rep\PYG{o}{\PYGZob{}}\PYG{o}{\PYGZcb{}}.rst
\end{sphinxVerbatim}


\chapter{Output section}
\label{\detokenize{04_Output:output-section}}\label{\detokenize{04_Output:output}}\label{\detokenize{04_Output::doc}}
\sphinxstylestrong{GENESIS} yields trajectory data (coordinates and velocities)
in the \sphinxstyleemphasis{DCD} file format regardless of the force field or MD algorithm.
\sphinxstylestrong{GENESIS} can also generate a restart file (\sphinxstyleemphasis{rstfile})
during or at the end of the simulation,
which can be used to restart and extend the simulation continuously.
Output frequency of each file (e.g., crdout\_period and velout\_period)
is specified in the \sphinxstylestrong{{[}DYNAMICS{]}} section in the case of the MD,
REMD, and RPATH simulations, or \sphinxstylestrong{{[}MINIMIZE{]}} section in the case of
the energy minimization.


\bigskip\hrule\bigskip



\section{General output files}
\label{\detokenize{04_Output:general-output-files}}
\sphinxstylestrong{dcdfile}
\begin{quote}

Filename for the coordinates trajectory data.
Coordinates are written in the DCD format,
which is commonly used in various MD software such as CHARMM and NAMD.
The filename must be given in the case of \sphinxcode{\sphinxupquote{crdout\_period \textgreater{} 0}}.
However, if \sphinxcode{\sphinxupquote{crdout\_period = 0}} is specified in the control file,
no \sphinxstylestrong{dcdfile} is generated, even if the filename is specified
in the \sphinxstylestrong{{[}OUTPUT{]}} section.
\end{quote}

\sphinxstylestrong{dcdvelfile}
\begin{quote}

Filename for the velocity trajectory data.
Velocities are written in the DCD format.
The filename must be given in the case of \sphinxcode{\sphinxupquote{velout\_period \textgreater{} 0}}.
However, if \sphinxcode{\sphinxupquote{velout\_period = 0}} is specified in the control file,
no \sphinxstylestrong{dcdvelfile} is generated, even if the filename is specified
in the \sphinxstylestrong{{[}OUTPUT{]}} section.
\end{quote}

\sphinxstylestrong{rstfile}
\begin{quote}

Filename for the restart data.
The rstfile contains coordinates, velocities, simulation box size, and so on.
This file can be used to extend the simulation continuously.
In addition, it can be used to switch the simulation algorithms
(e.g., from minimization to MD, from MD to REMD, from REMD to minimization, etc)
The filename must be given in the case of \sphinxcode{\sphinxupquote{rstout\_period \textgreater{} 0}}.
However, if \sphinxcode{\sphinxupquote{rstout\_period = 0}} is specified in the control file,
no \sphinxstylestrong{rstfile} is generated, even if the filename is specified
in the \sphinxstylestrong{{[}OUTPUT{]}} section.
\end{quote}

\sphinxstylestrong{pdbfile} (for ATDYN only)
\begin{quote}

Filename for the restart PDB file. This file is updated every \sphinxcode{\sphinxupquote{rstout\_period}} steps.
\end{quote}


\section{Output files in REMD and RPATH simulations}
\label{\detokenize{04_Output:output-files-in-remd-and-rpath-simulations}}
When the user performs REMD or RPATH simulations,
the user must include ‘\{\}’ in the output filename.
This \{\} is automatically replaced with the replica index.

\sphinxstylestrong{remfile} (only for REMD simulations)
\begin{quote}

This file contains parameter index data from the REMD simulation, which is
written for each replica every \sphinxcode{\sphinxupquote{exchange\_period}} steps.
This is used as an input file for the \sphinxstyleemphasis{remd\_convert} tool
to sort the coordinates trajectory data by parameters.
The filename must contain ‘\{\}’, which is automatically replaced with the replica index.
Note that the information about the parameter index as well as replica index
in the entire REMD simulation is written in the standard (single) output file
(see online Tutorials).
\end{quote}

\sphinxstylestrong{logfile} (only for REMD and RPATH simulations)
\begin{quote}

This file contains the energy trajectory data from the REMD or RPATH simulations, which is
written for each replica every \sphinxcode{\sphinxupquote{exchange\_period}} steps.
This is used as an input file for the \sphinxstyleemphasis{remd\_convert} tool
to sort the coordinates trajectory data by parameters.
The filename must contain ‘\{\}’, which is automatically replaced with the replica index.
\end{quote}

\sphinxstylestrong{rpathfile} (only for RPATH simulations)
\begin{quote}

This file contains the trajectory of image coordinates in the string method,
which are reference values used in the restraint functions.
Columns correspond to the collective variables, and rows are time steps.
This data is written with the same timing as the \sphinxstyleemphasis{dcdfile}.
For details, see {\hyperref[\detokenize{15_RPath:rpath}]{\sphinxcrossref{\DUrole{std,std-ref}{RPATH section}}}}.
\end{quote}


\section{Output file in GaMD simulations}
\label{\detokenize{04_Output:output-file-in-gamd-simulations}}
\sphinxstylestrong{gamdfile}
\begin{quote}

This file provides GaMD parameters determined during the GaMD simulation.
The filename must be given in the case of \sphinxcode{\sphinxupquote{update\_period \textgreater{} 0}} in \sphinxstylestrong{{[}GAMD{]}} section.
The GaMD simulation updates its parameters every \sphinxcode{\sphinxupquote{update\_period}}, and then updates parameters
are output to \sphinxstylestrong{gamdfile}.
The file includes the maximum, minimum, average, and deviation of the total potential or dihedral potential,
which are calculated within the interval \sphinxcode{\sphinxupquote{update\_period}}.
\end{quote}


\section{Output file in Vibrational analysis}
\label{\detokenize{04_Output:output-file-in-vibrational-analysis}}
\sphinxstylestrong{minfofile}
\begin{quote}

This file provides the coordinates and normal mode vectors of the molecules
specified for vibrational analysis. It is used in \sphinxstylestrong{SINDO} for visualizing
the vibrational motion. It is also an input file to start anharmonic
vibrational calculations. See {\hyperref[\detokenize{18_Vibration:vibration}]{\sphinxcrossref{\DUrole{std,std-ref}{Vibration section}}}} for the vibrational
analysis.
\end{quote}


\section{Output file in FEP simulations}
\label{\detokenize{04_Output:output-file-in-fep-simulations}}
\sphinxstylestrong{fepfile}
\begin{quote}

This file provides energy differences between adjacent windows in FEP simulations.
The filename must be given in the case of \sphinxcode{\sphinxupquote{fepout\_period \textgreater{} 0}} in \sphinxstylestrong{{[}ALCHEMY{]}} section.
However, if \sphinxcode{\sphinxupquote{fepout\_period = 0}} is specified in the control file,
no \sphinxstylestrong{fepfile} is generated, even if the filename is specified
in the \sphinxstylestrong{{[}OUTPUT{]}} section.
If FEP/\(\lambda\)-REMD simulations are performed
(i.e., both \sphinxstylestrong{{[}REMD{]}} and \sphinxstylestrong{{[}ALCHEMY{]}} sections are specified in the control file),
the filename must contain ‘\{\}’, which is automatically replaced with the replica index.
\end{quote}


\section{Examples}
\label{\detokenize{04_Output:examples}}
For normal MD simulations:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}OUTPUT\PYG{o}{]}
\PYG{n+nv}{dcdfile} \PYG{o}{=} run.dcd
\PYG{n+nv}{rstfile} \PYG{o}{=} run.rst
\end{sphinxVerbatim}

For REMD simulations:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}OUTPUT\PYG{o}{]}
\PYG{n+nv}{logfile} \PYG{o}{=} run\PYGZus{}rep\PYG{o}{\PYGZob{}}\PYG{o}{\PYGZcb{}}.log
\PYG{n+nv}{dcdfile} \PYG{o}{=} run\PYGZus{}rep\PYG{o}{\PYGZob{}}\PYG{o}{\PYGZcb{}}.dcd
\PYG{n+nv}{remfile} \PYG{o}{=} run\PYGZus{}rep\PYG{o}{\PYGZob{}}\PYG{o}{\PYGZcb{}}.rem
\PYG{n+nv}{rstfile} \PYG{o}{=} run\PYGZus{}rep\PYG{o}{\PYGZob{}}\PYG{o}{\PYGZcb{}}.rst
\end{sphinxVerbatim}


\chapter{Energy section}
\label{\detokenize{05_Energy:energy-section}}\label{\detokenize{05_Energy:energy}}\label{\detokenize{05_Energy::doc}}

\section{Force fields}
\label{\detokenize{05_Energy:force-fields}}
In general, potential energy function is given by:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}E(r) & = \sum_{\mathrm{bond}}K_b(b-b_0)^2+\sum_{\mathrm{angle}}K_{\theta}(\theta-\theta_0)^2  \\
     & + \sum_{\mathrm{dihedral}}K_{\phi}(1+\cos\left(n\phi-\delta)\right)+\sum_{\mathrm{improper}}K_{\phi_i}(\phi_i-\phi_{i,0})^2 \\
     & + \sum_{\mathrm{nonbond}}\epsilon\left[\left(\frac{R_{min,ij}}{r_{ij}}\right)^{12}-2\left(\frac{R_{min,ij}}{r_{ij}}\right)^6\right]+\sum_{\mathrm{nonbond}}\frac{q_iq_j}{\epsilon_1r_{ij}}\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(K_b\), \(K_{\theta}\), \(K_{\phi}\),
and \(K_{\phi_i}\) are the force constant of the bond, angle,
dihedral angle, and improper dihedral angle term, respectively, and
\(b_0\), \(\theta_0\), \(\phi_0\), and \(\phi_{i,0}\)
are corresponding equilibrium values.
\(\delta\) is a phase shift of the dihedral angle potential,
\(\epsilon\) is a Lennard-Jones potential well depth,
\(R_{min,ij}\) is a distance of the Lennard-Jones potential minimum,
\(q_i\) is an atomic charge, \(\epsilon_1\) is an effective
dielectric constant, and \(r_{ij}\) is a distance between two atoms.
The detailed formula and parameters in the potential energy function
depend on the force field and molecular model.


\bigskip\hrule\bigskip


\sphinxstylestrong{forcefield} \sphinxstyleemphasis{CHARMM / CHARMM19 / AMBER / GROAMBER / GROMARTINI / KBGO / CAGO / AAGO / RESIDCG}
\begin{quote}

\sphinxstylestrong{Default : CHARMM}

Type of the force field used for energy and force calculation.
For the AMBER force field, the scheme used in the GROMACS program package
is availble in addition to that used in the AMBER package.
In this case, calculation for the dispersion correction term and
truncation of the non-bonded energy term are different between AMBER and GROMACS.
\begin{itemize}
\item {} 
\sphinxstylestrong{CHARMM}: CHARMM force field with the all-atom model (CHARMM22, 27, 36, 36m) \sphinxcite{99_References:mackerell-1998} \sphinxcite{99_References:klauda-2010} \sphinxcite{99_References:best-2012} \sphinxcite{99_References:huang-2013}

\item {} 
\sphinxstylestrong{CHARMM19}: CHARMM force field with the united-atom model (\sphinxstylestrong{ATDYN} only)

\item {} 
\sphinxstylestrong{AMBER}: AMBER force field with the original AMBER scheme \sphinxcite{99_References:case-2005}

\item {} 
\sphinxstylestrong{GROAMBER}: AMBER force field with the GROMACS scheme

\item {} 
\sphinxstylestrong{GROMARTINI}: MARTINI model \sphinxcite{99_References:marrink-2007} \sphinxcite{99_References:monticelli-2008}

\item {} 
\sphinxstylestrong{KBGO}: model by Karanicolas and Brooks \sphinxcite{99_References:karanicolas-2002hl} \sphinxcite{99_References:karanicolas2003improved} (\sphinxstylestrong{ATDYN} only)

\item {} 
\sphinxstylestrong{CAGO}: C\(\alpha\) Go-model \sphinxcite{99_References:clementi-2000} (\sphinxstylestrong{ATDYN} only)

\item {} 
\sphinxstylestrong{AAGO}: All-atom Go-model \sphinxcite{99_References:whitford-2009}

\item {} 
\sphinxstylestrong{RESIDCG}: Residue-level coarse-grained models (\sphinxstylestrong{ATDYN} only)

\end{itemize}
\end{quote}


\section{Non-bonded interactions}
\label{\detokenize{05_Energy:non-bonded-interactions}}
Calculation of the non-bonded interaction is the most time consuming part in MD simulations.
Computational time for the non-bonded interaction terms without
any approximation is proportional to \(O(N^2)\).
To reduce the computational cost, a cut-off approximation is introduced,
where the energy and force calculation is truncated at a given cut-off value (keyword \sphinxstyleemphasis{cutoffdist}).
Simple truncation at the cut-off distance leads to discontinuous
energy and forces. So it is necessary to introduce a polynomial function
(so called \sphinxstyleemphasis{switching function}) that smoothly turn off the interaction from
another given value (so called \sphinxstyleemphasis{switch cut-off}), which is generally applied
to the van der Waals interactions (keyword \sphinxstyleemphasis{switchdist}).
There are two kinds of switching: “potential switch” and “force switch”.
In \sphinxstylestrong{GENESIS}, potential switching is turned on as the default.
However, in the case of the AMBER force field, potential switching is still turned off,
since the original AMBER program package is not using the potential switching.
To turn on the “force switching”, \sphinxcode{\sphinxupquote{vdw\_force\_switch=YES}} must be specified.
Note that the cut-off scheme for the electrostatic energy term is different from
that for the van der Waals energy term, where the former uses a shift function.
Such shift is turned on when \sphinxcode{\sphinxupquote{Electrostatic=Cutoff}} is specified.


\bigskip\hrule\bigskip


\sphinxstylestrong{electrostatic} \sphinxstyleemphasis{CUTOFF / PME}
\begin{quote}

\sphinxstylestrong{Default : PME}
\begin{itemize}
\item {} 
\sphinxstylestrong{CUTOFF}: Non-bonded interactions including the van der Waals interaction are just truncated at \sphinxstyleemphasis{cutoffdist}.

\item {} 
\sphinxstylestrong{PME}: Particle mesh Ewald (PME) method is employed for long-range interactions. This option is only availabe in the periodic boundary condition.

\end{itemize}
\end{quote}

\sphinxstylestrong{switchdist} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 10.0} (unit : Angstrom)

Switch-on distance for nonbonded interaction energy/force quenching.
If \sphinxstyleemphasis{switchdist} is set to be equal to \sphinxstyleemphasis{cutoffdist}, switching can be turned off.
Switching scheme depends on the selected force field, \sphinxstyleemphasis{vdw\_shift}, and \sphinxstyleemphasis{vdw\_force\_switch} parameters.
In the case of AMBER force field, this switching must be disabled, because the switching function is not available.
In the case of “forcefield = GROMARTINI” and “electrostatic = CUTOFF”, \sphinxstyleemphasis{switchdist} is used only in the van der Waals potential energy.
The switching-on distance for the electrostatic energy is automatically defined as 0.0.
\end{quote}

\sphinxstylestrong{cutoffdist} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 12.0} (unit : Angstrom)

Cut-off distance for the non-bonded interactions.
This distance must be larger than \sphinxstyleemphasis{switchdist}, while smaller than \sphinxstyleemphasis{pairlistdist}.
In the case of the AMBER force field, this value must be equal to \sphinxstyleemphasis{switchdist}.
\end{quote}

\sphinxstylestrong{pairlistdist} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 13.5} (unit : Angstrom)

Distance used to make a Verlet pair list for non-bonded interactions \sphinxcite{99_References:verlet-1967}.
This distance must be larger than \sphinxstyleemphasis{cutoffdist}.
\end{quote}

\sphinxstylestrong{dielec\_const} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 1.0}

Dielectric constant of the system.
Note that the distance dependent dielectric constant is not availabe in \sphinxstylestrong{GENESIS}.
\end{quote}

\sphinxstylestrong{vdw\_force\_switch} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

This paramter determines whether the force switch function for van der Waals
interactions is employed or not. \sphinxcite{99_References:steinbach-1994}
The users must take care about this parameter, when the CHARMM force field is used.
Typically, “vdw\_force\_switch=YES” should be specified in the case of CHARMM36.
\end{quote}

\sphinxstylestrong{vdw\_shift} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

This parameter determines whether the energy shift for the van der Waals interactions
is employed or not. If it is turned on, potential energy at the cut-off distance is
shifted by a constant value so as to nullify the energy at that distance,
instead of the default smooth quenching function.
This parameter is available only when “forcefield = GROAMBER” or “forcefield = GROMARTINI”.
\end{quote}

\sphinxstylestrong{dispersion\_corr} \sphinxstyleemphasis{NONE / ENERGY / EPRESS}
\begin{quote}

\sphinxstylestrong{Default : NONE} (automatically set to \sphinxstylestrong{EPRESS} in the case of AMBER)

This parameter determines how to deal with the long-range correction
about the cut-off for the van der Waals interactions.
Note that the formula used for the correction is different
between the GROMACS and AMBER schemes.
In the case of the CHARMM force filed, “dispersion\_corr=NONE” is always used.
\begin{itemize}
\item {} 
\sphinxstylestrong{NONE}: No correction is carried out.

\item {} 
\sphinxstylestrong{ENERGY}: Only energy correction is carried out.

\item {} 
\sphinxstylestrong{EPRESS}: Both energy and internal pressure corrections are carried out.

\end{itemize}
\end{quote}

\sphinxstylestrong{implicit\_solvent} \sphinxstyleemphasis{NONE / GBSA / EEF1 / IMM1 / IMIC} (\sphinxstylestrong{ATDYN} only)
\begin{quote}

\sphinxstylestrong{Default : NONE}

Use implicit solvent or not.
\begin{itemize}
\item {} 
\sphinxstylestrong{NONE}: Do not use implicit solvent model

\item {} 
\sphinxstylestrong{GBSA}: Use the GB/SA implicit water model (Only available with the CHARMM or AMBER all-atom force fields in non-boundary condition (“type=NOBC” in the \sphinxstylestrong{{[}BOUNDARY{]}} section). \sphinxcite{99_References:onufriev-2004} \sphinxcite{99_References:weiser-1999}

\item {} 
\sphinxstylestrong{EEF1}: Use the EEF1 implicit water model (Only available with the CHARMM force fields in NOBC) \sphinxcite{99_References:lazaridis-1999}

\item {} 
\sphinxstylestrong{IMM1}: Use the IMM1 implicit membrane model (Only available with the CHARMM force fields in NOBC) \sphinxcite{99_References:lazaridis-2003}

\item {} 
\sphinxstylestrong{IMIC}: Use the IMIC implicit micelle model (Only available with the CHARMM force fields in NOBC) \sphinxcite{99_References:mori-2020}

\end{itemize}
\end{quote}

\sphinxstylestrong{contact\_check} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

If this parameter is set to \sphinxstyleemphasis{YES}, length of all covalent bonds
as well as distance between non-bonded atom pairs are checked at the begining of the simulation.
If long covalent bonds or clashing atoms are detected, those atom indexes are displayed in the log file.
If \sphinxstyleemphasis{contact\_check} is turned on, \sphinxstyleemphasis{nonb\_limiter} is also automatically enabled.
If the users want to turn on only “contact\_check”,
please specify “contact\_check = YES” and “nonb\_limiter = NO” explicitly.
Note that this contact\_check does not work in the parallel-io scheme.
If you are using \sphinxstylestrong{SPDYN}, please see also \sphinxstyleemphasis{structure\_check}.
\end{quote}

\sphinxstylestrong{structure\_check} \sphinxstyleemphasis{NONE / FIRST / DOMAIN} (\sphinxstylestrong{SPDYN} only)
\begin{quote}

\sphinxstylestrong{Default : NONE}

If this parameter is set to FIRST or DOMAIN, length of all covalent bonds
as well as distance between non-bonded atom pairs are checked at the begininig or during the simulation.
This option is similar to \sphinxstyleemphasis{contact\_check},
but has an improved capability when the parallel-io scheme is employed.
In \sphinxstylestrong{SPDYN}, we recommend the users to use this option instead of \sphinxstyleemphasis{contact\_check}.
Since the structure check spends additional computational time,
the users had better turn off this option in the production run.
\begin{itemize}
\item {} 
\sphinxstylestrong{NONE}: Do not check the structure

\item {} 
\sphinxstylestrong{FIRST}: Check the structure only at the beginning of the simulation

\item {} 
\sphinxstylestrong{DOMAIN}: Check the structure whenever the pairlist is updated

\end{itemize}
\end{quote}

\sphinxstylestrong{nonb\_limiter} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO} (automatically set to be equal to \sphinxstylestrong{contact\_check})

If this parameter is set to \sphinxstyleemphasis{YES},
large force caused by the atomic clash is suppressed during the simulation.
Here, the atomic clash can be defined by \sphinxstyleemphasis{minimum\_contact} (see below).
If “contact\_check = YES” is specified, this parameter is automatically set to “YES”.
If the users want to turn on only “contact\_check”,
please specify “contact\_check = YES” and “nonb\_limiter = NO” explicitly.
This option is basically useful for the energy minimization or equilibration of the system.
However, we strongly recommend the users to turn off this option in the production run,
because suppression of large forces is an “unphysical” manipulation to avoid unstable simulations.
\end{quote}

\sphinxstylestrong{minimum\_contact} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.5} (unit : Angstrom)

This parameter defines the clash distance, when \sphinxcode{\sphinxupquote{contact\_check = YES}} is specified.
If the distance between the non-bonded atoms is less than this value,
energy and force are computed using this distance instead of the actual distance.
\end{quote}

\sphinxstylestrong{nonbond\_kernel}  \sphinxstyleemphasis{AUTOSELECT / GENERIC / FUGAKU / INTEL / GPU}
\begin{quote}

\sphinxstylestrong{Default: AUTOSELECT}

If this parameter is set to AUTOSELECT, the program automatically decide the kernel
for the real-space non-bonded interaction. Please remember that you should compile with GPU
option to define GPU as nonboned\_kernel.
\end{quote}


\section{Particle mesh Ewald method}
\label{\detokenize{05_Energy:particle-mesh-ewald-method}}
Electrostatic energy in the conventional Ewald sum method is expressed as:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}E_{elec}=\sum_{i<j}\frac{q_i q_j}{\epsilon_1} \frac{\text{erfc}(\alpha r_{ij})}{r_{ij}}+\frac{2\pi}{V}\sum_{{\left|\mathbf{G}\right|}^2\ne 0}\frac{\exp(-{\left|\mathbf{G}\right|}^2/4\alpha^2)}{\left|\mathbf{G}\right|^2}\sum_{ij}\frac{q_i q_j}{\epsilon_1}\exp(i{\mathbf{G}} \cdot \mathbf{r}_{ij})-\sum_{ij}\frac{q_i q_j}{\epsilon_1}\frac{\alpha}{\sqrt{\pi}}\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

Here, the cut-off scheme can be used for the first term, because it decreases
rapidly as distance between atoms increases. The third term is so called
\sphinxstyleemphasis{self-energy}, and is calculated only once. The second term can be rewritten as:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\sum_{{\left|\mathbf{G}\right|}^2\ne 0}\frac{\exp(-{\left|\mathbf{G}\right|}^2/4\alpha^2)}{\left|\mathbf{G}\right|^2} {\left|\mathbf{S}(\mathbf{G})\right|}^2\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where the structure factor \(\mathbf{S}(\mathbf{G})\) is defined as:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\mathbf{S}(\mathbf{G})=\sum_i q_i \exp(i\mathbf{G} \cdot \mathbf{r}_i)\end{split}
\end{equation*}\end{quote}

We cannot employ fast Fourier transformation (FFT) for the calculation of
\(\mathbf{S}(\mathbf{G})\) since atomic positions are usually not
equally spaced. In the smooth particle mesh Ewald (PME) method
\sphinxcite{99_References:darden-1993in} \sphinxcite{99_References:essmann-1995vj}, this structure factor is
approximated by using cardinal B-spline interpolation as:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\mathbf{S}(\mathbf{G})=\sum_i q_i \exp(i\mathbf{G} \cdot \mathbf{r}_i) \approx b_1(G_1)b_2(G_2)b_3(G_3)\mathbf{F}(\mathbf{Q})(G_1,G_2,G_3)\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(b_1(G_1)\), \(b_2(G_2)\), and \(b_3(G_3)\) are
the coefficients brought by the cardinal B-spline interpolation of order
\(n\) and \(\mathbf{Q}\) is a 3D tensor obtained by interpolating
atomic charges on the grids.
Since this \(\mathbf{Q}\) has equally spaced structure, its Fourier
transformation, \(\mathbf{F}(\mathbf{Q})\), can be calculated by using
FFT in the PME method.


\bigskip\hrule\bigskip


\sphinxstylestrong{pme\_alpha} \sphinxstyleemphasis{Real or auto}
\begin{quote}

\sphinxstylestrong{Default : auto}

Exponent of complementary error function.
If \sphinxcode{\sphinxupquote{pme\_alpha=auto}} is specified,
the value is automatically determined from \sphinxstyleemphasis{cutoffdist} and \sphinxstyleemphasis{pme\_alpha\_tol}.

\sphinxstyleemphasis{Note: The default of pme\_alpha was 0.34 in GENESIS ver. 1.1.0 or former.}
\end{quote}

\sphinxstylestrong{pme\_alpha\_tol} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 1.0e-5}

Tolerance to be used for determining \sphinxstyleemphasis{pme\_alpha}, when \sphinxcode{\sphinxupquote{pme\_alpha=auto}} is specified.
\end{quote}

\sphinxstylestrong{pme\_nspline} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 4}

B-spline interpolation order used for the evaluation of \(b_1 (G_1)\),
\(b_2 (G_2)\), \(b_3 (G_3)\), and \(\mathbf{Q}\).
The order must be \(>= 3\).
\end{quote}

\sphinxstylestrong{pme\_max\_spacing} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 1.2} (unit : Angstrom)

Max PME grid size used in the automatic grid number determination
This parameter is used only when \sphinxstyleemphasis{pme\_ngrid\_x}, \sphinxstyleemphasis{pme\_ngrid\_y},
and \sphinxstyleemphasis{pme\_grid\_z} are not given in the control file.
\end{quote}

\sphinxstylestrong{pme\_ngrid\_x} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : N/A (Optional)}

Number of FFT grid points along x dimension.
If not specified, program will determine an appropriate number of grids
using \sphinxcode{\sphinxupquote{pme\_max\_spacing}}.
\end{quote}

\sphinxstylestrong{pme\_ngrid\_y} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : N/A (Optional)}

Number of FFT grid points along y dimension.
If not specified, program will determine an appropriate number of grids
using \sphinxcode{\sphinxupquote{pme\_max\_spacing}}.
\end{quote}

\sphinxstylestrong{pme\_ngrid\_z} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : N/A (Optional)}

Number of FFT grid points along z dimension.
If not specified, program will determine an appropriate number of grids
using \sphinxcode{\sphinxupquote{pme\_max\_spacing}}.
\end{quote}

\sphinxstylestrong{pme\_multiple} \sphinxstyleemphasis{YES/NO} (\sphinxstylestrong{ATDYN} only)
\begin{quote}

\sphinxstylestrong{Default : NO}

IF pme\_multiple is set to YES, MPI processes are divided into two groups
to compute the PME real and reciprocal parts individually.
\end{quote}

\sphinxstylestrong{pme\_mul\_ratio} \sphinxstyleemphasis{Integer} (\sphinxstylestrong{ATDYN} only)
\begin{quote}

\sphinxstylestrong{Default : 1}

Ratio of the MPI processors for real and reciprocal PME term computations
(only used when “PME\_multiple=YES” is specified).
\end{quote}

\sphinxstylestrong{PME\_scheme} \sphinxstyleemphasis{AUTOSELECT / OPT\_1DALLTOAlL / NOOPT\_1DALLTOALL / OPT\_2DALLTOALL / NOOPT\_2DALLTOALL} (\sphinxstylestrong{SPDYN} only)
\begin{quote}

\sphinxstylestrong{Default : AUTOSELECTL}

\sphinxstyleemphasis{AUTOSELECT} chooses the best scheme by running in setup procedure.
Other schemes can be chosen manually according to your preference..
For \sphinxstyleemphasis{1DALLTOALL} and \sphinxstyleemphasis{2DALLTOALL}, see ref \sphinxcite{99_References:jung-2016} for details.
\end{quote}

\sphinxstylestrong{SPDYN} use OpenMP/MPI hybrid parallel fast Fourier transformation library, FFTE \sphinxcite{99_References:ffte-online}.
The number of PME grid points must be multiples of 2, 3, and 5 due to the restriction of this library.
Moreover, in \sphinxstylestrong{SPDYN}, there are several additional rules, which depends on the number of processes, in PME grid numbers.
In \sphinxstylestrong{SPDYN}, we first define domain numbers in each dimension such that product of them equals to the total number of
MPI processors. Let us assume that the domain numbers in each dimension are
\sphinxcode{\sphinxupquote{domain\_x}}, \sphinxcode{\sphinxupquote{domain\_y}}, and \sphinxcode{\sphinxupquote{domain\_z}}. The restriction condition of the grid numbers are as follows:
\begin{enumerate}
\def\theenumi{\arabic{enumi}}
\def\labelenumi{\theenumi )}
\makeatletter\def\p@enumii{\p@enumi \theenumi )}\makeatother
\item {} 
OPT\_1DALLTOALL or NOOPT\_1DALLTOALL:

\end{enumerate}
\begin{quote}

\sphinxcode{\sphinxupquote{pme\_ngrid\_x}} should be multiple of 2 \(\times\) \sphinxcode{\sphinxupquote{domain\_x}}.

\sphinxcode{\sphinxupquote{pme\_ngrid\_y}} should be multiple of \sphinxcode{\sphinxupquote{domain\_y}} \(\times\) \sphinxcode{\sphinxupquote{domain\_z}} if \sphinxcode{\sphinxupquote{domain\_z}} is an even number and multiple of \sphinxcode{\sphinxupquote{domain\_y}} \(\times\) \sphinxcode{\sphinxupquote{domain\_z}} \(\times\) 2 otherwise.

\sphinxcode{\sphinxupquote{pme\_ngrid\_z}} should be multiple of \sphinxcode{\sphinxupquote{domain\_x}} \(\times\) \sphinxcode{\sphinxupquote{domain\_z}} and multiple of \sphinxcode{\sphinxupquote{domain\_y}} \(\times\) \sphinxcode{\sphinxupquote{domain\_z}}.
\end{quote}
\begin{enumerate}
\def\theenumi{\arabic{enumi}}
\def\labelenumi{\theenumi )}
\makeatletter\def\p@enumii{\p@enumi \theenumi )}\makeatother
\setcounter{enumi}{1}
\item {} 
OPT\_2DALLTOALL or NOOPT\_2DALLTOAlL:

\end{enumerate}
\begin{quote}

\sphinxcode{\sphinxupquote{pme\_ngrid\_x}} should be multiple of 2 \(\times\) \sphinxcode{\sphinxupquote{domain\_x}}.

\sphinxcode{\sphinxupquote{pme\_ngrid\_y}} should be multiple of \sphinxcode{\sphinxupquote{domain\_y}} \(\times\) \sphinxcode{\sphinxupquote{domain\_z}} if \sphinxcode{\sphinxupquote{domain\_z}} is an even number and multiple of \sphinxcode{\sphinxupquote{domain\_y}} \(\times\) \sphinxcode{\sphinxupquote{domain\_z}} \(\times\) 2 otherwise.

\sphinxcode{\sphinxupquote{pme\_ngrid\_z}} should be multiple of \sphinxcode{\sphinxupquote{domain\_x}} \(\times\) \sphinxcode{\sphinxupquote{domain\_z}}.
\end{quote}
\begin{enumerate}
\def\theenumi{\arabic{enumi}}
\def\labelenumi{\theenumi )}
\makeatletter\def\p@enumii{\p@enumi \theenumi )}\makeatother
\setcounter{enumi}{2}
\item {} 
NOOPT\_1DALLTOALL or NOOPT\_2DALLTOALL:

\end{enumerate}
\begin{quote}

Cell size in each dimension divided by grid spacing should be greater than \sphinxcode{\sphinxupquote{pme\_nsplie}}.
\end{quote}
\begin{enumerate}
\def\theenumi{\arabic{enumi}}
\def\labelenumi{\theenumi )}
\makeatletter\def\p@enumii{\p@enumi \theenumi )}\makeatother
\setcounter{enumi}{3}
\item {} 
OPT\_1DALLTOALL or OPT\_2DALLTOALL:

\end{enumerate}
\begin{quote}

\sphinxcode{\sphinxupquote{pme\_ngrid\_{[}x,y,z{]}}} divided by \sphinxcode{\sphinxupquote{domain\_{[}x,y,z{]}}} should be greater than \(\times\) \sphinxcode{\sphinxupquote{pme\_nspline}}
\end{quote}


\section{Lookup table}
\label{\detokenize{05_Energy:lookup-table}}
The following keywords are relevant if CHARMM or AMBER force field is used.
For a linearly-interpolating lookup table, table points are assigned
at the unit interval of \(\text{cut-off}^2/r^2\)
and energy/gradients are evaluated as a function of \(b_2(G_2)\) \sphinxcite{99_References:jung-2013}.
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}F(r^2) \approx F_{\text{tab}}(L)+t(F_{\text{tab}}(L+1)-F_{\text{tab}}(L))\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}L=\text{INT}(\text{Density} \times r_v^2/r^2)\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

and
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}t=\text{Density} \times r_v^2/r^2-L\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

Linear interpolation is used if “Electrostatic=PME”.

Density is the number of points per a unit interval. Lookup table using cubic interpolation is different from that of linear interpolation. In the case of cubic interpolation, monotonic cubic Hermite polynomial  interpolation is used to impose the monotonicity of the energy value. Energy/gradients are evaluated as a function of \(r^2\) \sphinxcite{99_References:nilsson-2009} using four basis functions for the cubic Hermite spline : \(h_{00}(t)\), \(h_{10}(t)\), \(h_{01}(t)\), \(h_{11}(t)\)
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}F(r^2) & \approx F_{\text{tab}}(L-1)h_{00}(t)+\frac{F_{\text{tab}}(L-2) + F_{\text{tab}}(L-1)}{2} h_{10} \\ & + F_{\text{tab}}(L)h_{10}(t) + \frac{F_{\text{tab}}(L-1)+F_{\text{tab}}(L)}{2} h_{11}(t)\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}L=\text{INT}(\text{Density} \times r^2)\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

and
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}t=\text{Density} \times r^2-L\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

Cubic iterpolation is used if “Electrostatic=Cutoff”.


\section{Generalized Born/Solvent-Accessible Surface-Area model}
\label{\detokenize{05_Energy:generalized-born-solvent-accessible-surface-area-model}}
Implicit solvent model is useful to reduce computational cost
for the simulations of biomolecules \sphinxcite{99_References:mori-2016}. The GB/SA (Generalized
Born/Solvent accessible surface area) model is one of the popular
implicit solvent models, where the electrostatic contribution
to the solvation free energy (\(\Delta {G_{{\rm{elec}}}}\))
is computed with the GB theory \sphinxcite{99_References:still-1990}, and the non-polar contribution (\(\Delta {G_{{\rm{np}}}}\))
is calculated from the solvent accessible surface area \sphinxcite{99_References:eisenberg-1986}.
In the GB theory, solvent molecules
surrounding the solute are approximated as a continuum that has
the dielectric constant of \textasciitilde{}80. To date, various GB models have
been developed. In GENESIS, the OBC model \sphinxcite{99_References:onufriev-2004} and LCPO method \sphinxcite{99_References:weiser-1999} are
available in the calculations of the GB and SA energy terms, respectively.
Note that the GB/SA model is implemented in \sphinxstylestrong{ATDYN} but NOT \sphinxstylestrong{SPDYN}.
The solvation free energy is incorporated into the molecular
mechanics potential energy function as an effective energy term,
namely, \(U = U_{{\rm{FF}}} + \Delta {G_{{\rm{elec}}}} + \Delta {G_{{\rm{np}}}}\).


\subsection{GB energy term}
\label{\detokenize{05_Energy:gb-energy-term}}
In the GB theory, the solvation free energy of solute is given by
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\Delta {G_{{\rm{elec}}}} =  - \frac{1}{2}\left\{ {\frac{1}{{{\varepsilon _{\rm{p}}}}} - \frac{{\exp ( - \kappa {f_{ij}})}}{{{\varepsilon _{\rm{w}}}}}} \right\}\sum\limits_{i,j} {\frac{{{q_i}{q_j}}}{{{f_{ij}}}}},\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(\varepsilon_{{\rm{p}}}\) and \(\varepsilon_{{\rm{w}}}\) are the dielectric constants of
solute and solvent, respectively, \(q_i\) and \(q_j\) are the partial charges
on the \sphinxstyleemphasis{i}-th and \sphinxstyleemphasis{j}-th atoms, respectively. \(\kappa\) is the inverse of Debye length.
\(f_{ij}\) is the effective distance between the \sphinxstyleemphasis{i}- and \sphinxstyleemphasis{j}-th atoms,
which depends on the degree of burial of the atoms, and is given by
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}{f_{ij}} = \sqrt {r_{ij}^2 + {R_i}{R_j}\exp \left( {\frac{{ - r_{ij}^2}}{{4{R_i}{R_j}}}} \right)}.\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

Here, \(r_{ij}\) is the actual distance between the \sphinxstyleemphasis{i}- and \sphinxstyleemphasis{j}-th atoms,
and \(R_i\) is the effective Born radius of the \sphinxstyleemphasis{i}-th atom,
which is typically estimated in the Coulomb field approximation by
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}{\frac{1}{R_i}} = {\frac{1}{{{\rho _i}}} - \frac{1}{{4\pi }}\int_{{\rm{solute}},{\rm{ }}r > {\rho _i}} {\frac{1}{{{r^4}}}} dV}.\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

\(\rho_i\) is the radius of the \sphinxstyleemphasis{i}-th atom (mostly set to the atom’s van der Waals radius),
and the integral is carried out over the volume inside the solute but outside the \sphinxstyleemphasis{i}-th atom.
In the case of an isolated ion, \(R_i\) is equal to its van der Waals radius.
On the other hand, if the atom is buried inside a solute, \(R_i\) becomes larger,
resulting in larger \(f_{ij}\).
In the OBC model, the effective Born radius is approximated as
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}{\frac{1}{R_i}} = {\frac{1}{{\tilde \rho _i}} - \frac{1}{{{\rho _i}}}\tanh ({\alpha}{\Psi _i} - {\beta}\Psi _i^2 + {\gamma}\Psi _i^3)},\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \({\tilde \rho _i}\) is defied as \(\rho_i - \rho_0\) (intrinsic offset), and
\(\Psi_{i}\) describes the dgree of burial of the solute atom,
which is calculated from the pairwise descreening
function: \({\Psi _i} = {{\tilde \rho _i}}\sum\limits_j {{H(r_{ij})}}\) \sphinxcite{99_References:schaefer-1990}.


\subsection{SA energy term}
\label{\detokenize{05_Energy:sa-energy-term}}
In general, the non-polar contribution to the solvation free energy is calculated by
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\Delta {G_{{\rm{np}}}} = \sum\limits_i {{\gamma _i}{A_i}},\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(\gamma\) is the surface tension coefficient, and \(A_i\) is the surface area of the \sphinxstyleemphasis{i}-th atom.
In the LCPO method, \(A_i\) is calculated from a linear combination of the overlaps between the
neighboring atoms, given by
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}{A_i} = {P_{1i}}4\pi R_i^2 + {P_{2i}}\sum\limits_{j = 1}^n {{A_{ij}}}  + {P_{3i}}\sum\limits_{j = 1}^n {\sum\limits_{k = 1}^m {{A_{jk}}} }  + {P_{4i}}\sum\limits_{j = 1}^n {\left[ {{A_{ij}}\sum\limits_{j = 1}^n {\sum\limits_{k = 1}^m {{A_{jk}}} } } \right]}.\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

\(P_{1-4}\) are the empirical parameters determined for each atom type,
\(R_i\) is the radius of the \sphinxstyleemphasis{i}-th atom + probe radius (typically 1.4 Ang),
and \(A_{ij}\) is the area of the \sphinxstyleemphasis{i}-th atom buried inside the \sphinxstyleemphasis{j}-th atom, given by
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}{A_{ij}} = 2\pi {R_i}\left( {{R_i} - \frac{{{r_{ij}}}}{2} - \frac{{R_i^2 - R_j^2}}{{2{r_{ij}}}}} \right)\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(r_{ij}\) is the distance between the \sphinxstyleemphasis{i}- and \sphinxstyleemphasis{j}-th atoms.


\bigskip\hrule\bigskip


\sphinxstylestrong{gbsa\_eps\_solvent} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 78.5}

Dielectric constant of solvent \(\varepsilon_{{\rm{w}}}\).
\end{quote}

\sphinxstylestrong{gbsa\_eps\_solute} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 1.0}

Dielectric constant of solute \(\varepsilon_{{\rm{p}}}\).
\end{quote}

\sphinxstylestrong{gbsa\_alpha} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 1.0}

The empirical parameter \(\alpha\) in the equation for the effective Born radius calculation.
“gbsa\_alpha=0.8” for OBC1, and “gbsa\_alpha=1.0” for OBC2.
\end{quote}

\sphinxstylestrong{gbsa\_beta} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.8}

The empirical parameter \(\beta\) in the equation for the effective Born radius calculation.
“gbsa\_beta=0.0” for OBC1, and “gbsa\_beta=0.8” for OBC2.
\end{quote}

\sphinxstylestrong{gbsa\_gamma} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 4.85}

The empirical parameter \(\gamma\) in the equation for the effective Born radius calculation.
“gbsa\_gamma=2.91” for OBC1, and “gbsa\_gamma=4.85” for OBC2.
\end{quote}

\sphinxstylestrong{gbsa\_salt\_cons} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.2} (unit : mol/L)

Concentration of the monovalent salt solution.
\end{quote}

\sphinxstylestrong{gbsa\_vdw\_offset} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.09} (unit : Angstrom)

Intrinsic offset \(\rho_0\) for the van der Waals radius.
\end{quote}

\sphinxstylestrong{gbsa\_surf\_tens} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.005} (unit : kcal/mol/Angstrom$^{\text{2}}$)

Surface tension coefficient \(\gamma\) in the SA energy term.
\end{quote}

\begin{sphinxadmonition}{note}{Note:}
Debye length is calculated by
\({\kappa ^{ - 1}} = \sqrt {{\varepsilon _0}{\varepsilon _w}{k_B}T/2{N_A}{e^2}I}\), where
\sphinxstyleemphasis{T} is automatically set to the target temperature specified in the \sphinxstylestrong{{[}DYNAMICS{]}} section.
In the case of the energy minimization, \sphinxstyleemphasis{T} = 298.15 K is used. In the T-REMD simulations with
the GB/SA model, each replica has an individual Debye length depending on the assigned temperature.
\end{sphinxadmonition}


\section{EEF1, IMM1, and IMIC implicit solvent models}
\label{\detokenize{05_Energy:eef1-imm1-and-imic-implicit-solvent-models}}
In the EEF1 implicit solvent model \sphinxcite{99_References:lazaridis-1999}, the effective energy \sphinxstyleemphasis{W} of a solute molecule is defined as the sum of
the molecular mechanics potential energy \(E_{\rm{MM}}\) and solvation free energy \(\Delta {G_{{\rm{solv}}}}\), given by
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}W = E_{\rm{MM}} + \Delta {G_{{\rm{solv}}}},\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where
\begin{quote}

\vspace{-5mm}
\begin{align*}\!\begin{aligned}
\Delta {G_{{\rm{solv}}}} = \sum\limits_i {\Delta {G_i^{{\rm{ref}}}}} - \sum\limits_i {\sum\limits_{j \neq i} {g_i(r_{ij})V_j}},\\
g_i(r_{ij}) = {\frac{\Delta {G_i^{{\rm{free}}}}}{2\pi \sqrt{\pi} \lambda_i r_{ij}^2}} \exp \left\{ - \left( {\frac{r_{ij} - R_i}{\lambda_i}} \right)^2 \right\}.\\
\end{aligned}\end{align*}
\vspace{-3mm}
\end{quote}

\(r_{ij}\) is the distance between atoms \sphinxstyleemphasis{i} and \sphinxstyleemphasis{j}, and \(V_{j}\) is the volume
of the \sphinxstyleemphasis{j}-th atom. The function \(g_{i}\) is the density of the solvation free energy of the \sphinxstyleemphasis{i}-th atom,
defined with the van der Waals radius \(R_{i}\) and thickness of the first hydration shell \(\lambda_{i}\).
\(\Delta {G_i^{{\rm{ref}}}}\) is the solvation free energy of the atom when it is fully exposed to solvent.
\(\Delta {G_i^{{\rm{free}}}}\) is similar to \(\Delta {G_i^{{\rm{ref}}}}\),
but is determined to satisfy the zero solvation energy of deeply buried atoms.

In the IMM1 implicit membrane \sphinxcite{99_References:lazaridis-2003} and IMIC implicit micelle models \sphinxcite{99_References:mori-2020},
\(\Delta {G_i^{{\rm{free}}}}\) as well as \(\Delta {G_i^{{\rm{ref}}}}\) are defined as a combination
of the solvation free energies of the \sphinxstyleemphasis{i}-th solute atom in water and cyclohexane:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\Delta {G_i^{{\rm{ref}}}} &= f_i \Delta {G_i^{{\rm{ref,water}}}} + (1 - f_i) \Delta {G_i^{{\rm{ref,cyclohexane}}}},\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \sphinxstyleemphasis{f} is a function that describes the transition between water and cyclohexane phases.

In the IMM1 model, \(f_i\) is given by the sigmoidal function:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}f(z_i^{\prime}) = {\frac{z_i^{\prime n}}{1 + z_i^{\prime n}}},\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(z_i^{\prime} = |z_i|/(T/2)\), \(z_i\) is the \sphinxstyleemphasis{z}-coordinate of the \sphinxstyleemphasis{i}-th atom,
and \sphinxstyleemphasis{T} is the membrane thickness. \sphinxstyleemphasis{n} controls the steepness of the membrane-water interface.
In the IMM1 model, the membrane is centered at \(z = 0\).

In the IMIC model, the following function is used for \(f_i\):
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}f(d_i) = {\frac{1}{2}} \left\{ {\rm{tanh}}(sd_i) + 1 \right\},\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(d_i\) is the depth of the solute atom \sphinxstyleemphasis{i} from the micelle surface, and
\sphinxstyleemphasis{s} controls the steepness of the micelle-water interface.
The shape of the micelle is defined using a super-ellipsoid function:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\left\{ \left( {\frac{|x|}{a}} \right)^{\frac{2}{m_2}}  + \left( {\frac{|y|}{b}} \right)^{\frac{2}{m_2}} \right\}^{\frac{m_2}{m_1}} + \left( {\frac{|z|}{c}} \right)^{\frac{2}{m_1}} = 1.\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

\sphinxstyleemphasis{a}, \sphinxstyleemphasis{b}, and \sphinxstyleemphasis{c} are the semi-axes of the super-ellipsoid, and
\(m_1\) and \(m_2\) determine the shape of the cross section in the super-ellipsoid.
In the case of \(m_1 = m_2 = 1\), the equation gives an ordinary ellipsoid.
If \(0 < m_1 < 1\) and \(m_2 = 1\), the cross section in a plane perpendicular to
the \sphinxstyleemphasis{XY}-plane is expanded, keeping the semi-axes at the given lengths,
and the shape also resembles a bicelle or nanodisc.
If \(m_1 = 1\) and \(0 < m_2 < 1\), the cross section in a plane parallel to the \sphinxstyleemphasis{XY}-plane is expanded.
The shape becomes close to rectangle as both \(m_1\) and \(m_2\) decrease.
Note that \(m < 0\) or \(m > 1\) is not allowed, because it produces a non-micelle-like shape resembling an octahedron.
In the IMIC model, the micelle is centered at the origin of the system \((x, y, z) = (0, 0, 0)\).

In the IMM1 and IMIC models, a distance-dependent dielectric constant is used for the electrostatic interactions.
The dielectric constant depends on the positions of interacting atoms with respect to the membrane/micelle surface, defined as
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\epsilon = r^{ p + (1 - p) {\sqrt {f_i f_j}}},\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \sphinxstyleemphasis{r} is the distance between the \sphinxstyleemphasis{i}-th and \sphinxstyleemphasis{j}-th atoms, and \sphinxstyleemphasis{p} is an empirical parameter to adjust strength
of the interactions (\sphinxstyleemphasis{p} = 0.85 for CHARMM19 and 0.91 for CHARMM36).
Far from the membrane/micelle surface, the dielectric constant \(\epsilon\) is close to \sphinxstyleemphasis{r}, corresponding to the EEF1 model,
while in the membrane/micelle center, it provides strengthened interactions.
The IMIC model is nearly equivalent to the IMM1 model when \sphinxstyleemphasis{a} and \(b \rightarrow \infty\)  and \sphinxstyleemphasis{c} is half membrane thickness.

In the control file of GENESIS, the following parameters are specified,
and the other parameters such as \sphinxstyleemphasis{V}, \(\Delta {G_i^{{\rm{ref}}}}\),
\(\Delta {G_i^{{\rm{free}}}}\), and \(\lambda\) in the above equations
are read from the \sphinxstylestrong{eef1file}, which is set in the \sphinxstylestrong{{[}INPUT{]}} section (see {\hyperref[\detokenize{03_Input:input}]{\sphinxcrossref{\DUrole{std,std-ref}{Input section}}}}).
\sphinxstyleemphasis{R} is read from the \sphinxstylestrong{parfile}.


\bigskip\hrule\bigskip


\sphinxstylestrong{imm1\_memb\_thick} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 27.0} (unit : Angstrom)

Membrane thickness \sphinxstyleemphasis{T} in IMM1
\end{quote}

\sphinxstylestrong{imm1\_exponent\_n} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 10}

Steepness parameter \sphinxstyleemphasis{n} in IMM1
\end{quote}

\sphinxstylestrong{imm1\_factor\_a} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.91}

Adjustable empirical parameter \sphinxstyleemphasis{p} in IMM1 and IMIC.
\sphinxstyleemphasis{p} = 0.85 and 0.91 are recommended for CHARMM19 and CHARMM36, respectively.
\end{quote}

\sphinxstylestrong{imm1\_make\_pore}
\begin{quote}

\sphinxstylestrong{Default : NO}

Use IMM1-pore model \sphinxcite{99_References:lazaridis-2005}
\end{quote}

\sphinxstylestrong{imm1\_pore\_radius} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 5.0} (unit : Angstrom)

Aqueous pore radius in the IMM1-pore model
\end{quote}

\sphinxstylestrong{imic\_axis\_a} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 18.0} (unit : Angstrom)

Semi-axis \sphinxstyleemphasis{a} of the super-ellipsoid in IMIC
\end{quote}

\sphinxstylestrong{imic\_axis\_b} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 18.0} (unit : Angstrom)

Semi-axis \sphinxstyleemphasis{b} of the super-ellipsoid in IMIC
\end{quote}

\sphinxstylestrong{imic\_axis\_c} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 18.0} (unit : Angstrom)

Semi-axis \sphinxstyleemphasis{c} of the super-ellipsoid in IMIC
\end{quote}

\sphinxstylestrong{imic\_exponent\_m1} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 1.0}

Expansion parameter \(m_1\) in IMIC
\end{quote}

\sphinxstylestrong{imic\_exponent\_m2} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 1.0}

Expansion parameter \(m_2\) in IMIC
\end{quote}

\sphinxstylestrong{imic\_steepness} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.5}

Steepness parameter \sphinxstyleemphasis{s} in IMIC
\end{quote}


\section{Residue-level coarse-grained models}
\label{\detokenize{05_Energy:residue-level-coarse-grained-models}}
\sphinxstylestrong{Note:} here we only briefly describe the potential energy functions and
corresponding options of the available CG models in
GENESIS \sphinxcite{99_References:tan-ploscb2022}.  To prepare the topology and coordinate files,
please make use of the “GENESIS-CG-tool” \sphinxcite{99_References:tan-ploscb2022}, which is
included in GENESIS as a submodule and can be found from
“src/analysis/cg\_tool”.  It is also deposited as a Github repository, with
which a wiki page of the GENESIS-CG-tool can be found:
\sphinxurl{https://github.com/noinil/genesis\_cg\_tool/wiki}


\subsection{AICG2+ protein model}
\label{\detokenize{05_Energy:aicg2-protein-model}}
In the AICG2+ model for proteins, the energy function is given by: \sphinxcite{99_References:li-pnas2014}
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}E_{AICG2+}(\mathbf{r}) & = \sum_{\mathrm{bond}} K_b\left(b - b_{0}\right)^2 + V_{loc}^{flp} \\
     & + \sum_{\mathrm{1,3\ pair}} \varepsilon_{1,3} \exp\left(-\frac{(d-d_0)^2}{2w_{1,3}^2}\right) + \sum_{\mathrm{dihedral}} \varepsilon_{\phi}\exp \left(-\frac{(\phi - \phi_0)^2}{2w_{\phi}^2}\right) \\
     & + \sum_{\mathrm{native\ contact}} \varepsilon_{Go} \left[ 5\left(\frac{r_0}{r}\right)^{12} - 6 \left(\frac{r_0}{r}\right)^{10} \right] \\
     & + \sum_{\mathrm{non-native\ contact}} \varepsilon_{exv} \left[ \left(\frac{\sigma}{r}\right)^{12} - \frac{1}{2^{12}} \right].\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(K_{b}\), \(\varepsilon_{1,3}\), \(\varepsilon_{\phi}\),
\(\varepsilon_{Go}\), and \(\varepsilon_{exv}\) are the force constants
of the bond, 1,3 (the next neighbor) distance, dihedral angle, native contact,
and non-native contact terms, respectively; and \(b_{0}\), \(d_{0}\),
\(\phi_0\), and \(r_0\) are the native values of the corresponding
terms.  \(\sigma\) in the non-native contact term is the residue-type
dependent excluded volume radius. \(w_{1,3}\) and \(w_{\phi}\) are teh
widths of the Gaussian-type local potentials.  The statistical local potential
\(V_{loc}^{flp}\) is used to describe the intrinsical flexibility of
peptides and includes the following terms: \sphinxcite{99_References:terakawa-bj2011}
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}V_{loc}^{flp} = \sum_{\mathrm{angle}} -k_B T \ln \frac{P(\theta)}{\sin(\theta)}  + \sum_{\mathrm{dihedral}} - k_B T \ln P(\phi)\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(k_B\) is the Boltzmann constant, \(T\) is the temperature,
and \(P(\theta)\) and \(P(\phi)\) are the probability distributions of
the angles and dihedral angles, respectively.


\subsection{3SPN.2C DNA model}
\label{\detokenize{05_Energy:spn-2c-dna-model}}
Among the series of the 3SPN DNA models developed by de Pablo’s group
\sphinxcite{99_References:freeman-2011}\sphinxcite{99_References:hinckley-2013}\sphinxcite{99_References:freeman-2014}, 3SPN.2C is the
one for reproducing sequence-dependent curvature of double-stranded DNA
(dsDNA). \sphinxcite{99_References:freeman-2014} In this model, each nucleotide is represented by
three CG particles, P (phosphate), S (sugar), and B (base), respectively.  The
potential energy is given by:\sphinxcite{99_References:freeman-2014}
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}E_{3SPN.2C}(\mathbf{r}) & = \sum_{\mathrm{bond}} k_b (r - r_{0})^2 + 100 k_b (r - r_{0})^4 \\
    & + \sum_{\mathrm{angle}} k_\theta (\theta - \theta_{0})^2 \\
    & + \sum_{\mathrm{backbone\ dihedral}} -k_{\phi, Gaussian} \exp\big( \frac{-(\phi - \phi_0)^2}{2\sigma_{\phi}^2} \big) + \sum_{\mathrm{dihedral}} k_{\phi, periodic} \big[ 1+\cos(\phi - \phi_0) \big] \\
    & + \sum_{\mathrm{bstk}} U_m^{rep}(\epsilon_{bs}, \alpha_{bs}, r) + f(K_{bs}, \Delta\theta_{bs}) U_m^{attr} (\epsilon_{bs}, \alpha_{bs}, r) \\
    & + \sum_{\mathrm{bp}} U_m^{rep}(\epsilon_{bp}, \alpha_{bp}, r) + \frac{1}{2} \big( 1+\cos(\Delta \phi_1) \big) f(K_{bp}, \Delta\theta_{1}) f(K_{bp}, \Delta\theta_{2}) U_m^{attr} (\epsilon_{bp}, \alpha_{bp}, r) \\
    & + \sum_{\mathrm{cstk}} f(K_{BP}, \Delta\theta_{3}) f(K_{cs}, \Delta\theta_{cs}) U_m^{attr} (\epsilon_{cs}, \alpha_{cs}, r) \\
    & + \sum_{exv} \epsilon_r\bigg[ \Big(\frac{\sigma}{r} \Big)^{12} - 2 \Big(\frac{\sigma}{r} \Big)^6 \bigg]+ \epsilon_r \\
    & + \sum_{ele} \frac{q_i q_j e^{-r/\lambda_D}}{4\pi \epsilon_0 \epsilon(T, C) r}.\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where the first four lines are the local terms, for bond, angle, dihedral angle,
and the base-stacking between neighboring bases, respectively.  The non-local
terms include the base-pairing, cross-stacking, excluded volume, and
electrostatic interactions.  \(U_m^{rep}\) and \(U_m^{attr}\) are the
splited Morse-type potentials to describe the repulsive and attractive
interactions between DNA bases, while the \(f(K, \Delta\theta)\) are
bell-shaped modulating functions to smooth out the potentials.  The excluded
volume interactions are considered with a cutoff at \(\sigma\).  The
electrostatic interactions are modeled with the Debye-Hückel theory.  For more
details, please refer to reference \sphinxcite{99_References:freeman-2014}.


\subsection{Protein-DNA interaction models}
\label{\detokenize{05_Energy:protein-dna-interaction-models}}
We usually consider excluded volume and electrostatic terms as the
sequence-non-specific interactions between protein and DNA.  The excluded volume
term has the same form as the non-native contact in the AICG2+ model, and the
electrostatic term uses the Debye-Hückel potential as in the 3SPN.2C model. As
for the protein-DNA sequence-specific recognition, we used the PWMcos model for
the protein-DNA base interactions: \sphinxcite{99_References:tan-pwmcos}
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}E_{PWMcos}(\mathbf{r}) & = \sum_{i} \sum_{j} \sum_{m}\left[ V_{m,j}(b_i, \mathbf{r}) +  V_{m',j}(b_i, \mathbf{r}) \right],\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(i\) is the index of DNA base, \(j\) is the index of amino acid
residue (\(C_\alpha\)) that is forming contact with DNA in the reference
(native) complex structure, and \(m\) is the index of native contacts
between protein and DNA.  Each amino acid residue can form multiple native
contacts with DNA bases.  We consider contributions from both bases (indices
\(m\) and \(m'\)) in every base-pair.

The sequence dependent potential \(V_{m, j} (b_i, \mathbf{r})\) is defined as:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}V_{m, j} (b_i, \mathbf{r}) = \varepsilon_{PWM}(b_i) \cdot U_{mj} (i, \mathbf{r}),\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(N_m\) is the total number of contacts formed with the base pair
\(m-m'\), and \(\varepsilon_{PWM} (b_i)\) is based on the
position-weight-matrix and dependent on the base type of the \(i\)-th base:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\varepsilon_{PWM} = \gamma \left[  \frac{-k_B T}{N_m} \left( \log P_m(b) - \frac{1}{4} \sum_{b\in \{A, C, G, T\}} \log P_m(b) \right) + \varepsilon'  \right]\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(\gamma\) and \(\varepsilon'\) are two hyperparameters, which can
be calibrated by comparing simulated quantities with experimental results.

The second term in the formula of \(V_{m, j} (b_i, \mathbf{r})\) is a
modulating function:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}U_{mj}(i, \mathbf{r}) = f_{mj}(r)g_{1,mj} (\theta_1) g_{2, mj} (\theta_2) g_{3,mj} (\theta_{3})\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(r\) is the distance between the \(j\)-th \(C_\alpha\) and
the \(i\)-th DNA base.  \(\theta_1\), \(\theta_2\), and
\(\theta_3\) are the angle of sugar-base-\(C_\alpha\), the one between
vectors \(m-1\)-base-\(m+1\)-base and base-\(C_\alpha\), and the one
between vectors \(j-1\)-\(C_\alpha\)-\(j+1\)-\(C_\alpha\) and
base-\(C_\alpha\).  \(f\) is a Gaussian centered at the native value of
\(r\), and \(g\) is a bell-shaped function centered the native values of
the \(\theta\) s.  For more details, please refer to reference
\sphinxcite{99_References:tan-pwmcos}.

A similar potential is useful in the modeling of hydrogen-bonds formed between
proteins and DNA backbone phosphate groups, which has been successfully applied
in the studies of nucleosome dynamics \sphinxcite{99_References:niina-nuc2017}\sphinxcite{99_References:brandani-nuc2018}. We
also provide this model in GENESIS.


\subsection{HPS/KH IDR models}
\label{\detokenize{05_Energy:hps-kh-idr-models}}
GENESIS provides two models for intrinsically disordered proteins, namely the
HPS and KH IDR models \sphinxcite{99_References:dignon-plos2018}.  These two models share the same
potential function formulas and are different in the parameters.  The local
interactions include the bond terms.  The nonlocal interactions include the
electrostatic term (Debye-Hückel form, the same as described in the 3SPN.2C
sectionAshbaugh-Hatch type ) and the hydrophobicity-scale (HPS) term:
\sphinxcite{99_References:dignon-plos2018}
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\Phi(r) = \begin{cases} \Phi_{LJ} + (1+\lambda)\epsilon, & \mathrm{if}\ r \le 2^{1/6}\sigma \\ \lambda \Phi_{LJ}, & \mathrm{otherwise} \end{cases}\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(\lambda\) is the hydrophobicity value and \(\Phi_{LJ}\) is the standard Lennard-Jones potential:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\Phi_{LJ} =  4\epsilon \left[  \left(\frac{\sigma}{r}\right)^{12} - \left(\frac{\sigma}{r}\right)^6    \right].\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

For more details such as the parameter values in the HPS and KH models, please refer to reference \sphinxcite{99_References:dignon-plos2018}.

Notably, user can easily change the hydrophobicity parameters in the topology
files (see more descriptions in the wiki-page of the GENESIS-CG-tool).


\subsection{Structure and context-based RNA model}
\label{\detokenize{05_Energy:structure-and-context-based-rna-model}}
GENESIS also provides a structure and context based RNA model \sphinxcite{99_References:hori-jctc2012}, which has the potential functions:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}E_{RNA}(\mathbf{r}) & = \sum_{\mathrm{bond}} K_b\left(b - b_{0}\right)^2 + V_{loc}^{flp} \\
     & + \sum_{\mathrm{angle}} k_a \left(\theta - \theta_0\right)^2 \\
     & + \sum_{\mathrm{dihedral}} k_{\phi} \big[ 1-\cos(\phi - \phi_0) \big] + \frac{1}{2}k_{\phi} \big[ 1-\cos\left(3(\phi - \phi_0)\right) \big] \\
     & + \sum_{\mathrm{native\ contact}} \varepsilon_{Go} \left[ 5\left(\frac{r_0}{r}\right)^{12} - 6 \left(\frac{r_0}{r}\right)^{10} \right] \\
     & + \sum_{\mathrm{non-native\ contact}} \varepsilon_{exv} \left[ \left(\frac{\sigma}{r}\right)^{12} - \frac{1}{2^{12}} \right] \\
     & + \sum_{ele} \frac{q_i q_j e^{-r/\lambda_D}}{4\pi \epsilon_0 \epsilon(T, C) r}.\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where the first three lines are for the local terms (bond, angle, and dihedral
angles), the forth term is the structure-based Go-like potential for native
contacts, the fifth term is for the non-native contacts, and the sixth term is
the Debye-Hückel type electrostatic interaction.


\bigskip\hrule\bigskip


\sphinxstylestrong{cg\_cutoffdist\_ele} \sphinxstyleemphasis{Real} (unit: Angstrom)
\begin{quote}

\sphinxstylestrong{Default : 52.0}

Cutoff for Debye-Hückel type electrostatic interactions.
\end{quote}

\sphinxstylestrong{cg\_cutoffdist\_126} \sphinxstyleemphasis{Real} (unit: Angstrom)
\begin{quote}

\sphinxstylestrong{Default : 39.0}

Cutoff for 12-6 type Lennard-Jones interactions.
\end{quote}

\sphinxstylestrong{cg\_cutoffdist\_DNAbp} \sphinxstyleemphasis{Real} (unit: Angstrom)
\begin{quote}

\sphinxstylestrong{Default : 18.0}

Cutoff for DNA base-pairing interactions.
\end{quote}

\sphinxstylestrong{cg\_pairlistdist\_ele} \sphinxstyleemphasis{Real} (unit: Angstrom)
\begin{quote}

\sphinxstylestrong{Default : 57.0}

Distance for the pair list of Debye-Hückel type electrostatic interactions.
\end{quote}

\sphinxstylestrong{cg\_pairlistdist\_126} \sphinxstyleemphasis{Real} (unit: Angstrom)
\begin{quote}

\sphinxstylestrong{Default : 44.0}

Distance for the pair list of 12-6 type Lennard-Jones interactions.
\end{quote}

\sphinxstylestrong{cg\_pairlistdist\_PWMcos} \sphinxstyleemphasis{Real} (unit: Angstrom)
\begin{quote}

\sphinxstylestrong{Default : 23.0}

Distance for the pair list of PWMcos type protein-DNA interactions.
\end{quote}

\sphinxstylestrong{cg\_pairlistdist\_DNAbp} \sphinxstyleemphasis{Real} (unit: Angstrom)
\begin{quote}

\sphinxstylestrong{Default : 23.0}

Distance for the pair list of DNA base-pairing interactions.
\end{quote}

\sphinxstylestrong{cg\_pairlistdist\_exv} \sphinxstyleemphasis{Real} (unit: Angstrom)
\begin{quote}

\sphinxstylestrong{Default : 15.0}

Distance for the pair list of all the excluded volume interactions.
\end{quote}

\sphinxstylestrong{cg\_sol\_temperature} \sphinxstyleemphasis{Real} (unit: K)
\begin{quote}

\sphinxstylestrong{Default : 300.0}

Solution temperature used for the calculation of Debye length and dielectric
constant. \sphinxstylestrong{Note:} this option is only used when “temperature” in the “{[}
ENSEMBLE {]}” section is set to 0, otherwise the later is used.
\end{quote}

\sphinxstylestrong{cg\_sol\_ionic\_strength} \sphinxstyleemphasis{Real} (unit: M)
\begin{quote}

\sphinxstylestrong{Default : 0.15}

Ionic strength used for the calculation of Debye length and dielectric constant.
\end{quote}

\sphinxstylestrong{cg\_pro\_DNA\_ele\_scale\_Q} \sphinxstyleemphasis{Real} (unit: \(e^{-}\))
\begin{quote}

\sphinxstylestrong{Default : -1.0}

Charge of the CG phosphate in the 3SPN.2C DNA is set to this value when
protein-DNA electrostatic interactions are calculated.  \sphinxstylestrong{Note:} this option
does not change the DNA-DNA interaction, in which the phosphate charge is
\(-0.6e^{-}\).
\end{quote}

\sphinxstylestrong{cg\_exv\_sigma\_scaling} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 1.0}

The radii of CG particles for the excluded volume interactions (including
non-native contacts) are scaled by this value.  \sphinxstylestrong{Note:} this option does not
affect the DNA-DNA excluded volume interactions.
\end{quote}

\sphinxstylestrong{cg\_IDR\_HPS\_epsilon} \sphinxstyleemphasis{Real} (unit: kcal/mol)
\begin{quote}

\sphinxstylestrong{Default : 0.2}

The force constant \(\epsilon\) in the HPS-IDR model.
\end{quote}

\sphinxstylestrong{cg\_infinite\_DNA} \sphinxstyleemphasis{Boolean}
\begin{quote}

\sphinxstylestrong{Default : NO}

Use the infinite DNA model or not.
\end{quote}


\section{Examples}
\label{\detokenize{05_Energy:examples}}
Simulation with the CHARMM36 force field in the periodic boundary condition:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}ENERGY\PYG{o}{]}
\PYG{n+nv}{forcefield}       \PYG{o}{=} CHARMM  \PYG{c+c1}{\PYGZsh{} CHARMM force field}
\PYG{n+nv}{electrostatic}    \PYG{o}{=} PME     \PYG{c+c1}{\PYGZsh{} use Particle mesh Ewald method}
\PYG{n+nv}{switchdist}       \PYG{o}{=} \PYG{l+m}{10}.0    \PYG{c+c1}{\PYGZsh{} switch distance}
\PYG{n+nv}{cutoffdist}       \PYG{o}{=} \PYG{l+m}{12}.0    \PYG{c+c1}{\PYGZsh{} cutoff distance}
\PYG{n+nv}{pairlistdist}     \PYG{o}{=} \PYG{l+m}{13}.5    \PYG{c+c1}{\PYGZsh{} pair\PYGZhy{}list distance}
\PYG{n+nv}{vdw\PYGZus{}force\PYGZus{}switch} \PYG{o}{=} YES     \PYG{c+c1}{\PYGZsh{} force switch option for van der Waals}
\PYG{n+nv}{pme\PYGZus{}nspline}      \PYG{o}{=} \PYG{l+m}{4}       \PYG{c+c1}{\PYGZsh{} order of B\PYGZhy{}spline in [PME]}
\PYG{n+nv}{pme\PYGZus{}max\PYGZus{}spacing}  \PYG{o}{=} \PYG{l+m}{1}.2     \PYG{c+c1}{\PYGZsh{} max grid spacing allowed}
\end{sphinxVerbatim}

Simulation with the AMBER force field in the periodic boundary condition:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}ENERGY\PYG{o}{]}
\PYG{n+nv}{forcefield}       \PYG{o}{=} AMBER   \PYG{c+c1}{\PYGZsh{} AMBER force field}
\PYG{n+nv}{electrostatic}    \PYG{o}{=} PME     \PYG{c+c1}{\PYGZsh{} use Particle mesh Ewald method}
\PYG{n+nv}{switchdist}       \PYG{o}{=} \PYG{l+m}{8}.0     \PYG{c+c1}{\PYGZsh{} switch distance}
\PYG{n+nv}{cutoffdist}       \PYG{o}{=} \PYG{l+m}{8}.0     \PYG{c+c1}{\PYGZsh{} cutoff distance}
\PYG{n+nv}{pairlistdist}     \PYG{o}{=} \PYG{l+m}{9}.5     \PYG{c+c1}{\PYGZsh{} pair\PYGZhy{}list distance}
\PYG{n+nv}{pme\PYGZus{}nspline}      \PYG{o}{=} \PYG{l+m}{4}       \PYG{c+c1}{\PYGZsh{} order of B\PYGZhy{}spline in [PME]}
\PYG{n+nv}{pme\PYGZus{}max\PYGZus{}spacing}  \PYG{o}{=} \PYG{l+m}{1}.2     \PYG{c+c1}{\PYGZsh{} max grid spacing allowed}
\end{sphinxVerbatim}

Recommended options in the case of energy minimization (see {\hyperref[\detokenize{07_Minimize:minimize}]{\sphinxcrossref{\DUrole{std,std-ref}{Minimize section}}}}) for the initial structure with the CHARMM36 force field:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}ENERGY\PYG{o}{]}
\PYG{n+nv}{forcefield}       \PYG{o}{=} CHARMM  \PYG{c+c1}{\PYGZsh{} CHARMM force field}
\PYG{n+nv}{electrostatic}    \PYG{o}{=} PME     \PYG{c+c1}{\PYGZsh{} use Particle mesh Ewald method}
\PYG{n+nv}{switchdist}       \PYG{o}{=} \PYG{l+m}{10}.0    \PYG{c+c1}{\PYGZsh{} switch distance}
\PYG{n+nv}{cutoffdist}       \PYG{o}{=} \PYG{l+m}{12}.0    \PYG{c+c1}{\PYGZsh{} cutoff distance}
\PYG{n+nv}{pairlistdist}     \PYG{o}{=} \PYG{l+m}{13}.5    \PYG{c+c1}{\PYGZsh{} pair\PYGZhy{}list distance}
\PYG{n+nv}{vdw\PYGZus{}force\PYGZus{}switch} \PYG{o}{=} YES     \PYG{c+c1}{\PYGZsh{} force switch option for van der Waals}
\PYG{n+nv}{pme\PYGZus{}nspline}      \PYG{o}{=} \PYG{l+m}{4}       \PYG{c+c1}{\PYGZsh{} order of B\PYGZhy{}spline in [PME]}
\PYG{n+nv}{pme\PYGZus{}max\PYGZus{}spacing}  \PYG{o}{=} \PYG{l+m}{1}.2     \PYG{c+c1}{\PYGZsh{} max grid spacing allowed}
\PYG{n+nv}{contact\PYGZus{}check}    \PYG{o}{=} YES     \PYG{c+c1}{\PYGZsh{} check atomic clash}
\PYG{n+nv}{nonb\PYGZus{}limiter}     \PYG{o}{=} YES     \PYG{c+c1}{\PYGZsh{} avoid failure due to atomic clash}
\PYG{n+nv}{minimum\PYGZus{}contact}  \PYG{o}{=} \PYG{l+m}{0}.5     \PYG{c+c1}{\PYGZsh{} definition of atomic clash distance}
\end{sphinxVerbatim}

Simulations with the GB/SA implicit solvent model:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}ENERGY\PYG{o}{]}
\PYG{n+nv}{forcefield}       \PYG{o}{=} CHARMM  \PYG{c+c1}{\PYGZsh{} CHARMM force field}
\PYG{n+nv}{electrostatic}    \PYG{o}{=} CUTOFF  \PYG{c+c1}{\PYGZsh{} use cutoff scheme}
\PYG{n+nv}{switchdist}       \PYG{o}{=} \PYG{l+m}{23}.0    \PYG{c+c1}{\PYGZsh{} switch distance}
\PYG{n+nv}{cutoffdist}       \PYG{o}{=} \PYG{l+m}{25}.0    \PYG{c+c1}{\PYGZsh{} cutoff distance}
\PYG{n+nv}{pairlistdist}     \PYG{o}{=} \PYG{l+m}{27}.0    \PYG{c+c1}{\PYGZsh{} pair\PYGZhy{}list distance}
\PYG{n+nv}{implicit\PYGZus{}solvent} \PYG{o}{=} GBSA    \PYG{c+c1}{\PYGZsh{} Turn on GBSA calculation}
\PYG{n+nv}{gbsa\PYGZus{}eps\PYGZus{}solvent} \PYG{o}{=} \PYG{l+m}{78}.5    \PYG{c+c1}{\PYGZsh{} solvent dielectric constant in GB}
\PYG{n+nv}{gbsa\PYGZus{}eps\PYGZus{}solute}  \PYG{o}{=} \PYG{l+m}{1}.0     \PYG{c+c1}{\PYGZsh{} solute dielectric constant in GB}
\PYG{n+nv}{gbsa\PYGZus{}salt\PYGZus{}cons}   \PYG{o}{=} \PYG{l+m}{0}.2     \PYG{c+c1}{\PYGZsh{} salt concentration (mol/L) in GB}
\PYG{n+nv}{gbsa\PYGZus{}surf\PYGZus{}tens}   \PYG{o}{=} \PYG{l+m}{0}.005   \PYG{c+c1}{\PYGZsh{} surface tension (kcal/mol/A\PYGZca{}2) in SA}
\end{sphinxVerbatim}

Simulations with the IMM1 implicit membrane model:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}ENERGY\PYG{o}{]}
\PYG{n+nv}{forcefield}       \PYG{o}{=} CHARMM  \PYG{c+c1}{\PYGZsh{} CHARMM force field}
\PYG{n+nv}{electrostatic}    \PYG{o}{=} CUTOFF  \PYG{c+c1}{\PYGZsh{} use cutoff scheme}
\PYG{n+nv}{switchdist}       \PYG{o}{=} \PYG{l+m}{16}.0    \PYG{c+c1}{\PYGZsh{} switch distance}
\PYG{n+nv}{cutoffdist}       \PYG{o}{=} \PYG{l+m}{18}.0    \PYG{c+c1}{\PYGZsh{} cutoff distance}
\PYG{n+nv}{pairlistdist}     \PYG{o}{=} \PYG{l+m}{20}.0    \PYG{c+c1}{\PYGZsh{} pair\PYGZhy{}list distance}
\PYG{n+nv}{implicit\PYGZus{}solvent} \PYG{o}{=} IMM1    \PYG{c+c1}{\PYGZsh{} Turn on IMM1 calculation}
\PYG{n+nv}{imm1\PYGZus{}memb\PYGZus{}thick}  \PYG{o}{=} \PYG{l+m}{27}.0    \PYG{c+c1}{\PYGZsh{} membrane thickness in IMM1}
\end{sphinxVerbatim}

Simulations with the residue-level CG models

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}ENERGY\PYG{o}{]}
\PYG{n+nv}{forcefield}             \PYG{o}{=} RESIDCG
\PYG{n+nv}{electrostatic}          \PYG{o}{=} CUTOFF
\PYG{n+nv}{cg\PYGZus{}cutoffdist\PYGZus{}ele}      \PYG{o}{=} \PYG{l+m}{52}.0
\PYG{n+nv}{cg\PYGZus{}cutoffdist\PYGZus{}126}      \PYG{o}{=} \PYG{l+m}{39}.0
\PYG{n+nv}{cg\PYGZus{}cutoffdist\PYGZus{}DNAbp}    \PYG{o}{=} \PYG{l+m}{18}.0
\PYG{n+nv}{cg\PYGZus{}pairlistdist\PYGZus{}ele}    \PYG{o}{=} \PYG{l+m}{57}.0
\PYG{n+nv}{cg\PYGZus{}pairlistdist\PYGZus{}126}    \PYG{o}{=} \PYG{l+m}{44}.0
\PYG{n+nv}{cg\PYGZus{}pairlistdist\PYGZus{}PWMcos} \PYG{o}{=} \PYG{l+m}{23}.0
\PYG{n+nv}{cg\PYGZus{}pairlistdist\PYGZus{}DNAbp}  \PYG{o}{=} \PYG{l+m}{23}.0
\PYG{n+nv}{cg\PYGZus{}pairlistdist\PYGZus{}exv}    \PYG{o}{=} \PYG{l+m}{15}.0
\PYG{n+nv}{cg\PYGZus{}sol\PYGZus{}ionic\PYGZus{}strength}  \PYG{o}{=} \PYG{l+m}{0}.15
\PYG{n+nv}{cg\PYGZus{}pro\PYGZus{}DNA\PYGZus{}ele\PYGZus{}scale\PYGZus{}Q} \PYG{o}{=} \PYGZhy{}1.0
\PYG{n+nv}{cg\PYGZus{}IDR\PYGZus{}HPS\PYGZus{}epsilon}     \PYG{o}{=} \PYG{l+m}{0}.2
\end{sphinxVerbatim}


\chapter{Dynamics section}
\label{\detokenize{06_Dynamics:dynamics-section}}\label{\detokenize{06_Dynamics:dynamics}}\label{\detokenize{06_Dynamics::doc}}

\section{Molecular dynamics simulations}
\label{\detokenize{06_Dynamics:molecular-dynamics-simulations}}
In MD simulations, Newton’s equation of motion (\sphinxstyleemphasis{F} = \sphinxstyleemphasis{ma}) is integrated numerically,
where the force \sphinxstyleemphasis{F} is derived from the first derivative of the potential energy function
with respect to the atomic position.
To date, various integrators have been proposed.
In the leap-frog algorithm, velocities are updated with
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}{{\bf{v}}_i}(t + \frac{{\Delta t}}{2}) = {{\bf{v}}_i}(t - \frac{{\Delta t}}{2}) + \frac{{\Delta t}}{{{m_i}}}{{\bf{F}}_i}(t),\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

and coordinates are updated with
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}{{\bf{r}}_i}(t + \Delta t) = {{\bf{r}}_i}(t) + \Delta t{{\bf{v}}_i}(t + \frac{{\Delta t}}{2}).\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

In the velocity Verlet algorithm, coordinates and velocities are obtained at the same time.
The velocities are updated with
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}{{\bf{v}}_i}\left( t + \Delta t \right) = {{\bf{v}}_i}\left( t \right) + \frac{{\Delta {t}}}{{{2m_i}}}
\left( {{\bf{F}}_i}\left( t \right) + {{\bf{F}}_i}\left( t + {\Delta t} \right) \right).\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

and the coordinates are updated with
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}{{\bf{r}}_i}\left( t + \Delta t \right) = {{\bf{r}}_i}\left( t \right) + \Delta t{{\bf{v}}_i}\left( t \right) + \frac{{\Delta {t^2}}}{{{2m_i}}}{{\bf{F}}_i}\left( t \right).\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

These velocity/coordinate update are performed consecutively using the following procedure:
\begin{quote}

\vspace{-5mm}
\begin{align*}\!\begin{aligned}
{{\bf{v}}_i}\left( t + \frac{\Delta t}{2} \right) &= {{\bf{v}}_i}(t) &+& \frac{\Delta t}{2{m_i}}{{\bf{F}}_i}(t)\\
{{\bf{r}}_i}\left( t + \Delta t \right) &= {{\bf{r}}_i}\left( t \right) &+& \Delta t{{\bf{v}}_i}\left( t + \frac{\Delta t}{2} \right)\\
{{\bf{v}}_i}\left( t + \Delta t \right) &= {{\bf{v}}_i}\left( t + \frac{\Delta t}{2} \right) &+& \frac{\Delta t}{2{m_i}} {{\bf{F}}_i}\left( t + \Delta t \right)\\
\end{aligned}\end{align*}
\vspace{-3mm}
\end{quote}

In the case of multiple time step integration with r-RESPA, the force is split into slow and fast motion forces.
Slow motion force is less frequently evaluated to increase the perofrmance while keeping the accuracy.

In \sphinxstylestrong{ATDYN}, both leap-frog and velocity Verlet integrators are available.
In \sphinxstylestrong{SPDYN}, velocity Verlet integrator and multiple time step integrator with veclotiy Verlet type are available.
The users must pay attention to the \sphinxstylestrong{{[}ENSEMBLE{]}} section as well,
because the algorithms that control the temperature and pressure are involved in the integrator.
For details, see {\hyperref[\detokenize{09_Ensemble:ensemble}]{\sphinxcrossref{\DUrole{std,std-ref}{Ensemble section}}}}.


\bigskip\hrule\bigskip


\sphinxstylestrong{integrator} \sphinxstyleemphasis{LEAP / VVER / VRES / VVER\_CG}
\begin{quote}

\sphinxstylestrong{Default : VVER}
\begin{itemize}
\item {} 
\sphinxstylestrong{LEAP}: leap-frog integrator (\sphinxstylestrong{ATDYN} only).

\item {} 
\sphinxstylestrong{VVER}: velocity Verlet integrator.

\item {} 
\sphinxstylestrong{VRES}: RESPA integrator (\sphinxstylestrong{SPDYN} only).

\item {} 
\sphinxstylestrong{VVER\_CG}: velocity Verlet integrator for the coarse-grained simulations (\sphinxstylestrong{ATDYN} and Langevin thermostat only).

\end{itemize}
\end{quote}

\sphinxstylestrong{timestep} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.001} (unit : ps)

Time step in the MD run.
In general, timestep can be extended to 2 fs or longer,
when the SHAKE, RATTLE, or SETTLE algorithms are employed.
(see {\hyperref[\detokenize{08_Constraints:constraints}]{\sphinxcrossref{\DUrole{std,std-ref}{Constraints section}}}}).
\end{quote}

\sphinxstylestrong{nsteps} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 100}

Total number of steps in one MD run.
If “timestep=0.001” and “nsteps=1000000” are specified,
the users can carry out 1-ns MD simulation.
\end{quote}

\sphinxstylestrong{eneout\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 10}

Output frequency for the energy data. The trajectories are written
in the log file every \sphinxstylestrong{eneout\_period} steps during the simulation.
For example, if “timestep=0.001” and “eneout\_period=1000” are specified,
the energy is written every 1 ps.
\end{quote}

\sphinxstylestrong{crdout\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 0}

Output frequency for the coordinates data. The trajectories are written
in the “dcdfile” specified in the \sphinxstylestrong{{[}OUTPUT{]}} section
every \sphinxstylestrong{crdout\_period} steps during the simulation.
\end{quote}

\sphinxstylestrong{velout\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 0}

Output frequency for the velocities data. The trajectories are written
in the “dcdvelfile” specified in the \sphinxstylestrong{{[}OUTPUT{]}} section
every \sphinxstylestrong{velout\_period} steps during the simulation.
\end{quote}

\sphinxstylestrong{rstout\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 0}

Output frequency for the restart file. The restart information is written
in the “rstfile” specified in the \sphinxstylestrong{{[}OUTPUT{]}} section
every \sphinxstylestrong{rstout\_period} steps during the simulation.
\end{quote}

\begin{sphinxadmonition}{note}{Note:}
In the REMD or RPATH simulations, the value must be a multiple of \sphinxstyleemphasis{exchange\_period} (REMD) or \sphinxstyleemphasis{rpath\_period} (RPATH).
\end{sphinxadmonition}

\sphinxstylestrong{stoptr\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 10}

Frequency of removing translational and rotational motions of the whole system.
Note that the rotational motion is not removed when the periodic boundary condition is employed.
When you use positional restraints or RMSD restraints in the simulation,
you may have to take care about removal of those motions.
In some cases, such restraints can generate translational or rotational momentum in the system.
If the momentum is frequently removed, the dynamics can be significantly disturbed.
\end{quote}

\sphinxstylestrong{nbupdate\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 10}

Update frequency of the non-bonded pairlist.
\end{quote}

\sphinxstylestrong{elec\_long\_period} \sphinxstyleemphasis{Integer} (\sphinxstylestrong{VRES} in \sphinxstylestrong{SPDYN} only)
\begin{quote}

\sphinxstylestrong{Default : 1}

Frequency of long-range interaction calculation.
\end{quote}

\sphinxstylestrong{thermostat\_period} \sphinxstyleemphasis{Integer} (\sphinxstylestrong{VRES} in \sphinxstylestrong{SPDYN} only)
\begin{quote}

\sphinxstylestrong{Default : 1}

Frequency of thermostat integration. It must be multiple of \sphinxstylestrong{elec\_long\_period}.
\end{quote}

\sphinxstylestrong{barostat\_period} \sphinxstyleemphasis{Integer} (\sphinxstylestrong{VRES} in \sphinxstylestrong{SPDYN} only)
\begin{quote}

\sphinxstylestrong{Default : 1}

Frequency of barostat integration. It must be multiple of \sphinxstylestrong{thermostat\_period}.
\end{quote}

\sphinxstylestrong{initial\_time} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.0} (unit : ps)

Initial time of the MD run. Basically, you do not need to specify a certain value.
This option is useful in the case of the restart MD run, because the initial time is reset to 0 ps.
\end{quote}

\sphinxstylestrong{iseed} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : automatically generated according to the current date and time}

Seed for the pseudo-random number generator.
This random number seed is used in the Langevin and Bussi thermostats (see \sphinxtitleref{ensemble}).
If \sphinxstylestrong{iseed} is not specified in the control file,
it is automatically generated according to the current date and time.
In the restart MD run, the random number seed is taken over from rstfile.
However, if \sphinxstylestrong{iseed} value is specified in the control file in the restart run,
it is alternatively used, and the seed in rstfile is neglected.
\end{quote}

\sphinxstylestrong{verbose} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

Turn on or off the verbose output of the log information.
For example, if “verbose=YES” is specified,
virial and pressure of the system are written in the log file
even in the case of the NVE or NVT ensemble.
\end{quote}


\section{Hydrogen mass repartitioning (HMR)}
\label{\detokenize{06_Dynamics:hydrogen-mass-repartitioning-hmr}}
In \sphinxstylestrong{GENESIS2.0.0beta}, we can increase the time step in NVT/NPT conditions by evaluating temperature
and pressure in more accurate ways than conventional schemes. If we want to increase the time step, however,
we should be careful because there could be constraint errors. To avoid the error, we recommend the user to
make use of the HMR scheme with a large time step. In HMR, the mass of hydrogen atoms are increased by two to four
fold whereas the bonded hevay becomes lighter to conserver the total mass.


\bigskip\hrule\bigskip


\sphinxstylestrong{hydrogen\_mr} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

Turn on or off the usage of HMR
\end{quote}

\sphinxstylestrong{hmr\_ratio} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 3.0}

Mass scaling factor of hydrogen atoms
\end{quote}

\sphinxstylestrong{hmr\_ratio\_xh1} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 3.0}

Mass scaling facto of hydrogen atoms in XH1 group. If it is not written, scaling factor is decided by \sphinxstylestrong{hmr\_ratio}.
\end{quote}

\sphinxstylestrong{hmr\_target} \sphinxstyleemphasis{All / Solute}
\begin{quote}

\sphinxstylestrong{Default : All} (only when \sphinxstylestrong{hydrogen\_mr} = \sphinxstyleemphasis{YES})

Target of HMR application. If \sphinxstylestrong{hmr\_target} is \sphinxstyleemphasis{Solute}, HMR is not applied to the water molecules.
\end{quote}


\section{Simulated annealing and heating}
\label{\detokenize{06_Dynamics:simulated-annealing-and-heating}}
In simulated annelaing or heating protocol, the following keywords
are additinally specified in the conventional MD simulation.
In the protocol used in \sphinxstylestrong{GENESIS}, the target temperature is changed linearly.
Note that the protocol is available only in the \sphinxstyleemphasis{LEAP} integrator.


\bigskip\hrule\bigskip


\sphinxstylestrong{annealing} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

Turn on or off the simulated annealing or heating protocol.
\end{quote}

\sphinxstylestrong{anneal\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 0}

The target temperature is changed every \sphinxstylestrong{anneal\_period} steps during the simulation.
\end{quote}

\sphinxstylestrong{dtemperature} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.0} (unit : Kelvin)

Magnitute of changes of the target temperature.
If \sphinxstylestrong{dtemperature} \textgreater{} 0, the temperature is increased by \sphinxstylestrong{dtemperature} every \sphinxstylestrong{anneal\_period} steps.
If \sphinxstylestrong{dtemperature} \textless{} 0, the temperature is decreased.
\end{quote}


\section{Targeted MD and Steered MD simulations}
\label{\detokenize{06_Dynamics:targeted-md-and-steered-md-simulations}}
In GENESIS, targeted MD (TMD) and steered MD (SMD) methods are available.
These methods are useful to guide a protein structure towards a target.
In SMD, restraint forces (or steering forces) are applied on the selected atoms,
where the RMSD with respect to the target is changed during the MD simulation.
The restraint force is calculated from the derivative of the RMSD restraint potential:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}U = \frac{1}{2}k\left(RMSD(t)-RMSD_{0}(t)\right)^{2}\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(RMSD(t)\) is the instantaneous RMSD of the current coordinates
from the target coordinates, and \(RMSD_{0}\) is the target RMSD value.
The target RMSD value is changed linearly from the initial to target RMSD values:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}RMSD_0(t) = RMSD_{\text{initial}}+\frac{t}{T}\left(RMSD_{\text{final}}-RMSD_{\text{initial}}\right)\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(T\) is the total MD simulation time. Targeted MD (TMD),
originally suggested by J. Schlitter et al. \sphinxcite{99_References:schlitter-1994},
is different from SMD in that force constants are changed during MD
simulations. If the users perform SMD, there is a possibility observing the
large difference between the instantaneous RMSD and target RMSD.
In TMD, force constants are given by
Lagrangian multipliers to overcome the energy barrier between the
instantaneous and target RMSDs. Therefore, the users could find
trajectories where RMSD is almost identical to the target RMSD at
each time. In \sphinxstylestrong{{[}SELECTION{]}} section, the users select atoms involved
in RMSD calculations for SMD or TMD. Users should specify either
\sphinxstyleemphasis{RMSD} or \sphinxstyleemphasis{RMSDMASS} (mass-weighted RMSD) in \sphinxstylestrong{{[}RESTRAINTS{]}} section to
run TMD or SMD. In SMD, force constants defined in \sphinxstylestrong{{[}RESTRAINTS{]}}
section are used, but force constants are automatically determined using
Lagrangian multipliers during simulation in TMD.


\bigskip\hrule\bigskip


\sphinxstylestrong{target\_md} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

Turn on or off the targeted MD simulation.
\end{quote}

\sphinxstylestrong{steered\_md} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

Turn on or off the steered MD simulation.
\end{quote}

\sphinxstylestrong{initial\_rmsd} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.0} (unit : Angstrom)

Initial value of the reference rmsd.
If not specified explicitly, it is calculated from the initial and referce structures.
\end{quote}

\sphinxstylestrong{final\_rmsd} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.0} (unit : Angstrom)

Final value of the reference rmsd.
\end{quote}

\begin{sphinxadmonition}{note}{Note:}
In the RMSD restraint, structure fitting scheme is specified in the \sphinxstylestrong{{[}FITTING{]}}
section (see {\hyperref[\detokenize{13_Fitting:fitting}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting section}}}}). Since the default behavior was significantly
changed in ver. 1.1.5 (no fitting applied on the default setting),
the users of 1.1.4 or before must pay special attention on the fitting scheme.
In versions of 1.1.4 or before, structure fitting is automatically
applied for the atoms concerning restraint potential.
\end{sphinxadmonition}


\section{Examples}
\label{\detokenize{06_Dynamics:examples}}
100-ps MD simulation with the velocity Verlet integrator with the timestep of 2 fs:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}DYNAMICS\PYG{o}{]}
\PYG{n+nv}{integrator}        \PYG{o}{=}   VVER  \PYG{c+c1}{\PYGZsh{} velocity Verlet}
\PYG{n+nv}{nsteps}            \PYG{o}{=}  \PYG{l+m}{50000}  \PYG{c+c1}{\PYGZsh{} number of MD steps (100ps)}
\PYG{n+nv}{timestep}          \PYG{o}{=}  \PYG{l+m}{0}.002  \PYG{c+c1}{\PYGZsh{} timestep (2fs)}
\PYG{n+nv}{eneout\PYGZus{}period}     \PYG{o}{=}    \PYG{l+m}{500}  \PYG{c+c1}{\PYGZsh{} energy output period (1ps)}
\PYG{n+nv}{crdout\PYGZus{}period}     \PYG{o}{=}    \PYG{l+m}{500}  \PYG{c+c1}{\PYGZsh{} coordinates output period (1ps)}
\PYG{n+nv}{rstout\PYGZus{}period}     \PYG{o}{=}  \PYG{l+m}{50000}  \PYG{c+c1}{\PYGZsh{} restart output period}
\PYG{n+nv}{nbupdate\PYGZus{}period}   \PYG{o}{=}     \PYG{l+m}{10}  \PYG{c+c1}{\PYGZsh{} nonbond pair list update period}
\end{sphinxVerbatim}

100-ps MD simulation with the RESPA integrator with the timestep of 2.5 fs:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}DYNAMICS\PYG{o}{]}
\PYG{n+nv}{integrator}        \PYG{o}{=}   VRES  \PYG{c+c1}{\PYGZsh{} RESPA integrator}
\PYG{n+nv}{nsteps}            \PYG{o}{=}  \PYG{l+m}{40000}  \PYG{c+c1}{\PYGZsh{} number of MD steps (100ps)}
\PYG{n+nv}{timestep}          \PYG{o}{=} \PYG{l+m}{0}.0025  \PYG{c+c1}{\PYGZsh{} timestep (2.5fs)}
\PYG{n+nv}{eneout\PYGZus{}period}     \PYG{o}{=}    \PYG{l+m}{400}  \PYG{c+c1}{\PYGZsh{} energy output period (1ps)}
\PYG{n+nv}{crdout\PYGZus{}period}     \PYG{o}{=}    \PYG{l+m}{400}  \PYG{c+c1}{\PYGZsh{} coordinates output period (1ps)}
\PYG{n+nv}{rstout\PYGZus{}period}     \PYG{o}{=}  \PYG{l+m}{40000}  \PYG{c+c1}{\PYGZsh{} restart output period}
\PYG{n+nv}{nbupdate\PYGZus{}period}   \PYG{o}{=}     \PYG{l+m}{10}  \PYG{c+c1}{\PYGZsh{} nonbond pair list update period}
\PYG{n+nv}{elec\PYGZus{}long\PYGZus{}period}  \PYG{o}{=}      \PYG{l+m}{2}  \PYG{c+c1}{\PYGZsh{} period of reciprocal space calculation}
\PYG{n+nv}{thermostat\PYGZus{}period} \PYG{o}{=}     \PYG{l+m}{10}  \PYG{c+c1}{\PYGZsh{} period of thermostat update}
\PYG{n+nv}{barostat\PYGZus{}period}   \PYG{o}{=}     \PYG{l+m}{10}  \PYG{c+c1}{\PYGZsh{} period of barostat update}
\end{sphinxVerbatim}

The following is an example for simulated annelaing in the NVT ensemble (see {\hyperref[\detokenize{09_Ensemble:ensemble}]{\sphinxcrossref{\DUrole{std,std-ref}{Ensemble section}}}}),
where the temperature is decreased from 500 K by 2 K
every 250 steps in the 250,000-steps MD simulation (1 step = 2 fs).
Thus, the temperature eventually reaches to 300 K during 50 ps.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}DYNAMICS\PYG{o}{]}
\PYG{n+nv}{integrator}       \PYG{o}{=}   VVER   \PYG{c+c1}{\PYGZsh{} leap\PYGZhy{}frog integrator}
\PYG{n+nv}{nsteps}           \PYG{o}{=}  \PYG{l+m}{25000}   \PYG{c+c1}{\PYGZsh{} number of MD steps}
\PYG{n+nv}{timestep}         \PYG{o}{=}  \PYG{l+m}{0}.002   \PYG{c+c1}{\PYGZsh{} timestep (ps)}
\PYG{n+nv}{nbupdate\PYGZus{}period}  \PYG{o}{=}     \PYG{l+m}{10}   \PYG{c+c1}{\PYGZsh{} nonbond pair list update period}
\PYG{n+nv}{annealing}        \PYG{o}{=}    YES   \PYG{c+c1}{\PYGZsh{} simulated annealing}
\PYG{n+nv}{dtemperature}     \PYG{o}{=}   \PYGZhy{}2.0   \PYG{c+c1}{\PYGZsh{} delta temperature}
\PYG{n+nv}{anneal\PYGZus{}period}    \PYG{o}{=}    \PYG{l+m}{250}   \PYG{c+c1}{\PYGZsh{} temperature change period}

\PYG{o}{[}ENSEMBLE\PYG{o}{]}
\PYG{n+nv}{ensemble}         \PYG{o}{=} NVT      \PYG{c+c1}{\PYGZsh{} [NVT,NPT,NPAT,NPgT]}
\PYG{n+nv}{tpcontrol}        \PYG{o}{=} BUSSI    \PYG{c+c1}{\PYGZsh{} [BERENDSEN,NHC,BUSSI]}
\PYG{n+nv}{temperature}      \PYG{o}{=} \PYG{l+m}{500}.0    \PYG{c+c1}{\PYGZsh{} initial temperature (K)}
\end{sphinxVerbatim}


\chapter{Minimize section}
\label{\detokenize{07_Minimize:minimize-section}}\label{\detokenize{07_Minimize:minimize}}\label{\detokenize{07_Minimize::doc}}

\section{Energy minimization}
\label{\detokenize{07_Minimize:energy-minimization}}
In the \sphinxstylestrong{{[}MINIMIZE{]}} section, the user can select methods for energy minimization.
Currently, the steepest descent (SD) algorithm is available in \sphinxstylestrong{SPDYN} and \sphinxstylestrong{ATDYN},
and the limited memory version of Broyden-Fletcher-Goldfarb-Shano (LBFGS)
is additionally available in \sphinxstylestrong{ATDYN}.
Note that constraint algorithms such as SHAKE are not available
in the energy minimization scheme in \sphinxstylestrong{GENESIS}.
The energy minimization can be done with restraints (see {\hyperref[\detokenize{12_Restraints:restraints}]{\sphinxcrossref{\DUrole{std,std-ref}{Restraints section}}}}).

When the energy minimization is carried out for the initial structure,
it is strongly recommended to use the option “contact\_check=YES”
in the \sphinxstylestrong{{[}ENERGY{]}} section (see {\hyperref[\detokenize{05_Energy:energy}]{\sphinxcrossref{\DUrole{std,std-ref}{Energy section}}}}).
This is because the initial structure is usually artificial, and sometimes contains
atomic clashes, where the distance between atoms is very short.
Such strong interactions can generate huge forces on the atoms,
resulting in unstable calculations, which might cause memory errors.


\bigskip\hrule\bigskip


\sphinxstylestrong{method} \sphinxstyleemphasis{SD / LBFGS}
\begin{quote}

\sphinxstylestrong{Default : LBFGS} (for \sphinxstylestrong{ATDYN}), \sphinxstylestrong{SD} (for \sphinxstylestrong{SPDYN})

Algorithm of minimization.
\begin{itemize}
\item {} 
\sphinxstylestrong{SD} : Steepest descent method

\item {} 
\sphinxstylestrong{LBFGS} : Limited memory version of Broyden-Fletcher-Goldfarb-Shano method (\sphinxstylestrong{ATDYN} only)

\end{itemize}
\end{quote}

\sphinxstylestrong{nsteps} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 100}

Number of minimization steps.
\end{quote}

\sphinxstylestrong{eneout\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 10}

Frequency of energy outputs.
\end{quote}

\sphinxstylestrong{crdout\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 0}

Frequency of coordinates outputs.
\end{quote}

\sphinxstylestrong{rstout\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 0}

Frequency of restart file updates.
\end{quote}

\sphinxstylestrong{nbupdate\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 10}

Frequency of non-bonded pair-list updates
\end{quote}

\sphinxstylestrong{fixatm\_select\_index} \sphinxstyleemphasis{Integer} (\sphinxstylestrong{ATDYN} only)
\begin{quote}

\sphinxstylestrong{Default : N/A} (all atoms are minimized)

Index of an atom group to be fixed during minimization.
The index must be defined in \sphinxstylestrong{{[}SELECTION{]}} (see {\hyperref[\detokenize{11_Selection:selection}]{\sphinxcrossref{\DUrole{std,std-ref}{Selection section}}}}).
For example, if the user specifies \sphinxcode{\sphinxupquote{fixatm\_select\_index = 1}}, the reference atoms
should be members of \sphinxcode{\sphinxupquote{group1}} in the \sphinxstylestrong{{[}SELECTION{]}}.
\end{quote}

\sphinxstylestrong{tol\_rmsg} \sphinxstyleemphasis{Real} (\sphinxstylestrong{ATDYN} only)
\begin{quote}

\sphinxstylestrong{Default : 0.36} (unit : kcal/mol/Angstrom)

Tolerence of convergence for RMS gradient.
\end{quote}

\sphinxstylestrong{tol\_maxg} \sphinxstyleemphasis{Real} (\sphinxstylestrong{ATDYN} only)
\begin{quote}

\sphinxstylestrong{Default : 0.54} (unit : kcal/mol/Angstrom)

Tolerence of convergence for maximum gradient.
\end{quote}

\begin{sphinxadmonition}{note}{Note:}
In \sphinxstylestrong{ATDYN}, a minimization run stops when \sphinxstyleemphasis{both} RMSG and MAXG
become smaller than the tolerence values.
\end{sphinxadmonition}


\section{Steepest descent method}
\label{\detokenize{07_Minimize:steepest-descent-method}}
\sphinxstylestrong{force\_scale\_init} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.00005}

The initial value of the force scaling coefficient in the steepest descent method.
This value is also used as the minimum value of the scaling coefficient.
\end{quote}

\sphinxstylestrong{force\_scale\_max} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.0001}

Maximum value of the force scaling coefficient in the steepest
descent method.
\end{quote}


\section{LBFGS method}
\label{\detokenize{07_Minimize:lbfgs-method}}
\sphinxstylestrong{ncorrection} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 10}

Number of corrections to build the inverse Hessian.
\end{quote}

\sphinxstylestrong{lbfgs\_bnd} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : YES}

Set a boundary to move atoms in each step of minimization.
\end{quote}

\sphinxstylestrong{lbfgs\_bnd\_qmonly} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

Set the boundary only to QM atoms.
\end{quote}

\sphinxstylestrong{lbfgs\_bnd\_maxmove} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.1} (unit : Angstrom)

The maximum size of move in each step.
\end{quote}

\begin{sphinxadmonition}{note}{Note:}
LBFGS often makes a large move of atoms, especially, in the first
few steps, and creates a distorted structure. Although this is rarely a
problem in MM calculation, it may cause convergence problem in QM
calculation. \sphinxcode{\sphinxupquote{lbfgs\_bnd}} prevents a huge move and crush of atoms by
setting a maximum size of move. The size is set by \sphinxcode{\sphinxupquote{lbfgs\_bnd\_maxmove}}.
\end{sphinxadmonition}


\section{Macro/micro-iteration scheme in QM/MM}
\label{\detokenize{07_Minimize:macro-micro-iteration-scheme-in-qm-mm}}
In this scheme, the MM region is first minimized while holding the QM
region fixed. This step is called micro-iteration.  When the MM region
reaches the minima (or the maximum number of steps), the whole system
including the QM region is updated. This step is called macro-iteration.
Then, the MM region is minimized again with the new QM region. The
micro- and macro-iterations are repeated until the convergence is
reached.

This scheme requires time-consuming QM calculations only in the
macro-iteration. During the micro-iteration, ESP charges are used to
represent the electrostatic interaction between QM and MM region.
Therefore, it is by far more efficient than the usual scheme, and is
recommended to use when ESP charges are available. Currently, this
scheme works in combination with Gaussian.

The keywords in this subsection have no effect in MM calculations,
of course.


\bigskip\hrule\bigskip


\sphinxstylestrong{macro} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

Invoke macro/micro-iteration scheme if YES.
\end{quote}

\sphinxstylestrong{nsteps\_micro} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 100}

Number of minimization steps for micro-iteration.
\end{quote}

\sphinxstylestrong{tol\_rmsg\_micro} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.27} (unit : kcal/mol/Angstrom)

Tolerence of convergence for RMS gradient in micro-iteration.
\end{quote}

\sphinxstylestrong{tol\_maxg\_micro} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.41} (unit : kcal/mol/Angstrom)

Tolerence of convergence for maximum gradient in micro-iteration.
\end{quote}

\sphinxstylestrong{macro\_select\_index} \sphinxstyleemphasis{Integer}
\begin{quote}

Index of an atom group to be fixed in micro-iteration, and
minimized in macro-iteration.
The index must be defined in \sphinxstylestrong{{[}SELECTION{]}} (see {\hyperref[\detokenize{11_Selection:selection}]{\sphinxcrossref{\DUrole{std,std-ref}{Selection section}}}}).
QM atoms are selected by default.
\end{quote}


\section{Fixing ring penetrations and chirality errors}
\label{\detokenize{07_Minimize:fixing-ring-penetrations-and-chirality-errors}}
In the energy minimization of GENESIS, the users can automatically fix the ring penetrations or chirality errors in protein, DNA, and RNA.
The suspicious ring is detected based on the length of the covalent bonds consisting of the ring.
Note that this algorithm is currently available for CHARMM or CHARMM19 force fields.
If the users utilized this algorithm, please cite the paper (Mori et al., \sphinxstyleemphasis{J. Chem. Inf. Model.}, 2021 \sphinxcite{99_References:mori-2021-1}).

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=0.600\linewidth]{{Minimize_checkstructure}.png}
\caption{Algorithms for fixing ring penetrations and chirality errors.}\label{\detokenize{07_Minimize:id2}}\end{figure}


\bigskip\hrule\bigskip


\sphinxstylestrong{check\_structure} \sphinxstyleemphasis{YES / NO} (available for only CHARMM or CHARMM19)
\begin{quote}

\sphinxstylestrong{Default : NO}

Detect ring penetrations and chirality errors in the system.
\end{quote}

\sphinxstylestrong{fix\_ring\_error} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

Reduce the ring size of the suspicious ring in the initial structure to fix the ring penetration.
\end{quote}

\sphinxstylestrong{exclude\_ring\_grpid} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : N/A}

Space-separated list of the indexes of the detected suspicious ring group, which are neglected during the automatic error fixing.
\end{quote}

\sphinxstylestrong{fix\_chirality\_error} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

Invert the position of the hydrogen bond of the suspicious chiral center in the initial structure to fix the chirality error.
\end{quote}

\sphinxstylestrong{exclude\_chiral\_grpid} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : N/A}

Space-separated list of indexes of the detected suspicious chirality center, which are neglected during the automatic error fixing.
\end{quote}


\bigskip\hrule\bigskip


The basic usage of these options is as follows.
First, the users specify “check\_structure = YES”, “fix\_ring\_error = NO”, and “fix\_chirality\_error = NO” to just check the errors.
If there are no suspicious rings or chiral centers in the final snapshot, the following messages are displayed:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Check\PYGZus{}Ring\PYGZus{}Structure\PYGZgt{} Check ring structure

  No suspicious residue was detected.

Check\PYGZus{}Chirality\PYGZgt{} Check chirality

  No suspicious residue was detected.
\end{sphinxVerbatim}

If there are suspicious rings or chiral centers, some warning messages will be shown at the last part of the log message:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Check\PYGZus{}Ring\PYGZus{}Structure\PYGZgt{} Check ring structure
  suspicious ring group \PYG{n+nv}{id} \PYG{o}{=} \PYG{l+m}{20} : PRO  \PYG{l+m}{82} \PYG{o}{(}\PYG{n+nv}{atom} \PYG{o}{=} \PYG{l+m}{1353}\PYG{o}{)} max\PYGZus{}bond\PYGZus{}length ...
  suspicious ring group \PYG{n+nv}{id} \PYG{o}{=} \PYG{l+m}{23} : PHE  \PYG{l+m}{93} \PYG{o}{(}\PYG{n+nv}{atom} \PYG{o}{=} \PYG{l+m}{1542}\PYG{o}{)} max\PYGZus{}bond\PYGZus{}length ...
  suspicious ring group \PYG{n+nv}{id} \PYG{o}{=} \PYG{l+m}{49} : PHE \PYG{l+m}{206} \PYG{o}{(}\PYG{n+nv}{atom} \PYG{o}{=} \PYG{l+m}{3440}\PYG{o}{)} max\PYGZus{}bond\PYGZus{}length ...
  suspicious ring group \PYG{n+nv}{id} \PYG{o}{=} \PYG{l+m}{52} : PHE \PYG{l+m}{230} \PYG{o}{(}\PYG{n+nv}{atom} \PYG{o}{=} \PYG{l+m}{3810}\PYG{o}{)} max\PYGZus{}bond\PYGZus{}length ...

  WARNING!
  Some suspicious residues were detected. Minimization might be too short,
  or \PYG{l+s+s2}{\PYGZdq{}ring penetration\PYGZdq{}} might happen \PYG{k}{in} the above residues.
  Check the structure of those residues very carefully. If you found a ring
  penetration, try to perform the energy minimization again with the
  options \PYG{l+s+s2}{\PYGZdq{}check\PYGZus{}structure = YES\PYGZdq{}} and \PYG{l+s+s2}{\PYGZdq{}fix\PYGZus{}ring\PYGZus{}error = YES\PYGZdq{}} \PYG{k}{in} \PYG{o}{[}MINIMIZE\PYG{o}{]}.
  The energy minimization should start from the restart file obtained
  \PYG{k}{in} \PYG{l+s+s2}{\PYGZdq{}this\PYGZdq{}} run.
\end{sphinxVerbatim}

Please read the warning message very carefully.
In this example, the users first had better check the obtained structure around PRO82, PHE93, PHE206, and PHE230 using a molecular viewer like VMD.
Because this is just a warning message, there might be actually no errors.
But, if the users found some errors, energy minimization should be performed again, restarting from this run
with the options “check\_structure = YES” and “fix\_ring\_error = YES” or “fix\_chirality\_error = YES”.
If the users want to fix only PHE93 and PHE230, please add the option “exclude\_ring\_grpid = 20 49”,
where “20” and “49” are the “suspicious ring group id” of PRO82 and PHE206, respectively, shown in the warning message.
Note that the reduction of the size of the penetrated ring or inversion of the hydrogen atom in the bad chiral center is carried out for the initial structure.


\section{Examples}
\label{\detokenize{07_Minimize:examples}}
A 2,000-step energy minimization with the steepest descent method:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}MINIMIZE\PYG{o}{]}
\PYG{n+nv}{method}           \PYG{o}{=} SD      \PYG{c+c1}{\PYGZsh{} Steepest descent}
\PYG{n+nv}{nsteps}           \PYG{o}{=} \PYG{l+m}{2000}    \PYG{c+c1}{\PYGZsh{} number of minimization steps}
\PYG{n+nv}{eneout\PYGZus{}period}    \PYG{o}{=}   \PYG{l+m}{50}    \PYG{c+c1}{\PYGZsh{} energy output period}
\PYG{n+nv}{crdout\PYGZus{}period}    \PYG{o}{=}   \PYG{l+m}{50}    \PYG{c+c1}{\PYGZsh{} coordinates output period}
\PYG{n+nv}{rstout\PYGZus{}period}    \PYG{o}{=} \PYG{l+m}{2000}    \PYG{c+c1}{\PYGZsh{} restart output period}
\PYG{n+nv}{nbupdate\PYGZus{}period}  \PYG{o}{=}   \PYG{l+m}{10}    \PYG{c+c1}{\PYGZsh{} nonbond pair list update period}
\end{sphinxVerbatim}

An example of LBFGS optmization along with the macro/micro-iteration scheme:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}MINIMIZE\PYG{o}{]}
\PYG{n+nv}{method}             \PYG{o}{=} LBFGS
\PYG{n+nv}{nsteps}             \PYG{o}{=} \PYG{l+m}{500}  \PYG{c+c1}{\PYGZsh{} number of steps}
\PYG{n+nv}{eneout\PYGZus{}period}      \PYG{o}{=} \PYG{l+m}{5}    \PYG{c+c1}{\PYGZsh{} energy output period}
\PYG{n+nv}{crdout\PYGZus{}period}      \PYG{o}{=} \PYG{l+m}{5}    \PYG{c+c1}{\PYGZsh{} coordinates output period}
\PYG{n+nv}{rstout\PYGZus{}period}      \PYG{o}{=} \PYG{l+m}{5}    \PYG{c+c1}{\PYGZsh{} restart output period}
\PYG{n+nv}{nbupdate\PYGZus{}period}    \PYG{o}{=} \PYG{l+m}{1}    \PYG{c+c1}{\PYGZsh{} nonbond pair list update period}
\PYG{n+nv}{lbfgs\PYGZus{}bnd}          \PYG{o}{=} yes  \PYG{c+c1}{\PYGZsh{} set a boundary to move atoms}
\PYG{n+nv}{lbfgs\PYGZus{}bnd\PYGZus{}qmonly}   \PYG{o}{=} no   \PYG{c+c1}{\PYGZsh{} set the boundary only to QM atoms}
\PYG{n+nv}{lbfgs\PYGZus{}bnd\PYGZus{}maxmove}  \PYG{o}{=} \PYG{l+m}{0}.1  \PYG{c+c1}{\PYGZsh{} the max. size of move}
\PYG{n+nv}{macro}              \PYG{o}{=} yes  \PYG{c+c1}{\PYGZsh{} switch macro/micro\PYGZhy{}iteration scheme}
\PYG{n+nv}{nsteps\PYGZus{}micro}       \PYG{o}{=} \PYG{l+m}{100}  \PYG{c+c1}{\PYGZsh{} number of steps of micro\PYGZhy{}iteration}
\end{sphinxVerbatim}

Energy minimization with automatic fixing for ring penetrations and chirality errors:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}MINIMIZE\PYG{o}{]}
\PYG{n+nv}{method}              \PYG{o}{=} SD      \PYG{c+c1}{\PYGZsh{} Steepest descent}
\PYG{n+nv}{nsteps}              \PYG{o}{=} \PYG{l+m}{2000}    \PYG{c+c1}{\PYGZsh{} number of minimization steps}
\PYG{n+nv}{eneout\PYGZus{}period}       \PYG{o}{=}   \PYG{l+m}{50}    \PYG{c+c1}{\PYGZsh{} energy output period}
\PYG{n+nv}{crdout\PYGZus{}period}       \PYG{o}{=}   \PYG{l+m}{50}    \PYG{c+c1}{\PYGZsh{} coordinates output period}
\PYG{n+nv}{rstout\PYGZus{}period}       \PYG{o}{=} \PYG{l+m}{2000}    \PYG{c+c1}{\PYGZsh{} restart output period}
\PYG{n+nv}{nbupdate\PYGZus{}period}     \PYG{o}{=}   \PYG{l+m}{10}    \PYG{c+c1}{\PYGZsh{} nonbond pair list update period}
\PYG{n+nv}{check\PYGZus{}structure}     \PYG{o}{=}  YES    \PYG{c+c1}{\PYGZsh{} check ring penetration and chirality error}
\PYG{n+nv}{fix\PYGZus{}ring\PYGZus{}error}      \PYG{o}{=}  YES    \PYG{c+c1}{\PYGZsh{} automatically fix the ring penetrations}
\PYG{n+nv}{fix\PYGZus{}chirality\PYGZus{}error} \PYG{o}{=}  YES    \PYG{c+c1}{\PYGZsh{} automatically fix the chirality errors}
\end{sphinxVerbatim}


\chapter{Constraints section}
\label{\detokenize{08_Constraints:constraints-section}}\label{\detokenize{08_Constraints:constraints}}\label{\detokenize{08_Constraints::doc}}

\section{SHAKE/RATTLE algorithms}
\label{\detokenize{08_Constraints:shake-rattle-algorithms}}
In the \sphinxstylestrong{{[}CONSTRAINTS{]}} section, keywords related to bond constraints are specified.
In the leapfrog integrator, the SHAKE algorithm is applied for
covalent bonds involving hydrogen \sphinxcite{99_References:rycaert-1997}.
In the velocity Verlet and multiple time-step integrators,
not only SHAKE but also RATTLE are used \sphinxcite{99_References:andersen-1983}.
Note that bond constraint between heavy atoms is not available currently.


\bigskip\hrule\bigskip


\sphinxstylestrong{rigid\_bond} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

Turn on or off the SHAKE/RATTLE algorithms for covalent bonds involving hydrogen.
\end{quote}

\sphinxstylestrong{shake\_iteration} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 500}

Maximum number of iterations for SHAKE/RATTLE constraint.
If SHAKE/RATTLE does not converge within the given number of iterations,
the program terminates with an error message.
\end{quote}

\sphinxstylestrong{shake\_tolerance} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 1.0e-10} (unit : Angstrom)

Tolerance of SHAKE/RATTLE convergence.
\end{quote}

\sphinxstylestrong{hydrogen\_type} \sphinxstyleemphasis{NAME / MASS}
\begin{quote}

\sphinxstylestrong{Default : NAME}

This parameter defines how hydrogen atoms are detected. This parameter
is ignored when \sphinxstyleemphasis{rigid\_bond = NO}. Usually, the users do not need to
take care about this parameter.
\begin{itemize}
\item {} 
\sphinxstylestrong{MASS} : detect hydrogen only based on the atomic mass. If the mass
of an atom is less than \sphinxstyleemphasis{hydrogen\_mass\_upper\_bound} and greater than 0,
that atom is considered as a hydrogen.

\item {} 
\sphinxstylestrong{NAME} : detect hydrogen based on the atom name, type, and mass. If the mass
of an atom is less than \sphinxstyleemphasis{hydrogen\_mass\_upper\_bound} and the name
or type begins with ‘h’, ‘H’, ‘d’, or ‘D’, that atom is considered as
a hydrogen.

\end{itemize}


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabulary}{\linewidth}[t]{|T|T|T|T|}
\hline
\sphinxstyletheadfamily 
atom name (type)
&\sphinxstyletheadfamily 
mass
&\sphinxstyletheadfamily 
\sphinxstyleemphasis{NAME}
&\sphinxstyletheadfamily 
\sphinxstyleemphasis{MASS}
\\
\hline
HX
&
1.0
&
o
&
o
\\
\hline
XX
&
1.0
&
x
&
o
\\
\hline
HY
&
3.0
&
x
&
x
\\
\hline
YY
&
3.0
&
x
&
x
\\
\hline
\end{tabulary}
\par
\sphinxattableend\end{savenotes}

o: treated as hydrogen, x: not treated as hydrogen. Here, we assumed \sphinxstyleemphasis{hydrogen\_mass\_upper\_bound} 2.1.
\end{quote}

\sphinxstylestrong{hydrogen\_mass\_upper\_bound} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 2.1}

This parameter defines the upper limit of atomic mass to determine the hydrogen atom.
For exmaple, if you define it as 3.0, the atom with the atomic mass less than 3.0 is treated as a hydrogen.
You should write it in the case of hydrogen mass repartitioning scheme.
This option is available in \sphinxstylestrong{GENESIS 1.2} or later.
\end{quote}


\section{SETTLE algorithm}
\label{\detokenize{08_Constraints:settle-algorithm}}
\sphinxstylestrong{fast\_water} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : YES}

Turn on or off the SETTLE algorithm for the constraints of the water molecules \sphinxcite{99_References:miyamoto-1992}.
Although the default is “fast\_water=YES”, the users must specify “rigid\_bond=YES” to use the SETTLE algorithm.
If “rigid\_bond=YES” and “fast\_water=NO” are specified, the SHAKE/RATTLE algorithm is
applied to water molecules, which is not computationally efficient.
\end{quote}

\sphinxstylestrong{water\_model} \sphinxstyleemphasis{expression or NONE}
\begin{quote}

\sphinxstylestrong{Default : TIP3}

Residue name of the water molecule to be rigidified in the SETTLE algorithm.
In the case of the AMBER force field, “water\_model = WAT” must be specified.
\end{quote}

\begin{sphinxadmonition}{note}{Note:}
TIP4P water model is availabe in \sphinxstylestrong{GENESIS 1.2} or later.
In the case of using TIP4P water model, we regard it as rigid. In
molecular dynamics simulations, please define \sphinxstylestrong{rigid\_bond} and \sphinxstylestrong{fast\_water}
yes. In minimization, \sphinxstylestrong{{[}Constraints{]}} has not been used before, but now
you can define \sphinxstylestrong{fast\_water} yes when TIP4P water model is used, by
regarding TIP4P water molecule rigid. However, please keep in mind that
other parameters cannot be defined in minimizations, and constraints are not
applied except water molecules. TIP4P water model can be used only in SPDYN.
\end{sphinxadmonition}


\section{LINCS algorithm}
\label{\detokenize{08_Constraints:lincs-algorithm}}
\sphinxstylestrong{fast\_bond} \sphinxstyleemphasis{YES / NO} (\sphinxstylestrong{LEAP} integrator in \sphinxstylestrong{ATDYN} only)
\begin{quote}

\sphinxstylestrong{Default : NO}

Turn on or off the LINCS algorithm.
To use the LINCS algorithm, “rigid\_bond=YES” should be also specified.
\end{quote}

\sphinxstylestrong{lincs\_iteration} \sphinxstyleemphasis{Integer} (\sphinxstylestrong{ATDYN} only)
\begin{quote}

\sphinxstylestrong{Default : 1}

Number of iterations in the LINCS algorithm.
\end{quote}

\sphinxstylestrong{lincs\_order} \sphinxstyleemphasis{Integer} (\sphinxstylestrong{ATDYN} only)
\begin{quote}

\sphinxstylestrong{Default : 4}

Matrix expansion order in the LINCS algorithm.
\end{quote}


\section{Examples}
\label{\detokenize{08_Constraints:examples}}
In the case of the CHARMM force field:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}CONSTRAINTS\PYG{o}{]}
\PYG{n+nv}{rigid\PYGZus{}bond}  \PYG{o}{=} YES   \PYG{c+c1}{\PYGZsh{} Turn on SHAKE/RATTLE}
\PYG{n+nv}{fast\PYGZus{}water}  \PYG{o}{=} YES   \PYG{c+c1}{\PYGZsh{} Turn on SETTLE}
\end{sphinxVerbatim}

In the case of the AMBER force field:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}CONSTRAINTS\PYG{o}{]}
\PYG{n+nv}{rigid\PYGZus{}bond}  \PYG{o}{=} YES   \PYG{c+c1}{\PYGZsh{} Turn on SHAKE/RATTLE}
\PYG{n+nv}{fast\PYGZus{}water}  \PYG{o}{=} YES   \PYG{c+c1}{\PYGZsh{} Turn on SETTLE}
\PYG{n+nv}{water\PYGZus{}model} \PYG{o}{=} WAT   \PYG{c+c1}{\PYGZsh{} residue name of the rigid water}
\end{sphinxVerbatim}

Turn off all constraints in the system

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}CONSTRAINTS\PYG{o}{]}
\PYG{n+nv}{rigid\PYGZus{}bond}  \PYG{o}{=} NO
\PYG{n+nv}{fast\PYGZus{}water}  \PYG{o}{=} NO
\end{sphinxVerbatim}


\chapter{Ensemble section}
\label{\detokenize{09_Ensemble:ensemble-section}}\label{\detokenize{09_Ensemble:ensemble}}\label{\detokenize{09_Ensemble::doc}}

\section{Thermostat and barostat}
\label{\detokenize{09_Ensemble:thermostat-and-barostat}}
In the \sphinxstylestrong{{[}ENSEMBLE{]}} section, the type of ensemble, temperature and
pressure control algorithm, and parameters used in these algorithms
(such as temperature and pressure) can be specified.

In the Berendsen thermostat with weak coupling (“tpcontrol=berendsen”), velocities are rescaled by
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\lambda = \left [ 1 + \frac{\Delta t}{\tau} \left ( \frac{K_{\textrm{target}}}{K} - 1 \right) \right]^{\frac{1}{2}}\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

at every step where \({\tau}\) is the dampling time \sphinxcite{99_References:berendsen-1984}. It almost conserves dynamics
of NVE and one of the most popular thermostats. However, it fails to generate canonical distributions and
anomalous effects can happern \sphinxcite{99_References:braun-2018}. Bussi suggested a modification of Berendsen thermostat to
generate canonical distribution \sphinxcite{99_References:bussi-2007}. According to the scheme (“tpcontrol=bussi”),
\begin{quote}
\begin{quote}

the target temperature is not a fixed value but stochastically driven from the distributions:
\end{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}P \left( {K_{\textrm{target}}} \right ) \propto {K_{\textrm{target}}}^{\frac{N_{f}}{2}-1} e ^ {-\beta K_{\textrm{target}}} .\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

In addition, random force \({\Delta W}\) is applied to update temperature based on the instantaneousand target temperatures:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\Delta K = \left ( K_{\textrm{target}} - K \right ) \frac{\Delta t}{\tau} + 2 \sqrt{\frac{K K_{\textrm{target}}}{N_{f}}} \frac{\Delta W}{\tau} .\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

Another famous thermostat is suggested by Martyna et al., which is named as Nose-Hoover chain (NHC) \sphinxcite{99_References:martyna-1992}.
This thermostat was designed to solve non-ergodic progem in Nose-Hoover thermostat.
In this scheme, we solve the euqation of motion with additional textit\{m\} chains:
\begin{quote}

\vspace{-5mm}
\begin{align*}\!\begin{aligned}
\frac{d\mathbf{r}_{k}}{dt} & = \frac{\mathbf{p}_{k}}{m_{k}}\\
\frac{d\mathbf{p}_{k}}{dt} & = \mathbf{F}_{k} - \frac{p_{\eta_{1}}}{Q_{1}} \mathbf{p}_{k}\\
\frac{d \eta_{\alpha}}{dt} & = \frac{p_{\eta_{\alpha}}}{Q_{\alpha}}, \  \alpha=1,2,...,m\\
\frac{d p_{\eta_{1}}}{dt} & = K - K_{\textrm{target}} - \frac{p_{\eta_{2}}}{Q_{2}} p_{\eta_{1}}\\
\frac{d p_{\eta_{\alpha}}}{dt} & = \frac{p_{\eta_{\alpha-1}}}{Q_{\alpha-1}} - k_{B}T - \frac{p_{\eta_{\alpha + 1}}}{Q_{\alpha + 1}} p_{\eta_{\alpha}}, \ \alpha = 2,...,m-1\\
\frac{d p_{\eta_{m}}}{dt} & = \frac{p_{\eta_{m-1}}}{Q_{m-1}} - k_{B}T .\\
\end{aligned}\end{align*}
\vspace{-3mm}
\end{quote}

In \sphinxstylestrong{SPDYN}, we evaluate temperature in more accurate way than existing ones. With velocity Verlet
integration, two kinetic energy types can be used in temperature evaluations:
\begin{quote}

\vspace{-5mm}
\begin{align*}\!\begin{aligned}
K_{\textrm{full}} & = \sum_{k=1}^{N} \frac{\mathbf{p}_{k}\left( t \right) ^2}{2m_{k}}\\
K_{\textrm{half}} & = \frac{1}{2} \sum_{k=1}^{N} \left ( \frac{\mathbf{p}_{k} \left(t - \frac{\Delta t}{2} \right)^2}{2m_{k}} + \frac{\mathbf{p}_{k} \left( t + \frac{\Delta t}{2} \right)^2}{2m_{k}} \right)\\
\end{aligned}\end{align*}
\vspace{-3mm}
\end{quote}

These kinetic energies are accurate up to the first order of a time step and not recommended for a large time step.
In \sphinxstylestrong{SPDYN}, temperature is evaluted by combining the two kinetic energies:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}N_{f} k_{B} T = \frac{4}{3} \left < K_{\textrm{half}} (t) \right > + \frac{2}{3} \left < K_{\textrm{full}} (t) \right >\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

Temperature evaluted in this way is accurate up to the third order of the time step.

Barostat in \sphinxstylestrong{SPDYN} is based on MTK barostat type suggested by \sphinxcite{99_References:martyna-1996}
with three thermostats described above. To make use of kinetic energy at \(t + \frac{\Delta t}{2}\),
we recommend group temperature/pressure evaluations where kinetic energy and virial are evaluted by
considering XHn group one particle.

In the Langevin thermostat algorithm (“ensemble=NVT” with
“tpcontrol=LANGEVIN”), every particles are coupled with a viscous background
and a stochastic heat bath \sphinxcite{99_References:adelman-1976}:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\frac{d\mathbf{v}(t)}{dt} = \frac{\mathbf{F}(t)+\mathbf{R}(t)}{m}-\gamma \mathbf{v}(t)\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(\gamma\) is the thermostat friction parameter (\sphinxstyleemphasis{gamma\_t} keyword)
and \(\mathbf{R}(t)\) is the stochastic force.
In the Langevin thermostat and barostat method (“ensemble=NPT” with
“tpcontrol=LANGEVIN”),
the equation of motion is given by \sphinxcite{99_References:quigley-2004}:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\frac{d\mathbf{r}(t)}{dt} & = \mathbf{v}(t)+v_{\epsilon}\mathbf{r}(t) \\ \frac{d\mathbf{v}(t)}{dt} & = \frac{\mathbf{F}(t)+\mathbf{R}(t)}{m}-[\gamma_p+(1+\frac{3}{f})v_{\epsilon}]\mathbf{v}(t) \\ \frac{dv_{\epsilon}(t)}{dt} & = [3V(P(t)-P_0(t))+\frac{3K}{f}-\gamma_p v_{\epsilon} + R_p ] / p_{mass}\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(K\) is the kinetic energy, \(\gamma_p\) is the barostat
friction parameter (\sphinxstyleemphasis{gamma\_p} keyword), \(R_p\) is the stochastic
pressure variable.


\bigskip\hrule\bigskip


\sphinxstylestrong{ensemble} \sphinxstyleemphasis{NVE / NVT / NPT / NPAT / NPgT}
\begin{quote}

\sphinxstylestrong{Default : NVE}

Type of ensemble.
\begin{itemize}
\item {} 
\sphinxstylestrong{NVE}: Microcanonical ensemble.

\item {} 
\sphinxstylestrong{NVT}: Canonical ensemble.

\item {} 
\sphinxstylestrong{NPT}: Isothermal-isobaric ensemble.

\item {} 
\sphinxstylestrong{NPAT}: Constant area A (XY), pressure along the normal (Z), temperature
\sphinxcite{99_References:zhang-1995}. In this case, \sphinxstyleemphasis{isotropy} must be set to ‘XY-FIXED’ (see below).

\item {} 
\sphinxstylestrong{NPgT}: Constant surface-tension \(\gamma\) (XY), pressure along the normal (Z),
temperature \sphinxcite{99_References:zhang-1995}. In this case, \sphinxstyleemphasis{isotropy} must be set to ‘SEMI-ISO’ (see below).

\end{itemize}
\end{quote}

\sphinxstylestrong{temperature} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 298.15} (unit : Kelvin)

Initial and target temperature.
\end{quote}

\sphinxstylestrong{pressure} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 1.0} (unit : atm)

Target pressure in the NPT ensemble.
In the case of the NPAT and NPgT ensembles,
this is the pressure along the ‘Z’ axis.
\end{quote}

\sphinxstylestrong{group\_tp} \sphinxstyleemphasis{Yes / No}
\begin{quote}

\sphinxstylestrong{Default : Yes}
\begin{quote}

use of group temperature/pressure evaluation in thermostat/barostat.
\end{quote}
\end{quote}

\sphinxstylestrong{gamma} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.0} (unit : dyn/cm)

Target surface tension in NPgT ensemble.
\end{quote}

\sphinxstylestrong{tpcontrol} \sphinxstyleemphasis{NO / BERENDSEN / BUSSI / NHC / LANGEVIN}
\begin{quote}

\sphinxstylestrong{Default : NO}

Type of thermostat and barostat. The availabe algorithm depends on the integrator.
\begin{itemize}
\item {} 
\sphinxstylestrong{NO}: Do not use temperature/pressure control algorithm (for NVE only)

\item {} 
\sphinxstylestrong{BERENDSEN}: Berendsen thermostat/barostat \sphinxcite{99_References:berendsen-1984}

\item {} 
\sphinxstylestrong{BUSSI}: Bussi’s thermostat/barostat \sphinxcite{99_References:bussi-2007} \sphinxcite{99_References:bussi-2009}

\item {} 
\sphinxstylestrong{NHC}: Nose-Hoover chain thermostat with MTK barostat \sphinxcite{99_References:berendsen-1984} \sphinxcite{99_References:martyna-1996}

\item {} 
\sphinxstylestrong{LANGEVIN}: Langevin thermostat/barostat \sphinxcite{99_References:quigley-2004} (for ATDYN only)

\end{itemize}


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabulary}{\linewidth}[t]{|T|T|T|}
\hline
\sphinxstyletheadfamily 
integrator
&\sphinxstyletheadfamily 
ensemble
&\sphinxstyletheadfamily 
tpcontrol
\\
\hline\sphinxmultirow{3}{4}{%
\begin{varwidth}[t]{\sphinxcolwidth{1}{3}}
LEAP

(ATDYN)
\par
\vskip-\baselineskip\vbox{\hbox{\strut}}\end{varwidth}%
}%
&
NVT
&
BERENDSEN, LANGEVIN
\\
\cline{2-3}\sphinxtablestrut{4}&
NPT
&
BERENDSEN, LANGEVIN
\\
\cline{2-3}\sphinxtablestrut{4}&
NPAT/NPgT
&
BERENDSEN, LANGEVIN
\\
\hline\sphinxmultirow{3}{11}{%
\begin{varwidth}[t]{\sphinxcolwidth{1}{3}}
VVER

(ATDYN)
\par
\vskip-\baselineskip\vbox{\hbox{\strut}}\end{varwidth}%
}%
&
NVT
&
BERENDSEN, LANGEVIN, BUSSI
\\
\cline{2-3}\sphinxtablestrut{11}&
NPT
&
LANGEVIN, BUSSI
\\
\cline{2-3}\sphinxtablestrut{11}&
NPAT/NPgT
&
LANGEVIN
\\
\hline\sphinxmultirow{3}{18}{%
\begin{varwidth}[t]{\sphinxcolwidth{1}{3}}
VVER

(SPDYN)
\par
\vskip-\baselineskip\vbox{\hbox{\strut}}\end{varwidth}%
}%
&
NVT
&
BUSSI, BERENDSEN, NHC
\\
\cline{2-3}\sphinxtablestrut{18}&
NPT
&
BUSSI, BERENDSEN, NHC
\\
\cline{2-3}\sphinxtablestrut{18}&
NPAT/NPgT
&
BUSSI, BERENDSEN, NHC
\\
\hline\sphinxmultirow{3}{25}{%
\begin{varwidth}[t]{\sphinxcolwidth{1}{3}}
VRES

(SPDYN)
\par
\vskip-\baselineskip\vbox{\hbox{\strut}}\end{varwidth}%
}%
&
NVT
&
BUSSI, BERENDSEN, NHC
\\
\cline{2-3}\sphinxtablestrut{25}&
NPT
&
BUSSI, BERENDSEN, NHC
\\
\cline{2-3}\sphinxtablestrut{25}&
NPAT/NPgT
&
BUSSI, BERENDSEN, NHC
\\
\hline
\end{tabulary}
\par
\sphinxattableend\end{savenotes}
\end{quote}

\sphinxstylestrong{tau\_t} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 5.0} (unit : ps)

Temperature coupling time in the Berendsen and Bussi thermostats.
\end{quote}

\sphinxstylestrong{tau\_p} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 5.0} (unit : ps)

Pressure coupling time in the Berendsen and Bussi barostats.
\end{quote}

\sphinxstylestrong{compressibility} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.0000463} (unit : atm$^{\text{-1}}$)

Compressibility parameter in the Berendsen barostat.
\end{quote}

\sphinxstylestrong{gamma\_t} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 1.0} (unit : ps$^{\text{-1}}$)

Friction parameter of the Langevin thermostat.
\end{quote}

\sphinxstylestrong{gamma\_p} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.1} (unit : ps$^{\text{-1}}$)

Friction parameter of the Langevin barostat.
\end{quote}

\sphinxstylestrong{isotropy} \sphinxstyleemphasis{ISO / ANISO / SEMI-ISO / XY-FIXED}
\begin{quote}

\sphinxstylestrong{Default : ISO}

Isotropy of the simulation system.
This parameter specifies how X, Y, Z dimensions of the simulation
box change in NPT, NPgT, and NPAT ensembles.
\begin{itemize}
\item {} 
\sphinxstylestrong{ISO}: X, Y, and Z dimensions are coupled together.

\item {} 
\sphinxstylestrong{ANISO}: X, Y, and Z dimensions fluctuate independently.

\item {} 
\sphinxstylestrong{SEMI-ISO}: X, Y, and Z dimensions fluctuate, where the ratio of
X and Y dimensions are kept constant, and Z dimension can change
independently \sphinxcite{99_References:kandt-2007}. This setting with NPT or NPAT or NPgT
ensemble is expected to be useful for bio-membrane systems.

\item {} 
\sphinxstylestrong{XY-FIXED}: X and Y dimensions are fixed, while Z dimension can change
(NPAT only).

\end{itemize}
\end{quote}


\section{Examples}
\label{\detokenize{09_Ensemble:examples}}
NVT ensemble with Bussi thermostat:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}ENSEMBLE\PYG{o}{]}
\PYG{n+nv}{ensemble}    \PYG{o}{=} NVT       \PYG{c+c1}{\PYGZsh{} Canonical ensemble}
\PYG{n+nv}{tpcontrol}   \PYG{o}{=} BUSSI     \PYG{c+c1}{\PYGZsh{} Bussi thermostat}
\PYG{n+nv}{temperature} \PYG{o}{=} \PYG{l+m}{300}.0     \PYG{c+c1}{\PYGZsh{} target temperature (K)}
\end{sphinxVerbatim}

NPT ensemble with isotropic pressure coupling:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}ENSEMBLE\PYG{o}{]}
\PYG{n+nv}{ensemble}    \PYG{o}{=} NPT       \PYG{c+c1}{\PYGZsh{} Isothermal\PYGZhy{}isobaric ensemble}
\PYG{n+nv}{tpcontrol}   \PYG{o}{=} BUSSI     \PYG{c+c1}{\PYGZsh{} Bussi thermostat and barostat}
\PYG{n+nv}{temperature} \PYG{o}{=} \PYG{l+m}{300}.0     \PYG{c+c1}{\PYGZsh{} target temperature (K)}
\PYG{n+nv}{pressure}    \PYG{o}{=} \PYG{l+m}{1}.0       \PYG{c+c1}{\PYGZsh{} target pressure (atm)}
\end{sphinxVerbatim}

NPT ensemble with semi-isotropic pressure coupling, which is usually used for lipid bilayer systems:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}ENSEMBLE\PYG{o}{]}
\PYG{n+nv}{ensemble}    \PYG{o}{=} NPT       \PYG{c+c1}{\PYGZsh{} Isothermal\PYGZhy{}isobaric ensemble}
\PYG{n+nv}{tpcontrol}   \PYG{o}{=} BUSSI     \PYG{c+c1}{\PYGZsh{} Bussi thermostat and barostat}
\PYG{n+nv}{temperature} \PYG{o}{=} \PYG{l+m}{300}.0     \PYG{c+c1}{\PYGZsh{} target temperature (K)}
\PYG{n+nv}{pressure}    \PYG{o}{=} \PYG{l+m}{1}.0       \PYG{c+c1}{\PYGZsh{} target pressure (atm)}
\PYG{n+nv}{isotropy}    \PYG{o}{=} SEMI\PYGZhy{}ISO  \PYG{c+c1}{\PYGZsh{} Ratio of X to Y is kept constant}
\end{sphinxVerbatim}

NPAT ensemble:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}ENSEMBLE\PYG{o}{]}
\PYG{n+nv}{ensemble}    \PYG{o}{=} NPAT      \PYG{c+c1}{\PYGZsh{} Constant area ensemble}
\PYG{n+nv}{tpcontrol}   \PYG{o}{=} BUSSI     \PYG{c+c1}{\PYGZsh{} Bussi thermostat and barostat}
\PYG{n+nv}{temperature} \PYG{o}{=} \PYG{l+m}{300}.0     \PYG{c+c1}{\PYGZsh{} target temperature (K)}
\PYG{n+nv}{pressure}    \PYG{o}{=} \PYG{l+m}{1}.0       \PYG{c+c1}{\PYGZsh{} target normal pressure (atm)}
\PYG{n+nv}{isotropy}    \PYG{o}{=} XY\PYGZhy{}FIXED  \PYG{c+c1}{\PYGZsh{} the system area is kept constant}
\end{sphinxVerbatim}

NP\(\gamma\)T ensemble:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}ENSEMBLE\PYG{o}{]}
\PYG{n+nv}{ensemble}    \PYG{o}{=} NPgT      \PYG{c+c1}{\PYGZsh{} Constant surface\PYGZhy{}tension ensemble}
\PYG{n+nv}{tpcontrol}   \PYG{o}{=} BUSSI     \PYG{c+c1}{\PYGZsh{} Bussi thermostat and barostat}
\PYG{n+nv}{temperature} \PYG{o}{=} \PYG{l+m}{300}.0     \PYG{c+c1}{\PYGZsh{} target temperature (K)}
\PYG{n+nv}{pressure}    \PYG{o}{=} \PYG{l+m}{1}.0       \PYG{c+c1}{\PYGZsh{} target normal pressure (atm)}
\PYG{n+nv}{gamma}       \PYG{o}{=} \PYG{l+m}{200}.0     \PYG{c+c1}{\PYGZsh{} target surface tension (dyn/cm)}
\PYG{n+nv}{isotropy}    \PYG{o}{=} SEMI\PYGZhy{}ISO  \PYG{c+c1}{\PYGZsh{} Ratio of X to Y is kept constant}
\end{sphinxVerbatim}


\chapter{Boundary section}
\label{\detokenize{10_Boundary:boundary-section}}\label{\detokenize{10_Boundary:boundary}}\label{\detokenize{10_Boundary::doc}}

\section{Boundary condition}
\label{\detokenize{10_Boundary:boundary-condition}}
\sphinxstylestrong{type} \sphinxstyleemphasis{PBC / NOBC}
\begin{quote}

\sphinxstylestrong{Default : PBC}

Type of boundary condition.
\begin{itemize}
\item {} 
\sphinxstylestrong{PBC}: Periodic boundary condition (rectangular or cubic box)

\item {} 
\sphinxstylestrong{NOBC}: Non-boundary condition (vacuum system).
In \sphinxstylestrong{ATDYN}, \sphinxstylestrong{NOBC} is applicable to various force fields and models.
However, in \sphinxstylestrong{SPDYN}, it cannot be used.

\end{itemize}
\end{quote}

\sphinxstylestrong{box\_size\_x} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : N/A} (unit : Angstrom)

Box size along the x dimension.
\end{quote}

\sphinxstylestrong{box\_size\_y} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : N/A} (unit : Angstrom)

Box size along the y dimension.
\end{quote}

\sphinxstylestrong{box\_size\_z} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : N/A} (unit : Angstrom)

Box size along the z dimension.
\end{quote}

\sphinxstylestrong{local\_pbc} \sphinxstyleemphasis{Boolean}
\begin{quote}

\sphinxstylestrong{Default : NO}

Use PBC coordinates in force/energy calculations or not.  (only available for coarse-grained models and PBC in \sphinxstylestrong{ATDYN}, In \sphinxstylestrong{SPDYN}, \sphinxstylestrong{local\_pbc} is automatically defined in CHARMM and GROAMBER force field))
\end{quote}

\begin{sphinxadmonition}{note}{Note:}
If the simulation system has a periodic boundary condition (\sphinxstylestrong{PBC}),
the user must specify the box size in the control file
(at the energy minimization stage in most cases).
During the simulations, box size is saved in a restart file.
If the restart file is used as an input of the subsequent simulation,
the box size is overwritten with the restart information.
Note that in this case the box size given in the control file is ignored.
\end{sphinxadmonition}


\section{Domain decomposition}
\label{\detokenize{10_Boundary:domain-decomposition}}
\sphinxstylestrong{domain\_x} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : N/A (Optional)} (\sphinxstylestrong{SPDYN} only)

Number of domains along the x dimension.
\end{quote}

\sphinxstylestrong{domain\_y} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : N/A (Optional)} (\sphinxstylestrong{SPDYN} only)

Number of domains along the y dimension.
\end{quote}

\sphinxstylestrong{domain\_z} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : N/A (Optional)} (\sphinxstylestrong{SPDYN} only)

Number of domains along the z dimension.
\end{quote}

\begin{sphinxadmonition}{note}{Note:}
If number of domains (domain\_x, domain\_y, and domain\_z) are not specified
in the control file, they are automatically determined based on the
number of MPI processes.
When the user specifies the number of domains explicitly,
please make sure that the product of the domain numbers in each dimension
(i.e., domain\_x * domain\_y * domain\_z) is equal to the total number of
MPI processes.
\end{sphinxadmonition}


\section{Spherical potential}
\label{\detokenize{10_Boundary:spherical-potential}}
In MD simulations with \sphinxstylestrong{NOBC}, molecules may evaporate from a system,
and, once such an event happens, the molecule runs in the vacuum with
constant velocity to infinity. Therefore, it is useful to set a
potential which pulls the molecule back to the system.

In \sphinxstylestrong{ATDYN}, the users can set a spherical potential,
\begin{quote}
\begin{equation*}
\begin{split}V & = k (r_i - r_b)^n \hspace{10pt} ( r_i > r_b) \\
  & = 0,              \hspace{10pt} ( r_i \leq r_b)\end{split}
\end{equation*}\end{quote}

where \(k, n\) and \(r_b\) are a force constant, an exponent, and
a radius of the sphere, respectively, and \(r_i\) is the distance
between the \(i\)-th atom and the center of sphere,
\begin{quote}
\begin{equation*}
\begin{split}r_i = | \mathbf{x}_i - \mathbf{x}_0 | .\end{split}
\end{equation*}
\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=0.300\linewidth]{{fig10_1}.pdf}
\caption{An illustration of a combination of two spherical potentials (black thin circles),
which pulls back atoms that are out of the range (gray) towards the center of
sphere (1 and 2).}\label{\detokenize{10_Boundary:fig10-1}}\end{figure}
\end{quote}

Multiple spheres with different centers and radii can be combined to construct
the potential; for example, two spheres are combined in \hyperref[\detokenize{10_Boundary:fig10-1}]{Fig.\@ \ref{\detokenize{10_Boundary:fig10-1}}}. The
atoms that went out of the sphere (thin line) are pulled back to the nearest
center; the red atom to center 1 and the blue atoms to center 2.

The coordinates of the center can be specified in two ways. The first is
to set the center to a position of atoms in the initial structure (pdbfile)
using \sphinxstylestrong{{[}SELECTOR{]}}.  The other way is to directly specify coordinates of the
center in the input. See the description of options and the examples below for details.

The following options are available to set the spherical potential:


\bigskip\hrule\bigskip


\sphinxstylestrong{spherical\_pot} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

If YES (with type=NOBC), use the spherical boundary potential.
\end{quote}

\sphinxstylestrong{constant} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 10.0} (unit : \(\mathrm{kcal mol}^{-1}\))

The force constant of the potential.
\end{quote}

\sphinxstylestrong{exponent} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 2}

The exponent of the potential.
\end{quote}

\sphinxstylestrong{nindex} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 0}

The number of index, used with center\_select\_index \sphinxstyleemphasis{N}.
\end{quote}

\sphinxstylestrong{center\_select\_index} \sphinxtitleref{N} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : N/A}

The index of center in \sphinxstylestrong{{[}SELECTION{]}}
\end{quote}

\sphinxstylestrong{nfunction} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 0}

The number of function, used with center \sphinxstyleemphasis{N}.
\end{quote}

\sphinxstylestrong{center} \sphinxtitleref{N} \sphinxstyleemphasis{Real} \(\times 3\)
\begin{quote}

\sphinxstylestrong{Default : N/A}

The xyz coordinates of the center.
\end{quote}

\sphinxstylestrong{radius} \sphinxtitleref{N}
\begin{quote}

\sphinxstylestrong{Default : 0.0} (unit : Angstrom)

The radius of sphere.
\end{quote}

\sphinxstylestrong{fixatom} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : YES}

Atoms out of the sphere in the input structure are fixed.
\end{quote}

\sphinxstylestrong{fixlayer} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 1.0} (unit : Angstrom)

If fixatom = YES, atoms within this distance from the potential in the input structure are also fixed.
\end{quote}

\sphinxstylestrong{restart} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : YES}

Use the information in the restart file.
\end{quote}

\begin{sphinxadmonition}{note}{Note:}
The information of the sphere and fixed atoms are saved in a restart file. If the
information exists in rstfile, the options for the spherical potential in \sphinxstylestrong{{[}BOUNDARY{]}} will be
ignored. If you want to re-set the potential, you need to specify \sphinxstylestrong{restart} = NO.
\end{sphinxadmonition}


\section{Examples}
\label{\detokenize{10_Boundary:examples}}\begin{itemize}
\item {} 
Simulations in the gas-phase:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}BOUNDARY\PYG{o}{]}
\PYG{n+nb}{type}       \PYG{o}{=} NOBC      \PYG{c+c1}{\PYGZsh{} non\PYGZhy{}periodic system}
\end{sphinxVerbatim}

\item {} 
Simulations with the periodic boundary condition, where the
box size is set to 64 x 64 x 64.
In this case, the user should not use a restart file as an input,
because the box size in the control is overwritten with that in the restart file.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}BOUNDARY\PYG{o}{]}
\PYG{n+nb}{type}       \PYG{o}{=} PBC       \PYG{c+c1}{\PYGZsh{} periodic boundary condition}
\PYG{n+nv}{box\PYGZus{}size\PYGZus{}x} \PYG{o}{=} \PYG{l+m}{64}.0      \PYG{c+c1}{\PYGZsh{} Box size in the x dimension (Ang)}
\PYG{n+nv}{box\PYGZus{}size\PYGZus{}y} \PYG{o}{=} \PYG{l+m}{64}.0      \PYG{c+c1}{\PYGZsh{} Box size in the y dimension (Ang)}
\PYG{n+nv}{box\PYGZus{}size\PYGZus{}z} \PYG{o}{=} \PYG{l+m}{64}.0      \PYG{c+c1}{\PYGZsh{} Box size in the z dimension (Ang)}
\end{sphinxVerbatim}

\item {} 
Simulations with two spherical potentials around atom number 1 and 100 with a radius of 22.0 Angs.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}BOUNDARY\PYG{o}{]}
\PYG{n+nb}{type}          \PYG{o}{=} NOBC
\PYG{n+nv}{spherical\PYGZus{}pot} \PYG{o}{=} yes
\PYG{n+nv}{constant}      \PYG{o}{=} \PYG{l+m}{2}.0
\PYG{n+nv}{exponent}      \PYG{o}{=} \PYG{l+m}{2}
\PYG{n+nv}{nindex}        \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{center\PYGZus{}select\PYGZus{}index1} \PYG{o}{=} \PYG{l+m}{2}
\PYG{n+nv}{radius1}       \PYG{o}{=} \PYG{l+m}{22}.0
\PYG{n+nv}{fix\PYGZus{}layer}     \PYG{o}{=} \PYG{l+m}{0}.0
\PYG{n+nv}{fixatom}       \PYG{o}{=} no

\PYG{o}{[}SELECTION\PYG{o}{]}
...
\PYG{n+nv}{group2}         \PYG{o}{=} ano:1 or ano:100
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
Be careful not to set too many spheres because it may slow down
the performance. If you want to set the spheres around a protein, instead
of specifying all atoms in a protein, select part of the atoms, for example, by

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n+nv}{group2} \PYG{o}{=} segid:PROA and \PYG{n+nv}{an}\PYG{o}{=}CA
\end{sphinxVerbatim}
\end{sphinxadmonition}

\item {} 
Simulations with two spherical potentials. The center
coordinates are explicitly set by \sphinxstylestrong{center1} and \sphinxstylestrong{center2}.
With \sphinxstylestrong{fixatom} =yes and \sphinxstylestrong{fix\_layer} =1.0 Angs, the atoms that are
farther than 34 Angs from the centers are fixed.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}BOUNDARY\PYG{o}{]}
\PYG{n+nb}{type}          \PYG{o}{=} NOBC      \PYG{c+c1}{\PYGZsh{} [PBC,NOBC]}
\PYG{n+nv}{spherical\PYGZus{}pot} \PYG{o}{=} yes
\PYG{n+nv}{constant}      \PYG{o}{=} \PYG{l+m}{10}.0
\PYG{n+nv}{exponent}      \PYG{o}{=} \PYG{l+m}{2}
\PYG{n+nv}{nfunctions}    \PYG{o}{=} \PYG{l+m}{2}
\PYG{n+nv}{center1}       \PYG{o}{=}  \PYG{l+m}{17}.0, \PYG{l+m}{0}.0, \PYG{l+m}{0}.0    \PYG{c+c1}{\PYGZsh{} [x,y,z]}
\PYG{n+nv}{radius1}       \PYG{o}{=} \PYG{l+m}{35}.0
\PYG{n+nv}{center2}       \PYG{o}{=} \PYGZhy{}17.0, \PYG{l+m}{0}.0, \PYG{l+m}{0}.0    \PYG{c+c1}{\PYGZsh{} [x,y,z]}
\PYG{n+nv}{radius2}       \PYG{o}{=} \PYG{l+m}{35}.0
\PYG{n+nv}{fixatom}       \PYG{o}{=} YES
\PYG{n+nv}{fix\PYGZus{}layer}     \PYG{o}{=} \PYG{l+m}{1}.0
\end{sphinxVerbatim}

\end{itemize}


\chapter{Selection section}
\label{\detokenize{11_Selection:selection-section}}\label{\detokenize{11_Selection:selection}}\label{\detokenize{11_Selection::doc}}

\section{Atom selection}
\label{\detokenize{11_Selection:atom-selection}}
This section is used to select atoms, and define them as a group.
The user can select atoms accorging to their name, index, residue number, segment name, and so on.
The selected group index is used in other sections.
For example, restraint potential can be applied on the group selected in this section,
and the force constant of the potential is specified in the \sphinxstylestrong{{[}RESTRAINTS{]}} section.
\sphinxstylestrong{{[}SELECTION{]}} section is also used in the GENESIS analysis tools to specify the atoms to be analyzed.


\bigskip\hrule\bigskip


\sphinxstylestrong{group}\(\textbf{\textit{N}}\) \sphinxstyleemphasis{expression}
\begin{quote}

The user defines selected atoms as “group1”, “group2”, …, and “group:math:\sphinxtitleref{N}”.
Here, \sphinxstyleemphasis{N} must be a positive integer (\(N \geq 1\)).
The user selects atoms by using keywords and operators with a certain syntax (see table below).
Note that in the table mname (or moleculename, molname) is a molecule name defined by \sphinxstylestrong{mol\_name}.
\end{quote}

\sphinxstylestrong{mole\_name}\(\textbf{\textit{N}}\) \sphinxstyleemphasis{molecule} \sphinxstyleemphasis{starting-residue} \sphinxstyleemphasis{ending-residue}
\begin{quote}

The user defines a molecule by specifying its segment name, first and last residue numbers,
and residue name. \sphinxcode{\sphinxupquote{N}} must be a positive integer (\(N \geq 1\)).
The syntax for the residue selection is as follows:

\sphinxcode{\sphinxupquote{{[}segment name{]}:{[}residue number{]}:{[}residue name{]}}}

For details, see the example below.
\end{quote}

Table. Available keywords and operators in \sphinxcode{\sphinxupquote{group}}.


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabulary}{\linewidth}[t]{|T|T|T|T|}
\hline
\sphinxstyletheadfamily 
expression
&\sphinxstyletheadfamily 
meaning
&\sphinxstyletheadfamily 
example
&\sphinxstyletheadfamily 
other available expression
\\
\hline
an:\sphinxstyleemphasis{name}
&
atom name
&
an:CA
&
atomname, atom\_name
\\
\hline
ai:\sphinxstyleemphasis{number{[}-{[}number{]}{]}}
&
atom index
&
ai:1-5
&
atomindex, atomidx
\\
\hline
atno:\sphinxstyleemphasis{number{[}-{[}number{]}{]}}
&
atom number
&
atno:6
&
atomno
\\
\hline
rnam:\sphinxstyleemphasis{name}
&
residue name
&
rnam:GLY
&
residuename, resname
\\
\hline
rno:\sphinxstyleemphasis{number{[}-{[}number{]}{]}}
&
residue number
&
rno:1-5
&
residueno, resno
\\
\hline
mname:\sphinxstyleemphasis{name}
&
molecule name
&
mname:molA
&
moleculename, molname
\\
\hline
segid:\sphinxstyleemphasis{ID}
&
segment index
&
segid:PROA
&
segmentid, sid
\\
\hline
hydrogen
&
hydrogen atoms
&&
hydrogenatom
\\
\hline
heavy
&
heavy atoms
&&
heavyatom
\\
\hline
all
&
all atoms
&&
\sphinxcode{\sphinxupquote{*}}
\\
\hline
and
&
conjunction
&&
\&
\\
\hline
or
&
logical add
&&
\textbar{}
\\
\hline
not
&
negation
&&
!
\\
\hline
()
&
assemble
&&\\
\hline
\end{tabulary}
\par
\sphinxattableend\end{savenotes}

Table. Available keywords and operators in \sphinxcode{\sphinxupquote{group}} (continued).


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabulary}{\linewidth}[t]{|T|T|T|T|}
\hline
\sphinxstyletheadfamily 
expression
&\sphinxstyletheadfamily 
meaning
&\sphinxstyletheadfamily 
example
&\sphinxstyletheadfamily 
other available expression
\\
\hline
\sphinxstyleemphasis{X} around: \sphinxstyleemphasis{r}
&
atoms around \sphinxstyleemphasis{r} Angstrom of \sphinxstyleemphasis{X}
&
see below
&
around\_atoms
\\
\hline
\sphinxstyleemphasis{X} around\_res: \sphinxstyleemphasis{r}
&
residues around \sphinxstyleemphasis{r} Angstrom of \sphinxstyleemphasis{X}
&
see below
&
around\_residues
\\
\hline
\sphinxstyleemphasis{X} around\_mol: \sphinxstyleemphasis{r}
&
molecules around \sphinxstyleemphasis{r} Angstrom of \sphinxstyleemphasis{X}
&
see below
&
around\_molecules
\\
\hline
\end{tabulary}
\par
\sphinxattableend\end{savenotes}

\begin{sphinxadmonition}{note}{Note:}
\sphinxcode{\sphinxupquote{ai}} and \sphinxcode{\sphinxupquote{atno}} are slightly different. \sphinxcode{\sphinxupquote{ai}} indicates the atom index
which is sequentially re-numbered over all atoms in the system.
On the other hand, \sphinxcode{\sphinxupquote{atno}} is the index of atoms in the PDB file.
Atom index in PDB file (column 2) does not always start from 1, nor is numbered sequentially.
In such cases, \sphinxcode{\sphinxupquote{atno}} is useful to select atoms, although it is a very rare case.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
Atoms that are within a distance of a given atom (X) can be selected by \sphinxcode{\sphinxupquote{around}}.
Note that the coordinates in \sphinxcode{\sphinxupquote{reffile}} is used to judge the distance. If \sphinxcode{\sphinxupquote{reffile}}
is not present, those in input files (\sphinxcode{\sphinxupquote{pdbfile}}, \sphinxcode{\sphinxupquote{crdfile}}, etc.) are used instead.
Coordinates in \sphinxcode{\sphinxupquote{rstfile}} are never used.
\end{sphinxadmonition}


\section{Examples}
\label{\detokenize{11_Selection:examples}}
Select atoms based on their atom name, residue name, or residue number:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}     \PYG{o}{=} resno:1\PYGZhy{}60 and an:CA
\PYG{n+nv}{group2}     \PYG{o}{=} \PYG{o}{(}segid:PROA and not hydrogen\PYG{o}{)} \PYG{p}{\textbar{}} an:CA
\PYG{n+nv}{mole\PYGZus{}name1} \PYG{o}{=} molA PROA:1:TYR PROA:5:MET
\PYG{n+nv}{group3}     \PYG{o}{=} mname:molA and \PYG{o}{(}an:CA or an:C or an:O or an:N\PYG{o}{)}
\end{sphinxVerbatim}

Select atoms around an atom X. In the following examples, X = atom number 100.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}     \PYG{o}{=} atno:100 around:10.0
\PYG{n+nv}{group2}     \PYG{o}{=} atno:100 around\PYGZus{}res:10.0
\PYG{n+nv}{group3}     \PYG{o}{=} atno:100 around\PYGZus{}mol:10.0
\PYG{n+nv}{group4}     \PYG{o}{=} atno:100 around\PYGZus{}mol:10.0 or atno:100
\end{sphinxVerbatim}

In group1, atoms around 10.0 Angstrom of X are selected. Group 2 selects residues around
10.0 Angstrom of X, i.e., if the distance between X and any one of atoms in a residue is
less than 10.0 Angstrom, all atoms of the residue are selected. Group 3 is the same as
group 2, but for a molecule. Note that these commands do NOT select X itself.
In order to include X in the selection, add “or atno:100”, as in group 4.


\bigskip\hrule\bigskip


Select atoms around multiple atoms.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}     \PYG{o}{=} atno:100\PYGZhy{}101 around:10.0
\PYG{n+nv}{group2}     \PYG{o}{=} \PYG{o}{(}sid:PROT around\PYGZus{}res:10.0\PYG{o}{)} and rnam:TIP3
\PYG{n+nv}{group3}     \PYG{o}{=} \PYG{o}{(}rno:1 around:10.0\PYG{o}{)} or rno:1
\end{sphinxVerbatim}

Group 1 selects atoms around 10.0 Angstrom of atom 100 \sphinxstyleemphasis{or} 101. Note that it is NOT
“100 \sphinxstyleemphasis{and} 101” nor a center of 100 and 101. Group 2 is an example to select water
molecules around a protein (segname PROT). Group 3 selects not only the atoms around
residue1 but also the atoms of residue1.


\chapter{Restraints section}
\label{\detokenize{12_Restraints:restraints-section}}\label{\detokenize{12_Restraints:restraints}}\label{\detokenize{12_Restraints::doc}}

\section{Restraint potential}
\label{\detokenize{12_Restraints:restraint-potential}}
\sphinxstylestrong{{[}RESTRAINTS{]}} section contains keywords to define external
restraint functions. The restraint functions are applied to the
selected atom groups in \sphinxstylestrong{{[}SELECTION{]}} section to restrict the
motions of those atoms.

The potential energy of a restaint can be written as:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}U(x) = k\;(x-x_{0})^{n}\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(x\) is a variable (see bellow), \(x_0\) is a reference value,
\(k\) is a force constant, and \(n\) is an exponent factor.


\bigskip\hrule\bigskip



\subsection{General keywords}
\label{\detokenize{12_Restraints:general-keywords}}
\sphinxstylestrong{nfunctions} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: 0}

Number of restraint functions.
\end{quote}

\sphinxstylestrong{function}\(\textbf{\textit{N}}\) \sphinxstyleemphasis{POSI / DIST{[}MASS{]} / ANGLE{[}MASS{]} / DIHED{[}MASS{]} / RMSD{[}MASS{]} / PC{[}MASS{]} / EM}
\begin{quote}

\sphinxstylestrong{Default: N/A}

Type of restraint.
\begin{itemize}
\item {} 
\sphinxstylestrong{POSI}: positional restraint. The reference coordinates are given
by \sphinxstylestrong{reffile}, \sphinxstylestrong{ambreffile}, or \sphinxstylestrong{groreffile} in \sphinxstylestrong{{[}INPUT{]}}.
(see {\hyperref[\detokenize{03_Input:input}]{\sphinxcrossref{\DUrole{std,std-ref}{Input section}}}})

\item {} 
\sphinxstylestrong{DIST{[}MASS{]}}: distance restraint.

\item {} 
\sphinxstylestrong{ANGLE{[}MASS{]}}: angle restraint.

\item {} 
\sphinxstylestrong{DIHED{[}MASS{]}}: dihedral angle restraint.

\item {} 
\sphinxstylestrong{RMSD{[}MASS{]}}: RMSD restraint. \sphinxstyleemphasis{MASS} means mass-weighted RMSD.
Translational and rotational fitting to the reference coordiate
are done before calculating RMSD. The reference coodinate is specified
in the same manner as \sphinxstyleemphasis{POSI}.

\sphinxstyleemphasis{Important Notice (1.1.5 or later)}
Structural fitting method can be defined in \sphinxstylestrong{{[}FITTING{]}} section
({\hyperref[\detokenize{13_Fitting:fitting}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting section}}}}) on 1.1.5 or later. Users of GENESIS 1.1.4 or before
should pay special attention on the fitting scheme.
In versions 1.1.4 or before, translational and rotational fittings
were automatically applied for the atoms concerning RMSD restraint.
(same as the current default setting, \sphinxstyleemphasis{fitting\_method = TR+ROT})

\item {} 
\sphinxstylestrong{PC{[}MASS{]}}: principal component constraint. This option
requires \sphinxstyleemphasis{modefile} in the {\hyperref[\detokenize{03_Input:input}]{\sphinxcrossref{\DUrole{std,std-ref}{Input section}}}}.

\item {} 
\sphinxstylestrong{EM}: cryo-EM flexible fitting (see {\hyperref[\detokenize{19_Experiments:experiments}]{\sphinxcrossref{\DUrole{std,std-ref}{Experiments section}}}})

\end{itemize}

\sphinxstyleemphasis{DIST}, \sphinxstyleemphasis{ANGLE}, \sphinxstyleemphasis{DIHED} impose restraint on distance/angle/dihedral
defined by the selected groups. See \sphinxcode{\sphinxupquote{select\_indexN}} and examples
below for the specification.
\sphinxstyleemphasis{MASS} indicates that the force is applied to the center of mass of the
selected group.
When \sphinxstyleemphasis{MASS} is omitted, the force is applied to the geometric center of
the coordinates.
\sphinxstyleemphasis{MASS} keyword does nothing for groups consist of a single particle.

In \sphinxstylestrong{SPDYN}, \sphinxstyleemphasis{POSI} and \sphinxstyleemphasis{RMSD{[}MASS{]}} restraints are mutually exclusive;
you can use either one or none of them. Two different \sphinxstyleemphasis{POSI} restraints
might not be applied simultaneously, either.

\sphinxstyleemphasis{Notice: POSI, PC, and RMSD restraints can be influenced by the removal
of translational/rotational momentum. See also the notices in the
stoptr\_period parameter in the {[}DYNAMICS{]} section.}
\end{quote}

\sphinxstylestrong{constant}\(\textbf{\textit{N}}\) \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 0.0} (unit: depend on the restraint type)

Force constant of a restraint function.
The unit depends on the type of restraint.
Namely, \({\rm{kcal/mol/{\text{\AA}}^{\textit{n}}}}\) is used in the case of \sphinxstyleemphasis{DIST} and \sphinxstyleemphasis{RMSD},
while \({\rm{kcal/mol/{rad}^{\textit{n}}}}\) in the case of \sphinxstyleemphasis{ANGLE} and \sphinxstyleemphasis{DIHED},
where \(n\) is \sphinxstylestrong{exponent}\(\textbf{\textit{N}}\) specified in this section.
\end{quote}

\sphinxstylestrong{reference}\(\textbf{\textit{N}}\) \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 0.0} (unit: depend on the restraint type)

Reference value of a restraint function. For the positional restraint, the value is ignored.
The unit depends the type of restraint.
Namely, Angstrom is used in the case of \sphinxstyleemphasis{DIST},
while degree (NOT radian) is used in the case of \sphinxstyleemphasis{ANGLE} and \sphinxstyleemphasis{DIHED}.
\end{quote}

\sphinxstylestrong{select\_index}\(\textbf{\textit{N}}\) \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: N/A}

Index of an atom group, to which restraint potentials are applied.
The index must be defined in \sphinxstylestrong{{[}SELECTION{]}} (see {\hyperref[\detokenize{11_Selection:selection}]{\sphinxcrossref{\DUrole{std,std-ref}{Selection section}}}}).
For example, if you specify \sphinxcode{\sphinxupquote{select\_index1 = 1}}, this restraint function
is applied for \sphinxcode{\sphinxupquote{group1}} in the \sphinxstylestrong{{[}SELECTION{]}}.

Number of groups required depends on the type of the restraint function.
\begin{itemize}
\item {} 
\sphinxstyleemphasis{POSI/RMSD{[}MASS{]}}: 1

\item {} 
\sphinxstyleemphasis{DIST{[}MASS{]}}: \(2m\), where \(m = 1, 2, ...\)

\item {} 
\sphinxstyleemphasis{ANGLE{[}MASS{]}}: 3

\item {} 
\sphinxstyleemphasis{DIHED{[}MASS{]}}: 4

\item {} 
\sphinxstyleemphasis{PC{[}MASS{]}}: \(\ge 1\)

\end{itemize}

A group can contain more than single atom.
Suppose we have the following input.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}        \PYG{o}{=} ai:1\PYGZhy{}10
\PYG{n+nv}{group2}        \PYG{o}{=} ai:11\PYGZhy{}20

\PYG{o}{[}RESTRAINTS\PYG{o}{]}
\PYG{n+nv}{nfunctions}    \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{function1}     \PYG{o}{=} DIST
\PYG{n+nv}{constant1}     \PYG{o}{=} \PYG{l+m}{3}.0
\PYG{n+nv}{reference1}    \PYG{o}{=} \PYG{l+m}{10}.0
\PYG{n+nv}{select\PYGZus{}index1} \PYG{o}{=} \PYG{l+m}{1} \PYG{l+m}{2}
\end{sphinxVerbatim}

In this case, the distance restraint is applied for the distance between
geometric centers of group1 and group2. The calculated force is then
scattered to each atom. If \sphinxstyleemphasis{DISTMASS} is given instead of \sphinxstyleemphasis{DIST},
mass centers (mass-weighted average position) are used instead of
geometric centers (not mass-weighted average position).

In the case of \sphinxstyleemphasis{DIST{[}MASS{]}} restraint with more than 2 groups specified
(i.e. \(2m\) with \(m \geq 2\)),
the sum of \(m\) distances will be restrained. See \sphinxcode{\sphinxupquote{exponent\_dist}}
and \sphinxcode{\sphinxupquote{weight\_dist}} parameters for this distance summation.
However, this scheme might not be useful for the standard cases.
\end{quote}

\sphinxstylestrong{direction}\(\textbf{\textit{N}}\) \sphinxstyleemphasis{ALL / X / Y / Z}
\begin{quote}

\sphinxstylestrong{Default : ALL}

Direction of the \sphinxstyleemphasis{POSI} restraint. If \sphinxstyleemphasis{X} or \sphinxstyleemphasis{Y} or \sphinxstyleemphasis{Z} is specified, restraints
along the other two axes are not applied.
\end{quote}

\sphinxstylestrong{exponent}\(\textbf{\textit{N}}\) \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 2}

Exponent factor of the restraint function. The default is the harmonic.
This parameter does not work for \sphinxstyleemphasis{POSI} and \sphinxstyleemphasis{RMSD{[}MASS{]}} restraints
in \sphinxstylestrong{SPDYN}, where the default value, 2, is always used.
\end{quote}

\sphinxstylestrong{exponent\_dist}\(\textbf{\textit{N}}\) \sphinxstyleemphasis{Integer} (\sphinxstylestrong{DIST{[}MASS{]}} only)
\begin{quote}

\sphinxstylestrong{Default : 1}

Exponent factor used in the distance sum calculations.
The sum of distances is expressed as:
\(r_{\rm sum} = \sum_m{w|r_m|^n}\), where
\((1 \leq m \leq {\rm {num\ groups/2}})\),
\(n\) is \sphinxstylestrong{exponent\_dist}\(\textbf{\textit{N}}\),
and \(w\) is \sphinxstylestrong{weight\_dist}\(\textbf{\textit{N}}\).
\end{quote}

\sphinxstylestrong{weight\_dist}\(\textbf{\textit{N}}\) \sphinxstyleemphasis{Real} (\sphinxstylestrong{DIST{[}MASS{]}} only)
\begin{quote}

\sphinxstylestrong{Default : 1.0}

Weight factor used in the distance sum calculations.
\end{quote}

\sphinxstylestrong{mode}\(\textbf{\textit{N}}\) \sphinxstyleemphasis{Integer}
\begin{quote}

Specifies the mode index which is used for the PC (principal component)
restraint. For example, the 1st PC mode can be restrained by specifying
\sphinxcode{\sphinxupquote{mode1=1}}.
\end{quote}


\subsection{Pressure derived from restraints}
\label{\detokenize{12_Restraints:pressure-derived-from-restraints}}
Basically, the pressure calculated from the restraint potential is included
in an internal pressure, which is kept constant during the simulation in the NPT ensemble.
However, the pressure derived from positional and RMSD restraints are treated as an external pressure by default.
Keywords \sphinxstylestrong{pressure\_position} and \sphinxstylestrong{pressure\_rmsd} are used to include those pressures in the internal pressure.
If the simulation with POSI or RMSD restraint showed a strange behaviour
(especially, when a strong force constant is applied), turn on these options.

\sphinxstylestrong{pressure\_position} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

The virial terms from positional restraints are included in pressure evaluation.
\end{quote}

\sphinxstylestrong{pressure\_rmsd} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

The virial terms from RMSD restraints are included in pressure evaluation.
\end{quote}


\subsection{Advanced definition of restraints}
\label{\detokenize{12_Restraints:advanced-definition-of-restraints}}
Restraints can be also defined in an external input file (\sphinxstylestrong{localresfile}).
In this case, number of local restraints must NOT be included in \sphinxstylestrong{nfunctions}.
This option is availabe in \sphinxstylestrong{SPDYN} only.
For details, see {\hyperref[\detokenize{03_Input:input}]{\sphinxcrossref{\DUrole{std,std-ref}{Input section}}}}.


\subsection{Restraints in REUS simulations}
\label{\detokenize{12_Restraints:restraints-in-reus-simulations}}
If you employed a certain restraint term for REUS runs, \sphinxstyleemphasis{nreplica}
of force constants and reference values must be given as a space-separated list.
The above keywords except for \sphinxstylestrong{nfunctions}, \sphinxstylestrong{pressure\_position},
and \sphinxstylestrong{pressure\_rmsd}, must have a serial number,
‘N’, of the function (\(N \geq 1\)).
This serial number is referred when selecting restraints in REUS runs.
For details, see {\hyperref[\detokenize{14_REMD:remd}]{\sphinxcrossref{\DUrole{std,std-ref}{REMD section}}}}.


\section{Examples}
\label{\detokenize{12_Restraints:examples}}
Example of \sphinxstylestrong{{[}RESTRAINTS{]}} section:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}RESTRAINTS\PYG{o}{]}
\PYG{n+nv}{nfunctions}    \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{function1}     \PYG{o}{=} DIST
\PYG{n+nv}{reference1}    \PYG{o}{=} \PYG{l+m}{10}.0
\PYG{n+nv}{constant1}     \PYG{o}{=} \PYG{l+m}{2}.0
\PYG{n+nv}{select\PYGZus{}index1} \PYG{o}{=} \PYG{l+m}{1} \PYG{l+m}{2}        \PYG{c+c1}{\PYGZsh{} group1 and group2 in [SELECTION]}
\end{sphinxVerbatim}

Example of multiple restraints:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}RESTRAINTS\PYG{o}{]}
\PYG{n+nv}{nfunctions}    \PYG{o}{=} \PYG{l+m}{2}

\PYG{n+nv}{function1}     \PYG{o}{=} DIST
\PYG{n+nv}{constant1}     \PYG{o}{=} \PYG{l+m}{2}.0
\PYG{n+nv}{reference1}    \PYG{o}{=} \PYG{l+m}{10}.0       \PYG{c+c1}{\PYGZsh{} in angstrom}
\PYG{n+nv}{select\PYGZus{}index1} \PYG{o}{=} \PYG{l+m}{1} \PYG{l+m}{2}

\PYG{n+nv}{function2}     \PYG{o}{=} DIHED
\PYG{n+nv}{constant2}     \PYG{o}{=} \PYG{l+m}{3}.0
\PYG{n+nv}{reference2}    \PYG{o}{=} \PYG{l+m}{120}.0      \PYG{c+c1}{\PYGZsh{} in degrees}
\PYG{n+nv}{select\PYGZus{}index2} \PYG{o}{=} \PYG{l+m}{3} \PYG{l+m}{4} \PYG{l+m}{5} \PYG{l+m}{6}
\end{sphinxVerbatim}


\chapter{Fitting section}
\label{\detokenize{13_Fitting:fitting-section}}\label{\detokenize{13_Fitting:fitting}}\label{\detokenize{13_Fitting::doc}}

\section{Structure fitting}
\label{\detokenize{13_Fitting:structure-fitting}}
\sphinxstyleemphasis{(In GENESIS 1.1.5 or later only)}
Keywords in \sphinxstylestrong{{[}FITTING{]}} section define a structure superimposition scheme,
which is often employed in targeted MD, steered MD, or String method
(see {\hyperref[\detokenize{15_RPath:rpath}]{\sphinxcrossref{\DUrole{std,std-ref}{RPATH section}}}}) with positional restraint.
In the String method, the reference coordinate for fitting is given by \sphinxstylestrong{fitfile}
in the \sphinxstylestrong{{[}INPUT{]}} section. Otherwise (MD, MIN, REMD), the reference coordinate
is given by \sphinxstylestrong{reffile}, \sphinxstylestrong{ambreffile}, or \sphinxstylestrong{groreffile} in the
\sphinxstylestrong{{[}INPUT{]}} section. Note that this section is not related to
the cryo-EM flexible fitting (see {\hyperref[\detokenize{19_Experiments:experiments}]{\sphinxcrossref{\DUrole{std,std-ref}{Experiments section}}}})


\bigskip\hrule\bigskip


\sphinxstylestrong{fitting\_method} \sphinxstyleemphasis{NO / TR+ROT / XYTR+ZROT}
\begin{quote}

\sphinxstylestrong{Default: TR+ROT}

Type of fitting method.
\begin{itemize}
\item {} 
NO: No fitting routine is applied

\item {} 
TR+ROT: Remove both of translation and rotation

\item {} 
XYTR+ZROT: Remove translation in XY-plane and rotation along the Z-axis

\end{itemize}
\end{quote}

\sphinxstylestrong{fitting\_atom}: \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: N/A}

Index of an atom group which is to be fitted to the reference structure.
In RMSD restraints, Steered MD, or Targeted MD, this should be identical
to the group where the restraint potential is applied.
The index must be defined in \sphinxstylestrong{{[}SELECTION{]}} (see {\hyperref[\detokenize{11_Selection:selection}]{\sphinxcrossref{\DUrole{std,std-ref}{Selection section}}}}).
For example, if you specify \sphinxcode{\sphinxupquote{fitting\_atom = 1}}, the reference atoms
are members of \sphinxcode{\sphinxupquote{group1}} in the \sphinxstylestrong{{[}SELECTION{]}}.
\end{quote}

\sphinxstylestrong{mass\_weight}: \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default: NO}

If the parameter is set to \sphinxstyleemphasis{YES}, mass-weighted fitting is employed.
This parameter should be \sphinxstyleemphasis{YES} for RMSDCOM/PCCOM restraints
and should be \sphinxstyleemphasis{NO} for RMSD/PC restraints. Please make sure that this
parameter is correctly specified when you perform RMSD/RMSDCOM/PC/PCOM
type of calculations.
In the String method, mass-weighted superimposition is not supported.
\end{quote}

\sphinxstylestrong{force\_no\_fitting}: \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default: NO}

\sphinxstyleemphasis{This parameter must not be changed for standard MD runs.}
If the parameter is set to YES and fitting\_method is set to NO,
the fitting routine is turned off.
Translational and rotational fittings are
usually required to calculate correct RMSD values.
So GENESIS simulators (ATDYN and SPDYN) do not allow
\sphinxstyleemphasis{fitting\_method = NO} for simulations involving RMSD calculation
(targeted/steered MD, for example).
But such fitting is not desirable when generating initial
structure set for the String method using Cartesian coordinate as CV (see {\hyperref[\detokenize{15_RPath:rpath}]{\sphinxcrossref{\DUrole{std,std-ref}{RPATH section}}}}).
Actually, \sphinxstyleemphasis{fitting\_method = NO} was implemented just for this specific purpose.
If you are really want to turn off fittings of RMSD calculation for preparation of
initial structure set for String method,
please specify \sphinxstyleemphasis{fitting\_method = NO} and \sphinxstyleemphasis{force\_no\_fitting = YES}.
\end{quote}


\section{Examples}
\label{\detokenize{13_Fitting:examples}}
Example of \sphinxstylestrong{{[}FITTING{]}} section

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}FITTING\PYG{o}{]}
\PYG{n+nv}{fitting\PYGZus{}method} \PYG{o}{=} TR+ROT
\PYG{n+nv}{fitting\PYGZus{}atom}   \PYG{o}{=}  \PYG{l+m}{1}
\PYG{n+nv}{mass\PYGZus{}weight}    \PYG{o}{=}  NO
\end{sphinxVerbatim}


\chapter{REMD section}
\label{\detokenize{14_REMD:remd-section}}\label{\detokenize{14_REMD:remd}}\label{\detokenize{14_REMD::doc}}

\section{Replica-exchange molecular-dynamics simulation (REMD)}
\label{\detokenize{14_REMD:replica-exchange-molecular-dynamics-simulation-remd}}
In the \sphinxstylestrong{{[}REMD{]}} section, the users can specify keywords for Replica-Exchange
Molecular Dynamics (REMD) simulation.
REMD method is one of the enhanced conformational sampling methods
used for systems with rugged free-energy landscapes.
The original temperature-exchange method (T-REMD) is one of the most
widely used methods in biomolecules’ simulations
\sphinxcite{99_References:sugita-1999} \sphinxcite{99_References:mitsutake-2001}.
Here, replicas (or copies) of the original system are prepared,
and different temperatures are assigned to each replica.
Each replica runs in a canonical (NVT) or isobaric-isothermal (NPT)
ensemble, and the temperatures are periodically exchanged between
the neighboring replicas during a simulation.
Exchanging temperature enforces a random walk in temperature space,
allowing the system overcoming energy barriers and sampling much wider
conformational space.

In REMD methods, the transition probability of the replica exchange process is given by the usual Metropolis criterion,
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}w(X \to X') & = \mathrm{min}(1, \frac{P(X')}{P(X)}) = \mathrm{min}(1,\mathrm{exp}(-\Delta)).\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

In the T-REMD method, we have
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\Delta = (\beta_{m} - \beta_{n})\left\{ {E(q^{[j]}) -E(q^{[i]})} \right\},\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(E\) is the potential energy, \(q\) is the position of atoms,
\(\beta\) is the inverse temperature defined by \(\beta = 1/k_B T\),
\(i\) and \(j\) are the replica indexes, and \(m\) and \(n\)
are the parameter indexes.
After the replica exchange, atomic momenta are rescaled as follows:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}p^{[i]'} = \sqrt{\frac{T_{n}}{T_{m}}} p^{[i]}, \qquad p^{[j]'} = \sqrt{\frac{T_{m}}{T_{n}}} p^{[j]},\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(T\) is the temperature and \(p\) is the momenta of atoms.

The transition probability should be independent of the algorithms used:
i.e. constant temperature and constant pressure algorithms.
On the other hand, the momenta-rescaling scheme depends on the algorithm
used in the simulation.
If thermostat and barostat momenta are included in the equations of motion,
these variables should be also rescaled after replica exchange
\sphinxcite{99_References:mori-2010-1} \sphinxcite{99_References:mori-2010-2}.
In GENESIS, barostat momentum is rescaled in the case of T-REMD with
Langevin or Bussi method in NPT, NPAT, and NPgT ensembles.
For the other cases, only atomic momenta are rescaled.

In GENESIS, not only Temperature REMD but also pressure REMD \sphinxcite{99_References:okabe-2001},
surface-tension REMD \sphinxcite{99_References:mori-2013}, REUS (or Hamiltonian REMD) \sphinxcite{99_References:sugita-2000} \sphinxcite{99_References:fukunishi-2002}, replica exchange with solute tempering (REST) \sphinxcite{99_References:terakawa-rest2} \sphinxcite{99_References:kamiya-grest},
and their multi-dimensional combinations are available in both \sphinxstylestrong{ATDYN} and \sphinxstylestrong{SPDYN}.
Basically, these methods can be employed in the NVT, NPT, NPAT, NPgT ensembles,
except for the surface-tension REMD, which is only used in the NPgT ensemble.
REMD simulations in GENESIS require an MPI environment.
At least one MPI process must be assigned to one replica.
For example, when the user wants to employ 32 replicas,
\(32n\) MPI processes are required.

In the following parameters excluding \sphinxstyleemphasis{dimension}, \sphinxstyleemphasis{exchange\_period},
and \sphinxstyleemphasis{iseed}, the last character ‘N’ must be replaced with a positive
integer number (i.e. \(N \geq 1\)), which defines the index of
replica dimension. For example, \sphinxstyleemphasis{type1}, \sphinxstyleemphasis{nreplica1} are the replica type
and number of replicas for the first dimension, respectively.
For details, see the examples below.


\bigskip\hrule\bigskip


\sphinxstylestrong{dimension} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: 1}

Number of dimensions (i.e. number of parameter types to be exchanged)
\end{quote}

\sphinxstylestrong{type}\(\textbf{\textit{N}}\) \sphinxstyleemphasis{TEMPERATURE / PRESSURE / GAMMA / RESTRAINT / REST}
\begin{quote}

\sphinxstylestrong{Default: TEMPERATURE}

Type of parameter to be exchanged in the \(N\)-th dimension
\begin{itemize}
\item {} 
\sphinxstylestrong{TEMPERATURE}: Temperature REMD \sphinxcite{99_References:sugita-1999}

\item {} 
\sphinxstylestrong{PRESSURE}: Pressure REMD \sphinxcite{99_References:okabe-2001}

\item {} 
\sphinxstylestrong{GAMMA}: Surface-tension REMD \sphinxcite{99_References:mori-2013}

\item {} 
\sphinxstylestrong{RESTRAINT}: REUS (or Hamiltonian REMD) \sphinxcite{99_References:sugita-2000} \sphinxcite{99_References:fukunishi-2002}

\item {} 
\sphinxstylestrong{REST}: replica exchange with solute tempering (REST2 or gREST) \sphinxcite{99_References:terakawa-rest2} \sphinxcite{99_References:kamiya-grest}, which is totally different from the original version of REST \sphinxcite{99_References:berne-rest}. Currently, only AMBER and CHARMM force fields are supported.

\item {} 
\sphinxstylestrong{ALCHEMY}: FEP/\(\lambda\)-REMD \sphinxcite{99_References:jiang-2009}

\end{itemize}
\end{quote}

\sphinxstylestrong{nreplica}\(\textbf{\textit{N}}\) \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: 0}

Number of replicas (or parameters) in the \(N\)-th dimension
\end{quote}

\sphinxstylestrong{parameters}\(\textbf{\textit{N}}\) \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: N/A}

List of parameters for each replica in the \(N\)-th dimension.
Parameters must be given as a space-separated list,
and the total number of parameters must be equal to \sphinxstylestrong{nreplicaN}.
In case of REUS (type = RESTRAINT), parameters must be
specified in \sphinxstylestrong{{[}RESTRAINTS{]}} section (see the sample below).
In case of gREST (type = REST), these parameters are considered
as temperature of solute region.
Note that the order of the parameters in this list must NOT be changed
before and after the restart run, even if the parameters are exchanged
during the REMD simulation.
\end{quote}

\sphinxstylestrong{exchange\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: 100}

Frequency of the parameter exchange attempt.
If “exchange\_period = 0” is specified,
REMD simulation is carried out without parameter exchange,
which is useful to equilibrate the system in a condition
assinged to each replica before performing the production run.
\end{quote}

\sphinxstylestrong{cyclic\_params}\(\textbf{\textit{N}}\) \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default: NO}

Turn on or off the periodicity of the parameters in the \(N\)-th dimension.
If “cyclic\_paramsN = YES” is specified, the first and last parameters are
considered as neighbouring parameters.
This option can be applicabe to all parameter types.
Basically, this is useful in the case of REUS in dihedral angle space,
since the dihedral angle has a periodicity.
\end{quote}

\sphinxstylestrong{iseed} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: 3141592}

Random number seed in the replica exchange scheme.
If this is not specified explicitly, iseed is taken over from the restart file.
\end{quote}

\begin{sphinxadmonition}{note}{Note:}
In the {[}ENSEMBLE{]} section, there is also a parameter “temperature”.
In the T-REMD simulation, this temperature is ignored, even if it is specified explicitly.
Similarly, pressure and gamma in the {[}ENSEMBLE{]} section are ignored in the P-REMD
and surface-tension REMD simulations, respectively.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
When multi-dimensional REMD is carried out, parameters are exchanged alternatively.
For example, in TP-REMD (type1 = TEMPERATURE and type2 = PRESSURE),
there is a temperature exchange first, followed by a pressure exchange.
This is repeated during the simulations.
\end{sphinxadmonition}


\section{Replica-exchange umbrella-sampling (REUS)}
\label{\detokenize{14_REMD:replica-exchange-umbrella-sampling-reus}}
\sphinxstylestrong{rest\_function}\(\textbf{\textit{N}}\) (for \sphinxstylestrong{REUS} only)
\begin{quote}

Index of the restraint function to be used in the REUS simulation.
The detailed parameters in the restraint function (e.g., force constant and reference)
are defined in the \sphinxstylestrong{{[}RESTRAINTS{]}} section (see {\hyperref[\detokenize{12_Restraints:restraints}]{\sphinxcrossref{\DUrole{std,std-ref}{Restraints section}}}}).
Note that the order of the parameters in the \sphinxstylestrong{{[}RESTRAINTS{]}} section
must NOT be changed before and after the restart run,
even if the parameters are exchanged during the REUS simulation.

\sphinxstylestrong{GENESIS} supports not only on-grid but also off-grid schemes.
In the off-grid REUS, multiple restraints are merged into a single reaction coordinate (see example below).
Those restraints are defined in \sphinxstylestrong{{[}RESTRAINTS{]}} section,
where the number of parameters (const and reference) must be equal to \sphinxstyleemphasis{nreplicaN}.
Note that this kinds of combined axis can be used only for restraints,
other types (such as combined tempreature-pressure or temperature-restraint
coordinate) are not available currently.
\end{quote}

\begin{sphinxadmonition}{note}{Note:}
Positional restraint is not available for REUS.
In \sphinxstylestrong{SPDYN}, PCA restraint is not available for REUS.
The control file format was completely changed after verion 1.1.0,
since the off-grid REUS scheme was introduced.
When the users use the old control file, please be careful.
\end{sphinxadmonition}


\section{Replica-exchange with solute-tempering (gREST)}
\label{\detokenize{14_REMD:replica-exchange-with-solute-tempering-grest}}
\sphinxstylestrong{select\_index}\(\textbf{\textit{N}}\) \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: N/A}

Index of an atom group. The selected atoms are considered as “solute” in gREST.
The index must be defined in \sphinxstylestrong{{[}SELECTION{]}} (see {\hyperref[\detokenize{11_Selection:selection}]{\sphinxcrossref{\DUrole{std,std-ref}{Selection section}}}}).
\end{quote}

\sphinxstylestrong{param\_type}\(\textbf{\textit{N}}\) \sphinxstyleemphasis{ALL / BOND / ANGLE / UREY / DIHEDRAL / IMPROPER / CMAP / CHARGE / LJ}
\begin{quote}

\sphinxstylestrong{Default: ALL}

Solute energy terms for gREST \sphinxcite{99_References:kamiya-grest} simulations.
Energy terms selected by this parameter in the solute atom group
(defined by \sphinxstyleemphasis{select\_indexN}) are considered as “solute”
(scaled according to solute temperature) in
gREST. Other terms are considered as “solvent” (kept intact).
Solute-solvent terms are automatically determined from the solute
selection. You can specify multiple terms (see examples).
The parameter names are case-insensitive as follows:
\begin{itemize}
\item {} 
\sphinxstylestrong{ALL}: all the available energy terms.

\item {} 
\sphinxstylestrong{BOND}: (aliases: \sphinxstylestrong{B}, \sphinxstylestrong{BONDS}): 1-2 bonding terms.

\item {} 
\sphinxstylestrong{ANGLE}: (aliases: \sphinxstylestrong{A}, \sphinxstylestrong{ANGLES}): 1-2-3 angle terms.

\item {} 
\sphinxstylestrong{UREY}: (aliases: \sphinxstylestrong{U}, \sphinxstylestrong{UREYS}): Urey-Bradley terms.

\item {} 
\sphinxstylestrong{DIHEDRAL}: (aliases: \sphinxstylestrong{D}, \sphinxstylestrong{DIHEDRALS}): 1-2-3-4 dihedral terms.

\item {} 
\sphinxstylestrong{IMPROPER}: (aliases: \sphinxstylestrong{I}, \sphinxstylestrong{IMPROPERS}): improper torsion terms.

\item {} 
\sphinxstylestrong{CMAP}: (aliases: \sphinxstylestrong{CM}, \sphinxstylestrong{CMAPS}): CMAP terms.

\item {} 
\sphinxstylestrong{CHARGE}: (aliases: \sphinxstylestrong{C}, \sphinxstylestrong{CHARGES}): coulombic interaction terms.

\item {} 
\sphinxstylestrong{LJ}: (aliases: \sphinxstylestrong{L}, \sphinxstylestrong{LJS}): Lennard-Jones interaction terms.

\end{itemize}
\end{quote}

\begin{sphinxadmonition}{note}{Note:}
Note that restraint energy terms defined in \sphinxstylestrong{{[}RESTRAINTS{]}}
cannot be treated as solute terms.
They never be affected by gREST solute temperatures.
In SPDYN, water atoms cannot be specified as “solute” now.
This limitation will be removed in the future version.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
When the coulombic interaction terms are considered as the solute,
the solute region should have a net charge of 0 for an adequate
PME calculation.
\end{sphinxadmonition}


\section{Examples}
\label{\detokenize{14_REMD:examples}}
Basically, REMD simulations in \sphinxstylestrong{GENESIS} can be carried out by
just adding the \sphinxstylestrong{{[}REMD{]}} section
in the control file of a normal MD simulation.
For details, see the online Tutorial (\sphinxurl{https://www.r-ccs.riken.jp/labs/cbrt/tutorials2019/}).


\subsection{T-REMD}
\label{\detokenize{14_REMD:t-remd}}
If the users want to carry out T-REMD simulations with 4 replicas in the NVT ensemble,
where each replica has the temperature 298.15, 311.79, 321.18, or 330.82 K,
and replica exchange is attempted every 1000 steps,
the following section is added to the control file of a normal MD simulation in the NVT ensemble:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}REMD\PYG{o}{]}
\PYG{n+nv}{dimension}       \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{exchange\PYGZus{}period} \PYG{o}{=} \PYG{l+m}{1000}
\PYG{n+nv}{type1}           \PYG{o}{=} TEMPERATURE
\PYG{n+nv}{nreplica1}       \PYG{o}{=} \PYG{l+m}{4}
\PYG{n+nv}{parameters1}     \PYG{o}{=} \PYG{l+m}{298}.15 \PYG{l+m}{311}.79 \PYG{l+m}{321}.18 \PYG{l+m}{330}.82
\end{sphinxVerbatim}

As for the T-REMD simulation in the NPT ensemble, the users add this section
to the control file of a normal MD simulation in the NPT ensemble.
The REMD temperature generator (\sphinxurl{http://folding.bmc.uu.se/remd/}) is
a useful tool to set the target temperature of each replica.


\subsection{Two-dimensional REMD (T-REMD/REUS)}
\label{\detokenize{14_REMD:two-dimensional-remd-t-remd-reus}}
The following is an example of two-dimensinal REMD, where
temperature and restraint are exchanged alternatively,
The 1st dimension is T-REMD with 8 parameters,
and 2nd dimension is REUS in distance space with 4 parameters.
In total, 8 x 4 = 32 replicas are used:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}REMD\PYG{o}{]}
\PYG{n+nv}{dimension}       \PYG{o}{=} \PYG{l+m}{2}
\PYG{n+nv}{exchange\PYGZus{}period} \PYG{o}{=} \PYG{l+m}{1000}
\PYG{n+nv}{type1}           \PYG{o}{=} TEMPERATURE
\PYG{n+nv}{nreplica1}       \PYG{o}{=} \PYG{l+m}{8}
\PYG{n+nv}{parameters1}     \PYG{o}{=} \PYG{l+m}{298}.15 \PYG{l+m}{311}.79 \PYG{l+m}{321}.18 \PYG{l+m}{330}.82 \PYG{l+m}{340}.70 \PYG{l+m}{350}.83 \PYG{l+m}{361}.23 \PYG{l+m}{371}.89
\PYG{n+nv}{type2}           \PYG{o}{=} RESTRAINT
\PYG{n+nv}{nreplica2}       \PYG{o}{=} \PYG{l+m}{4}
\PYG{n+nv}{rest\PYGZus{}function2}  \PYG{o}{=} \PYG{l+m}{1}

\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}          \PYG{o}{=} ai:25
\PYG{n+nv}{group2}          \PYG{o}{=} ai:392

\PYG{o}{[}RESTRAINTS\PYG{o}{]}
\PYG{n+nv}{nfunctions}      \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{function1}       \PYG{o}{=} DIST
\PYG{n+nv}{constant1}       \PYG{o}{=}  \PYG{l+m}{2}.0   \PYG{l+m}{2}.0   \PYG{l+m}{2}.0   \PYG{l+m}{2}.0
\PYG{n+nv}{reference1}      \PYG{o}{=} \PYG{l+m}{10}.0  \PYG{l+m}{10}.5  \PYG{l+m}{11}.0  \PYG{l+m}{11}.5
\PYG{n+nv}{select\PYGZus{}index1}   \PYG{o}{=} \PYG{l+m}{1} \PYG{l+m}{2}
\end{sphinxVerbatim}

These sections are added to the control file of a normal MD simulation.


\subsection{Off-grid REUS}
\label{\detokenize{14_REMD:off-grid-reus}}
Example of off-grid REUS (merge two restraints into single reaction
coordinate), where distance and dihedral restraints are merged into
single reaction coordinate. First values of restraints ((2.0,10.0) for
distance, (10,-40) for dihedral) will be used for the first replica,
the fourth parameters ((2.0,11.5) for distance, (10,-10) for dihedral)
will be used for the fourth replica:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}REMD\PYG{o}{]}
\PYG{n+nv}{dimension}       \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{exchange\PYGZus{}period} \PYG{o}{=} \PYG{l+m}{1000}
\PYG{n+nv}{type1}           \PYG{o}{=} RESTRAINT           \PYG{c+c1}{\PYGZsh{} REUS}
\PYG{n+nv}{nreplica1}       \PYG{o}{=} \PYG{l+m}{4}
\PYG{n+nv}{rest\PYGZus{}function1}  \PYG{o}{=} \PYG{l+m}{1} \PYG{l+m}{2}                 \PYG{c+c1}{\PYGZsh{} off\PYGZhy{}grid REUS}

\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}          \PYG{o}{=} ai:25
\PYG{n+nv}{group2}          \PYG{o}{=} ai:392
\PYG{n+nv}{group3}          \PYG{o}{=} ai:72
\PYG{n+nv}{group4}          \PYG{o}{=} ai:73
\PYG{n+nv}{group5}          \PYG{o}{=} ai:74
\PYG{n+nv}{group6}          \PYG{o}{=} ai:75

\PYG{o}{[}RESTRAINTS\PYG{o}{]}
\PYG{n+nv}{nfunctions}      \PYG{o}{=} \PYG{l+m}{2}

\PYG{n+nv}{function1}       \PYG{o}{=} DIST
\PYG{n+nv}{constant1}       \PYG{o}{=}  \PYG{l+m}{2}.0  \PYG{l+m}{2}.0  \PYG{l+m}{2}.0  \PYG{l+m}{2}.0  \PYG{c+c1}{\PYGZsh{} num of values must be nreplica1}
\PYG{n+nv}{reference1}      \PYG{o}{=} \PYG{l+m}{10}.0 \PYG{l+m}{10}.5 \PYG{l+m}{11}.0 \PYG{l+m}{11}.5
\PYG{n+nv}{select\PYGZus{}index1}   \PYG{o}{=} \PYG{l+m}{1} \PYG{l+m}{2}

\PYG{n+nv}{function2}       \PYG{o}{=} DIHED
\PYG{n+nv}{constant2}       \PYG{o}{=}   \PYG{l+m}{10}   \PYG{l+m}{10}   \PYG{l+m}{10}   \PYG{l+m}{10}  \PYG{c+c1}{\PYGZsh{} num of values must be nreplica1}
\PYG{n+nv}{reference2}      \PYG{o}{=}  \PYGZhy{}40  \PYGZhy{}30  \PYGZhy{}20  \PYGZhy{}10
\PYG{n+nv}{select\PYGZus{}index2}   \PYG{o}{=} \PYG{l+m}{3} \PYG{l+m}{4} \PYG{l+m}{5} \PYG{l+m}{6}
\end{sphinxVerbatim}


\subsection{gREST}
\label{\detokenize{14_REMD:grest}}
In this example, the dihedral, CMAP, and LJ energy terms
in the selected atom groups are treated as “solute”.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}REMD\PYG{o}{]}
\PYG{n+nv}{dimension}       \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{exchange\PYGZus{}period} \PYG{o}{=} \PYG{l+m}{1000}
\PYG{n+nv}{type1}           \PYG{o}{=} REST
\PYG{n+nv}{nreplica1}       \PYG{o}{=} \PYG{l+m}{4}
\PYG{n+nv}{parameters1}     \PYG{o}{=} \PYG{l+m}{300}.0 \PYG{l+m}{310}.0 \PYG{l+m}{320}.0 \PYG{l+m}{330}.0  \PYG{c+c1}{\PYGZsh{} solute temperatures}
\PYG{n+nv}{param\PYGZus{}type1}     \PYG{o}{=} D CM L                   \PYG{c+c1}{\PYGZsh{} dihedral, CMAP, and LJ}
\PYG{n+nv}{select\PYGZus{}index1}   \PYG{o}{=} \PYG{l+m}{1}

\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}          \PYG{o}{=} ai:1\PYGZhy{}313
\end{sphinxVerbatim}

T-REMD in the two-dimensional REMD (T-REMD/REUS) may
be replaced with gREST (gREST/REUS \sphinxcite{99_References:re-2019}) to reduce the required number of replicas.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}REMD\PYG{o}{]}
\PYG{n+nv}{dimension}       \PYG{o}{=} \PYG{l+m}{2}
\PYG{n+nv}{exchange\PYGZus{}period} \PYG{o}{=} \PYG{l+m}{1000}
\PYG{n+nv}{type1}           \PYG{o}{=} REST
\PYG{n+nv}{nreplica1}       \PYG{o}{=} \PYG{l+m}{4}
\PYG{n+nv}{parameters1}     \PYG{o}{=} \PYG{l+m}{300}.0 \PYG{l+m}{310}.0 \PYG{l+m}{320}.0 \PYG{l+m}{330}.0  \PYG{c+c1}{\PYGZsh{} solute temperatures}
\PYG{n+nv}{param\PYGZus{}type1}     \PYG{o}{=} D CM L                   \PYG{c+c1}{\PYGZsh{} dihedral, CMAP, and LJ}
\PYG{n+nv}{select\PYGZus{}index1}   \PYG{o}{=} \PYG{l+m}{3}
\PYG{n+nv}{type2}           \PYG{o}{=} RESTRAINT
\PYG{n+nv}{nreplica2}       \PYG{o}{=} \PYG{l+m}{4}
\PYG{n+nv}{rest\PYGZus{}function2}  \PYG{o}{=} \PYG{l+m}{1}

\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}          \PYG{o}{=} ai:25
\PYG{n+nv}{group2}          \PYG{o}{=} ai:392
\PYG{n+nv}{group3}          \PYG{o}{=} ai:1\PYGZhy{}313

\PYG{o}{[}RESTRAINTS\PYG{o}{]}
\PYG{n+nv}{nfunctions}      \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{function1}       \PYG{o}{=} DIST
\PYG{n+nv}{constant1}       \PYG{o}{=}  \PYG{l+m}{2}.0   \PYG{l+m}{2}.0   \PYG{l+m}{2}.0   \PYG{l+m}{2}.0
\PYG{n+nv}{reference1}      \PYG{o}{=} \PYG{l+m}{10}.0  \PYG{l+m}{10}.5  \PYG{l+m}{11}.0  \PYG{l+m}{11}.5
\PYG{n+nv}{select\PYGZus{}index1}   \PYG{o}{=} \PYG{l+m}{1} \PYG{l+m}{2}
\end{sphinxVerbatim}


\chapter{RPATH section}
\label{\detokenize{15_RPath:rpath-section}}\label{\detokenize{15_RPath:rpath}}\label{\detokenize{15_RPath::doc}}

\section{Reaction Path Search}
\label{\detokenize{15_RPath:reaction-path-search}}
In the \sphinxstylestrong{{[}RPATH{]}} section, users can specify keywords for finding the
reaction path. The path search is carried out in two modes: the
minimum energy path (MEP) and the minimum free-energy path (MFEP).
The former searches for the energetically most favorable pathway on the
potential energy surface (PES), while the latter does the same on the
free-energy surface (FES).
The MEP search is used to find relatively fast
processes such as chemical reactions (very likely along with QM/MM), in
which the environment can be regarded more or less rigid.
On the other hand, the MFEP reveals large-scale conformational changes
of biomolecules by searching the path on a FES, where fast molecular
motions are averaged out.
In both cases, the path is
represented by a chain-of-replica, which evolves on the energy surface
so as to minimize the forces in the transverse direction.


\bigskip\hrule\bigskip


\sphinxstylestrong{rpathmode} \sphinxstyleemphasis{MFEP/MEP}
\begin{quote}

\sphinxstylestrong{Default: MFEP}

Specify \sphinxstylestrong{MFEP} or \sphinxstylestrong{MEP} to invoke the MFEP or MEP search.
\end{quote}


\bigskip\hrule\bigskip



\section{Minimum Free-Energy Path (MFEP) Search}
\label{\detokenize{15_RPath:minimum-free-energy-path-mfep-search}}
The MFEP search is invoked by specifying \sphinxstylestrong{rpathmode = MFEP}.
The path search is carried out by the string method, which is a powerful
sampling technique to find a path connecting two stable conformational
states.  This method is widely used for investigating large-scale
conformational changes of biomolecules where time-scale of the
transitions are not reachable in brute-force simulations.

There are three major algorithms in the string method:
the mean forces string method \sphinxcite{99_References:string-mean-forces},
the on-the-fly string method \sphinxcite{99_References:string-on-the-fly},
and the string method of swarms of trajectories \sphinxcite{99_References:string-swarms}.
Among these algorithgms, the mean forces
string method is available in \sphinxstylestrong{ATDYN} and \sphinxstylestrong{SPDYN} \sphinxcite{99_References:string-genesis}.

In the mean-forces string method, the pathway is represented by discretized
points (called images) in the collective variable (CV) space.
The current GENESIS supports distances, angles, dihedrals, Cartesian
coordinates, and principal components for CVs
(note that different types of CVs cannot be mixed in GENESIS. For example,
users cannot mix distance and angle).

In the calculation, each image is assigned to each replica, and
a replica samples mean forces and an average metric
tensor around its own image by short MD simulation (ps to ns length) with
restraints.
The restraints are imposed using the image coordinates as their reference values.
After the short simulation , each image is
evolved according to the mean force and metric tensor.
Then, smoothing and re-parametrization of images are performed
and go to the next cycle.

Image coordinates are written in rpath files (\sphinxstylestrong{rpathfile} keyword)
which user can specify in \sphinxstylestrong{{[}OUTPUT{]}} section.
This file provides the trajectory of image coordinates.
Columns correspond to CVs and rows are time steps.
These values are written at the same timing with \sphinxstylestrong{dcdfile}
(specified by \sphinxstylestrong{crdout\_period} in \sphinxstylestrong{{[}DYNAMICS{]}} section).

For the string method calculation, an initial pathway in the CV space
and atomistic coordinates around the pathway are required. For preparing these,
targeted or steered MD methods are recommended.


\bigskip\hrule\bigskip


\sphinxstylestrong{nreplica} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: 1}

Number of replicas (images) for representing the pathway.
\end{quote}

\sphinxstylestrong{rpath\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: 0}

Time-step period during which the mean-forces acting on the
images are evaluated. After evaluating the mean-forces,
the images are updated according to the mean-forces, then go to the
next cycle. If \sphinxstylestrong{rpath\_period = 0}, images are not updated. This
option is used for equilibration or umbrella sampling around the
pathway.
\end{quote}

\sphinxstylestrong{delta} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 0.0}

Step-size for steepest descent update of images.
\end{quote}

\sphinxstylestrong{smooth} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 0.0}

Smoothing parameter which controls the aggressiveness of the smoothing.
Values from 0.0 to 0.1 are recommended,
where “smooth = 0.0” means no-smoothing
\end{quote}

\sphinxstylestrong{rest\_function} \sphinxstyleemphasis{List of Integers}
\begin{quote}

\sphinxstylestrong{Default: N/A}

List of restraint function indices
defined in \sphinxstylestrong{{[}RESTRAINTS{]}} section (see {\hyperref[\detokenize{12_Restraints:restraints}]{\sphinxcrossref{\DUrole{std,std-ref}{Restraints section}}}}).
Specified restraints are defined as CVs, and \sphinxstyleemphasis{nreplica} images
(replicas) are created, where a set of corresponding restraint
reference values is assigned to each image.
Force constants in \sphinxstylestrong{{[}RESTRAINTS{]}} are also used for evaluation
of mean-forces.
\end{quote}

\sphinxstylestrong{fix\_terminal} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default: NO}

If \sphinxstylestrong{fix\_terminal = YES} is specified, the two terminal images are
always fixed and not updated. This is useful if the terminal images
correspond to crystal structures and users do not want to move them.
\end{quote}

\sphinxstylestrong{use\_restart} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : YES}

Restart file generated by the string method calculation includes the
last snapshot of images. If \sphinxstylestrong{use\_restart = YES} is specified,
the reference values in \sphinxstylestrong{{[}RESTRAINTS{]}} will be overwritten by
the values in the restart file. Note that force constants are not
overwritten.
\end{quote}

\begin{sphinxadmonition}{note}{Note:}
The following options are needed in the \sphinxstylestrong{{[}FITTING{]}} section when
Cartesian coordinates are used for CVs.

\sphinxstylestrong{fitting\_method} \sphinxstyleemphasis{TR+ROT / XYTR+ZROT / NO}
\begin{quote}

If this keyword is specified, rot-translational elements are
removed from the mean-force estimation by fitting instaneous
structures to the reference coordinates given by \sphinxstylestrong{fitfile}.
\end{quote}

\sphinxstylestrong{fitting\_atom} \sphinxstyleemphasis{List of Integers}
\begin{quote}

The user can specify index of an atom group which are fitted
to the reference structure.
Usually, the same atoms as CVs are selected.
\end{quote}
\end{sphinxadmonition}


\section{Minimum Energy Path (MEP) Search}
\label{\detokenize{15_RPath:minimum-energy-path-mep-search}}
The MEP search is available only in \sphinxstylestrong{ATDYN}.

The calculation is invoked by specifying \sphinxstylestrong{rpathmode = MEP} \sphinxcite{99_References:yagi-2021}.
Cartesian coordinates of atoms selected via \sphinxstylestrong{mepatm\_select\_index}
(denoted MEP atoms)  are employed for the path search.
Currently, other coordinates can not be used as CVs.
Starting from a set of images along an initial path, e.g., a linear
interpolation between the reactant and product,
the coordinates of the surrounding atoms are first energy minimized
with MEP atoms held fixed. (The {[}MINIMIZE{]} section is thus
required in the input as well.) Then, the forces
acting on MEP atoms are evaluated for each image, and the
images are evolved so as to minimize the forces in the transverse
direction by the string method \sphinxcite{99_References:string-min-ene-path}.
The process is repeated either until the convergence
threshold is met {[}variation in the energy (\sphinxstylestrong{tol\_energy}) and the
path length (\sphinxstylestrong{tol\_path}){]}, or
until the number of iterations reaches the maximum number (\sphinxstylestrong{ncycle}).

Another search algorithm, the nudged elastic band (NEB) method
\sphinxcite{99_References:nudged-elastic-band} is also implemented, which differs from
the string method in how the images evolve.  Note, however, that
the NEB is still experimental at this moment.


\bigskip\hrule\bigskip


\sphinxstylestrong{mepatm\_select\_index} \sphinxstyleemphasis{Integer}
\begin{quote}

Index of a group of atoms which is treated as MEP atoms.
The index must be defined in \sphinxstylestrong{{[}SELECTION{]}} (see {\hyperref[\detokenize{11_Selection:selection}]{\sphinxcrossref{\DUrole{std,std-ref}{Selection section}}}}).
\end{quote}

\sphinxstylestrong{ncycle} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: 1000}

Maximum number of cycle.
\end{quote}

\sphinxstylestrong{nreplica} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: 1}

Number of replicas (images) for representing the pathway.
\end{quote}

\begin{sphinxadmonition}{note}{Note:}
If MPI processes are larger than \sphinxstylestrong{nreplica}, the MPI processes must
be a multiple of \sphinxstylestrong{nreplica}. For example, if \sphinxstylestrong{nreplica = 16},
MPI processes must be 16, 32, 48, etc. If MPI processes are smaller
than \sphinxstylestrong{nreplica}, the MPI processes must be a divisor of \sphinxstylestrong{nreplica}.
For example, the calculation with \sphinxstylestrong{nreplica = 16}  can be performed
using 1, 2, 4, and 8 MPI processes.
\end{sphinxadmonition}

\sphinxstylestrong{eneout\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 10}

Frequency of the output of the energy profile and path length to the
standard output.
\end{quote}

\sphinxstylestrong{crdout\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 0}

Frequency of coordinates outputs. Note that coordinate outputs are turned
off for the minimization (\sphinxstylestrong{crdout\_period} in the {[}MINIMIZE{]} section).
\end{quote}

\sphinxstylestrong{rstout\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 0}

Frequency of restart file updates. Note that the updates are turned
off for the minimization (\sphinxstylestrong{rstout\_period} in the {[}MINIMIZE{]} section).
\end{quote}

\sphinxstylestrong{tol\_energy} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.01} (unit : kcal/mol)

Tolerence of convergence for the energy.
\end{quote}

\sphinxstylestrong{tol\_path} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.01} (unit : \(\text{\AA}\))

Tolerence of convergence for the path length.
\end{quote}

\sphinxstylestrong{massweightcoord} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

Use mass weighted Cartesian.
\end{quote}

\sphinxstylestrong{method} \sphinxstyleemphasis{STRING/NEB}
\begin{quote}

\sphinxstylestrong{Default: STRING}

Choose the algorithm of a MEP search.
\end{quote}


\bigskip\hrule\bigskip


Options for String.

\sphinxstylestrong{delta} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 0.001}

Step-size for steepest descent update of images.
\end{quote}


\bigskip\hrule\bigskip


Options for NEB.

\sphinxstylestrong{k\_spring} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 10.0} \({\rm{kcal/mol/{\text{\AA}}^{2}}}\)

Spring constant of the force that connects the images
\end{quote}

\sphinxstylestrong{ncorrection} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 10}

Number of corrections to build the inverse Hessian.
\end{quote}

\sphinxstylestrong{lbfgs\_bnd} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : YES}

Set a boundary to move atoms in each step of image update.
\end{quote}

\sphinxstylestrong{lbfgs\_bnd\_qmonly} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

Set the boundary only to QM atoms.
\end{quote}

\sphinxstylestrong{lbfgs\_bnd\_maxmove} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.1} (unit : \(\text{\AA}\))

The maximum size of move in each step.
\end{quote}


\bigskip\hrule\bigskip



\section{Examples}
\label{\detokenize{15_RPath:examples}}
Example of alanine-tripeptide with 16 replicas (images). Two dihedral
angles are specified as the collective variables.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}RPATH\PYG{o}{]}
\PYG{n+nv}{nreplica}          \PYG{o}{=} \PYG{l+m}{16}
\PYG{n+nv}{rpath\PYGZus{}period}      \PYG{o}{=} \PYG{l+m}{1000}
\PYG{n+nv}{delta}             \PYG{o}{=} \PYG{l+m}{0}.02
\PYG{n+nv}{smooth}            \PYG{o}{=} \PYG{l+m}{0}.0
\PYG{n+nv}{rest\PYGZus{}function}     \PYG{o}{=} \PYG{l+m}{1} \PYG{l+m}{2}

\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}        \PYG{o}{=} atomindex:15
\PYG{n+nv}{group2}        \PYG{o}{=} atomindex:17
\PYG{n+nv}{group3}        \PYG{o}{=} atomindex:19
\PYG{n+nv}{group4}        \PYG{o}{=} atomindex:25
\PYG{n+nv}{group5}        \PYG{o}{=} atomindex:27

\PYG{o}{[}RESTRAINTS\PYG{o}{]}
\PYG{n+nv}{nfunctions}    \PYG{o}{=} \PYG{l+m}{2}

\PYG{n+nv}{function1}     \PYG{o}{=} DIHED
\PYG{n+nv}{constant1}     \PYG{o}{=} \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+s+se}{\PYGZbs{}}
                \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0
\PYG{n+nv}{reference1}    \PYG{o}{=} \PYGZhy{}40.0 \PYGZhy{}40.0 \PYGZhy{}40.0 \PYGZhy{}40.0 \PYGZhy{}40.0 \PYGZhy{}40.0 \PYGZhy{}40.0 \PYGZhy{}40.0 \PYG{l+s+se}{\PYGZbs{}}
                \PYGZhy{}40.0 \PYGZhy{}40.0 \PYGZhy{}40.0 \PYGZhy{}40.0 \PYGZhy{}40.0 \PYGZhy{}40.0 \PYGZhy{}40.0 \PYGZhy{}40.0
\PYG{n+nv}{select\PYGZus{}index1} \PYG{o}{=} \PYG{l+m}{1} \PYG{l+m}{2} \PYG{l+m}{3} \PYG{l+m}{4}  \PYG{c+c1}{\PYGZsh{} PHI}

\PYG{n+nv}{function2}     \PYG{o}{=} DIHED
\PYG{n+nv}{constant2}     \PYG{o}{=} \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+s+se}{\PYGZbs{}}
                \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0 \PYG{l+m}{100}.0
\PYG{n+nv}{reference2}    \PYG{o}{=} \PYGZhy{}45.0 \PYGZhy{}33.0 \PYGZhy{}21.0 \PYGZhy{}9.0 \PYG{l+m}{3}.0 \PYG{l+m}{15}.0 \PYG{l+m}{27}.0 \PYG{l+m}{39}.0 \PYG{l+s+se}{\PYGZbs{}}
                 \PYG{l+m}{51}.0  \PYG{l+m}{63}.0  \PYG{l+m}{75}.0 \PYG{l+m}{87}.0 \PYG{l+m}{99}.0 \PYG{l+m}{111}.0 \PYG{l+m}{123}.0 \PYG{l+m}{135}.0
\PYG{n+nv}{select\PYGZus{}index2} \PYG{o}{=} \PYG{l+m}{2} \PYG{l+m}{3} \PYG{l+m}{4} \PYG{l+m}{5}  \PYG{c+c1}{\PYGZsh{} PSI}
\end{sphinxVerbatim}

Here is another example of Cartesian coordiante CVs for the same
alanine-tripeptide.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}INPUT\PYG{o}{]}
... skip ...
\PYG{n+nv}{rstfile} \PYG{o}{=} ../eq/\PYG{o}{\PYGZob{}}\PYG{o}{\PYGZcb{}}.rst
\PYG{n+nv}{reffile} \PYG{o}{=} \PYG{o}{\PYGZob{}}\PYG{o}{\PYGZcb{}}.pdb
\PYG{n+nv}{fitfile} \PYG{o}{=} fit.pdb

\PYG{o}{[}RPATH\PYG{o}{]}
\PYG{n+nv}{nreplica}          \PYG{o}{=} \PYG{l+m}{16}
\PYG{n+nv}{rpath\PYGZus{}period}      \PYG{o}{=} \PYG{l+m}{1000}
\PYG{n+nv}{delta}             \PYG{o}{=} \PYG{l+m}{0}.001
\PYG{n+nv}{smooth}            \PYG{o}{=} \PYG{l+m}{0}.00
\PYG{n+nv}{rest\PYGZus{}function}     \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{fix\PYGZus{}terminal}      \PYG{o}{=} NO

\PYG{o}{[}FITTING\PYG{o}{]}
\PYG{n+nv}{fitting\PYGZus{}method} \PYG{o}{=} TR+ROT
\PYG{n+nv}{fitting\PYGZus{}atom}   \PYG{o}{=} \PYG{l+m}{1}

\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}        \PYG{o}{=} ai:15 or ai:17 or ai:19 or ai:25 or ai:27

\PYG{o}{[}RESTRAINTS\PYG{o}{]}
\PYG{n+nv}{nfunctions}    \PYG{o}{=} \PYG{l+m}{1}

\PYG{n+nv}{function1}     \PYG{o}{=} POSI
\PYG{n+nv}{constant1}     \PYG{o}{=} \PYG{l+m}{10}.0 \PYG{l+m}{10}.0 \PYG{l+m}{10}.0 \PYG{l+m}{10}.0 \PYG{l+s+se}{\PYGZbs{}}
                \PYG{l+m}{10}.0 \PYG{l+m}{10}.0 \PYG{l+m}{10}.0 \PYG{l+m}{10}.0 \PYG{l+s+se}{\PYGZbs{}}
                \PYG{l+m}{10}.0 \PYG{l+m}{10}.0 \PYG{l+m}{10}.0 \PYG{l+m}{10}.0 \PYG{l+s+se}{\PYGZbs{}}
                \PYG{l+m}{10}.0 \PYG{l+m}{10}.0 \PYG{l+m}{10}.0 \PYG{l+m}{10}.0
\PYG{n+nv}{select\PYGZus{}index1} \PYG{o}{=} \PYG{l+m}{1}
\end{sphinxVerbatim}

Here is an example of a MEP search along with QM/MM

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}INPUT\PYG{o}{]}
\PYG{n+nv}{topfile} \PYG{o}{=} toppar/top\PYGZus{}all36\PYGZus{}prot.rtf, ...
\PYG{n+nv}{parfile} \PYG{o}{=} toppar/par\PYGZus{}all36\PYGZus{}prot.prm, ...
\PYG{n+nv}{psffile} \PYG{o}{=} prot.psf               \PYG{c+c1}{\PYGZsh{} protein structure file}
\PYG{n+nv}{reffile} \PYG{o}{=} prot.pdb               \PYG{c+c1}{\PYGZsh{} PDB file}
\PYG{n+nv}{pdbfile} \PYG{o}{=} initial/initial\PYG{o}{\PYGZob{}}\PYG{o}{\PYGZcb{}}.pdb  \PYG{c+c1}{\PYGZsh{} initial path}

\PYG{o}{[}OUTPUT\PYG{o}{]}
\PYG{n+nv}{dcdfile} \PYG{o}{=} mep\PYGZus{}\PYG{o}{\PYGZob{}}\PYG{o}{\PYGZcb{}}.dcd        \PYG{c+c1}{\PYGZsh{} coordinates}
\PYG{n+nv}{logfile} \PYG{o}{=} mep\PYGZus{}\PYG{o}{\PYGZob{}}\PYG{o}{\PYGZcb{}}.log        \PYG{c+c1}{\PYGZsh{} log files}
\PYG{n+nv}{rstfile} \PYG{o}{=} mep\PYGZus{}\PYG{o}{\PYGZob{}}\PYG{o}{\PYGZcb{}}.rst        \PYG{c+c1}{\PYGZsh{} restart file}
\PYG{n+nv}{rpathfile}  \PYG{o}{=} mep\PYGZus{}\PYG{o}{\PYGZob{}}\PYG{o}{\PYGZcb{}}.rpath   \PYG{c+c1}{\PYGZsh{} rpath file}

\PYG{o}{[}ENERGY\PYG{o}{]}
\PYG{n+nv}{forcefield}       \PYG{o}{=} CHARMM    \PYG{c+c1}{\PYGZsh{} CHARMM force field}
... skip ...

\PYG{o}{[}MINIMIZE\PYG{o}{]}
\PYG{n+nv}{method}              \PYG{o}{=} LBFGS  \PYG{c+c1}{\PYGZsh{} MIN using L\PYGZhy{}BFGS}
\PYG{n+nv}{nsteps}              \PYG{o}{=} \PYG{l+m}{100}    \PYG{c+c1}{\PYGZsh{} max. number of steps}
\PYG{n+nv}{eneout\PYGZus{}period}       \PYG{o}{=} \PYG{l+m}{5}      \PYG{c+c1}{\PYGZsh{} energy output period}
\PYG{n+nv}{fixatm\PYGZus{}select\PYGZus{}index} \PYG{o}{=} \PYG{l+m}{2}      \PYG{c+c1}{\PYGZsh{} fix the outer layer}
\PYG{n+nv}{macro}               \PYG{o}{=} yes    \PYG{c+c1}{\PYGZsh{} macro/micro iteration}

\PYG{o}{[}RPATH\PYG{o}{]}
\PYG{n+nv}{rpathmode}           \PYG{o}{=} MEP    \PYG{c+c1}{\PYGZsh{} MEP search}
\PYG{n+nv}{method}              \PYG{o}{=} STRING \PYG{c+c1}{\PYGZsh{} String method}
\PYG{n+nv}{delta}               \PYG{o}{=} \PYG{l+m}{0}.0005 \PYG{c+c1}{\PYGZsh{} step size}
\PYG{n+nv}{ncycle}              \PYG{o}{=} \PYG{l+m}{200}    \PYG{c+c1}{\PYGZsh{} max. number of cycle}
\PYG{n+nv}{nreplica}            \PYG{o}{=} \PYG{l+m}{16}     \PYG{c+c1}{\PYGZsh{} number of replica}
\PYG{n+nv}{eneout\PYGZus{}period}       \PYG{o}{=} \PYG{l+m}{1}      \PYG{c+c1}{\PYGZsh{} frequency of the energy output}
\PYG{n+nv}{crdout\PYGZus{}period}       \PYG{o}{=} \PYG{l+m}{1}      \PYG{c+c1}{\PYGZsh{} frequency of the coordinate output}
\PYG{n+nv}{rstout\PYGZus{}period}       \PYG{o}{=} \PYG{l+m}{1}      \PYG{c+c1}{\PYGZsh{} frequency of the restart update}
\PYG{n+nv}{fix\PYGZus{}terminal}        \PYG{o}{=} no     \PYG{c+c1}{\PYGZsh{} fix the terminal}
\PYG{n+nv}{massWeightCoord}     \PYG{o}{=} no     \PYG{c+c1}{\PYGZsh{} mass\PYGZhy{}weighted Cartesian}
\PYG{n+nv}{mepatm\PYGZus{}select\PYGZus{}index} \PYG{o}{=} \PYG{l+m}{1}      \PYG{c+c1}{\PYGZsh{} selection of the MEP atoms}

\PYG{o}{[}BOUNDARY\PYG{o}{]}
\PYG{n+nb}{type}          \PYG{o}{=} NOBC

\PYG{o}{[}QMMM\PYG{o}{]}
\PYG{n+nv}{qmtyp}              \PYG{o}{=} gaussian
\PYG{n+nv}{qmatm\PYGZus{}select\PYGZus{}index} \PYG{o}{=} \PYG{l+m}{1}
... skip ...

\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1} \PYG{o}{=} sid:DHA or \PYG{o}{(}sid:TIMA and \PYG{o}{(}rno:95 or rno:165\PYG{o}{)} and \PYG{l+s+se}{\PYGZbs{}}
         not \PYG{o}{(}an:CA \PYG{p}{\textbar{}} an:C \PYG{p}{\textbar{}} an:O \PYG{p}{\textbar{}} an:N \PYG{p}{\textbar{}} an:HN \PYG{p}{\textbar{}} an:HA\PYG{o}{)}\PYG{o}{)}
\PYG{n+nv}{group2} \PYG{o}{=} not \PYG{o}{(}sid:DHA  or sid:DHA  around\PYGZus{}res:6.0\PYG{o}{)}
\end{sphinxVerbatim}


\chapter{GAMD section}
\label{\detokenize{16_GaMD:gamd-section}}\label{\detokenize{16_GaMD:gamd}}\label{\detokenize{16_GaMD::doc}}

\section{Gaussian accelerated Molecular Dynamics}
\label{\detokenize{16_GaMD:gaussian-accelerated-molecular-dynamics}}
In the \sphinxstylestrong{{[}GAMD{]}} section, the users can specify keywords for Gaussian
accelerated Molecular Dynamics (GaMD) simulation.
The GaMD method
\sphinxcite{99_References:miao-2015}\sphinxcite{99_References:pang-2017}
accelerates the conformational sampling of biomolecules by adding
a harmonic boost potential to smooth their potential energy surface.
GaMD has the advantage that reaction coordinates do not need to be predefined,
thus setting up the system for the simulation is rather easy.
The use of the harmonic boost potential allows to recover the
unbiased free-energy changes through cumulant expansion to the second order,
which resolves the practical reweighting problem in the original accelerated MD method.

GaMD was developed as a potential-biasing method for enhanced sampling.
It accelerates the conformational sampling of a biomolecule by adding a
non-negative boost potential to the system potential energy \(U(\vec{x})\):
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}U'(\vec{x}) = U(\vec{x}) + \Delta U^{\mathrm{GaMD}}\left(U(\vec{x})\right),\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(\vec{x}\) is the configuration of the system,
\(U'(\vec{x})\) is the modified potential energy,
and \(\Delta U^{\mathrm{GaMD}}\) is the boost potential
depending only on \(U(\vec{x})\).

In conventional accelerated MD \sphinxcite{99_References:hamelberg-2004}\sphinxcite{99_References:hamelberg-2007}\sphinxcite{99_References:shen-2008},
the average of the Boltzmann factors of the boost potential terms appears in the
reweighting equation of the probability along the selected reaction coordinates,
causing a large statistical error.
In order to reduce the noise, GaMD uses a harmonic boost potential, which adopts
a positive value only when the system potential is lower than an energy threshold \(E\):
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\Delta U^{\mathrm{GaMD}}\left(U(\vec{x})\right) =
\begin{cases}
  \frac{1}{2}k\{E-U(\vec{x})\}^2 & (U(\vec{x}) < E) \\
  0 & (U(\vec{x}) \ge E)
\end{cases},\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(k\) is a harmonic force constant.
\(U'(\vec{x})\) should satisfy the following relationships \sphinxcite{99_References:miao-2015}\sphinxcite{99_References:pang-2017}:
\(U'(\vec{x}_1) < U'(\vec{x}_2)\) and \(U'(\vec{x}_2) - U'(\vec{x}_1) < U(\vec{x}_2) - U(\vec{x}_1)\)
if \(U(\vec{x}_1) < U(\vec{x}_2)\).
To keep the relationships, the threshold energy needs to be set as:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}U_{\mathrm{max}} \le E \le U_{\mathrm{min}} + \frac{1}{k},\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(U_{\mathrm{max}}\) and \(U_{\mathrm{min}}\) are
maximum and minimum energies of the system, respectively.
To ensure accurate reweighting, the deviation of the potential
must also satisfy the relation:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}k(E-U_{\mathrm{ave}}) \sigma_{U} \le \sigma_0,\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(U_{\mathrm{ave}}\) and \(\sigma_{U}\) are the average
and standard deviation of \(U(\vec{x})\), respectively.
\(\sigma_0\) is a user-specified upper limit.
\(k_0\) is defined as \(k_0 \equiv k(U_{\mathrm{max}}-U_{\mathrm{min}})\),
then \(0 < k_0 \le 1\).

When \(E\) is set to the lower bound \(U_{\mathrm{max}}\),
\(k_0\) is determined by
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}k_0 = \min \left(1,\frac{\sigma_0}{\sigma_U}\frac{U_{\mathrm{max}}-U_{\mathrm{min}}}{U_{\mathrm{max}}-U_{\mathrm{ave}}}\right)\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

When \(E\) is set to the upper bound \(U_{\mathrm{min}}+1/k\),
\(k_0\) is set to
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}k''_0 \equiv \left(1-\frac{\sigma_0}{\sigma_U}\right)\frac{U_{\mathrm{max}}-U_{\mathrm{min}}}{U_{\mathrm{ave}}-U_{\mathrm{min}}}\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

if \(0 < k''_0 < 1\), and \(k_0\) is set to 1 otherwise.

The above parameters
(\(U_{\mathrm{max}}\), \(U_{\mathrm{min}}\), \(U_{\mathrm{ave}}\), and \(\sigma_{U}\))
are determined from short-time simulations a priori.
When the distribution of the boost potential approaches Gaussian distribution,
the cumulant expansion of the average of \(\exp[\beta \Delta U^\mathrm{GaMD}]\)
to the second order provides a good approximation for the free energy \sphinxcite{99_References:miao-2014}.

GaMD can be combined with REUS in such a way that each replica in REUS is accelerated by the GaMD boost potential:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}U''_i(\vec{x}) &= U'(\vec{x}) + \Delta U_i^{\mathrm{REUS}}\left(\xi(\vec{x})\right) \\
&= U(\vec{x}) + \Delta U^{\mathrm{GaMD}}\left(U(\vec{x})\right) + \Delta U_i^{\mathrm{REUS}}\left(\xi(\vec{x})\right),\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(U''_i(\vec{x})\) is the modified potential energy of replica \(i\),
\(\Delta U_i^{\mathrm{REUS}}\) is the bias potential of REUS for replica \(i\), and
\(\xi(\vec{x})\) is the collective variable of REUS.
This method is referred to as Gaussian accelerated replica exchange umbrella sampling (GaREUS) \sphinxcite{99_References:oshima-2019}.
The parameters in the GaMD boost potential are used in all replicas of GaREUS simulations.
By using this combination, the simulated system in each replica becomes more flexible, or the energy
barrier irrelevant to the collective variable is lowered, enhancing the sampling efficiency.
When performing GaREUS simulations, the user must specify {[}REMD{]} section to use REUS
and define a collective variable in the {[}SELECTION{]} and {[}RESTRAINTS{]} sections.
Please check the example below.


\bigskip\hrule\bigskip


\sphinxstylestrong{gamd} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

Enable the GaMD method.
\end{quote}

\sphinxstylestrong{boost} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : YES}

Flag to apply GaMD boost to the system ().
If \sphinxstyleemphasis{boost = NO}, boost is not applied but GaMD parameters
are updated from the trajectory.
\end{quote}

\sphinxstylestrong{boost\_type} \sphinxstyleemphasis{DUAL / DIHEDRAL / POTENTIAL}
\begin{quote}

\sphinxstylestrong{Default: DUAL}

Type of boost.
\begin{itemize}
\item {} 
\sphinxstylestrong{DUAL}: Boost is applied on both the dihedral and total potential energies.

\item {} 
\sphinxstylestrong{DIHEDRAL}: Boost is applied on only the dihedral energy.

\item {} 
\sphinxstylestrong{POTENTIAL}: Boost is applied on only the total potential energy.

\end{itemize}
\end{quote}

\sphinxstylestrong{thresh\_type} \sphinxstyleemphasis{LOWER / HIGHER}
\begin{quote}

\sphinxstylestrong{Default: LOWER}

Type of threshold.
\begin{itemize}
\item {} 
\sphinxstylestrong{LOWER}: \(E\) is set to the lower bound \(E = U_{\text{max}}\).

\item {} 
\sphinxstylestrong{HIGHER}: \(E\) is set to its upper bound \(E = U_{\text{min}} + 1/k\).

\end{itemize}
\end{quote}

\sphinxstylestrong{update\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: 0}

Period of updating parameters in units of time step.
\end{quote}

\sphinxstylestrong{sigma0\_pot} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 6.0} (unit: kcal/mol)

Upper limit of the standard deviation of the total potential
boost (\(\sigma_0^{\mathrm{pot}}\)) that allows for
accurate reweighting.
\end{quote}

\sphinxstylestrong{pot\_max} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: -99999999.0} (unit: kcal/mol)

Maximum of the total potential energy of the system \(U_{\mathrm{max}}^{\mathrm{pot}}\).
\end{quote}

\sphinxstylestrong{pot\_min} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 99999999.0} (unit: kcal/mol)

Minimum of the total potential energy of the system \(U_{\mathrm{min}}^{\mathrm{pot}}\).
\end{quote}

\sphinxstylestrong{pot\_ave} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 0.0} (unit: kcal/mol)

Average of the total potential energy of the system \(U_{\mathrm{ave}}^{\mathrm{pot}}\).
\end{quote}

\sphinxstylestrong{pot\_dev} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 0.0} (unit: kcal/mol)

Standard deviation of the total potential energy of the system \(\sigma_{U}^{\mathrm{pot}}\).
\end{quote}

\sphinxstylestrong{sigma0\_dih} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 6.0} (unit: kcal/mol)

Upper limit of the standard deviation of the dihedral
boost (\(\sigma_0^{\mathrm{dih}}\)) that allows
for accurate reweighting.
\end{quote}

\sphinxstylestrong{dih\_max} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: -99999999.0} (unit: kcal/mol)

Maximum of the dihedral energy of the system \(U_{\mathrm{max}}^{\mathrm{dih}}\).
\end{quote}

\sphinxstylestrong{dih\_min} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 99999999.0} (unit: kcal/mol)

Minimum of the dihedral energy of the system \(U_{\mathrm{min}}^{\mathrm{dih}}\).
\end{quote}

\sphinxstylestrong{dih\_ave} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 0.0} (unit: kcal/mol)

Average of the dihedral energy of the system \(U_{\mathrm{ave}}^{\mathrm{dih}}\).
\end{quote}

\sphinxstylestrong{dih\_dev} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 0.0} (unit: kcal/mol)

Standard deviation of the dihedral energy of the system \(\sigma_{U}^{\mathrm{dih}}\).
\end{quote}


\section{Examples}
\label{\detokenize{16_GaMD:examples}}
Example of a GaMD simulation to determine initial parameters.
To obtain the initial guess of the boost potential,
(pot\_max, pot\_min, pot\_ave, pot\_dev, dih\_max, dih\_min, dih\_ave, dih\_dev)
are calculated from a short simulation without boosting.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}GAMD\PYG{o}{]}
\PYG{n+nv}{gamd}          \PYG{o}{=} yes
\PYG{n+nv}{boost}         \PYG{o}{=} no
\PYG{n+nv}{boost\PYGZus{}type}    \PYG{o}{=} DUAL
\PYG{n+nv}{thresh\PYGZus{}type}   \PYG{o}{=} LOWER
\PYG{n+nv}{sigma0\PYGZus{}pot}    \PYG{o}{=} \PYG{l+m}{6}.0
\PYG{n+nv}{sigma0\PYGZus{}dih}    \PYG{o}{=} \PYG{l+m}{6}.0
\PYG{n+nv}{update\PYGZus{}period} \PYG{o}{=} \PYG{l+m}{50000}
\end{sphinxVerbatim}

Example of a GaMD simulation updating parameters.
The boost potential is updated every \sphinxstyleemphasis{update\_period} during the simulation.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}GAMD\PYG{o}{]}
\PYG{n+nv}{gamd}          \PYG{o}{=} yes
\PYG{n+nv}{boost}         \PYG{o}{=} yes
\PYG{n+nv}{boost\PYGZus{}type}    \PYG{o}{=} DUAL
\PYG{n+nv}{thresh\PYGZus{}type}   \PYG{o}{=} LOWER
\PYG{n+nv}{sigma0\PYGZus{}pot}    \PYG{o}{=} \PYG{l+m}{6}.0
\PYG{n+nv}{sigma0\PYGZus{}dih}    \PYG{o}{=} \PYG{l+m}{6}.0
\PYG{n+nv}{update\PYGZus{}period} \PYG{o}{=} \PYG{l+m}{500}
\PYG{n+nv}{pot\PYGZus{}max}       \PYG{o}{=} \PYGZhy{}20935.8104
\PYG{n+nv}{pot\PYGZus{}min}       \PYG{o}{=} \PYGZhy{}21452.3778
\PYG{n+nv}{pot\PYGZus{}ave}       \PYG{o}{=} \PYGZhy{}21183.9911
\PYG{n+nv}{pot\PYGZus{}dev}       \PYG{o}{=} \PYG{l+m}{78}.1207
\PYG{n+nv}{dih\PYGZus{}max}       \PYG{o}{=} \PYG{l+m}{16}.4039
\PYG{n+nv}{dih\PYGZus{}min}       \PYG{o}{=} \PYG{l+m}{8}.5882
\PYG{n+nv}{dih\PYGZus{}ave}       \PYG{o}{=} \PYG{l+m}{11}.0343
\PYG{n+nv}{dih\PYGZus{}dev}       \PYG{o}{=} \PYG{l+m}{1}.0699
\end{sphinxVerbatim}

Example of a GaMD simulation for production.
In order to fix the parameters (pot\_max, pot\_min, pot\_ave, pot\_dev,
dih\_max, dih\_min, dih\_ave, dih\_dev), \sphinxstyleemphasis{update\_period} is set to 0.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}GAMD\PYG{o}{]}
\PYG{n+nv}{gamd}          \PYG{o}{=} yes
\PYG{n+nv}{boost}         \PYG{o}{=} yes
\PYG{n+nv}{boost\PYGZus{}type}    \PYG{o}{=} DUAL
\PYG{n+nv}{thresh\PYGZus{}type}   \PYG{o}{=} LOWER
\PYG{n+nv}{sigma0\PYGZus{}pot}    \PYG{o}{=} \PYG{l+m}{6}.0
\PYG{n+nv}{sigma0\PYGZus{}dih}    \PYG{o}{=} \PYG{l+m}{6}.0
\PYG{n+nv}{update\PYGZus{}period} \PYG{o}{=} \PYG{l+m}{0}
\PYG{n+nv}{pot\PYGZus{}max}       \PYG{o}{=} \PYGZhy{}20669.2404
\PYG{n+nv}{pot\PYGZus{}min}       \PYG{o}{=} \PYGZhy{}21452.3778
\PYG{n+nv}{pot\PYGZus{}ave}       \PYG{o}{=} \PYGZhy{}20861.5224
\PYG{n+nv}{pot\PYGZus{}dev}       \PYG{o}{=} \PYG{l+m}{48}.9241
\PYG{n+nv}{dih\PYGZus{}max}       \PYG{o}{=} \PYG{l+m}{23}.2783
\PYG{n+nv}{dih\PYGZus{}min}       \PYG{o}{=} \PYG{l+m}{8}.5882
\PYG{n+nv}{dih\PYGZus{}ave}       \PYG{o}{=} \PYG{l+m}{13}.3806
\PYG{n+nv}{dih\PYGZus{}dev}       \PYG{o}{=} \PYG{l+m}{1}.7287
\end{sphinxVerbatim}

Example of a GaREUS simulation.
The same GaMD parameters are applied in each replica of REUS.
After the simulation, the two-step reweighting procedure using the
multistate Bennett acceptance ratio method and the cumulant expansion
for the exponential average is required to obtain the unbiased free-energy landscapes.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}REMD\PYG{o}{]}
\PYG{n+nv}{dimension}        \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{exchange\PYGZus{}period}  \PYG{o}{=} \PYG{l+m}{5000}
\PYG{n+nv}{type1}            \PYG{o}{=} RESTRAINT
\PYG{n+nv}{nreplica1}        \PYG{o}{=} \PYG{l+m}{4}
\PYG{n+nv}{rest\PYGZus{}function1}   \PYG{o}{=} \PYG{l+m}{1}

\PYG{o}{[}GAMD\PYG{o}{]}
\PYG{n+nv}{gamd}          \PYG{o}{=} yes
\PYG{n+nv}{boost}         \PYG{o}{=} yes
\PYG{n+nv}{boost\PYGZus{}type}    \PYG{o}{=} DUAL
\PYG{n+nv}{thresh\PYGZus{}type}   \PYG{o}{=} LOWER
\PYG{n+nv}{sigma0\PYGZus{}pot}    \PYG{o}{=} \PYG{l+m}{6}.0
\PYG{n+nv}{sigma0\PYGZus{}dih}    \PYG{o}{=} \PYG{l+m}{6}.0
\PYG{n+nv}{update\PYGZus{}period} \PYG{o}{=} \PYG{l+m}{0}
\PYG{n+nv}{pot\PYGZus{}max}       \PYG{o}{=} \PYGZhy{}26491.7344
\PYG{n+nv}{pot\PYGZus{}min}       \PYG{o}{=} \PYGZhy{}27447.4316
\PYG{n+nv}{pot\PYGZus{}ave}       \PYG{o}{=} \PYGZhy{}26744.5742
\PYG{n+nv}{pot\PYGZus{}dev}       \PYG{o}{=} \PYG{l+m}{52}.5674
\PYG{n+nv}{dih\PYGZus{}max}       \PYG{o}{=} \PYG{l+m}{135}.8921
\PYG{n+nv}{dih\PYGZus{}min}       \PYG{o}{=} \PYG{l+m}{91}.2309
\PYG{n+nv}{dih\PYGZus{}ave}       \PYG{o}{=} \PYG{l+m}{116}.8572
\PYG{n+nv}{dih\PYGZus{}dev}       \PYG{o}{=} \PYG{l+m}{3}.6465

\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1} \PYG{o}{=} rno:1  and an:CA
\PYG{n+nv}{group2} \PYG{o}{=} rno:10 and an:CA

\PYG{o}{[}RESTRAINTS\PYG{o}{]}
\PYG{n+nv}{nfunctions}    \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{function1}     \PYG{o}{=} DISTMASS
\PYG{n+nv}{constant1}     \PYG{o}{=} \PYG{l+m}{1}.0 \PYG{l+m}{1}.0 \PYG{l+m}{1}.0 \PYG{l+m}{1}.0
\PYG{n+nv}{reference1}    \PYG{o}{=} \PYG{l+m}{5}.0 \PYG{l+m}{6}.0 \PYG{l+m}{7}.0 \PYG{l+m}{8}.0
\PYG{n+nv}{select\PYGZus{}index1} \PYG{o}{=} \PYG{l+m}{1} \PYG{l+m}{2}
\end{sphinxVerbatim}


\chapter{QMMM section}
\label{\detokenize{17_QMMM:qmmm-section}}\label{\detokenize{17_QMMM:qmmm}}\label{\detokenize{17_QMMM::doc}}

\section{Quantum mechanics/Molecular mechanics method (QM/MM)}
\label{\detokenize{17_QMMM:quantum-mechanics-molecular-mechanics-method-qm-mm}}
\sphinxstyleemphasis{QM/MM is available only in} \sphinxstylestrong{ATDYN}.

The QM/MM method, first proposed in seminal papers by Warshel, Levitt,
and Karplus \sphinxcite{99_References:warshel-1972} \sphinxcite{99_References:warshel-1976}, is a multi-scale
approach, which treats a partial region of interest (QM region) by
quantum chemistry, and the surrounding environment (MM region) by
force field.  The method is useful, in particular, when the QM region
involves an event that cannot be described by the standard force field;
for example, chemical reactions, spectroscopy, etc.

In the QM/MM method, the potential energy of the system is written as,
\begin{quote}
\begin{equation*}
\begin{split}V (\mathbf{R}_a, \mathbf{R}_m)
= V^\mathrm{QM} (\mathbf{R}_a, \mathbf{R}_m)
+ V^\mathrm{QM-MM}_\mathrm{LJ} (\mathbf{R}_a, \mathbf{R}_m)
+ V^\mathrm{MM} (\mathbf{R}_m),\end{split}
\end{equation*}\end{quote}

where \(\mathbf{R}_a\) and \(\mathbf{R}_m\) denote the
position of atoms in QM and MM regions, respectively.
\(V^\mathrm{QM-MM}_\mathrm{LJ}\) and \(V^{\mathrm{MM}}\)
are the Lennard-Jones interaction between QM-MM atoms and the
force field for MM atoms, respectively.  The QM energy,
\(V^\mathrm{QM}\), is written in
terms of the electronic energy and the Coulomb interaction
between nucleus-nucleus and nucleus-MM atoms,
\begin{quote}
\begin{equation*}
\begin{split}V^\mathrm{QM} (\mathbf{R}_a, \mathbf{R}_m) = E_e (\mathbf{R}_a, \mathbf{R}_m)
+ \sum_{a > a^\prime} \frac{Z_a Z_{a^\prime}}{r_{a{a^\prime}}}
+ \sum_{a, m} \frac{Z_a q_m}{r_{am}},\end{split}
\end{equation*}\end{quote}

where \(Z_a\) and \(q_m\) are the charge of nucleus and MM atoms,
respectively, and \(r_{a{a^\prime}}\) and \(r_{am}\) denote
the distantce between nucleus and nucleus-MM atoms, respectively.
The electronic energy is given by solving the Schrödinger
equation for electrons,
\begin{quote}
\begin{equation*}
\begin{split}\left[ - \frac{1}{2} \sum_i \nabla_i^2
+ \sum_{i>j} \frac{1}  {r_{ij}}
- \sum_{i,a} \frac{Z_a}{r_{ia}}
- \sum_{i,m} \frac{q_m}{r_{im}} \right] \ket{\Psi_e} = E_e \ket{\Psi_e},\end{split}
\end{equation*}\end{quote}

where \sphinxstyleemphasis{i}, \sphinxstyleemphasis{a}, and \sphinxstyleemphasis{m} are indices for electrons, nucleus, and MM
atoms, respectively, and \(r_{XY}\) denotes the distance between
particle \sphinxstyleemphasis{X} and \sphinxstyleemphasis{Y}.

\sphinxstylestrong{GENESIS} does not have a function to solve the electronic Schrödinger
equation, but rely on external QM programs, which provide
the QM energy, its derivatives, and other information.
The interface is currently avaliable for Gaussian, Q-Chem, TeraChem,
DFTB+, and QSimulate.
\begin{itemize}
\item {} 
\sphinxhref{http://gaussian.com}{Gaussian09/Gaussian16} (http://gaussian.com)

\item {} 
\sphinxhref{http://www.q-chem.com}{Q-Chem} (http://www.q-chem.com)

\item {} 
\sphinxhref{http://www.petachem.com}{TeraChem} (http://www.petachem.com)

\item {} 
\sphinxhref{https://www.dftbplus.org}{DFTB+} (https://www.dftbplus.org)

\item {} 
\sphinxhref{https://qsimulate.com/academic.html}{QSimulate} (https://qsimulate.com/academic.html)

\end{itemize}

GENESIS/QSimulate is interfaced via shared libraries and seamlessly uses
the MPI parallelization, thereby facilitating high performace QM/MM-MD
simulations \sphinxcite{99_References:yagi-2021}. A ready-to-use Singularity image is
provided by QSimualte Inc.  See \sphinxhref{https://qsimulate.com/academic.html}{QSimulate} (https://qsimulate.com/academic.html)
for further information.

Other QM programs are invoked via a system call function of Fortran. In this
scheme, the input file of a QM calculation is first generated, followed by
executing a script to run a QM program and reading the information from
QM output files.  For more information on the method and implementation,
see Ref.\sphinxcite{99_References:yagi-2019}.  Samples of a QM input file
(\sphinxstylestrong{qmcnt}) and a script (\sphinxstylestrong{qmexe}) are available in our
\sphinxhref{https://github.com/yagikiyoshi/QMMMscripts}{github} (https://github.com/yagikiyoshi/QMMMscripts)

In order to run QM/MM calculations, users add the \sphinxstylestrong{{[}QMMM{]}} section
in the control file. Avaliable options are listed in the following.


\bigskip\hrule\bigskip


\sphinxstylestrong{qmtyp} \sphinxstyleemphasis{DFTB+ / GAUSSIAN / QCHEM / TERACHEM / QSIMULATE}
\begin{quote}

The QM program to use in QM/MM calculations.
\end{quote}

\sphinxstylestrong{qmatm\_select\_index} \sphinxstyleemphasis{Integer}
\begin{quote}

Index of a group of atoms which is treated as QM atoms.
Link hydrogen atoms are automatically added based on a
bond connectivity (e.g., given by a PSF file). The index
must be defined in \sphinxstylestrong{{[}SELECTION{]}} (see {\hyperref[\detokenize{11_Selection:selection}]{\sphinxcrossref{\DUrole{std,std-ref}{Selection section}}}}).
\end{quote}

\sphinxstylestrong{qmcnt}  \sphinxstyleemphasis{Character}
\begin{quote}

A template input file of QM calculations.
\end{quote}

\sphinxstylestrong{qmexe} \sphinxstyleemphasis{Character}
\begin{quote}

A script file to execute a QM program.
\end{quote}

\sphinxstylestrong{workdir} \sphinxstyleemphasis{Character}
\begin{quote}

\sphinxstylestrong{Default : qmmm}

The name of a directory where QM input/output files are generated.
The replica ID is added after this name, e.g., qmmm.0, qmmm.1, etc.
\end{quote}

\sphinxstylestrong{basename} \sphinxstyleemphasis{Character}
\begin{quote}

\sphinxstylestrong{Default : N/A}

The basename of input / output files of QM calculations.
\end{quote}

\sphinxstylestrong{qmsave\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 1}

Frequency to save input / output files for QM calculations.
\end{quote}

\sphinxstylestrong{savedir} \sphinxstyleemphasis{Character}
\begin{quote}

\sphinxstylestrong{Default : N/A}

If present, QM files are copied from \sphinxstylestrong{workdir} to this directory.
It is typically the case that QM calculations are carried out
within a node, and the whole simulation (such as REMD) accross
nodes. Then, it is useful for a better performance to set
\sphinxstylestrong{workdir} to a local disk of each node with fast access (e.g.,
/dev/shm), and copy the QM files to \sphinxstylestrong{savedir} with a frequency
specified by qmsave\_period.
\end{quote}

\sphinxstylestrong{qmmaxtrial} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 1}

The maximum number of trial run for QM calculations.
When a QM calculation fails, \sphinxstylestrong{GENESIS} repeats the calculation until
the iteration reaches this number. The SCF threshold is lowered, if
the SCF threshold option is present in the QM control file.
\end{quote}

\sphinxstylestrong{exclude\_charge} \sphinxstyleemphasis{ATOM / GROUP / AMBER}
\begin{quote}

\sphinxstylestrong{Default: GROUP}

This option specifies how to exclude the MM charge in the vicinity of a
QM-MM boundary to avoid overpolarization of QM electron density.  When
the CHARMM force field is used, \sphinxstyleemphasis{ATOM} excludes only the charge of MM
link atom, while \sphinxstyleemphasis{GROUP} excludes the charges of all MM atoms that
belongs to the same group as a MM atom at the boundary. When the AMBER
force field is used, \sphinxstyleemphasis{AMBER} excludes the charge of MM link atom and
distributes it to rest of the system evenly.
\end{quote}

\begin{sphinxadmonition}{note}{Note:}
Since version 1.6.1, QM/MM supports both CHARMM and AMBER force field for
the MM.
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
The QM/MM calculation must be carried out in non-PBC. A non-PBC system
can be created from MD trajectory (pdb, dcd) using \sphinxstylestrong{qmmm\_generator} in
the analysis tool.
See \sphinxhref{https://www.r-ccs.riken.jp/labs/cbrt/tutorials2019/tutorial-16-1}{the tutorial of QM/MM} (https://www.r-ccs.riken.jp/labs/cbrt/tutorials2019/tutorial-16-1)
for more details.
\end{sphinxadmonition}


\section{Examples}
\label{\detokenize{17_QMMM:examples}}
In the following example, the atoms from \# 1 to 14 are selected as QM
atoms by \sphinxstylestrong{{[}SELECTION{]}} section. The QM program is Gaussian. A
directory \sphinxcode{\sphinxupquote{qmmm\_min}} is created, where input and output files for
Gaussian (jobXXXX.inp and jobXXXX.log) are saved every 10 steps.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}  \PYG{o}{=} atomno:1\PYGZhy{}14

\PYG{o}{[}QMMM\PYG{o}{]}
\PYG{n+nv}{qmatm\PYGZus{}select\PYGZus{}index}  \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{qmtyp}               \PYG{o}{=} gaussian
\PYG{n+nv}{qmcnt}               \PYG{o}{=} gaussian.com
\PYG{n+nv}{qmexe}               \PYG{o}{=} runGau.sh
\PYG{n+nv}{workdir}             \PYG{o}{=} qmmm\PYGZus{}min
\PYG{n+nv}{basename}            \PYG{o}{=} job
\PYG{n+nv}{qmsave\PYGZus{}period}       \PYG{o}{=} \PYG{l+m}{10}
\end{sphinxVerbatim}

The following example is for DFTB+. Because DFTB calculations
typically take \textless{} 1 sec per one snapshot, I/O to generate the
input and read output could be non-negligible.  It is
recommended to set \sphinxstylestrong{workdir} to a fast disk such as
\sphinxcode{\sphinxupquote{/dev/shm}}.  The input and output files of DFTB+ will be
copied to \sphinxcode{\sphinxupquote{qmmm\_min}} every 100 steps.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}QMMM\PYG{o}{]}
\PYG{n+nv}{qmatm\PYGZus{}select\PYGZus{}index} \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{qmtyp}              \PYG{o}{=} dftb+
\PYG{n+nv}{qmcnt}              \PYG{o}{=} dftb.hsd
\PYG{n+nv}{qmexe}              \PYG{o}{=} runDFTB.sh
\PYG{n+nv}{workdir}            \PYG{o}{=} /dev/shm/qmmm\PYGZus{}min
\PYG{n+nv}{savedir}            \PYG{o}{=} qmmm\PYGZus{}min
\PYG{n+nv}{basename}           \PYG{o}{=} job
\PYG{n+nv}{qmsave\PYGZus{}period}      \PYG{o}{=} \PYG{l+m}{100}
\end{sphinxVerbatim}


\chapter{Vibration section}
\label{\detokenize{18_Vibration:vibration-section}}\label{\detokenize{18_Vibration:vibration}}\label{\detokenize{18_Vibration::doc}}

\section{Vibrational analysis}
\label{\detokenize{18_Vibration:vibrational-analysis}}
\sphinxstyleemphasis{Vibration is available only in} \sphinxstylestrong{ATDYN}.

In the \sphinxstylestrong{{[}VIBRATION{]}} section, users can specify keywords for molecular
vibrational analysis. Vibrational analysis in \sphinxstylestrong{GENESIS} is done
for a subsystem, i.e., for a molecule of interest in the system.
In the subsystem’s space, a mass-weighted Hessian matrix is
generated and diagonalized to obtain normal modes (\(\mathbf{C}\))
and harmonic frequencies (\(\mathbf{\omega}\)),
\begin{quote}
\begin{equation*}
\begin{split}\mathbf{H C} = \mathbf{\omega C}\end{split}
\end{equation*}\end{quote}

where \(\mathbf{H}\) is the mass-weighted Hessian matrix,
\begin{quote}
\begin{equation*}
\begin{split}H_{ij} = \frac{1}{\sqrt{m_i m_j}}\frac{\partial^2 V}{\partial x_i \partial x_j},\end{split}
\end{equation*}\end{quote}

with the mass of the \sphinxstyleemphasis{i}-th atom, \(m_i\), and the potential energy, \sphinxstyleemphasis{V}.
The Hessian matrix is calculated by numerical differentiations of the gradient,
\begin{quote}
\begin{equation*}
\begin{split}\frac{\partial^2 V}{\partial x_i \partial x_j} \simeq
\frac{1}{2 \delta_i} \left( \frac{\partial V (+\delta_i)}{\partial x_j}
- \frac{\partial V(-\delta_i)}{\partial x_j} \right)\end{split}
\end{equation*}\end{quote}

This step requires 6 \sphinxstyleemphasis{N} number of gradient calculations, where \sphinxstyleemphasis{N} is
the number of atoms in the subsystem.  The gradient calculations are
parallelized by distributing over MPI processes.

The information is output to a minfo file, which can be visualized by a
molecular vibrational program, SINDO.


\bigskip\hrule\bigskip


\sphinxstylestrong{runmode} \sphinxstyleemphasis{HARM / QFF / GRID}
\begin{quote}

\sphinxstylestrong{Default : HARM}

Specifies the type of calculation. \sphinxstylestrong{HARM} invokes the harmonic analysis.
\sphinxstylestrong{QFF} and \sphinxstylestrong{GRID} are options for generating anharmonic potential.
\end{quote}

\sphinxstylestrong{nreplica} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 1}

The number of MPI processes.
\end{quote}

\sphinxstylestrong{vibatm\_select\_index} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: N/A}

Indices of a group of atoms to specify a target subsystem
for vibrational analyses.  The indices must be defined in
\sphinxstylestrong{{[}SELECTION{]}} (see {\hyperref[\detokenize{11_Selection:selection}]{\sphinxcrossref{\DUrole{std,std-ref}{Selection section}}}}).
\end{quote}

\sphinxstylestrong{output\_minfo\_atm} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: N/A}

Indices of a group of atoms, which are printed to a minfo file
in addition to the target subsystem. This option is useful
when one would like to visualize the atoms surrounding the
target subsystem.  The indices must be defined in
\sphinxstylestrong{{[}SELECTION{]}} (see {\hyperref[\detokenize{11_Selection:selection}]{\sphinxcrossref{\DUrole{std,std-ref}{Selection section}}}}).
\end{quote}

\sphinxstylestrong{diff\_stepsize} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.01} (unit: Angstrom)

The size of numerical differentiations when generating the Hessian matrix.
\end{quote}

\sphinxstylestrong{minfo\_folder} \sphinxstyleemphasis{Character}
\begin{quote}

\sphinxstylestrong{Default : minfo.files}

The name of a directory where intermediate minfo files are stored.
If the directory and intermediate files are present, the program restarts from
where it ended in the last run.
\end{quote}

\begin{sphinxadmonition}{note}{Note:}
The geometry of the subsystem needs to be optimized prior to the
vibrational analysis. RMSG \textless{} 0.35 kcal/mol/Angstrom is recommended.
\end{sphinxadmonition}

Furthermore, anharmonic vibrational calculations can be carried out by combining
SINDO and GENESIS. The following two options are used for generating anharmonic
potential energy surfaces. For more details, visit the website of
\sphinxhref{https://tms.riken.jp/en/research/software/sindo}{SINDO} (https://tms.riken.jp/en/research/software/sindo).


\bigskip\hrule\bigskip


\sphinxstylestrong{gridfile} \sphinxstyleemphasis{Character}
\begin{quote}

\sphinxstylestrong{Default : makeQFF.xyz} (for QFF) and \sphinxstylestrong{makeGrid.xyz} for GRID

The name of a file containing the XYZ coordinates of grid points for generating
the anharmonic PES. The xyz file is generated by MakePES module of SINDO.
\end{quote}

\sphinxstylestrong{datafile} \sphinxstyleemphasis{Character}
\begin{quote}

\sphinxstylestrong{Default : makeGrid.dat}

The name of a file containing the energy, dipole moment, etc. at grid points
specified by \sphinxstylestrong{gridfile}.  The file is used by SINDO for generating GRID potentials.
\end{quote}


\bigskip\hrule\bigskip



\section{Examples}
\label{\detokenize{18_Vibration:examples}}
In the following example, the vibrational analysis is performed for a
subsystem composed of atom number 5-8 (group1) and residue number 12-14
of segment “WAT” (group2). 4 MPI processes are used to calculate the
gradients at grid points of numerical differentiations.
The output is written to a minfo file, where the coordinates are given
not only for target atoms (group1 and group2) but also for the whole protein
(segid PROA).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}OUTPUT\PYG{o}{]}
\PYG{n+nv}{minfofile} \PYG{o}{=} qmmm\PYGZus{}vib.minfo

\PYG{o}{[}VIBRATION\PYG{o}{]}
\PYG{n+nv}{runmode}             \PYG{o}{=} HARM
\PYG{n+nv}{nreplica}            \PYG{o}{=} \PYG{l+m}{4}
\PYG{n+nv}{vibatm\PYGZus{}select\PYGZus{}index} \PYG{o}{=} \PYG{l+m}{1} \PYG{l+m}{2}
\PYG{n+nv}{output\PYGZus{}minfo\PYGZus{}atm}    \PYG{o}{=} \PYG{l+m}{3}

\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}  \PYG{o}{=} atomno:5\PYGZhy{}8
\PYG{n+nv}{group2}  \PYG{o}{=} segid:WAT and \PYG{o}{(}resno:12\PYGZhy{}14\PYG{o}{)}
\PYG{n+nv}{group3}  \PYG{o}{=} segid:PROA
\end{sphinxVerbatim}


\chapter{Experiments section}
\label{\detokenize{19_Experiments:experiments-section}}\label{\detokenize{19_Experiments:experiments}}\label{\detokenize{19_Experiments::doc}}

\section{Cryo-EM flexible fitting}
\label{\detokenize{19_Experiments:cryo-em-flexible-fitting}}

\subsection{Theory}
\label{\detokenize{19_Experiments:theory}}
Cryo-electron microscopy (Cryo-EM) is one of the powerful tools to determine
three-dimensional structures of biomolecules at near atomic resolution.
Flexible fitting has been widely utilized to model the atomic structure
from the experimental density map \sphinxcite{99_References:tama-2004}.
One of the commonly used methods is the MD-based flexible fitting \sphinxcite{99_References:orzechowski-2008} \sphinxcite{99_References:trabuco-2008}.
In the method, the total potential energy is defined as the
summation of a force field \(V_{\rm{FF}}\) and biasing potential \(V_{\rm{EM}}\)
that guides the protein structure towards the target density:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}{V_{{\rm{total}}}} = {V_{{\rm{FF}}}} + {V_{{\rm{EM}}}}\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

In the \sphinxstyleemphasis{c.c.}-based approach \sphinxcite{99_References:tama-2004}, one of the commonly used formulas for \(V_{\rm{EM}}\) is
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}{V_{{\rm{EM}}}} = k(1 - c.c.)\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \sphinxstyleemphasis{k} is the force constant, and \sphinxstyleemphasis{c.c.} is the cross-correlation
coefficient between the experimental and simulated EM density maps,
calculated as
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}c.c. = \frac{{\sum\limits_{ijk} {{\rho ^{\exp }}(i,j,k){\rho ^{{\rm{sim}}}}(i,j,k)} }}{{\sqrt {\sum\limits_{ijk} {{\rho ^{\exp }}{{(i,j,k)}^2}\sum\limits_{ijk} {{\rho ^{{\rm{sim}}}}{{(i,j,k)}^2}} } } }}\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

(\sphinxstyleemphasis{i}, \sphinxstyleemphasis{j}, \sphinxstyleemphasis{k}) is a voxel index in the density map,
and \(\rho ^{\exp }\) and \({\rho ^{{\rm{sim}}}}\) are
the experimental and simulated EM densities, respectively.

The simulated densities are usually computed using a
Gaussian mixture model, where a 3D Gaussian function is put
on the Cartesian coordinates of each target atom (i.e., protein atom),
and all contributions are integrated in each voxel of the map.
Here, several schemes have been proposed, in which the Gaussian
function is weighted with an atomic number \sphinxcite{99_References:topf-2008} or mass \sphinxcite{99_References:ishida-2014},
or it is simply applied to non-hydrogen atom \sphinxcite{99_References:tama-2004}.
In \sphinxstylestrong{GENESIS}, the last scheme is introduced. The simulated density of each voxel is defined as:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}{\rho ^{{\rm{sim}}}}(i.j,k) = \sum\limits_{n = 1}^N {\int {\int {\int_{{V_{ijk}}} {{g_n}(x,y,z){\rm{d}}x{\rm{d}}y{\rm{d}}z} } } }\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(V_{ijk}\) is the volume of the voxel, \sphinxstyleemphasis{N} is the total number of
non-hydrogen atoms in the system, and \sphinxstyleemphasis{n} is the index of the atom.
The Gaussian function \({g_n}(x,y,z)\) is given by
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}{g_n}(x,y,z) = \exp \left[ { - \frac{3}{{2{\sigma ^2}}}\left\{ {{{(x - {x_n})}^2} + {{(y - {y_n})}^2} + {{(z - {z_n})}^2}} \right\}} \right]\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(({x_n},{y_n},{z_n})\) are the coordinates of the \sphinxstyleemphasis{n}-th atom.
\(\sigma\) determines the width of the Gaussian function, and the
generated EM density has the resolution of \(2\sigma\) in the map.


\subsection{Control parameters}
\label{\detokenize{19_Experiments:control-parameters}}
In \sphinxstylestrong{GENESIS}, EM biasing force is treated as a kind of “Restraints” (see {\hyperref[\detokenize{12_Restraints:restraints}]{\sphinxcrossref{\DUrole{std,std-ref}{Restraints section}}}}).
The force constant of the biasing potential is given in the \sphinxstylestrong{{[}RESTRAINTS{]}} section
in a similar manner as the other restraint potentials,
where “functionN = EM” is specified for the restraint type (see below).
The unit of the force constant is kcal/mol.
In the cryo-EM flexible fitting, the users add the \sphinxstylestrong{{[}EXPERIMENTS{]}} section
in the control file, and specify the following keywords.
Note that the \sphinxstylestrong{{[}FITTING{]}} section (see {\hyperref[\detokenize{13_Fitting:fitting}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting section}}}}) is not related
to the cryo-EM flexible fitting.

The flexible fitting can be combined with various methods such as the replica-exchange
umbrella-sampling scheme (REUSfit) \sphinxcite{99_References:miyashita-2017},
all-atom Go-model (MDfit) \sphinxcite{99_References:whitford-2011}, and GB/SA implicit solvent model.
The method is parallelized with the hybrid MPI+OpenMP scheme
in both \sphinxstylestrong{ATDYN} and \sphinxstylestrong{SPDYN}, and also accelerated with GPGPU calculation in \sphinxstylestrong{SPDYN} \sphinxcite{99_References:mori-2019}.


\bigskip\hrule\bigskip


\sphinxstylestrong{emfit} \sphinxstyleemphasis{YES / NO}
\begin{quote}

\sphinxstylestrong{Default : NO}

Turn on or off the cryo-EM flexible fitting.
\end{quote}

\sphinxstylestrong{emfit\_target} \sphinxstyleemphasis{Character}
\begin{quote}

\sphinxstylestrong{Default : N/A}

The file name of the target EM density map.
The available file format is MRC/CCP4 (ver. 2000 or later) or SITUS (\sphinxurl{https://situs.biomachina.org/}),
which is automatically selected according to the file extension.
The file extension should be “.map”, “.mrc”, or “.ccp4” for MRC/CCP4, and “.sit” for SITUS.
\end{quote}

\sphinxstylestrong{emfit\_sigma} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 2.5} (unit : Angstrom)

Resolution parameter of the simulated map.
This is usually set to the half of the resolution of the target map.
For example, if the target map resolution is 5 Angstrom,
“emfit\_sigma=2.5” is a reasonable choice.
\end{quote}

\sphinxstylestrong{emfit\_tolerance} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.001}

This variable determines the tail length of the Gaussian function.
For example, if “emfit\_tolerance=0.001” is specified, the Gaussian function is truncated
to zero when it is less than 0.1\% of the maximum value.
Smaller value requires large computational cost.
\end{quote}

\sphinxstylestrong{emfit\_zero\_threshold} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default : 0.0}

This variable determines a threshold to set zero-densities in the target EM map.
If the density in a voxel of the target map is under a given “emfit\_zero\_threshold”,
the density is set to zero.
\end{quote}

\sphinxstylestrong{emfit\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default : 1}

Update frequency of the EM biasing force.
In the case of “emfit\_period=1”, the force is updated every step (slow but accurate).
\end{quote}


\subsection{Usage in SPDYN}
\label{\detokenize{19_Experiments:usage-in-spdyn}}
There are some limitations in the cryo-EM flexible fitting with SPDYN.
Here, we assume that the users perform the flexible fitting with explicit solvent under the periodic boundary condition (PBC).
In the PBC, there is a unit cell at the center of the system (red box in \hyperref[\detokenize{19_Experiments:cryo-spdyn}]{Fig.\@ \ref{\detokenize{19_Experiments:cryo-spdyn}}} left panel), which is surrounded by 26 image cells.
In GENESIS, the center of the unit cell is always at the origin \((X,Y,Z) = (0,0,0)\).
Thus, the coordinates of the edge of the unit cell is \((X,Y,Z) = 0.5 \times ( \pm {\rm{box\_size\_x,}} \pm {\rm{box\_size\_y,}} \pm {\rm{box\_size\_z}})\).
Please keep in mind that the “water box position” of the initial structure does NOT always correspond with the “unit cell position”.
If the user constructed the initial structure without considering the unit cell position, the center of mass of the system
might be largely shifted from the origin like \hyperref[\detokenize{19_Experiments:cryo-spdyn}]{Fig.\@ \ref{\detokenize{19_Experiments:cryo-spdyn}}} right panel, which is basically no problem in typical MD simulations.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=0.800\linewidth]{{cryofitting_spdyn}.png}
\caption{Limitation of the cryo-EM flexible fitting in SPDYN.}\label{\detokenize{19_Experiments:cryo-spdyn}}\end{figure}

However, as shown in \hyperref[\detokenize{19_Experiments:cryo-spdyn}]{Fig.\@ \ref{\detokenize{19_Experiments:cryo-spdyn}}} left panel, in the flexibe fitting with SPDYN,
all fitting atoms should satisfy the following condition due to parallelization algorithms implemented in SPDYN:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}{-0.5 \times {\rm{box\_size\_x}} + {\rm{margin}} < x < 0.5 \times {\rm{box\_size\_x}} - {\rm{margin}}}\\
{-0.5 \times {\rm{box\_size\_y}} + {\rm{margin}} < y < 0.5 \times {\rm{box\_size\_y}} - {\rm{margin}}}\\
{-0.5 \times {\rm{box\_size\_z}} + {\rm{margin}} < z < 0.5 \times {\rm{box\_size\_z}} - {\rm{margin}}}\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \sphinxstyleemphasis{x}, \sphinxstyleemphasis{y}, and \sphinxstyleemphasis{z} are the coordinates of each fitting atom. Here, the margin size should be larger than \(0.5 \times pairlistdist\).
If the fitting atoms are located outside this region, as shown in \hyperref[\detokenize{19_Experiments:cryo-spdyn}]{Fig.\@ \ref{\detokenize{19_Experiments:cryo-spdyn}}} right panel, correct flexible fitting calculations cannot be done.
In such cases, the users must translate the center of mass of the target protein and density map to the origin
by using other external tools. For the translation of the density map, \sphinxstyleemphasis{map2map} tool in SITUS is useful.
This kind of limitations does not exist in ATDYN.


\section{Examples}
\label{\detokenize{19_Experiments:examples}}

\subsection{Control parameters for the cryo-EM flexible fitting}
\label{\detokenize{19_Experiments:control-parameters-for-the-cryo-em-flexible-fitting}}
The following is an example of the cryo-EM flexible fitting using \sphinxstyleemphasis{k} = 10,000 kcal/mol
for the 4.1 Angstrom resolution map.
The other sections are common to the conventional MD simulations.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}          \PYG{o}{=} all and not hydrogen

\PYG{o}{[}RESTRAINTS\PYG{o}{]}
\PYG{n+nv}{nfunctions}      \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{function1}       \PYG{o}{=} EM              \PYG{c+c1}{\PYGZsh{} apply EM biasing potential}
\PYG{n+nv}{constant1}       \PYG{o}{=} \PYG{l+m}{10000}           \PYG{c+c1}{\PYGZsh{} force constant in Eem = k*(1 \PYGZhy{} c.c.)}
\PYG{n+nv}{select\PYGZus{}index1}   \PYG{o}{=} \PYG{l+m}{1}               \PYG{c+c1}{\PYGZsh{} apply force on protein heavy atoms}

\PYG{o}{[}EXPERIMENTS\PYG{o}{]}
\PYG{n+nv}{emfit}           \PYG{o}{=} YES             \PYG{c+c1}{\PYGZsh{} perform EM flexible fitting}
\PYG{n+nv}{emfit\PYGZus{}target}    \PYG{o}{=} emd\PYGZus{}8623.sit    \PYG{c+c1}{\PYGZsh{} target EM density map}
\PYG{n+nv}{emfit\PYGZus{}sigma}     \PYG{o}{=} \PYG{l+m}{2}.05            \PYG{c+c1}{\PYGZsh{} half of the map resolution (4.1 A)}
\PYG{n+nv}{emfit\PYGZus{}tolerance} \PYG{o}{=} \PYG{l+m}{0}.001           \PYG{c+c1}{\PYGZsh{} Tolerance for error (0.1\PYGZpc{})}
\PYG{n+nv}{emfit\PYGZus{}period}    \PYG{o}{=} \PYG{l+m}{1}               \PYG{c+c1}{\PYGZsh{} emfit force update period}
\end{sphinxVerbatim}

The following is an example of REUSfit using 8 replicas,
where the force constants 100\textendash{}800 kcal/mol are assigned to each replica,
and exchanged during the simulation (see also {\hyperref[\detokenize{14_REMD:remd}]{\sphinxcrossref{\DUrole{std,std-ref}{REMD section}}}}).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}REMD\PYG{o}{]}
\PYG{n+nv}{dimension}       \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{exchange\PYGZus{}period} \PYG{o}{=} \PYG{l+m}{1000}
\PYG{n+nv}{type1}           \PYG{o}{=} RESTRAINT
\PYG{n+nv}{nreplica1}       \PYG{o}{=} \PYG{l+m}{8}
\PYG{n+nv}{rest\PYGZus{}function1}  \PYG{o}{=} \PYG{l+m}{1}

\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}          \PYG{o}{=} all and not hydrogen

\PYG{o}{[}RESTRAINTS\PYG{o}{]}
\PYG{n+nv}{nfunctions}      \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{function1}       \PYG{o}{=} EM
\PYG{n+nv}{constant1}       \PYG{o}{=} \PYG{l+m}{100} \PYG{l+m}{200} \PYG{l+m}{300} \PYG{l+m}{400} \PYG{l+m}{500} \PYG{l+m}{600} \PYG{l+m}{700} \PYG{l+m}{800}
\PYG{n+nv}{select\PYGZus{}index1}   \PYG{o}{=} \PYG{l+m}{1}

\PYG{o}{[}EXPERIMENTS\PYG{o}{]}
\PYG{n+nv}{emfit}           \PYG{o}{=} YES
\PYG{n+nv}{emfit\PYGZus{}target}    \PYG{o}{=} target.sit
\PYG{n+nv}{emfit\PYGZus{}sigma}     \PYG{o}{=} \PYG{l+m}{5}
\PYG{n+nv}{emfit\PYGZus{}tolerance} \PYG{o}{=} \PYG{l+m}{0}.001
\PYG{n+nv}{emfit\PYGZus{}period}    \PYG{o}{=} \PYG{l+m}{1}
\end{sphinxVerbatim}


\subsection{Log messages in the cryo-EM flexible fitting}
\label{\detokenize{19_Experiments:log-messages-in-the-cryo-em-flexible-fitting}}
Here, we show examples of the log message obtained from the flexible fitting
in the NPT ensemble. In the case of ATDYN, the cross-correlation-coefficient (c.c.)
between the experimental and simulated density maps is displayed in the column “RESTR\_CVS001”,
if the EM biasing potential is specified in “function1” in the {[}RESTRAINTS{]} section:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}STEP5\PYG{o}{]} Perform Molecular Dynamics Simulation

INFO:      STEP           TIME      TOTAL\PYGZus{}ENE  POTENTIAL\PYGZus{}ENE     KINETIC\PYGZus{}ENE
           RMSG           BOND          ANGLE   UREY\PYGZhy{}BRADLEY        DIHEDRAL
       IMPROPER           CMAP        VDWAALS          ELECT RESTRAINT\PYGZus{}TOTAL
   RESTRAINT001   RESTR\PYGZus{}CVS001    TEMPERATURE         VOLUME            BOXX
           BOXY           BOXZ         VIRIAL       PRESSURE         PRESSXX
        PRESSYY        PRESSZZ
\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{} \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{} \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{} \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{} \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}
INFO:       \PYG{l+m}{500}         \PYG{l+m}{1}.0000    \PYGZhy{}93200.5069   \PYGZhy{}116131.3374      \PYG{l+m}{22930}.8305
         \PYG{l+m}{7}.6630       \PYG{l+m}{884}.5022      \PYG{l+m}{2334}.7396       \PYG{l+m}{304}.3716       \PYG{l+m}{2954}.2380
       \PYG{l+m}{186}.7872      \PYGZhy{}168.9763     \PYG{l+m}{10278}.5281   \PYGZhy{}133282.7986        \PYG{l+m}{377}.2708
       \PYG{l+m}{377}.2708         \PYG{l+m}{0}.8114       \PYG{l+m}{301}.5002    \PYG{l+m}{364999}.3065         \PYG{l+m}{89}.1785
        \PYG{l+m}{63}.9758        \PYG{l+m}{63}.9758    \PYGZhy{}15748.8082       \PYGZhy{}86.7133        \PYGZhy{}90.5768
       \PYGZhy{}73.8258       \PYGZhy{}95.7371
\end{sphinxVerbatim}

In the case of SPDYN, c.c. is displayed in the column “EMCORR”:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
INFO:      STEP           TIME      TOTAL\PYGZus{}ENE  POTENTIAL\PYGZus{}ENE     KINETIC\PYGZus{}ENE
           RMSG           BOND          ANGLE   UREY\PYGZhy{}BRADLEY        DIHEDRAL
       IMPROPER           CMAP        VDWAALS          ELECT          EMCORR
RESTRAINT\PYGZus{}TOTAL    TEMPERATURE         VOLUME           BOXX            BOXY
           BOXZ         VIRIAL       PRESSURE        PRESSXX         PRESSYY
        PRESSZZ
\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{} \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{} \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{} \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{} \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}
INFO:       \PYG{l+m}{500}         \PYG{l+m}{1}.0000    \PYGZhy{}93589.1620   \PYGZhy{}116347.3335      \PYG{l+m}{22758}.1715
         \PYG{l+m}{7}.5975       \PYG{l+m}{932}.4707      \PYG{l+m}{2369}.3892       \PYG{l+m}{300}.5418       \PYG{l+m}{2926}.0992
       \PYG{l+m}{171}.4987      \PYGZhy{}141.8994     \PYG{l+m}{10611}.0971   \PYGZhy{}133889.5003          \PYG{l+m}{0}.8135
       \PYG{l+m}{372}.9695       \PYG{l+m}{299}.2301    \PYG{l+m}{364820}.2154        \PYG{l+m}{89}.1639         \PYG{l+m}{63}.9654
        \PYG{l+m}{63}.9654    \PYGZhy{}14423.7954       \PYG{l+m}{140}.6472       \PYG{l+m}{311}.6641         \PYG{l+m}{19}.3734
        \PYG{l+m}{90}.9039
\end{sphinxVerbatim}


\chapter{Alchemy section}
\label{\detokenize{20_Alchemy:alchemy-section}}\label{\detokenize{20_Alchemy:alchemy}}\label{\detokenize{20_Alchemy::doc}}

\section{Free energy perturbation}
\label{\detokenize{20_Alchemy:free-energy-perturbation}}
In the \sphinxstylestrong{{[}Alchemy{]}} section, the users can specify keywords for the free-energy perturbation (FEP) method, which is one of the alchemical free energy calculations.
The FEP method calculates the free-energy difference between two states by gradually changing a part of the system from one state to another state.
Since the free energy depends only on the initial and final states, any intermediate states can be chosen, regardless of whether they are physically realizable or not.
If the intermediate states are not physically unrealizable but computationally realizable, the calculation and thermodynamic process are referred to as the alchemical free-energy calculation and alchemical process, respectively.
Using alchemical processes, the FEP method can be applied to a variety of free-energy calculations, such as solvation free energies, binding free energies, and free-energy changes upon protein mutations.
In particular, absolute binding free energies can be calculated by gradually vanishing a ligand of interest, while relative binding free energies can be calculated by gradually changing one ligand to another.

GENESIS enables to perform various alchemical calculations by implementing dual-topology and hybrid-topology approaches, soft-core potentials, and lambda-exchange calculations based on REMD.
GPGPU acceleration are also available in FEP simulations.
The short-range non-bonded interactions (LJ and PME real part) are calculated by GPU, while the long-range (PME reciprocal part) are calculated by CPU.
Currently, \sphinxstylestrong{SPDYN} supports only CHARMM and AMBER force fields for FEP simulations.
The FEP method is not available in \sphinxstylestrong{ATDYN}.
In this section, the FEP functions in GENESIS are briefly summarized and some examples are shown.


\subsection{Theory of FEP}
\label{\detokenize{20_Alchemy:theory-of-fep}}
The FEP method calculates the free-energy difference between two states, A and B, using the following equation.
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\Delta F &= F_{\text{B}} - F_{\text{A}} \\
    &= -k_{\text{BT}} \ln \frac{\int dx \exp [-\beta U_{\text{B}}(x)]}{\int dx \exp [-\beta U_{\text{A}}(x)]} \\
    &= -k_{\text{BT}} \ln \frac{\int dx \exp [-\beta U_{\text{A}}(x) - \beta \Delta U(x)]}{\int dx \exp [-\beta U_{\text{A}}(x)]} \\
    &= -k_{\text{BT}} \ln \left< \exp [-\beta \Delta U(x)]\right>_{\text{A}},\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(F\) and \(U\) are the free energy and potential energy of state A or B, \(\Delta U\) is the difference between \(U_{\text{A}}\) and \(U_{\text{B}}\), and \(x\) is the configuration of the system.
The bracket at the final line represents the ensemble average at state A.
This equation means that \(\Delta F\) can be estimated by sampling only equilibrium configurations of the state A.
However, if the difference between states A and B is large, there is little overlap of energy distributions (Left of \hyperref[\detokenize{20_Alchemy:intermediate-states}]{Fig.\@ \ref{\detokenize{20_Alchemy:intermediate-states}}}) and the configurations at state B are poorly sampled by simulations at the state A, leading to large statistical errors.
To reduce the errors, \(n-2\) intermediate states are inserted between states A and B to overlap energy distributions (Right of Left of \hyperref[\detokenize{20_Alchemy:intermediate-states}]{Fig.\@ \ref{\detokenize{20_Alchemy:intermediate-states}}}).
The potential energy of the intermediate state \(i\) is defined by \(U(\lambda_i) = (1-\lambda_i) U_{\text{A}} + \lambda_i U_{\text{B}}\), where \(\lambda_i\) is the scaling parameter for connecting the initial and final states.
By changing \(\lambda_i\) from A to B, states A and B can be connected smoothly.
\(\Delta F\) can be estimated by calculating the summation of free-energy changes between adjacent states:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}\Delta F &= \sum_{i=0}^{n-1}\Delta F_i \\
    &= -k_BT \sum_{i=0}^{n-1} \ln \left< \exp [-\beta (U(\lambda_i+1)-U(\lambda_i))]\right>,\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where the subscript \(\lambda_i\) represents the ensemble average at state \(i\).
The states of \(i = 0\) and \(n\) correspond to state A and B, respectively.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=0.700\linewidth]{{Alchemy_Fig1}.png}
\caption{Insertion of intermediate states. Some intermediate states are inserted between the reference state and the target state states to overlap energy distributions.}\label{\detokenize{20_Alchemy:intermediate-states}}\end{figure}


\subsection{Dual topology approach}
\label{\detokenize{20_Alchemy:dual-topology-approach}}
One of the most important applications of the FEP method is the calculation of protein-ligand binding affinity, which represents how strong a ligand binds to a protein.
In drug discovery, it is required to find a ligand that best binds to the target protein from a large number of chemical compounds.
The difference between binding affinities of two ligands, called the relative binding affinity, can be calculated by changing one ligand into another ligand during the simulation.
For example, consider the mutation from benzene to phenol (\hyperref[\detokenize{20_Alchemy:dual-topology}]{Fig.\@ \ref{\detokenize{20_Alchemy:dual-topology}}} (a)).
Benzene and phenol correspond to states A and B, respectively.
The atoms of benzene except for a hydrogen atom are common to both ligands, which have no need to be perturbed.
On the other hand, the H atom of benzene and the OH atoms of phenol are different in not only their force field parameters but also their topology.
To minimize perturbation and treat the topological difference, topologies of two ligands are unified such that the atoms with different topologies connect with the common atoms (\hyperref[\detokenize{20_Alchemy:dual-topology}]{Fig.\@ \ref{\detokenize{20_Alchemy:dual-topology}}} (b)).
This topology is called the dual topology, which consists of the common atoms, the atoms included in only state A (dualA in \hyperref[\detokenize{20_Alchemy:dual-topology}]{Fig.\@ \ref{\detokenize{20_Alchemy:dual-topology}}} (b)), and the atoms included in only state B (dualB in \hyperref[\detokenize{20_Alchemy:dual-topology}]{Fig.\@ \ref{\detokenize{20_Alchemy:dual-topology}}} (b)) \sphinxcite{99_References:gao-1989}\sphinxcite{99_References:pearlman-1994}\sphinxcite{99_References:axelsen-1998}.
The perturbation is applied to only the dualA and dualB parts.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=0.700\linewidth]{{Alchemy_Fig2}.png}
\caption{Dual topology approach. (a) Benzene and phenol. Common atoms, H atom of benzene, and OH atoms of phenol are in black, cyan, and green, respectively. (b) Dual topology of benzene and phenol. (c) The alchemical transformation from benzene to phenol.}\label{\detokenize{20_Alchemy:dual-topology}}\end{figure}

The free-energy change upon the mutation can be calculated by gradually switching the interactions of the dual-topology part from benzene to phenol (\hyperref[\detokenize{20_Alchemy:dual-topology}]{Fig.\@ \ref{\detokenize{20_Alchemy:dual-topology}}} (c)).
At state A, only the H atom exists in the dual-topology part, while the OH atoms do not interact with the other atoms in the system.
During the alchemical transformation, the H atom gradually disappears, whereas the OH atoms gradually appears.
At state B, only the OH atoms exist in the dual-topology part and interact with the other atoms.
The non-bonded potential energy is modified to connect smoothly state A to state B by introducing \(\lambda_{LJ}\) and \(\lambda_{elec}\):
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}U_{\text{nonbond}} &= U_{\text{LJ}}^{\text{other-other}} + U_{\text{LJ}}^{\text{other-common}} + U_{\text{LJ}}^{\text{common-common}} + U_{\text{elec}}^{\text{other-other}} + U_{\text{elec}}^{\text{other-common}} + U_{\text{elec}}^{\text{common-common}} \\
    & + \lambda_{\text{LJ}}^A (U_{\text{LJ}}^{\text{other-dualA}} + U_{\text{LJ}}^{\text{common-dualA}} + U_{\text{LJ}}^{\text{dualA-dualA}}) \\
    & + \lambda_{\text{LJ}}^B (U_{\text{LJ}}^{\text{other-dualB}} + U_{\text{LJ}}^{\text{common-dualB}} + U_{\text{LJ}}^{\text{dualB-dualB}}) \\
    & + \lambda_{\text{elec}}^A (U_{\text{elec}}^{\text{other-dualA}} + U_{\text{elec}}^{\text{common-dualA}} + U_{\text{elec}}^{\text{dualA-dualA}}) \\
    & + \lambda_{\text{elec}}^B (U_{\text{elec}}^{\text{other-dualB}} + U_{\text{elec}}^{\text{common-dualB}} + U_{\text{elec}}^{\text{dualB-dualB}})\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where “common”, “dualA”, “dualB”, and “other” in the superscripts respectively represent the common atoms, the atoms existing only at state A, the atoms existing only at state B, and the other molecules including solvent molecules, proteins, or other ligands.
For example, \(U_{\text{LJ}}^{\text{common-dualA}}\) represents the LJ interaction between a common atom and a dualA atom.
The potential energy at \(\lambda_{\text{LJ}}^{A}=1\), \(\lambda_{\text{elec}}^{A}=1\), \(\lambda_{\text{LJ}}^{B}=0\), and \(\lambda_{\text{elec}}^{B}=0\) corresponds to that of state A, while the energy at \(\lambda_{\text{LJ}}^{A}=0\), \(\lambda_{\text{elec}}^{A}=0\), \(\lambda_{\text{LJ}}^{B}=1\), and \(\lambda_{\text{elec}}^{B}=1\) corresponds to that of state B.
By gradually changing \(\lambda_{\text{LJ}}^{A}\), \(\lambda_{\text{elec}}^{A}\), \(\lambda_{\text{LJ}}^{B}\), and \(\lambda_{\text{elec}}^{B}\), states A and B can be connected smoothly.

In GENESIS, the dual-topology approach is available by specifying \sphinxstylestrong{fep\_topology = Dual} in the {[}ALCHEMY{]} section.
The users should select the atoms of the dual-topology part in the {[}SELECTION{]} section and assign their group numbers to \sphinxstylestrong{dualA} and \sphinxstylestrong{dualB} in the {[}ALCHEMY{]} section.
\(\lambda_{\text{LJ}}^{A}\), \(\lambda_{\text{elec}}^{A}\), \(\lambda_{\text{LJ}}^{B}\), and \(\lambda_{\text{elec}}^{B}\) can be specified by \sphinxstylestrong{lambljA}, \sphinxstylestrong{lambljB}, \sphinxstylestrong{lambelA}, and \sphinxstylestrong{lambelB}, respectively.
An example is shown below.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}ALCHEMY\PYG{o}{]}
\PYG{n+nv}{fep\PYGZus{}topology} \PYG{o}{=} Dual
\PYG{n+nv}{dualA}        \PYG{o}{=} \PYG{l+m}{1}  \PYG{c+c1}{\PYGZsh{} group1 in [SELECTION]}
\PYG{n+nv}{dualB}        \PYG{o}{=} \PYG{l+m}{2}  \PYG{c+c1}{\PYGZsh{} group2 in [SELECTION]}
\PYG{n+nv}{lambljA}      \PYG{o}{=} \PYG{l+m}{1}.00 \PYG{l+m}{0}.75 \PYG{l+m}{0}.50 \PYG{l+m}{0}.25 \PYG{l+m}{0}.00
\PYG{n+nv}{lambljB}      \PYG{o}{=} \PYG{l+m}{0}.00 \PYG{l+m}{0}.25 \PYG{l+m}{0}.50 \PYG{l+m}{0}.75 \PYG{l+m}{1}.00
\PYG{n+nv}{lambelA}      \PYG{o}{=} \PYG{l+m}{1}.00 \PYG{l+m}{0}.75 \PYG{l+m}{0}.50 \PYG{l+m}{0}.25 \PYG{l+m}{0}.00
\PYG{n+nv}{lambelB}      \PYG{o}{=} \PYG{l+m}{0}.00 \PYG{l+m}{0}.25 \PYG{l+m}{0}.50 \PYG{l+m}{0}.75 \PYG{l+m}{1}.00

\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}       \PYG{o}{=} ai:1    \PYG{c+c1}{\PYGZsh{} atoms in dual A}
\PYG{n+nv}{group2}       \PYG{o}{=} ai:2\PYGZhy{}3  \PYG{c+c1}{\PYGZsh{} atoms in dual B}
\end{sphinxVerbatim}

In this example, five sets of \(\lambda_{\text{LJ}}^{A}\), \(\lambda_{\text{elec}}^{A}\), \(\lambda_{\text{LJ}}^{B}\), and \(\lambda_{\text{elec}}^{B}\) are used to connect state A to state B.
In the {[}SELECTION{]} section, the H atom of benzene and the OH atoms of phenol are selected by group 1 and 2, respectively.
The group IDs are specified as \sphinxstylestrong{dualA = 1} and \sphinxstylestrong{dualB = 2}.


\subsection{Hybrid topology approach}
\label{\detokenize{20_Alchemy:hybrid-topology-approach}}
In the dual-topology approach, the force field parameters of common atoms are assumed to be the same in both states.
However, in general, they are different each other. \hyperref[\detokenize{20_Alchemy:hybrid-topology}]{Fig.\@ \ref{\detokenize{20_Alchemy:hybrid-topology}}} (a) shows that the charge distribution of the benzene ring of benzene is different from that of phenol.
The parameters of the common atoms for bond, angle, and dihedral are also different between two molecules.
To treat the difference of the force field parameters, the parts of the molecules with the same topology are superimposed (\hyperref[\detokenize{20_Alchemy:hybrid-topology}]{Fig.\@ \ref{\detokenize{20_Alchemy:hybrid-topology}}} (b)) \sphinxcite{99_References:jiang-2019}.
The superimposed part has a single topology, in which the parts corresponding to states A and B are referred to as “singleA” and “singleB”, respectively.
During FEP simulations, the single-topology part does not change its topology, but its force field parameters (charge, LJ, and internal bond) are gradually changed from state A to state B (\hyperref[\detokenize{20_Alchemy:hybrid-topology}]{Fig.\@ \ref{\detokenize{20_Alchemy:hybrid-topology}}} (c)).
In contrast, the other part has a dual topology, in which “dualA” and “dualB” correspond to states A and B, respectively.
In the dual-topology part, their topology is changed as well as their parameters (\hyperref[\detokenize{20_Alchemy:hybrid-topology}]{Fig.\@ \ref{\detokenize{20_Alchemy:hybrid-topology}}} (c)).

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=0.700\linewidth]{{Alchemy_Fig3}.png}
\caption{Hybrid topology approach. (a) Benzene and phenol. H atom of benzene and OH atoms of phenol are in cyan and green, respectively. The point charges on common atoms are shown in red and magenda, which are determined using Amber Tools \sphinxcite{99_References:amber18-program}. (b) Hybrid topology of benzene and phenol. (c) The alchemical transformation from benzene to phenol.}\label{\detokenize{20_Alchemy:hybrid-topology}}\end{figure}

In the hybrid topology approach, the potential energy is scaled by \(\lambda_{\text{LJ}}\), \(\lambda_{\text{elec}}\), and \(\lambda_{\text{bond}}\):
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}U_{\text{nonbond}} &= U_{\text{LJ}}^{\text{other-other}} + U_{\text{elec}}^{\text{other-other}} \\
    & + \lambda_{\text{LJ}}^A (U_{\text{LJ}}^{\text{other-singleA}} + U_{\text{LJ}}^{\text{other-dualA}} + U_{\text{LJ}}^{\text{singleA-singleA}} + U_{\text{LJ}}^{\text{singleA-dualA}} + U_{\text{LJ}}^{\text{dualA-dualA}}) \\
    & + \lambda_{\text{LJ}}^B (U_{\text{LJ}}^{\text{other-singleB}} + U_{\text{LJ}}^{\text{other-dualB}} + U_{\text{LJ}}^{\text{singleB-singleB}} + U_{\text{LJ}}^{\text{singleB-dualB}} + U_{\text{LJ}}^{\text{dualB-dualB}}) \\
    & + \lambda_{\text{elec}}^A (U_{\text{elec}}^{\text{other-singleA}} + U_{\text{elec}}^{\text{other-dualA}} + U_{\text{elec}}^{\text{singleA-singleA}} + U_{\text{elec}}^{\text{singleA-dualA}} + U_{\text{elec}}^{\text{dualA-dualA}}) \\
    & + \lambda_{\text{elec}}^B (U_{\text{elec}}^{\text{other-singleB}} + U_{\text{elec}}^{\text{other-dualB}} + U_{\text{elec}}^{\text{singleB-singleB}} + U_{\text{elec}}^{\text{singleB-dualB}} + U_{\text{elec}}^{\text{dualB-dualB}}) \\
\\
U_{\text{bond}} &= U_{\text{bond}}^{\text{other}} + U_{\text{bond}}^{\text{dualA}} + U_{\text{bond}}^{\text{dualB}} + U_{\text{bond}}^{\text{singleA-dualA}} + U_{\text{bond}}^{\text{singleB-dualB}} \\
    & + \lambda_{\text{bond}}^A (U_{\text{bond}}^{s\text{ingleA}}) \\
    & + \lambda_{\text{bond}}^B (U_{\text{bond}}^{\text{singleB}}) \\\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where “singleA”, “singleB”, “dualA”, and “dualB” in the superscripts respectively represent the atoms corresponding to “singleA”, “singleB”, “dualA”, and “dualB” parts, respectively, and “other” represents the other molecules including solvent molecules, proteins, or other ligands.
The potential energy at \(\lambda_{\text{LJ}}^{A}=1\), \(\lambda_{\text{elec}}^{A}=1\), \(\lambda_{\text{bond}}^{A}=1\), \(\lambda_{\text{LJ}}^{B}=0\), \(\lambda_{\text{elec}}^{B}=0\), and \(\lambda_{\text{bond}}^{B}=0\) corresponds to that of state A,
while the energy at \(\lambda_{\text{LJ}}^{A}=0\), \(\lambda_{\text{elec}}^{A}=0\), \(\lambda_{\text{bond}}^{A}=0\), \(\lambda_{\text{LJ}}^{B}=1\), \(\lambda_{\text{elec}}^{B}=1\), and \(\lambda_{\text{bond}}^{B}=1\) corresponds to that of state B.
By gradually changing the lambda values, states A and B can be connected smoothly.

In GENESIS, the hybrid-topology approach is available by specifying \sphinxstylestrong{fep\_topology = Hybrid} in the {[}ALCHEMY{]} section.
The users should select the atoms of the single-topology and dual-topology parts in the {[}SELECTION{]} section and assign their group numbers to \sphinxstylestrong{singleA}, \sphinxstylestrong{singleB}, \sphinxstylestrong{dualA}, and \sphinxstylestrong{dualB} in the {[}ALCHEMY{]} section.
\(\lambda_{\text{LJ}}^{A}\), \(\lambda_{\text{elec}}^{A}\), \(\lambda_{\text{bond}}^{A}\), \(\lambda_{\text{LJ}}^{B}\), \(\lambda_{\text{elec}}^{B}\), and \(\lambda_{\text{bond}}^{B}\) can be specified by \sphinxstylestrong{lambljA}, \sphinxstylestrong{lambljB}, \sphinxstylestrong{lambelA}, and \sphinxstylestrong{lambelB}, respectively.
An example is shown below.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}ALCHEMY\PYG{o}{]}
\PYG{n+nv}{fep\PYGZus{}topology} \PYG{o}{=} Hybrid
\PYG{n+nv}{singleA}      \PYG{o}{=} \PYG{l+m}{1}  \PYG{c+c1}{\PYGZsh{} group1 in [SELECTION]}
\PYG{n+nv}{singleB}      \PYG{o}{=} \PYG{l+m}{2}  \PYG{c+c1}{\PYGZsh{} group2 in [SELECTION]}
\PYG{n+nv}{dualA}        \PYG{o}{=} \PYG{l+m}{3}  \PYG{c+c1}{\PYGZsh{} group3 in [SELECTION]}
\PYG{n+nv}{dualB}        \PYG{o}{=} \PYG{l+m}{4}  \PYG{c+c1}{\PYGZsh{} group4 in [SELECTION]}
\PYG{n+nv}{lambljA}      \PYG{o}{=} \PYG{l+m}{1}.00 \PYG{l+m}{0}.75 \PYG{l+m}{0}.50 \PYG{l+m}{0}.25 \PYG{l+m}{0}.00
\PYG{n+nv}{lambljB}      \PYG{o}{=} \PYG{l+m}{0}.00 \PYG{l+m}{0}.25 \PYG{l+m}{0}.50 \PYG{l+m}{0}.75 \PYG{l+m}{1}.00
\PYG{n+nv}{lambelA}      \PYG{o}{=} \PYG{l+m}{1}.00 \PYG{l+m}{0}.75 \PYG{l+m}{0}.50 \PYG{l+m}{0}.25 \PYG{l+m}{0}.00
\PYG{n+nv}{lambelB}      \PYG{o}{=} \PYG{l+m}{0}.00 \PYG{l+m}{0}.25 \PYG{l+m}{0}.50 \PYG{l+m}{0}.75 \PYG{l+m}{1}.00
\PYG{n+nv}{lambbondA}    \PYG{o}{=} \PYG{l+m}{1}.00 \PYG{l+m}{0}.75 \PYG{l+m}{0}.50 \PYG{l+m}{0}.25 \PYG{l+m}{0}.00
\PYG{n+nv}{lambbondB}    \PYG{o}{=} \PYG{l+m}{0}.00 \PYG{l+m}{0}.25 \PYG{l+m}{0}.50 \PYG{l+m}{0}.75 \PYG{l+m}{1}.00

\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}       \PYG{o}{=} ai:1\PYGZhy{}11   \PYG{c+c1}{\PYGZsh{} atoms in single A}
\PYG{n+nv}{group2}       \PYG{o}{=} ai:13\PYGZhy{}23  \PYG{c+c1}{\PYGZsh{} atoms in single B}
\PYG{n+nv}{group3}       \PYG{o}{=} ai:12     \PYG{c+c1}{\PYGZsh{} atoms in dual A}
\PYG{n+nv}{group4}       \PYG{o}{=} ai:24\PYGZhy{}25  \PYG{c+c1}{\PYGZsh{} atoms in dual B}
\end{sphinxVerbatim}

In this example, five sets of the lambda values are used to connect state A to state B.
In the {[}SELECTION{]} section,
the benzene ring of benzene, the benzene ring of phenol, the H atom of benzene, and the OH atoms of phenol are selected by group 1, 2, 3, and 4, respectively.
The group IDs are specified as \sphinxstylestrong{singleA = 1}, \sphinxstylestrong{singleB = 2}, \sphinxstylestrong{dualA = 3}, and \sphinxstylestrong{dualB = 4}.


\subsection{Soft core potentials}
\label{\detokenize{20_Alchemy:soft-core-potentials}}
Close to the end point of alchemical calculations (\(\lambda_{\text{LJ}} = 0\) or 1),
overlaps between perturbed atoms or between perturbed and non-perturbed atoms occur, causing large energy change.
The system becomes unstable due to the overlaps and the simulations might be stopped, which is called the end point catastrophe.
To avoid the catastrophe, the soft core is introduced to the LJ potential \sphinxcite{99_References:zacharias-1994}:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}U_{\text{LJ}}(r_{ij}, \lambda_{\text{LJ}}) = 4 \lambda_{\text{LJ}} \epsilon \left[ \left( \frac{\sigma^2}{r_{ij}^2+\alpha_{sc} (1-\lambda_{\text{LJ}})} \right)^6 - \left( \frac{\sigma^2}{r_{ij}^2+\alpha_{sc} (1-\lambda_{\text{LJ}})} \right)^3 \right],\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(\alpha_{\text{sc}}\) is the parameter for the soft-core potential.
In the potential, \(r_{ij}^2\) is shifted to \(\alpha_{\text{sc}} (1-\lambda_{\text{LJ}})\), which weakens the repulsive part in the LJ potential when \(\lambda_{\text{LJ}}\) approaches 0 (\hyperref[\detokenize{20_Alchemy:lj-softcore}]{Fig.\@ \ref{\detokenize{20_Alchemy:lj-softcore}}}).
Since the soft-core potential corresponds to the original LJ potential at the end point of \(\lambda_{\text{LJ}}\):
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}U_{\text{LJ}}(r_{ij}, \lambda_{\text{LJ}}=1) &= 4 \epsilon \left[ \left( \frac{\sigma}{r_{ij}} \right)^{12} - \left( \frac{\sigma}{r_{ij})} \right)^6 \right], \\
U_{\text{LJ}}(r_{ij}, \lambda_{\text{LJ}}=0) &= 0,\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

the soft-core modification in the LJ potential does not affect the free-energy calculation.
\(\alpha_{\text{sc}}\) can be specified by a keyword \sphinxstylestrong{sc\_alpha} in the GENESIS control file.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=0.700\linewidth]{{Alchemy_Fig4}.png}
\caption{Soft core potential for Lennard-Jones interaction.}\label{\detokenize{20_Alchemy:lj-softcore}}\end{figure}

In GENESIS, the soft core is also applied to the electrostatic potential \sphinxcite{99_References:steinbrecher-2011}:
\begin{quote}

\vspace{-5mm}
\begin{equation*}
\begin{split}U_{\text{elec}}(r_{ij}, \lambda_{\text{elec}}) = \lambda_{\text{elec}} \frac{q_i q_j \text{erfc} (\alpha \sqrt{r_{ij}^2 + \beta_{sc} (1-\lambda_{\text{elec}})})}{\epsilon \sqrt{r_{ij}^2 + \beta_{sc} (1-\lambda_{\text{elec}})}} + \lambda_{\text{elec}} (\text{PME reciprocal and self terms}),\end{split}
\end{equation*}
\vspace{-3mm}
\end{quote}

where \(\beta_{\text{sc}}\) is the parameter for the electrostatic soft-core potential.
In the potential, \(r_{ij}^2\) is also shifted to \(\beta_{\text{sc}} (1-\lambda_{\text{elec}})\) like the LJ soft-core potential, which softens disruptions due to overlaps of point charges.
This soft-core potential is almost the same as used in Amber \sphinxcite{99_References:steinbrecher-2011}.
\(\beta_{\text{sc}}\) can be specified by a keyword \sphinxstylestrong{sc\_beta} in the GENESIS control file.


\subsection{FEP/\protect\(\lambda\protect\)-REMD}
\label{\detokenize{20_Alchemy:fep-lambda-remd}}
To enhance the sampling efficiency, the FEP simulations at different \(\lambda\) values are coupled using the Hamiltonian replica exchange method \sphinxcite{99_References:sugita-2000}\sphinxcite{99_References:fukunishi-2002}.
This method is called FEP/\(\lambda\)-REMD or \(\lambda\)-exchange FEP \sphinxcite{99_References:jiang-2009}.
Replicas run in parallel and exchange their parameters at fixed intervals during the simulation.
The exchanges between adjacent replicas are accepted or rejected according to Metropolis’s criterion.
In FEP/\(\lambda\)-REMD simulations, {[}REMD{]} section is also required.
\sphinxstylestrong{type1 = alchemy} is set to exchange the lambda values.
The following is an example of the control file for the FEP/\(\lambda\)-REMD simulation.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}REMD\PYG{o}{]}
\PYG{n+nv}{dimension}       \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{exchange\PYGZus{}period} \PYG{o}{=} \PYG{l+m}{500}
\PYG{n+nv}{type1}           \PYG{o}{=} alchemy
\PYG{n+nv}{nreplica1}       \PYG{o}{=} \PYG{l+m}{5}

\PYG{o}{[}ALCHEMY\PYG{o}{]}
\PYG{n+nv}{fep\PYGZus{}topology} \PYG{o}{=} Hybrid
\PYG{n+nv}{singleA}      \PYG{o}{=} \PYG{l+m}{1}  \PYG{c+c1}{\PYGZsh{} group1 in [SELECTION]}
\PYG{n+nv}{singleB}      \PYG{o}{=} \PYG{l+m}{2}  \PYG{c+c1}{\PYGZsh{} group2 in [SELECTION]}
\PYG{n+nv}{dualA}        \PYG{o}{=} \PYG{l+m}{3}  \PYG{c+c1}{\PYGZsh{} group3 in [SELECTION]}
\PYG{n+nv}{dualB}        \PYG{o}{=} \PYG{l+m}{4}  \PYG{c+c1}{\PYGZsh{} group4 in [SELECTION]}
\PYG{n+nv}{lambljA}      \PYG{o}{=} \PYG{l+m}{1}.00 \PYG{l+m}{0}.75 \PYG{l+m}{0}.50 \PYG{l+m}{0}.25 \PYG{l+m}{0}.00
\PYG{n+nv}{lambljB}      \PYG{o}{=} \PYG{l+m}{0}.00 \PYG{l+m}{0}.25 \PYG{l+m}{0}.50 \PYG{l+m}{0}.75 \PYG{l+m}{1}.00
\PYG{n+nv}{lambelA}      \PYG{o}{=} \PYG{l+m}{1}.00 \PYG{l+m}{0}.75 \PYG{l+m}{0}.50 \PYG{l+m}{0}.25 \PYG{l+m}{0}.00
\PYG{n+nv}{lambelB}      \PYG{o}{=} \PYG{l+m}{0}.00 \PYG{l+m}{0}.25 \PYG{l+m}{0}.50 \PYG{l+m}{0}.75 \PYG{l+m}{1}.00
\PYG{n+nv}{lambbondA}    \PYG{o}{=} \PYG{l+m}{1}.00 \PYG{l+m}{0}.75 \PYG{l+m}{0}.50 \PYG{l+m}{0}.25 \PYG{l+m}{0}.00
\PYG{n+nv}{lambbondB}    \PYG{o}{=} \PYG{l+m}{0}.00 \PYG{l+m}{0}.25 \PYG{l+m}{0}.50 \PYG{l+m}{0}.75 \PYG{l+m}{1}.00

\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}       \PYG{o}{=} ai:1\PYGZhy{}11   \PYG{c+c1}{\PYGZsh{} atoms in single A}
\PYG{n+nv}{group2}       \PYG{o}{=} ai:13\PYGZhy{}23  \PYG{c+c1}{\PYGZsh{} atoms in single B}
\PYG{n+nv}{group3}       \PYG{o}{=} ai:12     \PYG{c+c1}{\PYGZsh{} atoms in dual A}
\PYG{n+nv}{group4}       \PYG{o}{=} ai:24\PYGZhy{}25  \PYG{c+c1}{\PYGZsh{} atoms in dual B}
\end{sphinxVerbatim}


\section{Parameters for alchemy section}
\label{\detokenize{20_Alchemy:parameters-for-alchemy-section}}
\sphinxstylestrong{fep\_direction} \sphinxstyleemphasis{Bothsides / Forward / Reverse}
\begin{quote}

\sphinxstylestrong{Default: Bothsides}

Direction of calculation of energy difference.
When the current state of the simulation or the replica is \(i\),
GENESIS outputs the energy difference between state \(i\) and the adjacent state at a frequency determined by \sphinxstylestrong{fepout\_period}.
If \sphinxstylestrong{fep\_direction = Forward}, the energy difference between states \(i\) and \(i+1\) is output.
If \sphinxstylestrong{fep\_direction = Reverse}, the energy difference between states \(i\) and \(i-1\) is output.
If \sphinxstylestrong{fep\_direction = Bothsides}, the energy differences between states \(i\) and \(i-1\) and between states \(i\) and \(i+1\) are output.
\end{quote}

\sphinxstylestrong{fepout\_period} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: 0}

Period of outputting energy differences.
\end{quote}

\sphinxstylestrong{fep\_topology} \sphinxstyleemphasis{Dual / Hybrid}
\begin{quote}

\sphinxstylestrong{Default: Hybrid}

Topology of perturbed region.
If \sphinxstylestrong{fep\_topology = Dual}, the dual-topology approach is used.
If \sphinxstylestrong{fep\_topology = Hybrid}, the hybrid-topology approach is used.
\end{quote}

\sphinxstylestrong{singleA} \sphinxstyleemphasis{Integer} or \sphinxstyleemphasis{None}
\begin{quote}

\sphinxstylestrong{Default: 0}

Group index for the single topology region of state A.
If \sphinxstylestrong{singleA = None}, calculations for the region are skipped.
If \sphinxstylestrong{fep\_topology = Hybrid}, this parameter must be specified.
\end{quote}

\sphinxstylestrong{singleB} \sphinxstyleemphasis{Integer} or \sphinxstyleemphasis{None}
\begin{quote}

\sphinxstylestrong{Default: 0}

Group index for the single topology region of state B.
If \sphinxstylestrong{singleB = None}, calculations for the region are skipped.
If \sphinxstylestrong{fep\_topology = Hybrid}, this parameter must be specified.
\end{quote}

\sphinxstylestrong{dualA} \sphinxstyleemphasis{Integer} or \sphinxstyleemphasis{None}
\begin{quote}

\sphinxstylestrong{Default: 0}

Group index for the dual topology region of state A.
If \sphinxstylestrong{dualA = None}, calculations for the region are skipped.
\end{quote}

\sphinxstylestrong{dualB} \sphinxstyleemphasis{Integer} or \sphinxstyleemphasis{None}
\begin{quote}

\sphinxstylestrong{Default: 0}

Group index for the dual topology region of state B.
If \sphinxstylestrong{dualB = None}, calculations for the region are skipped.
\end{quote}

\sphinxstylestrong{sc\_alpha} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 5.0} (dimensionless)

Parameter for the soft-core potential for the Lennard-Jones interaction.
\end{quote}

\sphinxstylestrong{sc\_beta} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 0.5} (dimensionless)

Parameter for the soft-core potential for the electrostatic interaction.
\end{quote}

\sphinxstylestrong{equilsteps} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: 0}

Number of steps of equilibration at each lambda window.
If \sphinxstylestrong{equilsteps \textgreater{} 0}, equilibration run is performed until the time step reaches \sphinxstylestrong{equilsteps}.
During the equilibration, energy differences are not outputted.
After the equilibration, production run is performed with outputting energy differences until the time step reaches \sphinxstylestrong{timesteps + equilsteps}.
\end{quote}

\sphinxstylestrong{lambljA} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 1.0}

Scaling parameters for Lennard-Jones interactions in state A (\(\lambda_{\text{LJ}}^A\)).
\end{quote}

\sphinxstylestrong{lambljB} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 1.0}

Scaling parameters for Lennard-Jones interactions in state B (\(\lambda_{\text{LJ}}^B\)).
\end{quote}

\sphinxstylestrong{lambelA} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 1.0}

Scaling parameters for electrostatic interactions in state A (\(\lambda_{\text{elec}}^A\)).
\end{quote}

\sphinxstylestrong{lambelB} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 1.0}

Scaling parameters for electrostatic interactions in state B (\(\lambda_{\text{elec}}^B\)).
\end{quote}

\sphinxstylestrong{lambbondA} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 1.0}

Scaling parameters for bonded interactions in state A (\(\lambda_{\text{bond}}^A\)).
\end{quote}

\sphinxstylestrong{lambbondB} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 1.0}

Scaling parameters for bonded interactions in state B (\(\lambda_{\text{bond}}^B\)).
\end{quote}

\sphinxstylestrong{lambrest} \sphinxstyleemphasis{Real}
\begin{quote}

\sphinxstylestrong{Default: 1.0}

Scaling parameters for restraint interactions (\(\lambda_{\text{rest}}\)).
\end{quote}

\sphinxstylestrong{fep\_md\_type} \sphinxstyleemphasis{Serial / Single / Parallel}
\begin{quote}

\sphinxstylestrong{Default: Serial}

Type of FEP simulation.
If \sphinxstylestrong{fep\_md\_type = Serial}, FEP simulations are performed with changing lambda values specified in \sphinxstylestrong{lambljA}, \sphinxstylestrong{lambljB}, \sphinxstylestrong{lambelA}, etc. For example, if 0.0, 0.5, and 1.0 are specified in \sphinxstylestrong{lambljA}, GENESIS first performs the FEP simulation with \sphinxstylestrong{lambljA = 0.0}, subsequently performs the FEP simulation with \sphinxstylestrong{lambljA = 0.5}, and finally performs the FEP simulation with \sphinxstylestrong{lambljA = 1.0}.
If \sphinxstylestrong{fep\_md\_type = Single}, a FEP simulation is performed with the lambda window specified in \sphinxstylestrong{ref\_lambid}.
If \sphinxstylestrong{fep\_md\_type = Parallel}, each lambda window is simulated in parallel. In this case, \sphinxstylestrong{{[}REMD{]}} section must be specified.
\end{quote}

\sphinxstylestrong{ref\_lambid} \sphinxstyleemphasis{Integer}
\begin{quote}

\sphinxstylestrong{Default: 0}

Reference window id for a single FEP MD simulation.
If \sphinxstylestrong{fep\_md\_type = Single}, \sphinxstylestrong{ref\_lambid} must be specified.
\end{quote}


\section{Examples}
\label{\detokenize{20_Alchemy:examples}}
Example of a calculation of the solvation free energy of a ligand.
The solvation free energy corresponds to the free-energy change upon the transfer of the ligand from vacuum to solvent.
In state A (= in solvent) the ligand fully interacts with solvent molecules, whereas in state B (= in vacuum) those interactions vanishes.
To perform such calculation, the dual topology is employed, and \sphinxstylestrong{dualA} is set to the group ID of the selected ligand, while \sphinxstylestrong{dualB} is set to \sphinxstylestrong{NONE}.
\sphinxstylestrong{dualB = NONE} means that there is no ligand in the system at state B.
\sphinxstylestrong{lambljA}, \sphinxstylestrong{lambljB}, \sphinxstylestrong{lambelA}, and \sphinxstylestrong{lambelB} should be zero at state B.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}ALCHEMY\PYG{o}{]}
\PYG{n+nv}{fep\PYGZus{}direction} \PYG{o}{=} BothSides
\PYG{n+nv}{fep\PYGZus{}topology}  \PYG{o}{=} Dual
\PYG{n+nv}{singleA}       \PYG{o}{=} NONE
\PYG{n+nv}{singleB}       \PYG{o}{=} NONE
\PYG{n+nv}{dualA}         \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{dualB}         \PYG{o}{=} NONE
\PYG{n+nv}{fepout\PYGZus{}period} \PYG{o}{=} \PYG{l+m}{500}
\PYG{n+nv}{equilsteps}    \PYG{o}{=} \PYG{l+m}{0}
\PYG{n+nv}{sc\PYGZus{}alpha}      \PYG{o}{=} \PYG{l+m}{5}.0
\PYG{n+nv}{sc\PYGZus{}beta}       \PYG{o}{=} \PYG{l+m}{0}.5
\PYG{n+nv}{lambljA}       \PYG{o}{=} \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{0}.750 \PYG{l+m}{0}.500 \PYG{l+m}{0}.250 \PYG{l+m}{0}.000
\PYG{n+nv}{lambljB}       \PYG{o}{=} \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000
\PYG{n+nv}{lambelA}       \PYG{o}{=} \PYG{l+m}{1}.000 \PYG{l+m}{0}.750 \PYG{l+m}{0}.500 \PYG{l+m}{0}.250 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000
\PYG{n+nv}{lambelB}       \PYG{o}{=} \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000
\PYG{n+nv}{lambbondA}     \PYG{o}{=} \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{1}.000
\PYG{n+nv}{lambbondB}     \PYG{o}{=} \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000 \PYG{l+m}{0}.000
\PYG{n+nv}{lambrest}      \PYG{o}{=} \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{1}.000 \PYG{l+m}{1}.000

\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1} \PYG{o}{=} segid:LIG
\end{sphinxVerbatim}

Example of an alchemical transformation between two ligands using serial FEP simulations.
When the {[}REMD{]} section is not specified and more than one lambda values are specified,
GENESIS performs serial calculations by changing lambda values.
If \sphinxstylestrong{fep\_direction} is \sphinxstylestrong{Bothsides}, \sphinxstylestrong{lambljA}, \sphinxstylestrong{lambljB}, \sphinxstylestrong{lambelA}, \sphinxstylestrong{lambelB}, \sphinxstylestrong{lambbondA},
and \sphinxstylestrong{lambbondB} are first set to the leftmost values,
which are “1.00”, “0.00”, “1.00”, “0.00”, “1.00”, and “0.00”, respectively, in the below example.
After the \sphinxstylestrong{equilsteps + nsteps}-steps FEP simulation is performed with the set of the lambda values, the lambda values are changed to the second values from the left.
In this way, GENESIS performs FEP simulations, changing lambda values.
When the FEP simulation with the rightmost values of lambda finishes, GENESIS stops the calculation.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}ALCHEMY\PYG{o}{]}
\PYG{n+nv}{fep\PYGZus{}direction} \PYG{o}{=} BothSides
\PYG{n+nv}{fep\PYGZus{}topology}  \PYG{o}{=} Hybrid
\PYG{n+nv}{singleA}       \PYG{o}{=} \PYG{l+m}{1}  \PYG{c+c1}{\PYGZsh{} group1 in [SELECTION]}
\PYG{n+nv}{singleB}       \PYG{o}{=} \PYG{l+m}{2}  \PYG{c+c1}{\PYGZsh{} group2 in [SELECTION]}
\PYG{n+nv}{dualA}         \PYG{o}{=} \PYG{l+m}{3}  \PYG{c+c1}{\PYGZsh{} group3 in [SELECTION]}
\PYG{n+nv}{dualB}         \PYG{o}{=} \PYG{l+m}{4}  \PYG{c+c1}{\PYGZsh{} group4 in [SELECTION]}
\PYG{n+nv}{fepout\PYGZus{}period} \PYG{o}{=} \PYG{l+m}{500}
\PYG{n+nv}{equilsteps}    \PYG{o}{=} \PYG{l+m}{0}
\PYG{n+nv}{sc\PYGZus{}alpha}      \PYG{o}{=} \PYG{l+m}{5}.0
\PYG{n+nv}{sc\PYGZus{}beta}       \PYG{o}{=} \PYG{l+m}{5}.0
\PYG{n+nv}{lambljA}       \PYG{o}{=} \PYG{l+m}{1}.00 \PYG{l+m}{0}.75 \PYG{l+m}{0}.50 \PYG{l+m}{0}.25 \PYG{l+m}{0}.00
\PYG{n+nv}{lambljB}       \PYG{o}{=} \PYG{l+m}{0}.00 \PYG{l+m}{0}.25 \PYG{l+m}{0}.50 \PYG{l+m}{0}.75 \PYG{l+m}{1}.00
\PYG{n+nv}{lambelA}       \PYG{o}{=} \PYG{l+m}{1}.00 \PYG{l+m}{0}.75 \PYG{l+m}{0}.50 \PYG{l+m}{0}.25 \PYG{l+m}{0}.00
\PYG{n+nv}{lambelB}       \PYG{o}{=} \PYG{l+m}{0}.00 \PYG{l+m}{0}.25 \PYG{l+m}{0}.50 \PYG{l+m}{0}.75 \PYG{l+m}{1}.00
\PYG{n+nv}{lambbondA}     \PYG{o}{=} \PYG{l+m}{1}.00 \PYG{l+m}{0}.75 \PYG{l+m}{0}.50 \PYG{l+m}{0}.25 \PYG{l+m}{0}.00
\PYG{n+nv}{lambbondB}     \PYG{o}{=} \PYG{l+m}{0}.00 \PYG{l+m}{0}.25 \PYG{l+m}{0}.50 \PYG{l+m}{0}.75 \PYG{l+m}{1}.00

\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}        \PYG{o}{=} ai:1\PYGZhy{}11   \PYG{c+c1}{\PYGZsh{} atoms in single A}
\PYG{n+nv}{group2}        \PYG{o}{=} ai:13\PYGZhy{}23  \PYG{c+c1}{\PYGZsh{} atoms in single B}
\PYG{n+nv}{group3}        \PYG{o}{=} ai:12     \PYG{c+c1}{\PYGZsh{} atoms in dual A}
\PYG{n+nv}{group4}        \PYG{o}{=} ai:24\PYGZhy{}25  \PYG{c+c1}{\PYGZsh{} atoms in dual B}
\end{sphinxVerbatim}

Example of an alchemical transformation between two ligands at a set of lambda values.
If the user wants to perform a FEP simulation with a specified set of lambda values,
set \sphinxstylestrong{fep\_md\_type} to \sphinxstylestrong{Single} and assign the ID of the set of lambda values to \sphinxstylestrong{ref\_lambid}.
In the following example, \sphinxstylestrong{ref\_lambid} is set to 3, which means that the third column of the lambda values:
\sphinxstylestrong{lambljA = 0.5}, \sphinxstylestrong{lambljB = 0.5}, \sphinxstylestrong{lambelA = 0.5}, \sphinxstylestrong{lambelB = 0.5}, \sphinxstylestrong{lambbondA = 0.5}, and \sphinxstylestrong{lambbondB = 0.5}.
If \sphinxstylestrong{fep\_direction = Bothsides}, the energy differences between “\sphinxstylestrong{ref\_lambid}”-th and “\sphinxstylestrong{ref\_lambid} -1”-th columns and between “\sphinxstylestrong{ref\_lambid}”-th and “\sphinxstylestrong{ref\_lambid} +1”-th columns are outputted into the fepout file.
By using these function, the user can independently perform FEP simulations with different lambda values in parallel.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}ALCHEMY\PYG{o}{]}
\PYG{n+nv}{fep\PYGZus{}direction}   \PYG{o}{=} BothSides
\PYG{n+nv}{fep\PYGZus{}topology}    \PYG{o}{=} Hybrid
\PYG{n+nv}{fep\PYGZus{}md\PYGZus{}type}     \PYG{o}{=} Single
\PYG{n+nv}{ref\PYGZus{}lambid}      \PYG{o}{=} \PYG{l+m}{3}
\PYG{n+nv}{singleA}         \PYG{o}{=} \PYG{l+m}{1}  \PYG{c+c1}{\PYGZsh{} group1 in [SELECTION]}
\PYG{n+nv}{singleB}         \PYG{o}{=} \PYG{l+m}{2}  \PYG{c+c1}{\PYGZsh{} group2 in [SELECTION]}
\PYG{n+nv}{dualA}           \PYG{o}{=} \PYG{l+m}{3}  \PYG{c+c1}{\PYGZsh{} group3 in [SELECTION]}
\PYG{n+nv}{dualB}           \PYG{o}{=} \PYG{l+m}{4}  \PYG{c+c1}{\PYGZsh{} group4 in [SELECTION]}
\PYG{n+nv}{fepout\PYGZus{}period}   \PYG{o}{=} \PYG{l+m}{500}
\PYG{n+nv}{equilsteps}      \PYG{o}{=} \PYG{l+m}{0}
\PYG{n+nv}{sc\PYGZus{}alpha}        \PYG{o}{=} \PYG{l+m}{5}.0
\PYG{n+nv}{sc\PYGZus{}beta}         \PYG{o}{=} \PYG{l+m}{5}.0
\PYG{n+nv}{lambljA}         \PYG{o}{=} \PYG{l+m}{1}.00 \PYG{l+m}{0}.75 \PYG{l+m}{0}.50 \PYG{l+m}{0}.25 \PYG{l+m}{0}.00
\PYG{n+nv}{lambljB}         \PYG{o}{=} \PYG{l+m}{0}.00 \PYG{l+m}{0}.25 \PYG{l+m}{0}.50 \PYG{l+m}{0}.75 \PYG{l+m}{1}.00
\PYG{n+nv}{lambelA}         \PYG{o}{=} \PYG{l+m}{1}.00 \PYG{l+m}{0}.75 \PYG{l+m}{0}.50 \PYG{l+m}{0}.25 \PYG{l+m}{0}.00
\PYG{n+nv}{lambelB}         \PYG{o}{=} \PYG{l+m}{0}.00 \PYG{l+m}{0}.25 \PYG{l+m}{0}.50 \PYG{l+m}{0}.75 \PYG{l+m}{1}.00
\PYG{n+nv}{lambbondA}       \PYG{o}{=} \PYG{l+m}{1}.00 \PYG{l+m}{0}.75 \PYG{l+m}{0}.50 \PYG{l+m}{0}.25 \PYG{l+m}{0}.00
\PYG{n+nv}{lambbondB}       \PYG{o}{=} \PYG{l+m}{0}.00 \PYG{l+m}{0}.25 \PYG{l+m}{0}.50 \PYG{l+m}{0}.75 \PYG{l+m}{1}.00

\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}          \PYG{o}{=} ai:1\PYGZhy{}11   \PYG{c+c1}{\PYGZsh{} atoms in single A}
\PYG{n+nv}{group2}          \PYG{o}{=} ai:13\PYGZhy{}23  \PYG{c+c1}{\PYGZsh{} atoms in single B}
\PYG{n+nv}{group3}          \PYG{o}{=} ai:12     \PYG{c+c1}{\PYGZsh{} atoms in dual A}
\PYG{n+nv}{group4}          \PYG{o}{=} ai:24\PYGZhy{}25  \PYG{c+c1}{\PYGZsh{} atoms in dual B}
\end{sphinxVerbatim}

Example of an alchemical transformation between two ligands using a parallel FEP simulation.
When the {[}REMD{]} section is specified, GENESIS performs the FEP/\(\lambda\)-REMD simulation.
Each lambda value in \sphinxstylestrong{lambljA}, \sphinxstylestrong{lambljB}, \sphinxstylestrong{lambelA}, \sphinxstylestrong{lambelB}, \sphinxstylestrong{lambbondA}, and \sphinxstylestrong{lambbondB}
is assigned to each replica, and the FEP simulation in each replica is performed in parallel.
The lambda values are exchanged at fixed intervals specified by \sphinxstylestrong{exchange\_period} during the simulation.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{[}REMD\PYG{o}{]}
\PYG{n+nv}{dimension}       \PYG{o}{=} \PYG{l+m}{1}
\PYG{n+nv}{exchange\PYGZus{}period} \PYG{o}{=} \PYG{l+m}{1000}
\PYG{n+nv}{type1}           \PYG{o}{=} alchemy
\PYG{n+nv}{nreplica1}       \PYG{o}{=} \PYG{l+m}{5}

\PYG{o}{[}ALCHEMY\PYG{o}{]}
\PYG{n+nv}{fep\PYGZus{}direction}   \PYG{o}{=} BothSides
\PYG{n+nv}{fep\PYGZus{}topology}    \PYG{o}{=} Hybrid
\PYG{n+nv}{singleA}         \PYG{o}{=} \PYG{l+m}{1}  \PYG{c+c1}{\PYGZsh{} group1 in [SELECTION]}
\PYG{n+nv}{singleB}         \PYG{o}{=} \PYG{l+m}{2}  \PYG{c+c1}{\PYGZsh{} group2 in [SELECTION]}
\PYG{n+nv}{dualA}           \PYG{o}{=} \PYG{l+m}{3}  \PYG{c+c1}{\PYGZsh{} group3 in [SELECTION]}
\PYG{n+nv}{dualB}           \PYG{o}{=} \PYG{l+m}{4}  \PYG{c+c1}{\PYGZsh{} group4 in [SELECTION]}
\PYG{n+nv}{fepout\PYGZus{}period}   \PYG{o}{=} \PYG{l+m}{500}
\PYG{n+nv}{equilsteps}      \PYG{o}{=} \PYG{l+m}{0}
\PYG{n+nv}{sc\PYGZus{}alpha}        \PYG{o}{=} \PYG{l+m}{5}.0
\PYG{n+nv}{sc\PYGZus{}beta}         \PYG{o}{=} \PYG{l+m}{5}.0
\PYG{n+nv}{lambljA}         \PYG{o}{=} \PYG{l+m}{1}.00 \PYG{l+m}{0}.75 \PYG{l+m}{0}.50 \PYG{l+m}{0}.25 \PYG{l+m}{0}.00
\PYG{n+nv}{lambljB}         \PYG{o}{=} \PYG{l+m}{0}.00 \PYG{l+m}{0}.25 \PYG{l+m}{0}.50 \PYG{l+m}{0}.75 \PYG{l+m}{1}.00
\PYG{n+nv}{lambelA}         \PYG{o}{=} \PYG{l+m}{1}.00 \PYG{l+m}{0}.75 \PYG{l+m}{0}.50 \PYG{l+m}{0}.25 \PYG{l+m}{0}.00
\PYG{n+nv}{lambelB}         \PYG{o}{=} \PYG{l+m}{0}.00 \PYG{l+m}{0}.25 \PYG{l+m}{0}.50 \PYG{l+m}{0}.75 \PYG{l+m}{1}.00
\PYG{n+nv}{lambbondA}       \PYG{o}{=} \PYG{l+m}{1}.00 \PYG{l+m}{0}.75 \PYG{l+m}{0}.50 \PYG{l+m}{0}.25 \PYG{l+m}{0}.00
\PYG{n+nv}{lambbondB}       \PYG{o}{=} \PYG{l+m}{0}.00 \PYG{l+m}{0}.25 \PYG{l+m}{0}.50 \PYG{l+m}{0}.75 \PYG{l+m}{1}.00

\PYG{o}{[}SELECTION\PYG{o}{]}
\PYG{n+nv}{group1}          \PYG{o}{=} ai:1\PYGZhy{}11   \PYG{c+c1}{\PYGZsh{} atoms in single A}
\PYG{n+nv}{group2}          \PYG{o}{=} ai:13\PYGZhy{}23  \PYG{c+c1}{\PYGZsh{} atoms in single B}
\PYG{n+nv}{group3}          \PYG{o}{=} ai:12     \PYG{c+c1}{\PYGZsh{} atoms in dual A}
\PYG{n+nv}{group4}          \PYG{o}{=} ai:24\PYGZhy{}25  \PYG{c+c1}{\PYGZsh{} atoms in dual B}
\end{sphinxVerbatim}


\chapter{Trouble Shooting}
\label{\detokenize{97_Trouble_shooting:trouble-shooting}}\label{\detokenize{97_Trouble_shooting:trouble}}\label{\detokenize{97_Trouble_shooting::doc}}
The followings are representative error messages that the users can frequently encounter during the simulations.
We describe possible reasons for each error message, and provide suggestions to solve the problem.

\sphinxstylestrong{Compute\_Shake\textgreater{} SHAKE algorithm failed to converge}
\begin{quote}

This message indicates that constraint for the rigid bond using the SHAKE algorithm (see {\hyperref[\detokenize{08_Constraints:constraints}]{\sphinxcrossref{\DUrole{std,std-ref}{Constraints section}}}}) was failed due to some reasons.
In most cases, SHAKE errors are originated from insufficient equilibration, bad initial structure, or bad input parameters.
We recommend the users to check the following points:
\begin{itemize}
\item {} 
Reconsider the equilibration scheme. More moderate equilibration might be needed.
For example, heating the system from 0 K, using a shorter timestep (e.g., 1.0 fs),
or performing long energy minimization is a possible solution.

\item {} 
Check the initial structure very carefully.
One of the frequent mistakes in the initial structure modeling is “ring penetration” of covalent bonds.
One covalent bond might be somehow inserted into an aromatic ring.
Solve the ring penetration first, and then try the simulation again.

\item {} 
Some force field parameters are missing or wrong, which can easily cause unstable simulations.

\end{itemize}
\end{quote}

\sphinxstylestrong{Check\_Atom\_Coord\textgreater{} Some atoms have large clashes}
\begin{quote}

This message indicates that there is an atom pair whose distance is zero or close to zero.
Those atom indexes and distance are displayed in a warning message: “WARNING: too short distance:”.
This situation is not allowed, especially in \sphinxstylestrong{SPDYN}, since it can cause a numerical error in the lookup table method.
Check the initial structure first. Even if you cannot see such atomic clashes,
there may be a clash between the atoms in the unit cell and image cells in the case of the periodic boundary condition.
One of the automatic solutions is to specify “contact\_check = YES” in the control file (see {\hyperref[\detokenize{05_Energy:energy}]{\sphinxcrossref{\DUrole{std,std-ref}{Energy section}}}}).
However, this cannot work well, if the distance is exactly zero.
In such cases, the problem should be solved by the users themselves.
For example, the users may have to slightly move the clashing atoms manually, or specify larger or smaller box size, or
rebuild the initial structure more carefully.
\end{quote}

\sphinxstylestrong{Setup\_Processor\_Number\textgreater{} Cannot define domains and cells. Smaller MPI processors, or shorter pairlistdist, or larger boxsize should be used}
\begin{quote}

This message indicates that the total number of MPI processors used in your calculation is not appropriate for your system.
The users had better understand relations between the system size and number of MPI processors.
In \sphinxstylestrong{SPDYN}, the system is divided into several domains for parallel computation,
where the number of domains must be equal to the number of MPI processors (see {\hyperref[\detokenize{02_Available_Programs:available-programs}]{\sphinxcrossref{\DUrole{std,std-ref}{Available Programs}}}}).
In most cases, this message tells you that the system could not be divided into the specified number of domains.
Although there are mainly three solutions for this problem, first one is the most recommended way:
\begin{itemize}
\item {} 
Use smaller number of MPI processors. If it can work, the previous number was too large to handle the system.

\item {} 
Use shorter pairlistdist. This treatment can make a domain size smaller, allowing to use a larger number of MPI processors.
However, this is not recommended, if you are already using a recommended parameter set for
switchdist, cutoffdist, and pairlistdist (e.g., 10, 12, and 13.5 \(\text{\AA}\) in the CHARMM force field)

\item {} 
Build a larger initial structure by adding solvent molecules in the system,
which may allow the users to divide the system into the desired number of domains.

\end{itemize}
\end{quote}

\sphinxstylestrong{Update\_Boundary\_Pbc\textgreater{} too small boxsize/pairdist. larger boxsize or shorter pairdist should be used.}
\begin{quote}

This message indicates that your system is too small to handle in the periodic boundary condition.
In \sphinxstylestrong{ATDYN}, cell-linked list method is used to make non-bonded pairlists,
where the cell size is determined to be close to and larger than the pairlist distance given in the control file.
In addition, the total number of cells in x, y, and z dimensions must be at least three.
\sphinxstylestrong{SPDYN} has a similar lower limitation in the available box size.
Therefore, in order to solve this problem, the users may have to set a shorter pairlistdist,
or build a larger system by adding much solvent molecules.
\end{quote}

\sphinxstylestrong{Compute\_Energy\_Experimental\_Restraint\_Emfit\textgreater{} Gaussian kernel is extending outside the map box}
\begin{quote}

This message indicates that the simulated densities were generated outside the target density map.
If atoms to be fitted are located near the edge of the target density map, this error can frequently happen.
\begin{itemize}
\item {} 
Create a larger density map by adding an enough margin to the map,
which can be easily accomplished with the “voledit” tool in SITUS (\sphinxurl{https://situs.biomachina.org/}).

\item {} 
Examine a normal MD simulation by turning off the EM biasing potential (emfit = NO).
If the simulation is not stable, there is an issue in the molecular mechanics calculation
rather than the biasing potential calculation.
In such cases, please check the initial structure carefully.
There might be large clashes between some atoms, which can cause explosion of
the target molecule, and push some atoms out of the density map.
The problems to be solved are almost same with those in the SHAKE errors (see above).

\end{itemize}
\end{quote}

\sphinxstylestrong{Compute\_Energy\_Restraints\_Pos\textgreater{} Positional restraint energy is too big}
\begin{quote}

This message indicates that some atoms to be restrained are significantly deviated
from the reference position, indicating that the restraint might not be properly applied to such atoms.
This situation is not allowed in \sphinxstylestrong{SPDYN}.
\begin{itemize}
\item {} 
Use a larger force constant to keep their position near the reference.

\item {} 
Turn off the positional restraint for such atoms if it is not essential.

\end{itemize}
\end{quote}


\chapter{Appendix}
\label{\detokenize{98_Appendix:appendix}}\label{\detokenize{98_Appendix:id1}}\label{\detokenize{98_Appendix::doc}}

\section{Install the requirements in Linux}
\label{\detokenize{98_Appendix:install-the-requirements-in-linux}}
In the first sub-section, we explain how to install the requirements
using the package manager “apt” in Ubuntu/Debian.
If you want to install them from the source codes,
or if you want to use other Linux systems like CentOS,
please see the second sub-section.


\subsection{For Ubuntu/Debian users}
\label{\detokenize{98_Appendix:for-ubuntu-debian-users}}

\subsubsection{GNU compilers and build tools}
\label{\detokenize{98_Appendix:gnu-compilers-and-build-tools}}
First, we install compilers and build tools.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} sudo apt update
\PYGZdl{} sudo apt install build\PYGZhy{}essential
\PYGZdl{} sudo apt install gfortran autoconf automake
\end{sphinxVerbatim}


\subsubsection{OpenMPI}
\label{\detokenize{98_Appendix:openmpi}}
Then, we install OpenMPI. Note that the development version (XXX-dev) should be installed.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} sudo apt install openmpi\PYGZhy{}bin libopenmpi\PYGZhy{}dev

\PYGZdl{} which mpirun mpif90 mpicc
/usr/bin/mpirun
/usr/bin/mpif90
/usr/bin/mpicc
\end{sphinxVerbatim}


\subsubsection{LAPACK/BLAS libraries}
\label{\detokenize{98_Appendix:lapack-blas-libraries}}
Finally, we install LAPACK/BLAS libraries. Again, development version (XXX-dev) is installed.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} sudo apt install liblapack\PYGZhy{}dev

\PYGZdl{} ls /usr/lib/x86\PYGZus{}64\PYGZhy{}linux\PYGZhy{}gnu/liblapack.*
\PYGZdl{} ls /usr/lib/x86\PYGZus{}64\PYGZhy{}linux\PYGZhy{}gnu/libblas.*
\end{sphinxVerbatim}

\clearpage


\subsection{For CentOS/RedHat users}
\label{\detokenize{98_Appendix:for-centos-redhat-users}}
Here, we explain how to install OpenMPI and LAPACK/BLAS libraries from the source codes.
We assume that the users already installed GNU compilers.
The following schemes are commonly applicable to typical Linux systems including CentOS and Red Hat.


\subsubsection{OpenMPI}
\label{\detokenize{98_Appendix:id2}}
The source code of OpenMPI is availabe in \sphinxurl{https://www.open-mpi.org/}.
The following commands install OpenMPI 3.1.5 in the user’s local directory
“\$HOME/Software/mpi” as an example.
Here, we use GNU compilers (gcc, g++, and gfortran).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{cd} \PYG{n+nv}{\PYGZdl{}HOME}
\PYGZdl{} mkdir Software
\PYGZdl{} \PYG{n+nb}{cd} Software

\PYGZdl{} mkdir build
\PYGZdl{} \PYG{n+nb}{cd} build

\PYGZdl{} wget https://download.open\PYGZhy{}mpi.org/release/open\PYGZhy{}mpi/v3.1/openmpi\PYGZhy{}3.1.5.tar.gz

\PYGZdl{} tar \PYGZhy{}xvf openmpi\PYGZhy{}3.1.5.tar.gz
\PYGZdl{} \PYG{n+nb}{cd} openmpi\PYGZhy{}3.1.5

\PYGZdl{} ./configure \PYGZhy{}\PYGZhy{}prefix\PYG{o}{=}\PYG{n+nv}{\PYGZdl{}HOME}/Software/mpi \PYG{n+nv}{CC}\PYG{o}{=}gcc \PYG{n+nv}{CXX}\PYG{o}{=}g++ \PYG{n+nv}{F77}\PYG{o}{=}gfortran \PYG{n+nv}{FC}\PYG{o}{=}gfortran

\PYGZdl{} make all
\PYGZdl{} make install
\end{sphinxVerbatim}

The following information is added in “\textasciitilde{}/.bash\_profile” (or “\textasciitilde{}/.bashrc”).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n+nv}{MPIROOT}\PYG{o}{=}\PYG{n+nv}{\PYGZdl{}HOME}/Software/mpi
\PYG{n+nb}{export} \PYG{n+nv}{PATH}\PYG{o}{=}\PYG{n+nv}{\PYGZdl{}MPIROOT}/bin:\PYG{n+nv}{\PYGZdl{}PATH}
\PYG{n+nb}{export} \PYG{n+nv}{LD\PYGZus{}LIBRARY\PYGZus{}PATH}\PYG{o}{=}\PYG{n+nv}{\PYGZdl{}MPIROOT}/lib:\PYG{n+nv}{\PYGZdl{}LD\PYGZus{}LIBRARY\PYGZus{}PATH}
\PYG{n+nb}{export} \PYG{n+nv}{MANPATH}\PYG{o}{=}\PYG{n+nv}{\PYGZdl{}MPIROOT}/share/man:\PYG{n+nv}{\PYGZdl{}MANPATH}
\end{sphinxVerbatim}

Launch another terminal window or reload “\textasciitilde{}/.bash\_profile” (or “\textasciitilde{}/.bashrc”):

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{source} \PYGZti{}/.bash\PYGZus{}profile
\end{sphinxVerbatim}

The OpenMPI tools should be installed in “\$HOME/Software/mpi/bin”.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} which mpirun mpif90 mpicc
\PYGZti{}/Software/mpi/bin/mpirun
\PYGZti{}/Software/mpi/bin/mpif90
\PYGZti{}/Software/mpi/bin/mpicc
\end{sphinxVerbatim}

If you want to uninstall OpenMPI, just remove the directory “mpi” in “Software”.


\subsubsection{LAPACK/BLAS libraries}
\label{\detokenize{98_Appendix:id3}}
The source code of LAPACK/BLAS is availabe in \sphinxurl{http://www.netlib.org/lapack/}.
The following commands install LAPACK 3.8.0
in the user’s local directory “\$HOME/Software/lapack-3.8.0” as an example.
The BLAS library is also installed. We use GNU compilers (gcc and gfortran).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{cd} \PYG{n+nv}{\PYGZdl{}HOME}/Software
\PYGZdl{} wget http://www.netlib.org/lapack/lapack\PYGZhy{}3.8.0.tar.gz
\PYGZdl{} tar \PYGZhy{}xvf lapack\PYGZhy{}3.8.0.tar.gz
\PYGZdl{} \PYG{n+nb}{cd} lapack\PYGZhy{}3.8.0

\PYGZdl{} cp make.inc.example make.inc
\PYGZdl{} make blaslib
\PYGZdl{} make lapacklib

\PYGZdl{} ls lib*
liblapack.a  librefblas.a

\PYGZdl{} ln \PYGZhy{}s librefblas.a ./libblas.a
\end{sphinxVerbatim}

The following information is added in “\textasciitilde{}/.bash\_profile” (or “\textasciitilde{}/.bashrc”).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n+nb}{export} \PYG{n+nv}{LAPACK\PYGZus{}PATH}\PYG{o}{=}\PYG{n+nv}{\PYGZdl{}HOME}/Software/lapack\PYGZhy{}3.8.0
\end{sphinxVerbatim}

Launch another terminal window or reload “\textasciitilde{}/.bash\_profile” (or “\textasciitilde{}/.bashrc”):

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{source} \PYGZti{}/.bash\PYGZus{}profile
\end{sphinxVerbatim}

If you want to uninstall LAPACK/BLAS, just remove the directory “lapack-3.8.0” in “Software”.

\clearpage


\section{Install the requirements in Mac}
\label{\detokenize{98_Appendix:install-the-requirements-in-mac}}
We recommend the Mac users to utilize “Xcode” for the installation of GENESIS,
and also to install “OpenMPI” from the source code to avoid a “clang” problem (see below).


\subsection{Install general tools}
\label{\detokenize{98_Appendix:install-general-tools}}

\subsubsection{Xcode and Homebrew}
\label{\detokenize{98_Appendix:xcode-and-homebrew}}
“Xcode” is available in the Mac App Store (\sphinxurl{https://developer.apple.com/xcode/}),
and it is free of charge.
After the installation of Xcode, all tasks described below will be done on “Terminal”.
The “Terminal app” is in the “Utilities” folder in Applications.
Please launch the Terminal. This terminal is almost same with that in Linux.

We recommend you to further install “Homebrew”, which enables easy installation
of various tools such as compilers. If you have already installed “MacPorts”,
you do not need to install “Homebrew” to avoid a conflict between “Homebrew” and “MacPorts”.
In the Homebrew website (\sphinxurl{https://brew.sh/}), you can find a long command
like “\sphinxcode{\sphinxupquote{/usr/bin/ruby -e "\$(curl -fsSL https://...}}”.
To install homebrew, execute that command in the Terminal prompt.


\subsubsection{GNU compilers and build tools}
\label{\detokenize{98_Appendix:id4}}
First, we install “gcc”, “autoconf”, “automake”, and other tools via homebrew:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} brew install gcc
\PYGZdl{} brew install autoconf
\PYGZdl{} brew install automake
\PYGZdl{} brew install wget
\end{sphinxVerbatim}

To confirm the installation of “gcc”, let us type the following commands:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} which gcc
/usr/bin/gcc

\PYGZdl{} gcc \PYGZhy{}\PYGZhy{}version
...
Apple LLVM version \PYG{l+m}{10}.0.1 \PYG{o}{(}clang\PYGZhy{}1001.0.46.4\PYG{o}{)}
\end{sphinxVerbatim}

These messages tell us that “gcc” is installed in the “/usr/bin” directory.
However, this gcc is not a “real” GNU compiler, and it is linked to another compiler “clang”.
If you use this gcc for the installation of OpenMPI,
it can cause a trouble in compiling \sphinxstylestrong{GENESIS} with a certain option.
Therefore, you have to use a “real” GNU compiler, which is actually installed in “/usr/local/bin”.
For example, if you have installed gcc ver. 9, you can find it as “gcc-9” in “/usr/local/bin”.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ls /usr/local/bin/gcc*
/usr/local/bin/gcc\PYGZhy{}9      /usr/local/bin/gcc\PYGZhy{}ar\PYGZhy{}9      ...

\PYGZdl{} gcc\PYGZhy{}9 \PYGZhy{}\PYGZhy{}version
gcc\PYGZhy{}9 \PYG{o}{(}Homebrew GCC \PYG{l+m}{9}.2.0\PYG{o}{)} \PYG{l+m}{9}.2.0
\end{sphinxVerbatim}


\subsection{Install libraries}
\label{\detokenize{98_Appendix:install-libraries}}

\subsubsection{OpenMPI}
\label{\detokenize{98_Appendix:id5}}
We then install “OpenMPI”. We specify “real” GNU compilers explicitly in the configure command.
The following commands install OpenMPI in the user’s local directory “\$HOME/Software/mpi”.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{cd} \PYG{n+nv}{\PYGZdl{}HOME}
\PYGZdl{} mkdir Software
\PYGZdl{} \PYG{n+nb}{cd} Software
\PYGZdl{} mkdir build
\PYGZdl{} \PYG{n+nb}{cd} build

\PYGZdl{} wget https://download.open\PYGZhy{}mpi.org/release/open\PYGZhy{}mpi/v3.1/openmpi\PYGZhy{}3.1.5.tar.gz

\PYGZdl{} tar \PYGZhy{}xvf openmpi\PYGZhy{}3.1.5.tar.gz
\PYGZdl{} \PYG{n+nb}{cd} openmpi\PYGZhy{}3.1.5

\PYGZdl{} ./configure \PYGZhy{}\PYGZhy{}prefix\PYG{o}{=}\PYG{n+nv}{\PYGZdl{}HOME}/Software/mpi \PYG{n+nv}{CC}\PYG{o}{=}gcc\PYGZhy{}9 \PYG{n+nv}{CXX}\PYG{o}{=}g++\PYGZhy{}9 \PYG{n+nv}{F77}\PYG{o}{=}gfortran\PYGZhy{}9 \PYG{n+nv}{FC}\PYG{o}{=}gfortran\PYGZhy{}9

\PYGZdl{} make all
\PYGZdl{} make install
\end{sphinxVerbatim}

The following information is added in “\textasciitilde{}/.bash\_profile” (or “\textasciitilde{}/.bashrc”).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n+nv}{MPIROOT}\PYG{o}{=}\PYG{n+nv}{\PYGZdl{}HOME}/Software/mpi
\PYG{n+nb}{export} \PYG{n+nv}{PATH}\PYG{o}{=}\PYG{n+nv}{\PYGZdl{}MPIROOT}/bin:\PYG{n+nv}{\PYGZdl{}PATH}
\PYG{n+nb}{export} \PYG{n+nv}{LD\PYGZus{}LIBRARY\PYGZus{}PATH}\PYG{o}{=}\PYG{n+nv}{\PYGZdl{}MPIROOT}/lib:\PYG{n+nv}{\PYGZdl{}LD\PYGZus{}LIBRARY\PYGZus{}PATH}
\PYG{n+nb}{export} \PYG{n+nv}{MANPATH}\PYG{o}{=}\PYG{n+nv}{\PYGZdl{}MPIROOT}/share/man:\PYG{n+nv}{\PYGZdl{}MANPATH}
\end{sphinxVerbatim}

Launch another terminal window or reload “\textasciitilde{}/.bash\_profile” (or “\textasciitilde{}/.bashrc”):

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{source} \PYGZti{}/.bash\PYGZus{}profile
\end{sphinxVerbatim}

The OpenMPI tools should be installed in “\$HOME/Software/mpi/bin”.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} which mpirun mpif90 mpicc
/Users/\PYG{o}{[}username\PYG{o}{]}/Software/mpi/bin/mpirun
/Users/\PYG{o}{[}username\PYG{o}{]}/Software/mpi/bin/mpif90
/Users/\PYG{o}{[}username\PYG{o}{]}/Software/mpi/bin/mpicc
\end{sphinxVerbatim}

Make sure that “mpicc” and “mpif90” are linked to “gcc-9” and “gfortran-9”, respectively.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} mpicc \PYGZhy{}\PYGZhy{}version
gcc\PYGZhy{}9 \PYG{o}{(}Homebrew GCC \PYG{l+m}{9}.2.0\PYG{o}{)} \PYG{l+m}{9}.2.0

\PYGZdl{} mpif90 \PYGZhy{}\PYGZhy{}version
GNU Fortran \PYG{o}{(}Homebrew GCC \PYG{l+m}{9}.2.0\PYG{o}{)} \PYG{l+m}{9}.2.0
\end{sphinxVerbatim}

If you want to uninstall OpenMPI, just remove the directory “mpi” in “Software”.


\subsubsection{LAPACK/BLAS libraries}
\label{\detokenize{98_Appendix:id6}}
Finally, we install LAPACK and BLAS libraries.
Again, “real” GNU compilers are used for the install.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{cd} \PYG{n+nv}{\PYGZdl{}HOME}/Software
\PYGZdl{} wget http://www.netlib.org/lapack/lapack\PYGZhy{}3.8.0.tar.gz
\PYGZdl{} tar \PYGZhy{}xvf lapack\PYGZhy{}3.8.0.tar.gz
\PYGZdl{} \PYG{n+nb}{cd} lapack\PYGZhy{}3.8.0

\PYGZdl{} cp make.inc.example make.inc
\end{sphinxVerbatim}

In the “make.inc” file, there are three lines to be modified:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{}  CC is the C compiler, normally invoked with options CFLAGS.}
\PYG{c+c1}{\PYGZsh{}}
\PYG{n+nv}{CC}     \PYG{o}{=} gcc\PYGZhy{}9
...

\PYG{c+c1}{\PYGZsh{}  should not compile LAPACK with flags such as \PYGZhy{}ffpe\PYGZhy{}trap=overflow.}
\PYG{c+c1}{\PYGZsh{}}
\PYG{n+nv}{FORTRAN} \PYG{o}{=} gfortran\PYGZhy{}9
...

\PYG{c+c1}{\PYGZsh{}  load options for your machine.}
\PYG{c+c1}{\PYGZsh{}}
\PYG{n+nv}{LOADER}   \PYG{o}{=} gfortran\PYGZhy{}9
...
\end{sphinxVerbatim}

After the modification, we install BLAS and LAPACK libraries:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} make blaslib
\PYGZdl{} make lapacklib

\PYGZdl{} ls lib*
liblapack.a  librefblas.a

\PYGZdl{} ln \PYGZhy{}s librefblas.a ./libblas.a
\end{sphinxVerbatim}

The following information is added in “\textasciitilde{}/.bash\_profile” (or “\textasciitilde{}/.bashrc”).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n+nb}{export} \PYG{n+nv}{LAPACK\PYGZus{}PATH}\PYG{o}{=}\PYG{n+nv}{\PYGZdl{}HOME}/Software/lapack\PYGZhy{}3.8.0
\end{sphinxVerbatim}

Launch another terminal window or reload “\textasciitilde{}/.bash\_profile” (or “\textasciitilde{}/.bashrc”):

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} \PYG{n+nb}{source} \PYGZti{}/.bash\PYGZus{}profile
\end{sphinxVerbatim}

If you want to uninstall LAPACK/BLAS, just remove the directory “lapack-3.8.0” in “Software”.
\phantomsection\label{\detokenize{99_References:zreferences}}


\begin{sphinxthebibliography}{100}
\bibitem[1]{99_References:amber18-program}
D.A. Case, I.Y. Ben-Shalom, S.R. Brozell, D.S. Cerutti, T.E. Cheatham III, V.W.D. Cruzeiro, T.A. Darden, R.E. Duke, D. Ghoreishi, M.K. Gilson, H. Gohlke, A.W. Goetz, D. Greene, R Harris, N. Homeyer, S. Izadi, A. Kovalenko, T. Kurtzman, T.S. Lee, S. LeGrand, P. Li, C. Lin, J. Liu, T. Luchko, R. Luo, D.J. Mermelstein, K.M. Merz, Y. Miao, G. Monard, C. Nguyen, H. Nguyen, I. Omelyan, A. Onufriev, F. Pan, R. Qi, D.R. Roe, A. Roitberg, C. Sagui, S. Schott-Verdugo, J. Shen, C.L. Simmerling, J. Smith, R. Salomon-Ferrer, J. Swails, R.C. Walker, J. Wang, H. Wei, R.M. Wolf, X. Wu, L. Xiao, D.M. York, and P.A. Kollman. Amber18. University of California, San Francisco, 2018.
\bibitem[2]{99_References:charmm-program}
B. R. Brooks, C. L. Brooks, A. D. Mackerell, L. Nilsson, R. J. Petrella, B. Roux, Y. Won, G. Archontis, C. Bartels, S. Boresch, A. Caflisch, L. Caves, Q. Cui, A. R. Dinner, M. Feig, S. Fischer, J. Gao, M. Hodoscek, W. Im, K. Kuczera, T. Lazaridis, J. Ma, V. Ovchinnikov, E. Paci, R. W. Pastor, C. B. Post, J. Z. Pu, M. Schaefer, B. Tidor, R. M. Venable, H. L. Woodcock, X. Wu, W. Yang, D. M. York, and M. Karplus. CHARMM: The biomolecular simulation program. \sphinxstyleemphasis{J. Comput. Chem.}, 30:1545\textendash{}1614, 2009. URL: \sphinxurl{http://dx.doi.org/10.1002/jcc.21287}, \sphinxhref{https://doi.org/10.1002/jcc.21287}{doi:10.1002/jcc.21287} (https://doi.org/10.1002/jcc.21287).
\bibitem[3]{99_References:gromacs-program}
S. Pronk, S. Páll, R. Schulz, P. Larsson, P. Bjelkmar, R. Apostolov, M. R. Shirts, J. C. Smith, P. M. Kasson, D. van der Spoel, B. Hess, and E. Lindahl. GROMACS 4.5: a high-throughput and highly parallel open source molecular simulation toolkit. \sphinxstyleemphasis{Bioinformatics}, 29:845\textendash{}854, 2013.
\bibitem[4]{99_References:desmond-program}
K. J. Bowers, E. Chow, H. Xu, R. O. Dror, M. P. Eastwood, B. A. Gregersen, J. L. Klepeis, I. Kolossvary, M. A. Moraes, F. D. Sacerdoti, J. K. Salmon, Y. Shan, and D. E. Shaw. Scalable algorithms for molecular dynamics simulations on commodity clusters. In \sphinxstyleemphasis{SC 2006 Conference, Proceedings of the ACM/IEEE}, 11\textendash{}17. IEEE, 2006.
\bibitem[5]{99_References:namd-program}
J. C. Phillips, R. Braun, W. Wang, J. Gumbart, E. Tajkhorshid, E. Villa, C. Chipot, R. D. Skeel, L. Kalé, and K. Schulten. Scalable molecular dynamics with NAMD. \sphinxstyleemphasis{J. Comput. Chem.}, 26:1781\textendash{}1802, 2005. URL: \sphinxurl{http://dx.doi.org/10.1002/jcc.20289}, \sphinxhref{https://doi.org/10.1002/jcc.20289}{doi:10.1002/jcc.20289} (https://doi.org/10.1002/jcc.20289).
\bibitem[6]{99_References:case-2005}
D. A. Case, T. E. Cheatham, T. Darden, H. Gohlke, R. Luo, K. M. Merz, A. Onufriev, C. Simmerling, B. Wang, and R. J. Woods. The Amber biomolecular simulation programs. \sphinxstyleemphasis{J. Comput. Chem.}, 26:1668\textendash{}1688, 2005.
\bibitem[7]{99_References:mackerell-1998}
A. D. MacKerell, D. Bashford, M. Bellott, R. L. Dunbrack, J. D. Evanseck, M. J. Field, S. Fischer, J. Gao, H. Guo, S. Ha, D. Joseph-McCarthy, L. Kuchnir, K. Kuczera, F. T. K. Lau, C. Mattos, S. Michnick, T. Ngo, D. T. Nguyen, B. Prodhom, W. E. Reiher, B. Roux, M. Schlenkrich, J. C. Smith, R. Stote, J. Straub, M. Watanabe, J. Wiorkiewicz-Kuczera, D. Yin, and M. Karplus. All-atom empirical potential for molecular modeling and dynamics studies of proteins. \sphinxstyleemphasis{J. Phys. Chem. B}, 102:3586\textendash{}3616, 1998.
\bibitem[8]{99_References:mackerell-2004}
A. D. MacKerell, M. Feig, and C. L. Brooks. Improved treatment of the protein backbone in empirical force fields. \sphinxstyleemphasis{J. Am. Chem. Soc.}, 126:698\textendash{}699, 2004.
\bibitem[9]{99_References:jorgensen-1996}
W. L. Jorgensen, D. S. Maxwell, and J. Tirado-Rives. Development and Testing of the OPLS All-Atom Force Field on Conformational Energetics and Properties of Organic Liquids. \sphinxstyleemphasis{J. Am. Chem. Soc.}, 118:11225\textendash{}11236, 1996.
\bibitem[10]{99_References:oostenbrink-2004}
C. Oostenbrink, A. Villa, A. E. Mark, and W. F. Van Gunsteren. A biomolecular force field based on the free enthalpy of hydration and solvation: The GROMOS force-field parameter sets 53A5 and 53A6. \sphinxstyleemphasis{J. Comput. Chem.}, 25:1656\textendash{}1676, 2004.
\bibitem[11]{99_References:jung-2013}
J. Jung, T. Mori, and Y. Sugita. Efficient lookup table using a linear function of inverse distance squared. \sphinxstyleemphasis{J. Comput. Chem.}, 34:2412\textendash{}2420, 2013.
\bibitem[12]{99_References:jung-2014}
J. Jung, T. Mori, and Y. Sugita. Midpoint cell method for hybrid (MPI+OpenMP) parallelization of molecular dynamics simulations. \sphinxstyleemphasis{J. Comput. Chem.}, 35:1064\textendash{}1072, 2014.
\bibitem[13]{99_References:huang-2017}
J. Huang, S. Rauscher, G. Nawrocki, T. Ran, M. Feig, B. L. de Groot, H. Grubmüller, and A. D. MacKerell Jr. CHARMM36m: an improved force field for folded and intrinsically disordered proteins. \sphinxstyleemphasis{Nat. Methods}, 14:71\textendash{}73, 2017.
\bibitem[14]{99_References:humphrey-1996}
W. Humphrey, A. Dalke, and K. Schulten. VMD: Visual molecular dynamics. \sphinxstyleemphasis{J. Mol. Graph.}, 14:33\textendash{}38, 1996.
\bibitem[15]{99_References:jo-2008}
S. Jo, T. Kim, V. G. Iyer, and W. Im. CHARMM GUI: A web based graphical user interface for CHARMM. \sphinxstyleemphasis{J. Comput. Chem.}, 29:1859\textendash{}1865, 2008.
\bibitem[16]{99_References:cornell-1995}
W. D. Cornell, P. Cieplak, C. I. Bayly, I. R. Gould, K. M. Merz, D. M. Ferguson, D. C. Spellmeyer, T. Fox, J. W. Caldwell, and P. A. Kollman. A Second Generation Force Field for the Simulation of Proteins, Nucleic Acids, and Organic Molecules. \sphinxstyleemphasis{J. Am. Chem. Soc.}, 117:5179\textendash{}5197, 1995.
\bibitem[17]{99_References:taketomi-1975}
H. Taketomi, Y. Ueda, and N. Go. Studies on protein folding, unfolding and fluctuations by computer simulation. \sphinxstyleemphasis{nt. J. Peptide Proteins Res.}, 7:445\textendash{}459, 1975.
\bibitem[18]{99_References:marrink-2007}
S.J. Marrink, H.J. Risselada, S. Yefimov, D.P. Tieleman, and A.H. de Vries. The MARTINI forcefield: coarse grained model for biomolecular simulations. \sphinxstyleemphasis{J. Phys. Chem. B}, 111:7812\textendash{}7824, 2007.
\bibitem[19]{99_References:whitford-2009}
P. C. Whitford, J. K. Noel, S. Gosavi, A. Schug, K. Y. Sanbonmatsu, and J. N. Onuchic. An all-atom structure-based potential for proteins: Bridging minimal models with all-atom empirical forcefields. \sphinxstyleemphasis{Proteins: Structure, Function, and Bioinformatics}, 75:430\textendash{}441, 2009.
\bibitem[20]{99_References:noel-2010}
J. K. Noel, P. C. Whitford, K. Y. Sanbonmatsu, and J. N. Onuchic. SMOG@ctbp: simplified deployment of structure based models in GROMACS. \sphinxstyleemphasis{Nucleic Acids Res.}, 38:W657\textendash{}W661, 2010.
\bibitem[21]{99_References:noel-2016}
J. K. Noel, M. Levi, M. Raghunathan, H. Lammert, R. L. Hayes, J. N. Onuchic, and P. C. Whitford. SMOG 2: A Versatile Software Package for Generating Structure Based Models. \sphinxstyleemphasis{PLoS Comput. Biol.}, 12:e1004794, 2016.
\bibitem[22]{99_References:karanicolas-2002hl}
J. Karanicolas and C. L. Brooks, III. The origins of asymmetry in the folding transition states of protein L and protein G. \sphinxstyleemphasis{Protein Sci.}, 11:2351\textendash{}2361, 2002.
\bibitem[23]{99_References:karanicolas2003improved}
J. Karanicolas and C. L. Brooks III. Improved Go-like models demonstrate the robustness of protein folding mechanisms towards non-native interactions. \sphinxstyleemphasis{J. Mol. Biol.}, 334:309\textendash{}325, 2003.
\bibitem[24]{99_References:feig-2004}
M. Feig, J. Karanicolas, and C. L. III Brooks. MMTSB Tool Set: enhanced sampling and multiscale modeling methods for applications in structural biology. \sphinxstyleemphasis{J. Mol. Graph, Model.}, 22:377\textendash{}395, 2004.
\bibitem[25]{99_References:charmm-online}
CHARMM. http://www.charmm.org/.
\bibitem[26]{99_References:amber-online}
AMBER. http://ambermd.org/.
\bibitem[27]{99_References:gromacs-online}
Gromacs. http://www.gromacs.org/.
\bibitem[28]{99_References:klauda-2010}
J. B. Klauda, R. M. Venable, J. A. Freites, J. W. O’Connor, D. J. Tobias, C. Mondragon-Ramirez, I. Vorobyov, and R. W. Pastor. Update of the charmm all-atom additive force field for lipids: validation on six lipid types. \sphinxstyleemphasis{J. Phys. Chem. B}, 114:7830\textendash{}7843, 2010.
\bibitem[29]{99_References:best-2012}
R. B. Best, X. Zhu, J. Shim, P. E. M. Lopes, J. Mittal, M. Feig, and A. D. MacKerell. Optimization of the additive CHARMM all-atom protein force field targeting improved sampling of the backbone \textbackslash{}phi , \textbackslash{}psi and side-Chain \textbackslash{}chi 1 and \textbackslash{}chi 2 dihedral angles. \sphinxstyleemphasis{J. Chem. Theo. Comput.}, 8:3257\textendash{}3273, 2012.
\bibitem[30]{99_References:huang-2013}
J. Huang and A. D. MacKerell. CHARMM36 all-atom additive protein force field: Validation based on comparison to NMR data. \sphinxstyleemphasis{J. Comput. Chem.}, 34:2135\textendash{}2145, 2013.
\bibitem[31]{99_References:monticelli-2008}
L. Monticelli, S.K. Kandasamy, X. Periole, R.G. Larson, D.P. Tieleman, and S.J. Marrink. The MARTINI coarse grained forcefield: extension to proteins. \sphinxstyleemphasis{J. Chem. Theo. Comput.}, 4:819\textendash{}834, 2008.
\bibitem[32]{99_References:clementi-2000}
C. Clementi, H. Nymeyer, and J. Onuchic. Topological and energetic factors: what determines the structural details of the transition state ensemble and “en-route” intermediates for protein folding? an investigation for small globular proteins. \sphinxstyleemphasis{J. Mol. Biol.}, 298:937\textendash{}953, 2000.
\bibitem[33]{99_References:verlet-1967}
L. Verlet. Computer Experiments on Classical Fluids .I. Thermodynamical Properties of Lennard-Jones Molecules. \sphinxstyleemphasis{Phys. Rev.}, 159:98\textendash{}103, 1967.
\bibitem[34]{99_References:steinbach-1994}
P. J. Steinbach and B. R. Brooks. New Spherical-Cutoff Methods for Long-Range Forces in Macromolecular Simulation. \sphinxstyleemphasis{J. Comput. Chem.}, 15:667\textendash{}683, 1994.
\bibitem[35]{99_References:onufriev-2004}
A. Onufriev, D. Bashford, and D. A. Case. Exploring protein native states and large scale conformational changes with a modified generalized born model. \sphinxstyleemphasis{Proteins}, 55:383\textendash{}394, 2004.
\bibitem[36]{99_References:weiser-1999}
J. Weiser, P. S. Shenkin, and W. C. Still. Approximate atomic surfaces from linear combinations of pairwise overlaps (LCPO). \sphinxstyleemphasis{J. Comput. Chem.}, 20:217\textendash{}230, 1999.
\bibitem[37]{99_References:lazaridis-1999}
T. Lazaridis and M. Karplus. Effective energy function for proteins in solution. \sphinxstyleemphasis{Proteins}, 35:133\textendash{}152, 1999.
\bibitem[38]{99_References:lazaridis-2003}
T. Lazaridis. Effective energy function for proteins in lipid membranes. \sphinxstyleemphasis{Proteins}, 52:176\textendash{}192, 2003.
\bibitem[39]{99_References:mori-2020}
T. Mori and Y. Sugita. Implicit Micelle Model for Membrane Proteins Using Superellipsoid Approximation. \sphinxstyleemphasis{J. Chem. Theory Comput.}, 16:711\textendash{}724, 2020.
\bibitem[40]{99_References:darden-1993in}
T. Darden, D. York, and L. Pedersen. Particle mesh Ewald: An Nlog(N) method for Ewald sums in large systems. \sphinxstyleemphasis{J. Chem. Phys.}, 98:10089\textendash{}10092, 1993.
\bibitem[41]{99_References:essmann-1995vj}
U. Essmann, L. Perera, M. L. Berkowitz, T. Darden, H. Lee, and L. G. Pedersen. A smooth particle mesh Ewald method. \sphinxstyleemphasis{J. Chem. Phys.}, 103:8577\textendash{}8593, 1995.
\bibitem[42]{99_References:jung-2016}
J. Jung, C. Kobayashi, T. Imamura, and Y. Sugita. Parallel implementation of 3D FFT with volumetric decomposition schemes for efficient molecular dynamics simulations. \sphinxstyleemphasis{Comp. Phys. Comm.}, 200:57\textendash{}65, 2016.
\bibitem[43]{99_References:ffte-online}
D. Takahashi. FFTE: A Fast Fourier Transform Package. http://www.ffte.jp/.
\bibitem[44]{99_References:nilsson-2009}
L. Nilsson. Efficient Table Lookup Without Inverse Square Roots for Calculation of Pair Wise Atomic Interactions in Classical Simulations. \sphinxstyleemphasis{J. Comput. Chem.}, 30:1490\textendash{}1498, 2009.
\bibitem[45]{99_References:mori-2016}
T. Mori, N. Miyashita, W. Im, M. Feig, and Y. Sugita. Molecular dynamics simulations of biological membranes and membrane proteins using enhanced conformational sampling algorithms. \sphinxstyleemphasis{BBA-Biomembranes}, 1858:1635\textendash{}1651, 2016.
\bibitem[46]{99_References:still-1990}
W. C. Still, A. Tempczyk, R. C. Hawley, and T. Hendrickson. Semianalytical treatment of solvation for molecular mechanics and dynamics. \sphinxstyleemphasis{J. Am. Chem. Soc.}, 112:6127\textendash{}6129, 1990.
\bibitem[47]{99_References:eisenberg-1986}
D. Eisenberg and A. D. McLachlan. Solvation energy in protein folding and binding. \sphinxstyleemphasis{Nature}, 319:199\textendash{}203, 1986.
\bibitem[48]{99_References:schaefer-1990}
M. Schaefer and C. Froemmel. A Precise Analytical Method for Calculating the Electrostatic Energy of Macromolecules in Aqueous Solution. \sphinxstyleemphasis{J. Mol. Biol.}, 216:1045\textendash{}1066, 1990.
\bibitem[49]{99_References:lazaridis-2005}
T. Lazaridis. Structural Determinants of Transmembrane beta-Barrels. \sphinxstyleemphasis{J. Chem. Theory Comput.}, 1:716\textendash{}722, 2005.
\bibitem[50]{99_References:tan-ploscb2022}
C. Tan, J. Jung, C. Kobayashi, D. Ugarte La Torre, S. Takada, and Y. Sugita. Implementation of residue-level coarse-grained models in GENESIS for large-scale molecular dynamics simulations. \sphinxstyleemphasis{PLoS Comput. Biol.}, 18(4):e1009578, 2022.
\bibitem[51]{99_References:li-pnas2014}
W. Li, W. Wang, and S. Takada. Energy landscape views for interplays among folding, binding, and allostery of calmodulin domains. \sphinxstyleemphasis{Proc. Nati. Acad. Sci.}, 111(29):10550\textendash{}10555, 2014.
\bibitem[52]{99_References:terakawa-bj2011}
T. Terakawa and S. Takada. Multiscale Ensemble Modeling of Intrinsically Disordered Proteins: p53 N-Terminal Domain. \sphinxstyleemphasis{Biophys. J.}, 101(6):1450\textendash{}1458, 2011.
\bibitem[53]{99_References:freeman-2011}
G. S. Freeman, D. M. Hinckley, and J. J. de Pablo. A coarse-grain three-site-per-nucleotide model for DNA with explicit ions. \sphinxstyleemphasis{J. Chem. Phys.}, 135(16):165104, 2011.
\bibitem[54]{99_References:hinckley-2013}
D. M. Hinckley, G. S. Freeman, J. K. Whitmer, and J. J. de Pablo. An experimentally-informed coarse-grained 3-site-per-nucleotide model of DNA: Structure, thermodynamics, and dynamics of hybridization. \sphinxstyleemphasis{J. Chem. Phys.}, 139(14):144903, 2013.
\bibitem[55]{99_References:freeman-2014}
G. S. Freeman, D. M. Hinckley, J. P. Lequieu, J. K. Whitmer, and J. J. de Pablo. Coarse-grained modeling of DNA curvature. \sphinxstyleemphasis{J. Chem. Phys.}, 141(16):165103, 2014.
\bibitem[56]{99_References:tan-pwmcos}
C. Tan and S. Takada. Dynamic and Structural Modeling of the Specificity in Protein\textendash{}DNA Interactions Guided by Binding Assay and Structure Data. \sphinxstyleemphasis{J. Chem. Theory Comput.}, 14(7):3877\textendash{}3889, 2018.
\bibitem[57]{99_References:niina-nuc2017}
T. Niina, G. B. Brandani, C. Tan, and S. Takada. Sequence-dependent nucleosome sliding in rotation-coupled and uncoupled modes revealed by molecular simulations. \sphinxstyleemphasis{PLoS Comput. Biol.}, 13(12):e1005880, 2017.
\bibitem[58]{99_References:brandani-nuc2018}
G. B. Brandani, T. Niina, C. Tan, and S. Takada. DNA sliding in nucleosomes via twist defect propagation revealed by molecular simulations. \sphinxstyleemphasis{Nucleic Acids Res.}, 46(6):2788\textendash{}2801, 2018.
\bibitem[59]{99_References:dignon-plos2018}
G. L. Dignon, W. Zheng, Y. C. Kim, R. B. Best, and J. Mittal. Sequence determinants of protein phase behavior from a coarse-grained model. \sphinxstyleemphasis{PLoS Comput. Biol.}, 14(1):e1005941, 2018.
\bibitem[60]{99_References:hori-jctc2012}
N. Hori and S. Takada. Coarse-Grained Structure-Based Model for RNA-Protein Complexes Developed by Fluctuation Matching. \sphinxstyleemphasis{J. Chem. Theory Comput.}, 8(9):3384\textendash{}3394, 2012.
\bibitem[61]{99_References:schlitter-1994}
J. Schlitter, M. Engels, and P. Kruger. Targeted molecular dynamics: a new approach for searching pathways of conformational transitions. \sphinxstyleemphasis{J. Mol. Graph.}, 12:84\textendash{}89, 1994.
\bibitem[62]{99_References:mori-2021-1}
T. Mori, G. Terashi, D. Matsuoka, D. Kihara, and Y. Sugita. Efficient Flexible Fitting Refinement with Automatic Error Fixing for De Novo Structure Modeling from Cryo-EM Density Maps. \sphinxstyleemphasis{J. Chem. Inf. Model.}, 61:3516\textendash{}3528, 2021.
\bibitem[63]{99_References:rycaert-1997}
J. P. Ryckaert, G. Ciccotti, and H. J. C. Berendsen. Numerical-Integration of Cartesian Equations of Motion of a System with Constraints - Molecular-Dynamics of N-Alkanes. \sphinxstyleemphasis{J. Comput. Chem.}, 23:327\textendash{}341, 1977.
\bibitem[64]{99_References:andersen-1983}
H. C. Andersen. Rattle - a Velocity Version of the Shake Algorithm for Molecular-Dynamics Calculations. \sphinxstyleemphasis{J. Comput. Chem.}, 52:24\textendash{}34, 1983.
\bibitem[65]{99_References:miyamoto-1992}
S. Miyamoto and P. A. Kollman. Settle - an Analytical Version of the Shake and Rattle Algorithm for Rigid Water Models. \sphinxstyleemphasis{J. Comput. Chem.}, 13:952\textendash{}962, 1992.
\bibitem[66]{99_References:berendsen-1984}
H. J. C. Berendsen, J. P. M. Postma, W. F. Vangunsteren, A. Dinola, and J. R. Haak. Molecular-Dynamics with Coupling to an External Bath. \sphinxstyleemphasis{J. Chem. Phys.}, 81:3684\textendash{}3690, 1984.
\bibitem[67]{99_References:braun-2018}
E. Braun, S. M. Moosavi, and S. Smit. Anomalous Effects of Velocity Rescaling Algorithms: The Flying Ice Cube Effect Revisited. \sphinxstyleemphasis{J. Chem. Theory Comput.}, 14:5262\textendash{}5272, 2018.
\bibitem[68]{99_References:bussi-2007}
G. Bussi, D. Donadio, and M. Parrinello. Canonical sampling through velocity rescaling. \sphinxstyleemphasis{J. Chem. Phys.}, 126:014101, 2007.
\bibitem[69]{99_References:martyna-1992}
G. J. Martyna, M. L. Klein, and M. Tuckerman. Nose-Hoover Chains - the Canonical Ensemble Via Continuous Dynamics. \sphinxstyleemphasis{J. Chem. Phys.}, 97:2635\textendash{}2643, 1992.
\bibitem[70]{99_References:martyna-1996}
G. J. Martyna, M. E. Tuckerman, D. J. Tobias, and D. J. Klein. Explicit reversible integrators for extended systems dynamics. \sphinxstyleemphasis{Mol. Phys.}, 87:1117, 1996.
\bibitem[71]{99_References:adelman-1976}
S. A. Adelman and J. D. Doll. Generalized Langevin Equation Approach for Atom-Solid-Surface Scattering - General Formulation for Classical Scattering Off Harmonic Solids. \sphinxstyleemphasis{J. Chem. Phys.}, 64:2375\textendash{}2388, 1976.
\bibitem[72]{99_References:quigley-2004}
D. Quigley and M. I. J. Probert. Langevin dynamics in constant pressure extended systems. \sphinxstyleemphasis{J. Chem. Phys.}, 120:11432\textendash{}11441, 2004.
\bibitem[73]{99_References:zhang-1995}
Y. Zhang, S. E. Feller, B. R. Brooks, and Pastor R. W. Computer simulation of liquid/liquid interfaces. I. Theory and application to octane/water. \sphinxstyleemphasis{J. Chem. Phys.}, 103:10252\textendash{}10266, 1995.
\bibitem[74]{99_References:bussi-2009}
G. Bussi, T. Zykova-Timan, and M. Parrinello. Isothermal-isobaric molecular dynamics using stochastic velocity rescaling. \sphinxstyleemphasis{J. Chem. Phys.}, 130:074101, 2009.
\bibitem[75]{99_References:kandt-2007}
C. Kandt, W. L. Ash, and D. P. Tieleman. Setting up and running molecular dynamics simulations of membrane proteins. \sphinxstyleemphasis{Method}, 41:475\textendash{}488, 2007.
\bibitem[76]{99_References:sugita-1999}
Y. Sugita and Y. Okamoto. Replica-exchange molecular dynamics method for protein folding. \sphinxstyleemphasis{Chem. Phys. Lett.}, 314:141\textendash{}151, 1999.
\bibitem[77]{99_References:mitsutake-2001}
A. Mitsutake, Y. Sugita, and Y. Okamoto. Generalized-ensemble algorithms for molecular simulations of biopolymers. \sphinxstyleemphasis{Biopolymers}, 60:96\textendash{}123, 2001.
\bibitem[78]{99_References:mori-2010-1}
Y. Mori and Y. Okamoto. Generalized-ensemble algorithms for the isobaric-isothermal ensemble. \sphinxstyleemphasis{J. Phys. Soc. Jpn.}, 79:074003, 2010.
\bibitem[79]{99_References:mori-2010-2}
Y. Mori and Y. Okamoto. Replica-exchange molecular dynamics simulations for various constant temperature algorithms. \sphinxstyleemphasis{J. Phys. Soc. Jpn.}, 79:074001, 2010.
\bibitem[80]{99_References:okabe-2001}
T. Okabe, M. Kawata, Y. Okamoto, and M. Mikami. Replica-exchange Monte Carlo method for the isobaric-isothermal ensemble. \sphinxstyleemphasis{Chem. Phys. Lett.}, 335:435\textendash{}439, 2001.
\bibitem[81]{99_References:mori-2013}
T. Mori, J. Jung, and Y. Sugita. Surface-tension replica-exchange molecular dynamics method for enhanced sampling of biological membrane systems. \sphinxstyleemphasis{J. Chem. Theory. Comput.}, 9:5629\textendash{}5640, 2013.
\bibitem[82]{99_References:sugita-2000}
Y. Sugita, A. Kitao, and Y. Okamoto. Multidimensional replica-exchange method for free-energy calculations. \sphinxstyleemphasis{J. Chem. Phys.}, 113:6042\textendash{}6051, 2000.
\bibitem[83]{99_References:fukunishi-2002}
H. Fukunishi, O. Watanabe, and S. Takada. On the Hamiltonian replica exchange method for efficient sampling of biomolecular systems: Application to protein structure prediction. \sphinxstyleemphasis{J. Chem. Phys.}, 116:9058\textendash{}9067, 2002.
\bibitem[84]{99_References:terakawa-rest2}
T. Terakawa, T. Kameda, and S. Takada. On Easy Implementation of a Variant of the Replica Exchange with Solute Tempering in GROMACS. \sphinxstyleemphasis{J. Comput. Chem.}, 32:1228\textendash{}1234, 2011.
\bibitem[85]{99_References:kamiya-grest}
M. Kamiya and Y. Sugita. Flexible selection of the solute region in replica exchange with solute tempering: Application to protein-folding simulations. \sphinxstyleemphasis{J. Chem. Phys.}, 149:072304, 2018.
\bibitem[86]{99_References:berne-rest}
P. Liu, B. Kim, R. A. Friesner, and B. J. Berne. Replica exchange with solute tempering: A method for sampling biological systems in explicit water. \sphinxstyleemphasis{Proc. Natl. Acad. Sci. USA}, 102:13749\textendash{}13754, 2005.
\bibitem[87]{99_References:jiang-2009}
W. Jiang, M. Hodoscek, and B. Roux. Computation of Absolute Hydration and Binding Free Energy with Free Energy Perturbation Distributed Replica-Exchange Molecular Dynamics. \sphinxstyleemphasis{J. Chem. Theory Comput.}, 5:2583\textendash{}2588, 2009.
\bibitem[88]{99_References:re-2019}
S. Re, H. Oshima, K. Kasahara, M. Kamiya, and Y. Sugita. Encounter complexes and hidden poses of kinase-inhibitor binding on the free-energy landscape. \sphinxstyleemphasis{Proc. Natl. Acad. Sci. USA}, 116:18404\textendash{}18409, 2019.
\bibitem[89]{99_References:string-mean-forces}
L Maragliano, A. Fischer, E. Vanden-Eijnden, and G. Ciccotti. String method in collective variables: minimum free energy paths and isocommittor surfaces. \sphinxstyleemphasis{J. Chem. Phys.}, 125:24106, 2006.
\bibitem[90]{99_References:string-on-the-fly}
L. Maragliano and E. Vanden-Eijnden. On-the-fly string method for minimum free energy paths calculation. \sphinxstyleemphasis{Chem. Phys. Lett.}, 446:182\textendash{}190, 2007.
\bibitem[91]{99_References:string-swarms}
A. C Pan, D. Sezer, and B. Roux. Finding transition pathways using the string method with swarms of trajectories. \sphinxstyleemphasis{J. Phys. Chem. B}, 112:3432\textendash{}3440, 2008.
\bibitem[92]{99_References:string-genesis}
Y. Matsunaga, Y. Komuro, C. Kobayashi, J. Jung, T. Mori, and Y. Sugita. Dimensionality of Collective Variables for Describing Conformational Changes of a Multi-Domain Protein. \sphinxstyleemphasis{J. Phys. Chem. Lett.}, 7:1446\textendash{}1451, 2016.
\bibitem[93]{99_References:yagi-2021}
K. Yagi, S. Ito, and Y. Sugita. Exploring the Minimum-Energy Pathways and Free-Energy Profiles of Enzymatic Reactions with QM/MM Calculations. \sphinxstyleemphasis{J. Phys. Chem. B}, 125:4701\textendash{}4713, 2021.
\bibitem[94]{99_References:string-min-ene-path}
W. E, W. Ren, and E. Vanden-Eijnden. Simplified and improved string method for computing the minimum energy paths in barrier-crossing events. \sphinxstyleemphasis{J. Chem. Phys.}, 126:164103, 2007.
\bibitem[95]{99_References:nudged-elastic-band}
D. Sheppard, R. Terrell, and G. Henkelman. Optimization methods for finding minimum energy paths. \sphinxstyleemphasis{J. Chem. Phys.}, 128:134106, 2008.
\bibitem[96]{99_References:miao-2015}
Y. Miao, V. A. Feher, and J. A. McCammon. Gaussian Accelerated Molecular Dynamics: Unconstrained Enhanced Sampling and Free Energy Calculation. \sphinxstyleemphasis{J. Chem. Theory Comput.}, 11:3584\textendash{}3595, 2015.
\bibitem[97]{99_References:pang-2017}
Y. T. Pang, Y. Miao, Y. Wang, and J. A. McCammon. Gaussian Accelerated Molecular Dynamics in NAMD. \sphinxstyleemphasis{J. Chem. Theory Comput.}, 13:9\textendash{}19, 2017.
\bibitem[98]{99_References:hamelberg-2004}
D. Hamelberg, J. Mongan, and J. A. McCammon. Accelerated Molecular Dynamics: A Promising and Efficient Simulation Method for Biomolecules. \sphinxstyleemphasis{J. Chem. Phys.}, 120:11919\textendash{}11929, 2004.
\bibitem[99]{99_References:hamelberg-2007}
D. Hamelberg, C. A. F. de Oliveira, and J. A. McCammon. Sampling of Slow Diffusive Conformational Transitions with Accelerated Molecular Dynamics. \sphinxstyleemphasis{J. Chem. Phys.}, 127:155102, 2007.
\bibitem[100]{99_References:shen-2008}
T. Shen and D. Hamelberg. A Statistical Analysis of the Precision of Reweighting-Based Simulations. \sphinxstyleemphasis{J. Chem. Phys.}, 129:034103, 2008.
\bibitem[101]{99_References:miao-2014}
Y. Miao, W. Sinko, L. Pierce, D. Bucher, R. C. Walker, and J. A. McCammon. Improved Reweighting of Accelerated Molecular Dynamics Simulations for Free Energy Calculation. \sphinxstyleemphasis{J. Chem. Theory Comput.}, 10:2677\textendash{}2689, 2014.
\bibitem[102]{99_References:oshima-2019}
H. Oshima, S. Re, and Y. Sugita. Replica-Exchange Umbrella Sampling Combined with Gaussian Accelerated Molecular Dynamics for Free-Energy Calculation of Biomolecules. \sphinxstyleemphasis{J. Chem. Theory Comput.}, 2019. URL: \sphinxurl{https://pubs.acs.org/doi/10.1021/acs.jctc.9b00761}, \sphinxhref{https://doi.org/10.1021/acs.jctc.9b00761}{doi:10.1021/acs.jctc.9b00761} (https://doi.org/10.1021/acs.jctc.9b00761).
\bibitem[103]{99_References:warshel-1972}
A. Warshel and M. Karplus. Calculation of Ground and Excited State Potential Surfaces of Conjugated Molecules. I. Formulation and Parametrization. \sphinxstyleemphasis{J. Am. Chem. Soc.}, 94:5612\textendash{}5625, 1972.
\bibitem[104]{99_References:warshel-1976}
A. Warshel and M. Levitt. Theoretical studies of enzymic reactions: Dielectric, electrostatic and steric stabilization of the carbonium ion in the reaction of lysozyme. \sphinxstyleemphasis{J. Mol. Biol.}, 103:227\textendash{}249, 1976.
\bibitem[105]{99_References:yagi-2019}
K. Yagi, K. Yamada, C. Kobayashi, and Y. Sugita. Anharmonic Vibrational Analysis of Biomolecules and Solvated Molecules Using Hybrid QM/MM Computations. \sphinxstyleemphasis{J. Chem. Theory Comput.}, 15:1924\textendash{}1938, 2019.
\bibitem[106]{99_References:tama-2004}
F. Tama, O. Miyashita, and C. L. Brooks. Flexible multi scale fitting of atomic structures into low resolution electron density maps with elastic network normal mode analysis. \sphinxstyleemphasis{J. Mol. Biol.}, 337:985\textendash{}999, 2004.
\bibitem[107]{99_References:orzechowski-2008}
M. Orzechowski and F. Tama. Flexible fitting of high resolution X ray structures into cryoelectron microscopy maps using biased molecular dynamics simulations. \sphinxstyleemphasis{Biophys. J.}, 95:5692\textendash{}5705, 2008.
\bibitem[108]{99_References:trabuco-2008}
L. G. Trabuco, E. Villa, K. Mitra, J. Frank, and K. Schulten. Flexible fitting of atomic structures into electron microscopy maps using molecular dynamics. \sphinxstyleemphasis{Structure}, 16:673\textendash{}683, 2008.
\bibitem[109]{99_References:topf-2008}
M. Topf, K. Lasker, B. Webb, H. Wolfson, W. Chiu, and A. Sali. Protein structure fitting and refinement guided by cryo EM density. \sphinxstyleemphasis{Structure}, 16:295\textendash{}307, 2008.
\bibitem[110]{99_References:ishida-2014}
H. Ishida and A. Matsumoto. Free energy landscape of reverse tRNA translocation through the ribosome analyzed by electron microscopy density maps and molecular dynamics simulations. \sphinxstyleemphasis{PloS one}, 9:e101951, 2014.
\bibitem[111]{99_References:miyashita-2017}
O. Miyashita, C. Kobayashi, T. Mori, Y. Sugita, and F. Tama. Flexible fitting to cryo-EM density map using ensemble molecular dynamics simulations. \sphinxstyleemphasis{J. Comput. Chem.}, 38:1447\textendash{}1461, 2017.
\bibitem[112]{99_References:whitford-2011}
P. C. Whitford, A. Ahmed, Y. Yu, Hennelly, S. P., F. Tama, Spahn, C. M., J. N. Onuchic, and K. Y. Sanbonmatsu. Excited states of ribosome translocation revealed through integrative molecular modeling. \sphinxstyleemphasis{Proc. Natl. Acad. Sci. U.S.A.}, 108:18943\textendash{}18948, 2011.
\bibitem[113]{99_References:mori-2019}
T. Mori, M. Kulik, O. Miyashita, J. Jung, F. Tama, and Y. Sugita. Acceleration of cryo-EM flexible fitting for large biomolecular systems by efficient space partitioning. \sphinxstyleemphasis{Structure}, 27:161\textendash{}174.e3, 2019.
\bibitem[114]{99_References:gao-1989}
J. Gao, K. Kuczera, B. Tidor, and M. Karplus. Hidden thermodynamics of mutant proteins: A molecular dynamics analysis. \sphinxstyleemphasis{Science}, 244:1069\textendash{}1072, 1989.
\bibitem[115]{99_References:pearlman-1994}
D. A. Pearlman. A comparison of alternative approaches to free energy calculations. \sphinxstyleemphasis{J. Phys. Chem.}, 98:1487\textendash{}1493, 1994.
\bibitem[116]{99_References:axelsen-1998}
P. H. Axelsen and D. Li. Improved convergence in dual-topology free energy calculations through use of harmonic restraints. \sphinxstyleemphasis{J. Comput. Chem.}, 19:1278\textendash{}1283, 1998.
\bibitem[117]{99_References:jiang-2019}
W. Jiang, C. Chipot, and B. Roux. Computing relative binding 791 affinity of ligands to receptor: An effective hybrid single-dual-topology free-energy perturbation approach in NAMD. \sphinxstyleemphasis{J. Chem. Inf. Model.}, 59:3794\textendash{}3802, 2019.
\bibitem[118]{99_References:zacharias-1994}
M. Zacharias, T. P. Straatsma, and J. A. McCammon. Separation‐shifted scaling, a new scaling method for Lennard‐Jones interactions in thermodynamic integration. \sphinxstyleemphasis{J. Chem. Phys.}, 100:9025\textendash{}9031, 1994.
\bibitem[119]{99_References:steinbrecher-2011}
T. Steinbrecher, I. Joung, and D. A. Case. Soft-core potentials in thermodynamic integration: Comparing one- and two-step transformations. \sphinxstyleemphasis{J. Comput. Chem.}, 32:3253\textendash{}3263, 2011.
\end{sphinxthebibliography}



\renewcommand{\indexname}{Index}
\printindex
\end{document}